<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<!--
############################################################################
# LICENSE INFO:                                                            #
############################################################################
#    This file is part of CAMPARI.                                         #
#                                                                          #
#    Version 2.0                                                           #
#                                                                          #
#    Copyright (C) 2014, The CAMPARI development team (current and former  #
#                        contributors)                                     #
#                        Andreas Vitalis, Adam Steffen, Rohit Pappu, Hoang #
#                        Tran, Albert Mao, Xiaoling Wang, Jose Pulido,     #
#                        Nicholas Lyle, Nicolas Bloechliger                #
#                                                                          #
#    Website: http://sourceforge.net/projects/campari/                     #
#                                                                          #
#    CAMPARI is free software: you can redistribute it and/or modify       #
#    it under the terms of the GNU General Public License as published by  #
#    the Free Software Foundation, either version 3 of the License, or     #
#    (at your option) any later version.                                   #
#                                                                          #
#    CAMPARI is distributed in the hope that it will be useful,            #
#    but WITHOUT ANY WARRANTY; without even the implied warranty of        #
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
#    GNU General Public License for more details.                          #
#                                                                          #
#    You should have received a copy of the GNU General Public License     #
#    along with CAMPARI.  If not, see <http://www.gnu.org/licenses/>.      #
############################################################################
# AUTHORSHIP INFO:                                                         #
############################################################################
#                                                                          #
# MAIN AUTHOR:   Andreas Vitalis                                           #
# CONTRIBUTIONS: Adam Steffen                                              #
#                                                                          #
############################################################################
-->
<html xmlns="http://www.w3.org/1999/xhtml" dir="ltr" lang="en-US" xml:lang="en">
<head>
    <!--
    Designed by CAMPARI Development Group
    Base template (without user's data) checked by http://validator.w3.org : "This page is valid XHTML 1.0 Transitional"
    -->
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"></meta>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge"></meta>
    <title>CAMPARI Input Files</title>
    <link rel="stylesheet" href="style_red.css" type="text/css" media="screen"></link>
</head>
<body>
<div id="art-page-background-simple-gradient"></div>
<div id="art-main">
    <!-- Header -->
    <div class="art-Sheet2">
        
        <!-- Banner Art -->
        <div class="art-Header">
            <div class="art-Header-png"></div>
        </div>
                
        <!-- Top Navigation Bar -->
        <div class="art-nav">
            <div class="l"></div>
            <div class="r"></div>
            <ul class="art-menu">
                <li><a href="index.html"><span class="l"></span><span class="r"></span><span class="t">Home</span></a></li>
                <li><a href="documentation.html" class=" active"><span class="l"></span><span class="r"></span><span class="t">Documentation</span></a></li>
                <li><a href="references.html"><span class="l"></span><span class="r"></span><span class="t">References</span></a></li>
                <li><a href="contact.html"><span class="l"></span><span class="r"></span><span class="t">Support</span></a></li>
                <li><a href="download.html"><span class="l"></span><span class="r"></span><span class="t">Download</span></a></li>
            </ul>
        </div>
            
        <!-- Top Sub Navigation Bar -->
        <div class="nav-menu">
            <ul class="navlist">
                <li><a href="documentation.html">Overview</a></li>
                <li><a href="install.html">Install</a></li>
                <li><a href="tutorials.html">Tutorials</a></li>
                <li><a href="inputfiles.html" class=" active">Inputfiles</a></li>
                <li><a href="keywords.html">Keywords</a></li>
                <li><a href="outputfiles.html">Outputfiles</a></li>
                <li><a href="parameters.html">Parameters</a></li>
                <li><a href="development.html">Development</a></li>
                <li></li>
            </ul>
        </div> 
    </div>
        
    <!-- Main Page Contents -->
    <div class="art-Sheet">
        <div class="art-Sheet-bl"></div>
        <div class="art-Sheet-br"></div>
        <div class="art-Sheet-bc"></div>
        <div class="art-Sheet-cl"></div>
        <div class="art-Sheet-cr"></div>
        <div class="art-Sheet-cc"></div>
        <div class="art-Sheet-body">
            
            <div class="art-contentLayout">
                <!-- Page Contents -->
                <div class="art-content">
                <div class="art-Post">
                <div class="art-Post-body">

<div class="art-PostContent">
<br/>
<h2><a class="NoHi" id="TOP">Input Files for CAMPARI Runs</a></h2>
<hr/>
<h3>Classes of Input Files:<br/>
</h3>
<ol>
  <li><a href="inputfiles.html#Sec1">Files defining the system:</a></li>
  <li><a href="inputfiles.html#Sec2">Files altering the system:</a></li>
  <li><a href="inputfiles.html#Sec3">Files redefining standard terms of the energy function:</a></li>
  <li><a href="inputfiles.html#Sec4">Files defining auxiliary terms to the energy function:</a></li>
  <li><a href="inputfiles.html#Sec5">Files relevant for global behavior of analysis functionality:</a></li>
  <li><a href="inputfiles.html#Sec6">Files relevant to specific analysis routines:</a></li>
  <li><a href="inputfiles.html#Sec7">Files relevant to the NetCDF analysis mode:</a></li>
</ol>
<br/>
<h3>Notes on Building CAMPARI Input Files</h3><br/>
<u><b>IMPORTANT COMMENT FOR MPI RUNS:</b></u><br/>
<br/>
While the key-file only needs to be visible to the master node, most
data-files (like the steric grids, the backbone segment file or of
course the sequence information) need to be read by all nodes. Since
the execution is forced to be strictly sequential, it is absolutely
valid to point them all to a single copy of those files, <i>i.e.</i>, jobs can
very well be run in a pseudo-single-CPU setting on a cluster-wide
NFS-mount. Otherwise, consistency between nodes with regard to the
contents of those files <b>has to be ensured by the user</b> (or unpredictable
crashes loom).<br/>
<br/>
<br/>
<br/>
<br/>
<h3><a class="NoHi" id="Sec1">Section 1: Files defining the system</a></h3>
<hr style="width: 100%; height: 2px;"></hr> (<a href="inputfiles.html#TOP">back to top</a>)<br/>
<h4><a class="NoHi" id="FMCSC_SEQFILE">Sequence input file:</a></h4>
Keyword: <b><a href="keywords.html#SEQFILE">FMCSC_SEQFILE</a></b><br/>
<br/>
Aside from the <a href="keywords.html#PARAMETERS">parameter file</a>, this file is the only one that is always required.<br/><br/>
This input file specifies the sequence of the system to be simulated or analyzed. Every residue (the base organizational unit) 
occupies one line and uses a base three-letter representation. There is a set of polymer residues and small molecules
that is supported in nearly every type of run CAMPARI can perform since CAMPARI possesses an internal representation of those,
<i>i.e.</i>, it is aware of molecular topology and intrinsic flexibility. These are described next. However, it is also possible
to obtained limited support for other residues and molecules including complex polymers.
Here, the topology and internal flexibility of each unsupported entity are inferred from a structural input file and by analogy.
The details of the inference from structural input are provided <a href="keywords.html#PDB_TEMPLATE">elsewhere</a>,
but an overview of what is needed in the sequence file is given below the sections on natively supported residues.<br/>
<br/>
<b><u>Polypeptide residues:</u></b><br/>
<br/>
Three-letter code, all twenty common amino acids
are supported:<br/>
<br/>
<big><code>GLY&nbsp; ALA&nbsp; VAL&nbsp; LEU&nbsp; ILE&nbsp;
PRO&nbsp; MET&nbsp; SER&nbsp; THR&nbsp; ASN&nbsp; GLN&nbsp; CYS&nbsp;
PHE&nbsp; TYR&nbsp; TRP&nbsp; HIE/HID&nbsp; ASP&nbsp; GLU&nbsp;
ARG&nbsp; LYS<br/></code>&nbsp;</big><br/>
In addition to the histidine tautomers (HID, HIE), alternative protonation
states can be modeled for the following residues:<br/>
ASP &rarr; ASH (aspartic acid by addition of a proton, pK<sub>a</sub> in free form ~3.9)<br/>
CYS &rarr; CYX (thiolate anion by removal of a proton, pK<sub>a</sub> of CYS in free form ~8.0)<br/>
GLU &rarr; GLH (glutamic acid by addition of a proton, pK<sub>a</sub> in free form ~4.1)<br/>
HIE/HID &rarr; HIP (protonated histidine (imidazolium cation) by addition of a protein, pK<sub>a</sub> in free form ~6.0)<br/>
LYS &rarr; LYD (neutral <i>n</i>-butylamine side chain by removal of a proton, pK<sub>a</sub> of LYS in free form ~10.5)<br/>
TYR &rarr; TYO (phenolate anion by removal of a proton, pK<sub>a</sub> of TYR in free form ~10.5)<br/>
Note that the neutral form of arginine is a species of almost zero practical relevance due to
the pK<sub>a</sub> of ~12.5. It is not supported by CAMPARI at the moment.
By adding _N, _C, or _D any residue can be made C-terminal,
N-terminal (each charged) or the chirality can be changed to D(R)
(except GLY). To achieve both or to make a free amino acid simply use
both modifiers (<i>i.e.</i>, _C_D or _D_C would give a C-terminal D-amino
acid).<br/>
<br/>
Non-standard amino acids:<br/>
AIB (alpha-amino isobutyric acid), ABA (alpha-amino butyric acid), NVA
(norvaline = alpha-amino valeric acid), NLE (norleucine = alpha-amino
capronic acid), ORN (ornithine = alpha,delta-diamino valeric acid), DAB
(alpha,gamma-diamino butyric acid), PTR (hydrogen phosphate ester(-) of tyrosine), TPO (hydrogen phosphate ester(-) of threonine),
SEP (hydrogen phosphate ester(-) of serine), KAC (NZ-acetylated lysine), KM1 (NZ-monomethylated lysine(+)),
KM2 (NZ,NZ-dimethylated lysine(+)), KM3 (NZ,NZ,NZ-trimethylated lysine(+)). The same modifiers are available.<br/>
<br/>
It should be noted that quenching the tautomer isomerization for residues HID, HIE, ASH, GLH,
SEP, PTR, and TPO can represent a severe approximation both thermodynamically and kinetically. Histidine is the only
example for which two alternative tautomers can be built explicitly (HID/HIE). However, they do remain fixed species 
during simulations (quenching still occurs). Lastly, the deprotonated versions of the methylated lysines KM1 and KM2 are currently
not supported despite having pKa values closer to the physiological range than lysine itself.
<br/>
<br/>
<b>Internal degrees of freedom:</b><br/>
<br/>
<u>Backbone:</u><br/>
All non N-terminal residues: &#969; (CA<sub>-1</sub>,C<sub>-1</sub>,N,CA); in case of residue -1 being ACE CA<sub>-1</sub> is CH3; in case of FOR CA<sub>-1</sub> is H<br/>
<br/>
All except PRO: &#966; (C<sub>-1</sub>,N,CA,C); in case of N-terminal residues C<sub>-1</sub> is 1HN<br/>
<br/>
All: &#968; (N,CA,C,N<sub>+1</sub>); in case of C-terminal residues N<sub>+1</sub> is 2OXT<br/>
<br/>
PRO: Puckering (seven non-redundant degrees of freedom): torsions C<sub>-1</sub>-N-CA-C, C<sub>-1</sub>-N-CA-CB, N-CA-CB-CG, CB-CA-N-CD and angles N-CA-CB, CA-CB-CG, CA-N-CD<br/>
<br/>
<u>Sidechain:</u><br/>
AIB, ALA, GLY, PRO: none<br/>
ABA, ARG, ASH, ASN, ASP, DAB, GLH, GLN, GLU, HID, HIE, HIP, LEU, LYD, LYS, KAC, KM1, KM2, KM3, MET, NVA, NLE, ORN, PHE, PTR, TRP, TYO, TYR:  &#967;<sub>1</sub> (N,CA,CB,CG)<br/>
SER, SEP: &#967;<sub>1</sub> (N,CA,CB,OG)<br/>
THR, TPO: &#967;<sub>1</sub> (N,CA,CB,OG1)<br/>
CYS, CYX: &#967;<sub>1</sub> (N,CA,CB,SG)<br/>
VAL, ILE: &#967;<sub>1</sub> (N,CA,CB,CG1)<br/>
HID, HIE, HIP, PHE, PTR, TRP, TYO, TYR: &#967;<sub>2</sub> (CA,CB,CG,CD1)<br/>
ARG, GLH, GLN, GLU, LYD, LYS, KAC, KM1, KM2, KM3, NVA, NLE, ORN: &#967;<sub>2</sub> (CA,CB,CG,CD)<br/>
ASH, ASN, ASP: &#967;<sub>2</sub> (CA,CB,CG,OD1)<br/>
LEU: &#967;<sub>2</sub> (CA,CB,CG,CD1)<br/>
ILE: &#967;<sub>2</sub> (CA,CB,CG1,CD1)<br/>
MET: &#967;<sub>2</sub> (CA,CB,CG,SD)<br/>
SER: &#967;<sub>2</sub> (CA,CB,OG,HG)<br/>
THR: &#967;<sub>2</sub> (CA,CB,OG1,HG1)<br/>
CYS: &#967;<sub>2</sub> (CA,CB,SG,HG)<br/>
DAB: &#967;<sub>2</sub> (CA,CB,CG,ND)<br/>
SEP: &#967;<sub>2</sub> (CA,CB,OG,P)<br/>
TPO: &#967;<sub>2</sub> (CA,CB,OG1,P)<br/>
ASH: &#967;<sub>3</sub> (CB,CG,OD2,HD2)<br/>
ARG, ORN: &#967;<sub>3</sub> (CB,CG,CD,NE)<br/>
LYD, LYS, KAC, KM1, KM2, KM3, NLE: &#967;<sub>3</sub> (CB,CG,CD,CE)<br/>
ORN:  &#967;<sub>3</sub> (CB,CG,CD,NE)<br/>
GLH, GLN, GLU: &#967;<sub>3</sub> (CB,CG,CD,OE1)<br/>
MET: &#967;<sub>3</sub> (CB,CG,SD,CE)<br/>
TYR: &#967;<sub>3</sub> (CE2,CZ,OH,HH)<br/>
PTR: &#967;<sub>3</sub> (CE2,CZ,OZ,P)<br/>
SEP: &#967;<sub>3</sub> (CB,OG,P,OH)<br/>
TPO: &#967;<sub>3</sub> (CB,OG1,P,OH)<br/>
ARG: &#967;<sub>4</sub> (CG,CD,NE,CZ)<br/>
LYD, LYS, KAC, KM1, KM2, KM3: &#967;<sub>4</sub> (CG,CD,CE,NZ)<br/>
GLH: &#967;<sub>4</sub> (CG,CD,OE2,HE2)<br/>
PTR: &#967;<sub>4</sub> (CZ,OZ,P,OH)<br/>
SEP: &#967;<sub>4</sub> (OG,P,OH,HOP)<br/>
TPO: &#967;<sub>4</sub> (OG1,P,OH,HOP)<br/>
LYD: &#967;<sub>5</sub> (CD,CE,NZ,1HZ)<br/>
KAC: &#967;<sub>5</sub> (CD,CE,NZ,CH)<br/>
KM1, KM2, KM3: &#967;<sub>5</sub> (CD,CE,NZ,CH1)<br/>
PTR: &#967;<sub>5</sub> (OZ,P,OH,HOP)<br/>
KAC: &#967;<sub>6</sub> (CE,NZ,CH,CI)<br/><br/>

Note that the "missing" torsions are those assumed to be quasi-redundant, <i>e.g.</i> the dihedral angles formed across
the three CZ to nitrogen bonds in the guanidino group of arginine. They can be sampled, however, both in torsional dynamics
(&rarr; <a href="keywords.html#TMD_UNKMODE">TMD_UNKMODE</a>) and in Monte Carlo runs (&rarr; <a href="keywords.html#OTHERFREQ">OTHERFREQ</a>).<br/>
<br/>
<b><u>Polypeptide cap residues:</u></b><br/>
<br/>
FOR (N-terminal COH-unit)&nbsp; ACE (N-terminal acetyl-unit)&nbsp; NH2
(C-terminal amide)&nbsp; NME (C-terminal N-methyl amide)<br/><br/>
No modifiers are
available, <i>i.e.</i>, FOR and ACE are always N-terminal, NH2 and NME are always
C-terminal. It is not recommended to directly concatenate two caps to
build small molecule amides.<br/>
<br/>
<b>Internal degrees of freedom:</b><br/>
<br/>
<u>Backbone:</u><br/>
FOR, ACE, NH2: none<br/>
NME: &#969; (CA<sub>-1</sub>,C<sub>-1</sub>,N,CH3)<br/>
<br/>
Note that the "missing" torsions are those assumed to be quasi-redundant, <i>e.g.</i>,
the dihedral angles formed across the C-CH3 bond in ACE. They can be sampled, however, both in torsional dynamics
(&rarr; <a href="keywords.html#TMD_UNKMODE">TMD_UNKMODE</a>) and in Monte Carlo runs (&rarr; <a href="keywords.html#OTHERFREQ">OTHERFREQ</a>).<br/>
<br/>
<b><u>Nucleic acid residues:</u></b><br/>
<br/>
In terms of nucleic acids twelve 5'-nucleotides are supported (<i>i.e.</i>, the
base element is XM-5'P)<br/>
<br/>
<big><samp>D5P&nbsp; DPC&nbsp; DPU&nbsp; DPT&nbsp; DPA&nbsp;
DPG&nbsp; R5P&nbsp; RPC&nbsp; RPU&nbsp; RPT&nbsp; RPA&nbsp; RPG<br/><br/></samp></big>
The DPX are the common deoxyribonucleotides, while
the RPX are the ribonucleotides D5P and R5P are empty
sugar-5'-phosphates (<i>i.e.</i>, instead of the glcyosamine bond, the C1' has
its native hydroxyl group). By adding _N or _C any residue can be made
into a 5'- or 3'-capping residue, respectively. 5'-terminal residues
generated in this way always carry a free 5'-phosphate group (singly
charged). 3'-terminal residues generated in this way carry only a free
3'-hydroxyl.<br/><br/>
Similar to a comment made above for polypeptide residues,
it should be noted that quenching the tautomer isomerization for residues 
carrying a free 5'-phosphate(-) may represent a severe approximation in both
a thermodynamic and kinetic sense.<br/>
<br/>
<b>Internal degrees of freedom:</b><br/>
<br/>
<span style="text-decoration: underline;">Backbone:</span><br/>
non-5'-terminal: nuc_bb_1(C3*-O3P-P-O5P) - note that in CAMPARI
convention only C3* is part of the preceding residue<br/>
<br/>
5'-terminal: nuc_bb_1(O5P-P-O3P-HOP)<br/>
<br/>
All: nuc_bb_2(O3P-P-O5P-C5*), nuc_bb_3(P-O5P-C5*-C4*),
nuc_bb_4(O5P-C5*-C4*-C3*)<br/>
<br/>
Non-3'-terminal: nuc_bb_5(C4*-C3*-O3P-P) - note that in CAMPARI
convention O3P and P are part of the following residue<br/>
<br/>
3'-terminal: nuc_bb_5(C4*-C3*-O3*-HO3*)<br/>
<br/>
All: Sugar puckering (seven degrees of freedom): torsions C5*-C4*-C3*-O3P/O3*, C5*-C4*-C3*-C2*, C4*-C3*-C2*-C1*, C2*-C3*-C4*-O4* and angles C4*-C3*-C2*, C3*-C2*-C1*, C3*-C4*-O4*<br/>
<br/>
<span style="text-decoration: underline;">Sidechain:</span><br/>
&nbsp;RPC, RPU, RPT: &#967;<sub>1</sub>(C3*,C2*,O2*,HO2*), &#967;<sub>2</sub>(C2*,C1*,N1,C2)<br/>
&nbsp;RPT only: &#967;<sub>3</sub>(C4,C5,C5M,1H5M)<br/>
&nbsp;RPG, RPA: &#967;<sub>1</sub>(C3*,C2*,O2*,HO2*), &#967;<sub>2</sub>(C2*,C1*,N9,C4)<br/>
&nbsp;R5P: &#967;<sub>1</sub>(C3*,C2*,O2*,HO2*), &#967;<sub>2</sub>(C2*,C1*,O1*,HO1*)<br/>
&nbsp;DPC, DPU, DPT: &#967;<sub>1</sub>(C2*,C1*,N1,C2)<br/>
&nbsp;DPT only: &#967;<sub>2</sub>(C4,C5,C5M,1H5M)<br/>
&nbsp;DPG, DPA: &#967;<sub>1</sub>(C2*,C1*,N9,C4)<br/>
&nbsp;D5P: &#967;<sub>1</sub>(C2*,C1*,O1*,HO1*)<br/><br/>
Note that the "missing" torsions are those assumed to be quasi-redundant, <i>e.g.</i>
the dihedral angles defined by the NH<sub>2</sub> groups in DPA, DPG, <i>etc.</i> They can be sampled, however, both in torsional dynamics
(&rarr; <a href="keywords.html#TMD_UNKMODE">TMD_UNKMODE</a>) and in Monte Carlo runs (&rarr; <a href="keywords.html#OTHERFREQ">OTHERFREQ</a>).<br/>
<br/>
<span style="text-decoration: underline;"><b>Nucleic acid cap
residues:</b></span><br/>
<br/>
As 5'-terminal capping groups, 12 nucleosides are also supported: <br/>
<br/>
<big><code>DIB&nbsp; DIC&nbsp; DIU&nbsp; DIT&nbsp; DIA&nbsp;
DIG&nbsp; RIB&nbsp; RIC&nbsp; RIU&nbsp; RIT&nbsp; RIA&nbsp; RIG<br/><br/></code></big>
DIB and RIB are deoxyribose and ribose,
respectively, <i>i.e.</i>, empty nucleosides, <i>i.e.</i>, sugars. The remaining DIX
and RIX are the standard nucleosides with the 5 canonical bases. Note
that nucleosides cannot be made C-terminal, <i>i.e.</i>, it is not possible
(yet) to build free nucleosides using this capping residues.<br/>
<br/>
<b>Internal degrees of freedom:</b><br/>
<br/>
For backbone:<br/>
&nbsp;&nbsp;&nbsp; nuc_bb_1(HO5*-O5*-C5*-C4*)<br/>
&nbsp;&nbsp;&nbsp; nuc_bb_2(O5*-C5*-C4*-C3*)<br/>
&nbsp;&nbsp;&nbsp; nuc_bb_3(C4*-C3*-O3P-P)<br/>
Sugar puckering and sidechain degrees of freedom are identical to full nucleotides (see above).<br/>
<br/>
<span style="text-decoration: underline;"><b>Small molecules:</b></span><br/>
<br/>
NA+ (sodium ion), CL- (chloride ion), SPC (<a href="references.html#ref1_22">SPC or SPC/E water</a>),
T3P (<a href="references.html#ref1_10">TIP3P water</a>), T4P (<a href="references.html#ref1_10">TIP4P water</a>), T4E (<a href="references.html#ref1_21">TIP4P-Ewald water</a>),
T5P (<a href="references.html#ref1_13">TIP5P water</a>), URE (urea), NMF (Z-N-methylformamide (<i>i.e.</i>,
polar H and O are trans)), NMA (Z-N-methylacetamide), ACA (acetamide),
PPA (propionamide), FOA (formamide), DMA (N,N'-dimethylacetamide), CH4
(methane), MOH (methanol), PCR (p-cresol), K+ (potassium ion), BR-
(bromide in), CS+ (caesium ion), I- (iodide ion), O2 (molecular oxygen: note that
diatomic and linear molecules in general are not yet supported in <a href="keywords.html#DYNAMICS">dynamics</a>),
NH4 (ammonium ion), AC- (acetate ion), GDN (guanidinium cation), NO3 (nitrate ion), PCL (perchlorate ion), 1MN (methylammonium ion),
2MN (dimethylammonium ion), CYT (cytosine), URA
(uracil), THY (thymine), PUR (purine), ADE (adenine), GUA (guanine), PRP (propane), NBU (n-butane), IBU (2-methylpropane),
EMT (ethylmethylthioether), EOH (ethanol), MSH (methanethiol), BEN (benzene), TOL (toluene), NAP (naphthalene),
IMD (&#948;-protonated 2-methylimidazole), IME (&#949;-protonated 2-methylimidazole), MIN (3-methylindol)<br/>
<br/>
<b>Degrees of freedom:</b><br/>
&nbsp;&nbsp;&nbsp; O2, SPC, T3P, T4P, T5P, T4E, URE, CH4, FOA, BEN, NAP, PUR, ADE, GUA, CYT, URA: RBC<br/>
&nbsp;&nbsp;&nbsp; NA+, CL-, K+, BR-, CS+, I-, NH4, GDN, NO3, LCP: RBC<br/>
&nbsp;&nbsp;&nbsp; ACA: RBC, &#967;<sub>1</sub> (N,C,CCT,1HCT)<br/>
&nbsp;&nbsp;&nbsp; AC-: RBC, &#967;<sub>1</sub> (1OXT,C,CH3,1H)<br/>
&nbsp;&nbsp;&nbsp; MOH: RBC, &#967;<sub>1</sub> (1H,C,O,HO)<br/>
&nbsp;&nbsp;&nbsp; THY: RBC, &#967;<sub>1</sub> (1HT,CT,C5,C6)<br/>
&nbsp;&nbsp;&nbsp; 1MN: RBC, &#967;<sub>1</sub> (1HN,N,CT,1HT)<br/>
&nbsp;&nbsp;&nbsp; NMF: RBC, &#969; (O,C,N,CCT), &#967;<sub>1</sub> (C,N,CNT,1HNT)<br/>
&nbsp;&nbsp;&nbsp; NMA: RBC, &#969; (O,C,N,CCT), &#967;<sub>1</sub> (C,N,CNT,1HNT), &#967;<sub>2</sub> (N,C,CCT,1HCT)<br/>
&nbsp;&nbsp;&nbsp; PPA: RBC, &#967;<sub>1</sub> (N,C,CCT,CBT), &#967;<sub>2</sub> (C,CCT,CBT,1HBT)<br/>
&nbsp;&nbsp;&nbsp; PCR: RBC, &#967;<sub>1</sub> (C21,C1,CT,1HCT), &#967;<sub>2</sub> (C31,C4,O,HO)<br/>
&nbsp;&nbsp;&nbsp; DMA: RBC, &#967;<sub>1</sub> (C,N,CZT,1HZT), &#967;<sub>2</sub>, (C,N,CET,1HET), &#967;<sub>3</sub> (N,C,CCT,1HCT)<br/>
&nbsp;&nbsp;&nbsp; PRP: RBC, &#967;<sub>1</sub> (2CT,CB,1CT,1HT1), &#967;<sub>2</sub> (1CT,CB,2CT,1HT2)<br/>
&nbsp;&nbsp;&nbsp; 2MN: RBC, &#967;<sub>1</sub> (2CT,N,1CT,1HT1), &#967;<sub>2</sub> (1CT,N,2CT,1HT2)<br/>
&nbsp;&nbsp;&nbsp; IBU: RBC, &#967;<sub>1</sub> (2CT,CB,1CT,1HT1), &#967;<sub>2</sub> (3CT,CB,2CT,1HT2), &#967;<sub>3</sub> (2CT,CB,3CT,1HT3)<br/>
&nbsp;&nbsp;&nbsp; NBU: RBC, &#967;<sub>1</sub> (1CT,1CB,2CB,2CT), &#967;<sub>2</sub> (2CB,1CB,1CT,1HT1), &#967;<sub>3</sub> (1CB,2CB,2CT,1HT2)<br/>
&nbsp;&nbsp;&nbsp; EMT: RBC, &#967;<sub>1</sub> (CCT,CB,S,CNT), &#967;<sub>2</sub> (CB,S,CNT,1HNT), &#967;<sub>3</sub> (S,CB,CCT,1HCT)<br/>
&nbsp;&nbsp;&nbsp; EOH: RBC, &#967;<sub>1</sub> (O,CB,CT,1HT), &#967;<sub>2</sub> (CT,CB,O,HO)<br/>
&nbsp;&nbsp;&nbsp; MSH: RBC, &#967;<sub>1</sub> (1H,C,S,1HS)<br/>
&nbsp;&nbsp;&nbsp; IMD, IME: RBC, &#967;<sub>1</sub> (N2,C1,CT,1HT)<br/>
&nbsp;&nbsp;&nbsp; TOL: RBC, &#967;<sub>1</sub> (C2,C1,CT,1HT)<br/>
&nbsp;&nbsp;&nbsp; MIN: RBC, &#967;<sub>1</sub> (C2,C3,CT,1HT)<br/><br/>
Note that the methyl spins fall away if united-atom representation is used (see <a href="keywords.html#UAMODEL">FMCSC_UAMODEL</a>) and that the remaining &#967;s may be shifted down (relevant for EOH, PCR). Note as well that the different water
models listed have different numbers of sites (3-5), different covalent geometry, and are all assumed to be 
completely rigid internally. Of course, a match of the published models still requires using the correct 
<a href="keywords.html#energy_terms">energy functions</a> and <a href="parameters.html">parameters</a>.
As with all other systems, the "missing" torsions are those assumed to be quasi-redundant, <i>e.g.</i>
the dihedral angles defined by the NH<sub>2</sub> groups in primary amides. They can be sampled, however, both in torsional dynamics
(&rarr; <a href="keywords.html#TMD_UNKMODE">TMD_UNKMODE</a>) and in Monte Carlo runs (&rarr; <a href="keywords.html#OTHERFREQ">OTHERFREQ</a>).
<br/>
<br/>
<b><u>Chemical crosslinks:</u></b><br/>
<br/>
<b>Disulfide linkages:</b><br/>
<br/>
The only currently supported type of crosslink are disulfide linkages between two cysteine residues
on one or two polypeptide chains. They are specified by identifying the linked cysteine residue with
the lower residue index number, and - separated by a blank - appending on the same line the residue
index number for the target linkage residue. The target can either be on the same or in a different
molecule.<br/>
Note that several limitations exist:<br/>
<ol>
<li>Intramolecular crosslinks cannot be so complicated as to create a topology in which multiple ring constraints
depend on each other completely. This is analyzed by CAMPARI automatically and the program will terminate
abnormally if it identifies such a situation.</li>
<li>Intermolecular crosslinks cannot create ring-like topologies but are otherwise unrestricted. This limitation
is more likely to be overcome in the near future than the prior one.</li>
<li>Two residues which are direct sequence neighbors can be crosslinked but their arrangement cannot be constructed
randomly (see <a href="keywords.html#RANDOMIZE">FMCSC_RANDOMIZE</a>). Thus, for such a crosslink to be simulated,
<a href="keywords.html#PDBFILE">(partial) structural input</a> is required. The reason is that the default 
bond lengths and angles do not allow exact closure of this cycle.</li>
<li>Depending on sampling algorithm and choice of methodology (&rarr; <a href="keywords.html#CRLK_MODE">keyword
FMCSC_CRLK_MODE</a>), crosslinks may cause undesirable side effects.</li>
</ol>
<br/>
<b><u>Other (unsupported) residues:</u></b><br/>
<br/>
As was alluded to above, CAMPARI offers the option to infer the topology and further characteristics of residues and
molecules that it has no internal reference representation of.
The ultimate requirement for using this feature is the presence of a sane structural input file from which the 
topology can be inferred (&rarr; <a href="keywords.html#PDBFILE">FMCSC_PDBFILE</a> or <a href="keywords.html#PDB_TEMPLATE">FMCSC_PDB_TEMPLATE</a>).
Then, the residue parsing from said structural input file simply has to be transferred to the sequence file.
If the unknown unit is meant to be the start or end of a polymer or a single-residue small molecule, then the
corresponding flags have to be appended as usual ("_N", "_C", or "_N_C"). Note that the parsing can be changed 
as long as concerted changes are made to both the structural input and sequence files. Note that the terminal
flags are, i) always mandatory, ii) the chiral flag ("_D") is not supported since the complete topology
is inferred from structural input anyway, and iii) terminal residues still have to differ in their base name
if they occur in terminal positions (this is in addition to having to use the "_N" and "_C" indicators).<br/>
<br/><br/>
<u><b>Examples:<br/></b></u><br/>
The sequence input file then simply has to list valid molecules, <i>i.e.</i>,
at least one N- and one C-terminal residue (which can be the same)
eventually with intra-chain residues. Every sequence input has to be
terminated by an END.<br/><br/>
<span class="textborder">Alanine dipeptide:</span><br/><br/>
<big><code>ACE<br/>
ALA<br/>
NME<br/>
END</code></big><br/><br/>
<span class="textborder">An ion pair:</span><br/><br/>
<big><code>NA+<br/>
CL-<br/>
END</code></big><br/><br/>
<span class="textborder">A bad input file (because the alanine-based peptide has no termini):</span><br/><br/>
<big><code>ALA<br/>
ALA<br/>
ALA<br/>
URE<br/>
URE<br/>
URE<br/>
END</code></big><br/><br/>
<span class="textborder">A correct way to create an intermolecular disulfide linkage:</span><br/><br/>
<big><code>
GLY_N<br/>
GLY<br/>
CYS 8<br/>
GLY<br/>
PRO<br/>
GLY<br/>
ALA<br/>
CYS<br/>
NME<br/>
CL-<br/>
END</code></big><br/><br/>
<span class="textborder">A correct way to ask for analysis for a trajectory with the unsupported
single-residue small molecule GMP:</span><br/><br/>
<big><code>
CL-<br/>
CL-<br/>
CL-<br/>
CL-<br/>
NA+<br/>
NA+<br/>
NA+<br/>
NA+<br/>
NA+<br/>
GMP_N_C<br/>
END</code></big><br/><br/>
<span class="textborder">A bad input file (impossible crosslink):</span><br/><br/>
<big><code>
ACE<br/>
CYS 6<br/>
NME<br/>
ACE<br/>
CYS<br/>
NME<br/>
END</code></big><br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_PDBFILE">Structural input file:</a></h4>
Keyword: <b><a href="keywords.html#PDBFILE">FMCSC_PDBFILE</a></b><br/>
<br/>
Starting structures for simulations can only be supplied in pdb format. These are largely standard pdb files.
CAMPARI expects to find a continuous section of entries of type "ATOM", "HETATM", "ANISOU", or "TER", and any other 
records terminate the processing of input for a given snapshot. It is therefore necessary to remove any additional records
embedded within the coordinates section. The use of "MODEL" and "ENDMDL" indicators should be consistent. 
It is generally good practice to remove all superfluous coordinate information from the input file to assist
the software. This is critical when trying to make sure that any missing tails of biopolymers can be added when
<a href="keywords.html#FMCSC_PDB_READMODE">FMCSC_PDB_READMODE</a> is 3. Information is superfluous if it does not
provide coordinates for entries in the <a href="inputfiles.html#FMCSC_SEQFILE">sequence file</a>.<br/>
The parsing of pdb files always requires that atom and residue names be understood by CAMPARI. While the software does a fair amount of translation
(&rarr; <a href="keywords.html#PDB_R_CONV">FMCSC_PDB_R_CONV</a>), manual
adjustment may sometimes be required. However, files in standard pdb convention that are not missing relevant coordinates
should generally be processable "as is." Note that the pdb file is assumed to be in fixed format, <i>i.e.</i>
column shifts are never tolerable. It is, however, possible to redefine the expected fixed input
format using keyword <a href="keywords.html#PDB_INPUTSTRING">FMCSC_PDB_INPUTSTRING</a>, which is
useful for dealing with files generated by CAMPARI itself or custom scripts that are
<a href="keywords.html#PDB_OUTPUTSTRING">not in standard pdb format</a>.<br/>
The target (built-in) names can be constructed 
from the biotype section in the parameter files, or - simpler - by consulting
a CAMPARI-generated pdb file from a dummy run of the system of interest (see for example
<a href="outputfiles.html#basename_START.pdb">here</a>) as long as 
keyword <a href="keywords.html#PDB_W_CONV">FMCSC_PDB_W_CONV</a> is set to 1.
A number of default translations of residue names occur, <i>e.g.</i>, "HSE" to "HIE" or "HOH" to either "SPC"
or "T3P" depending on keyword <a href="keywords.html#PDB_R_CONV">FMCSC_PDB_R_CONV</a>. They can be looked up
in subroutine "pdbcorrect" in source file <a href="../source/readpdb.f90">"readpdb.f90"</a>. The most important
feature here is the "soft" recognition of ambiguous PDB residue names, which occur prominently
for histidine ("HIS") and nucleic acids ("G", <i>etc</i>). CAMPARI requires the
<a href="inputfiles.html#FMCSC_SEQFILE">sequence file</a> to be explicit about variants (<i>e.g.</i>, "DPG" for 
deoxyguanosinephosphate) and will ignore or add atoms from/to the PDB file depending on this.<br/>
If certain atoms in a recognized (built-in) residue are not found (predominantly applies to hydrogen
&rarr; <a href="keywords.html#PDB_HMODE">FMCSC_PDB_HMODE</a>),
the molecular geometry is built according to default assumptions
(<i>e.g.</i>, <a href="references.html#ref17_1">reference</a> for heavy atoms).
Premature termination and/or corrupt structures will occur if important information is missing.
While translation support for the formatting 
of multiple hydrogens on the same site (carbon, nitrogen) exists with regards
to the position of the index label, it is important that such hydrogens be in fact
systematically numbered. A four-character string like ' HB ' for a sidechain hydrogen
atom of an alanine residue would not be read at all, <i>i.e.</i>, the reading utility has no capability processing
identical atom names for the same residue (and will usually exit with an error in such a case).
There is some additional flexibility in interpreting atom names in general as long as they are unique on a per-residue basis. This happens: i)
through simple automatic conversions for typical cases for which two or more standards exist
(<i>e.g.</i>, "OT1" and "OT2" are understood as carboxylate oxygen atoms of a C-terminal polypeptide residue); and ii)
through the keyword <a href="keywords.html#PDB_R_CONV">FMCSC_PDB_R_CONV</a>.<br/>
<br/>
In addition to the use of pdb files to define starting conformations for simulations, 
CAMPARI can also use pdb trajectories in <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a>. If structural input corresponds
to a trajectory file in a <a href="keywords.html#PDB_FORMAT">format</a> other than pdb, a pdb file may still play a role
in serving as a <a href="keywords.html#PDB_TEMPLATE">template for remapping atom names</a>.
These two cases (trajectory analysis and starting a simulation from an input structure) are handled separately below:<br/>
<br/>
<ol>
<li>Analysis runs (see <a href="keywords.html#PDBANALYZE">FMCSC_PDBANALYZE</a>)<br/>
<br/>
To provide maximally accurate analysis, any trajectory is processed
purely by reading in Cartesian coordinates for all atoms. If the trajectory is in
pdb format (using the MODEL/ENDMDL syntax) or requires a <a href="keywords.html#PDB_TEMPLATE">template</a>,
this necessitates that atom names be understood by CAMPARI as outlined above.
For simulation trajectory analyses it is generally recommended to utilize one
of the binary formats (<a href="keywords.html#XTCFILE">FMCSC_XTCFILE</a>, <a href="keywords.html#DCDFILE">FMCSC_DCDFILE</a>,
and <a href="keywords.html#NETCDFFILE">FMCSC_NETCDFFILE</a>) and possibly a suitable pdb-template
(&rarr; <a href="keywords.html#PDB_TEMPLATE">FMCSC_PDB_TEMPLATE</a>) that is processed
in exactly the same way as other pdb input files but serves to provide a map
between CAMPARI intrinsic order of atoms and the aforementioned binary trajectory
file in use. Both binary trajectory file and the template (if required) need to be complete (in this context, the chosen level of representation
may become important &rarr; <a href="keywords.html#UAMODEL">FMCSC_UAMODEL</a>).<br/>
If the analysis run occurs in parallel (see <a href="keywords.html#REMC">elsewhere for details</a>), CAMPARI
expects not a single trajectory, but a set of systematically named trajectories. This applies to all supported 
formats in trajectory analysis mode.<br/><br/>
<u>Portion of an example key-file:</u><br/><br/>
<big><code>
FMCSC_DCDFILE /home/dummy/simulation.dcd<br/>
FMCSC_REPLICAS 4<br/>
FMCSC_PDB_FORMAT 4<br/>
FMCSC_REMC 1<br/>
FMCSC_PDBANALYZE 1<br/>
</code></big><br/>
In this example, the MPI-version of CAMPARI would look for 4 individual trajectory files named "N_000_simulation.dcd",
"N_001_simulation.dcd", "N_002_simulation.dcd", and "N_003_simulation.dcd" all located in the directory "/home/dummy/". 
This means that the specified name does not actually correspond to an existing file. The choice of prefix is consistent
with the output prefix CAMPARI uses for example in replica-exchange simulations. Note that, if a  <a href="keywords.html#PDB_TEMPLATE">template</a>
is required or desired for other reasons, the template file is not treated in the same way, <i>i.e.</i>, all replicas will read
and process the same file in the same location.<br/><br/>
</li>
<li>Runs with a starting structure read from a pdb file<br/>
<br/>
The choice here depends on the sampled degrees of freedom. For runs in Cartesian degrees
of freedom (&rarr; <a href="keywords.html#CARTINT">FMCSC_CARTINT</a>), small distortions are allowed to relax, and it
is recommended to deal with the peculiarities of pdb files in the same way as described above.
This is selected by choosing keyword <a href="keywords.html#PDB_READMODE">FMCSC_PDB_READMODE</a>
to be 2. There is an important caveat, however, which is the <a href="keywords.html#SHAKESET">treatment of constraints</a>.
Keyword <a href="keywords.html#SHAKEFROM">FMCSC_SHAKEFROM</a> is critical in this regard.
Furthermore, subsequent restarts of simulations from pdb files may still create unnecessary noise, which
is avoidable with the help of keywords <a href="keywords.html#PDB_INPUTSTRING">FMCSC_PDB_INPUTSTRING</a> and 
<a href="keywords.html#PDB_OUTPUTSTRING">FMCSC_PDB_OUTPUTSTRING</a>.<br/>
Because of the limited precision of pdb files, CAMPARI also offers an alternative that is designed
for runs in torsional/rigid-body space starting from CAMPARI-generated pdb files but is largely 
obsolete now (and continues to exist primarily for backwards compatibility reasons).
This mode is meant to prevent the distortion of covalent, molecular
geometries of frozen degrees of freedom by limited precision.
By choosing keyword <a href="keywords.html#PDB_READMODE">FMCSC_PDB_READMODE</a>
to be 1, simulation runs can rebuild the molecules using CAMPARI's default geometry.
This means that the Cartesian coordinates of the atoms defining the sampled degrees of
freedom are simply used to determine the value for a given degree of
freedom. These values then seed the molecular building routines. Depending on where the input structure is from
and the <a href="keywords.html#PDB_INPUTSTRING">precision of the pdb input</a>,
this will almost always lead to a different structure than the one found in
the pdb file (with an error increasing along any polymer chains present). 
Thus, this mode is largely obsolete and will produce satisfactory results only when a)
the pdb file was generated by CAMPARI in a run not relying on external input itself
and a <a href="keywords.html#PDB_INPUTSTRING">custom precision</a>
format is used; or b) the pdb file contains only small molecules (3 atoms or less).
Sometimes, it may be possible to relax slightly distorted conformations using the
torsional restraint options (see options 3 and 4 for <a href="inputfiles.html#FMCSC_TORFILE">the corresponding input file</a>).
All comments about naming conventions apply as before. It is important to keep
in mind that switching between the options for <a href="keywords.html#PDB_READMODE">FMCSC_PDB_READMODE</a>
can imply that sampling in <a href="keywords.html#CARTINT">rigid-body / torsional space</a> occurs with a
different, underlying (rigid) geometry in place.</li>
</ol>
In general, the processing of the pdb file will abort as soon as there is a sequence mismatch, <i>i.e.</i>,
the pdb file contains information that cannot be matched with <a href="inputfiles.html#FMCSC_SEQFILE">sequence input</a>.
What happens with any missing parts will depend on the choice for keyword <a href="keywords.html#RANDOMIZE">FMCSC_RANDOMIZE</a>.
If <a href="keywords.html#PDB_READMODE">FMCSC_PDB_READMODE</a> is 3, CAMPARI allows the random construction
of parts of polymers (N- and C-terminal tails), which are often missing in database information (conversely, gaps in the middle
of chains cannot currently be processed automatically). To make sure that this procedure is successful, the pdb
file should fulfill some additional requirements. These and other hints are stated next:<br/>
<ul>
<li>All superfluous coordinate information must be removed for CAMPARI to read "over" a gap. For example,
if the pdb file contains information for the stretch ...CGLLKM and the user is interested in creating a model
with the C-terminal tail ...CGLLKSSSSSHHHHHHH, the information for the methionine residue ("M") must be removed
for CAMPARI to keep reading any subsequent information in the pdb file (for example, if there is another protein
chain).</li>
<li>The matching procedure of sequence stretches between pdb and <a href="inputfiles.html#FMCSC_SEQFILE">sequence input</a>
is greedy (as long as possible) and prefers to map to the first possible residue in the sequence (in case there are ties in length).
This is a potential issue with low complexity sequences or very incomplete structural information (<i>e.g.</i>, bound peptides
in crystal structures).</li>
<li>Many pdb files contain partial structural information for a residue. If there are backbone atoms missing,
it is usually better to remove all information for this residue from the pdb file in case the residue is terminal
or directly adjacent. If the residue is in the middle of a chain that is otherwise meant to be read in, it will most likely
be required to manually rebuild the missing backbone atom(s) or to introduce an artificial chain break.
In general, if any atoms in a residue 
CAMPARI expects to read from pdb are rebuilt (because they are missing or the geometry is off (&rarr;
<a href="keywords.html#PDB_TOLERANCE_A">FMCSC_PDB_TOLERANCE_A</a> and
<a href="keywords.html#PDB_TOLERANCE_B">FMCSC_PDB_TOLERANCE_B</a>), this will most likely corrupt the pdb-provided conformation
of atoms that depend on it (which is why the consequences for backbone atoms are much more dramatic). Because
of CAMPARI's default N-to-C (or 5'-to-3') building direction, this is usually not a concern for C-terminal tails.</li>
<li>As a related point, it is recommended to delete all atoms in the pdb that are branched off (side chain) but for which one or
more of the atoms connecting it to the main chain are missing. For example, if in a polypeptide glutamine residue the
CB atom is missing in the pdb, ATOM records for CG, CD, <i>etc.</i> atoms should also be deleted. CAMPARI has no facility
to infer the position of atoms based on arbitrary reference atoms, and the aforementioned scenario is very likely to lead
to severe violations of covalent geometries at the missing atom.</li>
<li>When adding missing tails to nucleic acids, it is important to keep in mind that CAMPARI internally groups the 3'
oxygen bound to the phosphate to the same residue as the phosphate, which differs from standard convention
(see <a href="keywords.html#PDB_NUCMODE2">FMCSC_PDB_NUCMODE</a> for details). While the read-in step is adjusted
automatically, this does mean that the oxygen atom should be there, either as the only atom in the previous residue
named " O3*" or as the first atom in the first residue to be read with either name "2O3*" or " O3*"
depending on whether in the input pdb file the 3'-oxygen is part of the same residue or not (duplicate atom names are not
allowed and will cause an error). Conversely, if the 
first 5' residue to be read from pdb misses this oxygen atom, or CAMPARI fails to read it, it will be rebuilt, which will
almost certainly destroy the absolute coordinates of the pdb input for this molecule.</li>
<li>It can lead to corrupt geometries if some but not all atoms of a flexible ring are read from pdb input,
which is relevant for example for sugars in polynucleotides and proline or related residues, in particular N-terminal ones
(where the amine hydrogens are usually missing). </li>
</ul>
Sometimes, either <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> or 
actual simulations contain entities (residues, molecules) that are not part of CAMPARI's
built-in database. In such a case, keyword <a href="keywords.html#PDB_READMODE">FMCSC_PDB_READMODE</a>
must necessarily be either 2 or 3 (use Cartesian coordinates), CAMPARI possesses the facility to infer the topology and 
other characteristics of such unsupported residues (for a list of built-in residues,
see <a href="inputfiles.html#FMCSC_SEQFILE">above</a>). The requirements that this pdb file
has to fulfill are as follows:<br/>
<ol type='a'>
<li>It must contain a representation of the unknown part of the system
that is physically feasible and chemically canonical, <i>i.e.</i>, is free of steric
clashes and complies to bond length intervals that are typical for the atoms in their most common
hybridization states. This requires guessing the identity of the chemical element encoded
by a given atom name. For the common elements (C, N, O, H, S, P), it is generally the 14<sup>th</sup> column
in the input file that matters (this is the second column of the four-character atom string in the default
format &rarr; these numbers may change if a <a href="keywords.html#PDB_INPUTSTRING">custom format</a> is used). Recognition 
of rare elements can be triggered by starting the atom name in the 13<sup>th</sup> column, <i>e.g.</i>,
to trigger identification of a copper atom, one should use <tt>"CU  "</tt> for the atom name. Conversely,
the name <tt>" CU "</tt> would be identified as a carbon atom. The correct mass will only be retained
if the <a href="keywords.html#PARAMETERS">parameter file in use</a> offers an appropriate <a href="parameters.html#S2_LJ-types">atom (LJ) type</a>
for this chemical element.
</li>
<li>The unsupported residues must be in a structure that provides, at least, for each one the immediate
sequence context of residues in the same chain. For example, if simulations of a dye-labeled cysteine residue
are desired, and the context of the cysteine residue in the protein sequence of interest is QCA, then the  
pdb input file (whether provided as a <a href="keywords.html#PDB_TEMPLATE">template</a> or not), must contain
this particular stretch in full. This implies that <i>all</i> atoms in the surrounding residues, here glutamine
and alanine, must be present and read correctly. CAMPARI will expect atoms based on its intrinsic representation and
the choice for keyword <a href="keywords.html#UAMODEL">FMCSC_UAMODEL</a>. This completeness requirement is fulfilled trivially
if the surrounding residues are themselves unsupported or if no polymer context exists (<i>i.e.</i>, the 
unsupported residue is a small molecule).</li>
<li>Atoms within the unsupported residue(s) must be ordered such that a meaningful topology
can be built, <i>i.e.</i>, it must not be the case that an atom comes before all other
atoms it is covalently bound to (naturally, with the exception of the very first one). Otherwise,
resultant topologies will be garbled, and garbled topologies can severely limit the access of
the unsupported entities to CAMPARI's sampling engines. In particular, inter-residue linkages should always make
sure that the linkage atom is the very first one in a residue.</li>
<li>All atom names in an unsupported residue should be unique. If there are two residues of exactly the same type
(<i>i.e.</i>, they have the same name, the same number, order and names of atoms, and they occur in
the same molecular context (for a polymer, terminal residues are distinct from non-terminal ones)), then
the two residues will be treated as if of identical type (which simplifies several parameter patches).</li>
<li>Names of chemically equivalent atoms (like the various hydrogen atoms in a methyl group) should
be named such that the last three characters of the corresponding field are identical, and the first
character uses a numbering starting at 1. This helps with consistency in interpretation and can improve
naming in output pdb files.</li>
<li>For unknown residues whose basic architecture resembles a supported polymer type, it is
recommended to follow the atom order and nomenclature used by CAMPARI (or in terms of nomenclature by the pdb file if it
conforms to a <a href="keywords.html#PDB_R_CONV">convention</a> CAMPARI understands). This enables
CAMPARI to infer for example which dihedral angles in an unsupported polypeptide residue such as 
oxidized methionine would correspond to the &phi;/&psi;/&omega;-angles. This is important in ensuring
that the unsupported residue has maximal access to sampling and analysis routines that are specialized 
for this polymer type (<i>e.g.</i>, <a href="keywords.html#DSSPCALC">DSSP analysis</a> for polypeptides, or 
<a href="keywords.html#SUGARFREQ">sugar pucker sampling</a> for polynucleotides).</li>
</ol>
Note that not all of the above guidelines are stringent requirements. For certain types of runs or analyses,
it may suffice to pay less attention to the formatting. 
Further details on how CAMPARI utilizes the information in the input pdb file are provided
<a href="keywords.html#PDB_TEMPLATE">elsewhere</a>. Whatever the application, it is of course required
 to use options 2 or 3 for <a href="keywords.html#PDB_READMODE">FMCSC_PDB_READMODE</a> if unsupported residues
are present.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_RESTART">Restart file:</a></h4>
Keyword: <b><a href="keywords.html#RESTART">FMCSC_RESTART</a></b><br/>
<br/>
Required for: Restarting a previously terminated or aborted simulation &rarr; <a href="keywords.html#RESTART">FMCSC_RESTART</a><br/><br/>
Rather than specifying the starting conformation by a limited-precision input file such as <a href="inputfiles.html#FMCSC_PDBFILE">pdb files explained above</a>, it is more appropriate to extend or restart simulations with the help of
restart files. These are human-readable and can therefore be edited if needed. The name of the <a href="outputfiles.html#%7Bbasename%7D_1.rst">restart file</a> is assumed to be systematically constructed from the <a href="keywords.html#BASENAME">default name of the run</a> and should correspond to {basename}.rst. 
Details on restarting runs are provided <a href="keywords.html#RESTART">elsewhere</a>.<br/>
<!--<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_FYCFILE">Torsional input file:</a></h4>
Keyword: <b><a href="keywords.html#FYCFILE">FMCSC_FYCFILE</a></b><br/>
<br/>
Build a structure containing a single molecule from a set of internal degrees of freedom (and a sequence file of
course). This option is near-deprecated and will exit fatally if multiple molecules are present. The reader
expects a header line first, otherwise the file format is the same as that of output file
<a href="outputfiles.html#FYC.dat">FYC.dat</a>. In general, usage is <b>not</b> recommended at this point in time.<br/>-->
<br/>
<br/>
<br/>
<h3><a class="NoHi" id="Sec2">Section 2: Files altering the system</a></h3>
<hr style="width: 100%; height: 2px;"></hr> (<a href="inputfiles.html#TOP">back to top</a>)<br/>
<h4><a class="NoHi" id="FMCSC_PARTICLEFLUCFILE">Particle number fluctuation file:</a></h4>
Keyword: <b><a href="keywords.html#PARTICLEFLUCFILE">FMCSC_PARTICLEFLUCFILE</a></b><br/>
<br/>
Required for: Use of a simulation ensemble with fluctuating particle numbers, <i>i.e.</i>, cases where <a href="keywords.html#ENSEMBLE">FMCSC_ENSEMBLE</a> is set to 5 or 6 (grand canonical or (pseudo-)semi-grand ensemble).<br/><br/>

First line: One integer, X, that specifies the number of particle types whose numbers can fluctuate<br/>
Each of the next X lines contains three numbers separated by whitespace:
<ul>
<li>An integer specifying the ID of a particle type whose numbers will be allowed to fluctuate.
The type ID starts from 1 and follows the order of the sequence file. See below for an example.</li>
<li>An integer specifying the initial number of particles of this type to place in the system. If the 
<a href="keywords.html#GRANDMODE">implementation mode</a> is chosen as the default, this initial number
is also taken to be the expected (bulk) number of particles of this type.</li>
<li>A real number specifying either the excess (<a href="keywords.html#GRANDMODE">FMCSC_GRANDMODE</a> is 2)
or the absolute chemical potential (<a href="keywords.html#GRANDMODE">FMCSC_GRANDMODE</a> is 1) for this
particle type in kcal/mol.</li>
</ul>

<p><strong>Note on specifying ghost particles in the sequence file:</strong></p>

<p>CAMPARI, like most molecular simulation packages, was initially designed and implemented with the assumption that particle numbers in a simulated system always remain constant.
To enable the concept of fluctuating particle numbers, a requirement for grand canonical simulations, we employ a mechanism of "ghost particles": the lengths of internal arrays that track particles are still constant, but now, each particle has a flag indicating whether it is "present" or "ghosted".
"Ghosted" particles do not interact with each other or with "present" particles, but retain their intra-particle interactions and their interactions with the implicit solvent.
The grand canonical Monte Carlo moves that insert or delete particles from the system are implemented as operations that change the present/ghosted status of a particle.
The trajectories are output in such a way that ghosted particles are translated in the z-direction away from the "real" box allowing both clean visualization and subsequent parsing for external analysis.</p>

<p>The consequence is that the end user must include sufficient ghost particles in the sequence input file.
For every particle type that can fluctuate, the end user must specify enough copies of that particle such that the number present in the system never exceeds the number in the sequence file.
Unfortunately, one does not necessarily know the maximum number that will be present in advance, and including a huge number of ghost particles to be on the safe side may considerably reduce computational efficiency because 1) ghosted particles still require evaluation of purely intramolecular energies, and 2) Monte Carlo moves will be expended on ghosted particles (this is how inserted particles have their degrees of freedom randomized).
Therefore, some trial and error will likely be necessary to determine the appropriate number of ghost particles to specify.</p>

<p>Example: Suppose the sequence input file is</p>

<pre>
ACE
ALA
ALA
ALA
ALA
NME
URE
URE
URE
URE
URE
URE
URE
URE
URE
URE
URE
URE
END</pre>

<p>and the particle number fluctuation file is (with <a href="keywords.html#GRANDMODE">FMCSC_GRANDMODE</a> being 2)</p>

<pre>1<br/>2   5 -0.19</pre>

The particle number fluctuation file specifies that there is 1 (the integer on the first line) type of particle whose numbers shall be allowed to fluctuate.
The type ID is 2, which corresponds to urea (the Ac-(Ala)<sub>4</sub>-Nme polypeptide has type ID = 1).
Initially, five urea molecules will be present in the simulation box, and the remaining seven will be ghosted. This 
sets a bulk concentration that can be computed from the box parameters.
The excess chemical potential of urea is -0.19 kcal/mol. In a binary system such as the one
specified above (urea+peptide), a preliminary simulation of the single-component system (urea) should have
yielded the value provided for the excess chemical potential by showing it to be consistent with the expected bulk concentration. 
Note that if the number of urea molecules present reaches twelve, and another grand canonical insertion move is accepted, the simulation becomes invalid.
Such an event would be reported to log-output, but will not cause the simulation to terminate.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_FRZFILE">Constraint input file:</a></h4>
Keyword: <b><a href="keywords.html#FRZFILE">FMCSC_FRZFILE</a></b><br/>
<br/>
This input file allows selected degrees of freedom to be constrained. Here, constraints can be applied only
to variables that are explicit degrees of freedom of the system. These are then explicitly excluded from 
the lists constituting eligible entities for Monte Caro moves, and are not integrated in <a href="keywords.html#DYNAMICS">gradient-based</a> 
methods in <a href="keywords.html#CARTINT">internal coordinate space</a>. This is notably distinct and conceptually
much simpler than the inclusion of indirect geometric constraints that depend on more than one native degree of freedom
(via Lagrange multipliers). The latter, theoretically more general approach to constraints is currently available only in 
Cartesian dynamics runs (&rarr; <a href="keywords.html#SHAKESET">SHAKESET</a>, <i>e.g.</i>,
also with <a href="inputfiles.html#FMCSC_SHAKEFILE">file-based option</a>).<br/>
For simple, explicit constraints supplied via FMCSC_FRZFILE, the first letter in the file
indicates the mode:<br/>
<br/>
'T' or 't':<br/>
(molecule type-based constraints):<br/>
Generally the input is of the type:<br/>
-----------------<br/>
<span style="font-style: italic;">i&nbsp;&nbsp; XXX<br/>
-----------------<br/>
</span>Here, <i>i</i> denotes the number of the molecule type (as
defined by sequence input), and <i>XXX</i> is one of ALL, RBC, INT, FYO, CHI.<br/>
&nbsp;ALL: Constrain all degrees of freedom for all molecules of that
type<br/>
&nbsp;RBC: Constrain only rigid-body degrees of freedom for all molecules of that type<br/>
&nbsp;INT: Constrain only internal (torsional) degrees of freedom for all molecules of that type<br/>
&nbsp;FYO: Constrain only backbone degrees of freedom for all molecules of that type<br/>
&nbsp;CHI: Constrain only sidechain degrees of freedom for all molecules of that type<br/>
&nbsp;PUC: Constrain pucker state for all relevant residues in all molecules of that type (only relevant in MC)<br/>
&nbsp;OTH: Constrain only dihedral angle degrees of freedom that are not native to CAMPARI, <i>e.g.</i>, those in
<a href="keywords.html#OTHERUNKFREQ">unsupported residues</a><br/>
<br/>
'M' or 'm':<br/>
Identical only that a molecule number is given instead of a molecule
type number.<br/>
<br/>
'R' or 'r':<br/>
Identical only that a residue number is given instead of a molecule
type number. Note that when ALL or RBC is selected in this mode, the
rigid-body degrees of freedom for the whole molecule to which the
selected residue belongs will be constrained.<br/><br/>
Depending on whether the run is a pure Monte Carlo simulation, uses a gradient-based technique, or a hybrid scheme,
the interpretation of these requests may differ slightly. In Monte Carlo, the constraint implies that 
the all relevant degrees of freedom within the specified molecules or residues are removed from all
sampling lists associated with active move types that could potentially perturb the degrees of freedom
in question. This means that, for example, the "FYO" request on a polypeptide residue could remove
the residue from several sampling lists including <a href="keywords.html#OMEGAFREQ">omega moves</a>, 
<a href="keywords.html#PIVOTMODE">standard phi/psi moves</a> and all types of
<a href="keywords.html#CRFREQ">polypeptide concerted rotation moves</a>. For the latter in particular,
this may render many other residues ineligible for these samplers. Conversely, requesting "CHI" for a 
lysine residue while <a href="keywords.html#CHIFREQ">sidechain moves</a> and  <a href="keywords.html#OTHERNATFREQ">OTHER moves on native CAMPARI
degrees of freedom</a> are inactive would be redundant since the chosen move set automatically constrains
these degrees of freedom.<br/>
In internal coordinate space dynamics runs, the interpretation is simpler 
and a constraint request will lead to the selected degrees of freedom no longer being included in
the numerical integration scheme. This simplicity is also the reason why there is an additional mode
exclusively available to this type of runs:<br/>
<br/>
'A or 'a':<br/>
(Z matrix line-based constraints):<br/>
Input is of the type:<br/>
-----------------<br/>
<span style="font-style: italic;">i&nbsp;&nbsp; *<br/>
-----------------<br/></span>
Here, <i>i</i> denotes an atomic index (Z matrix line) that should correspond either to an atom
defining a rotatable dihedral angle or to the first atom in a molecule. For the former,
just the selected degree of freedom is constrained in internal coordinate
space dynamics if "<i>*</i>" is "INT". If the run is a hybrid run also featuring MC segments, the MC moves will <b>not</b> honor
these constraints unless the move set is explicitly designed to do so. The correct atomic
indices can be obtained most easily by turning on the corresponding report: <a href="keywords.html#TMDREPORT">FMCSC_TMDREPORT</a>.
It is theoretically possible to also infer it by consulting the documentation on <a href="inputfiles.html#FMCSC_SEQFILE">sequence input above</a>
and the <a href="outputfiles.html#basename_START.int">reference Z matrix and pdb file(s) written by CAMPARI</a>.<br/>
Conversely, if <i>i</i> is the first atom in a molecule, CAMPARI expects one of 14 alternative strings:
XYZ, XY, XZ, YZ, X, Y, Z, ABC, AB, AC, BC, A, B, or C. The letters X, Y, and Z represent translational degrees 
of freedom of the center of mass of that molecule along the 3 cardinal axes. The letters A, B, and C represent
the rotational degrees of freedom around these same axes. Because strings are checked generously, users
are advised to keep the formatting of the file "clean," which means only using numbers and one of the 15 permitted
strings listed above. Rotational degrees of freedom are only available
for polyatomic molecules of course. These option provide maximally fine-grained control over the rigid-body
degrees of freedom. Note that it is of course rarely useful to eliminate a subset of translational or 
rotational axes. This is mostly for debugging (possible real applications include the emulation of 2D or even 1D systems
similar to the Cartesian case below). If the run is a hybrid run also featuring MC segments, the MC moves <b>cannot</b> 
in general honor these constraints because there is no way of splitting coordinate axes in MC rigid body moves.
<br/>
Finally, for runs in <a href="keywords.html#CARTINT">Cartesian space</a> (dynamics only), there is an
analogous set of constraints available, <i>i.e.</i>, constraints preventing explicitly the integration
of one ore more positional coordinates in one or several atoms. The mode is again atom-based (note that
this mode is mutually exclusive with the Z matrix line-based constraints above):<br/>
<br/>
'A or 'a':<br/>
(explicit Cartesian constraints):<br/>
Input is of the type:<br/>
-----------------<br/>
<span style="font-style: italic;">i&nbsp;&nbsp; *<br/>
-----------------<br/></span>
Here, <i>i</i> denotes an atomic index, and "<i>*</i>" is either XYZ, XY, XZ, YZ, X, Y, or Z. In any
of the options, all listed elements of the position vectors are constrained. As is true in general,
constraints can be cumulative. Full positional constraints of this type can be used to construct
crude boundary conditions for condensed phase simulations. More exotic applications would be
the inclusion of fixed surfaces in the system or carrying out a simulation in 2D.<br/>
It is very important to note that these constraints are mutually incompatible with
<a href="keywords.html#SHAKESET">standard geometric constraints</a>. This means that no atom constrained
via FMCSC_FRZFILE must occur in any holonomic constraint group. Note as well that this option
is restricted to Cartesian dynamics, which implies that hybrid runs in mixed Cartesian/internal coordinate
space are most likely even less of a good idea than in general if this type of constraint is present.<br/>
<br/>
<u>Examples</u>:<br/>
----------<br/>
<big><code>M<br/>
1 CHI<br/>
3 RBC</code></big><br/>
----------<br/>
If the run has an <a href="keywords.html#CARTINT">internal coordinate space component</a>,
this input file would constrain sidechain angles in molecule #1 and the
rigid-body movement of molecule #3. Note that the program will ignore
most meaningless requests but will complain if all degrees of freedom
of a certain type are eliminated through constraints. These move set sanity checks
naturally have no effect in pure dynamics calculations.<br/>
----------<br/>
<big><code>A<br/>
144 XY<br/>
145 XY</code></big><br/>
----------<br/>
If this run has a <a href="keywords.html#CARTINT">Cartesian space dynamics</a> component, this input
file would constrain the x- and y-coordinates of atoms 144 and 145. For this request to be legal,
these two atoms must not participate in any holonomic constraint group (CAMPARI will exit otherwise).
Conversely, if this is an internal coordinate space dynamics runs, CAMPARI would eliminate translations
along x- and y-axes of those two molecules whose first atoms have index 144 and 145, respectively.
If those atoms do not correspond to the first atoms in molecules, these constraint requests would simply be ignored.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_SHAKEFILE">Holonomic constraints input file for Cartesian dynamics:</a></h4>
Keyword: <b><a href="keywords.html#SHAKEFILE">FMCSC_SHAKEFILE</a></b><br/>
<br/>
Required for: Custom constraints on internal distances in Cartesian dynamics &rarr; <a href="keywords.html#SHAKESET">FMCSC_SHAKESET</a><br/><br/>
This is a relatively simple input file used to manually determine
which interatomic distances to constrain during a Cartesian dynamics simulation.
Canonical selections are available via keyword input (&rarr; <a href="keywords.html#SHAKESET">FMCSC_SHAKESET</a>),
but those may sometimes not fit the user's needs. The list available 
through this input file is composed of distances only (no explicit angular constraints are
currently supported via the file-based input method). CAMPARI will parse the list to identify
groups of coupled constraints to be enforced by an appropriate <a href="keywords.html#SHAKEMETHOD">FMCSC_SHAKEMETHOD</a>.
The actual values the distances should be constrained is not provided through this input file
(options are available via keyword <a href="keywords.html#SHAKEFROM">FMCSC_SHAKEFROM</a>).<br/>
There are two modes as follows:<br/>
<br/>
<b>Mode 'A':</b>
&nbsp;
Two numbers per line after the first line ('T'), where the numbers
specify the index of two atoms whose distance is to be constrained. Note that
it is generally permissible for these atoms to be part of different molecules
or to be separated by many bonds within a single molecule. Note that 
user-selected, geometric constraints can become a problem if a)
the coupling between constraints becomes intractable (a general problem also observed for
bond angle constraints &rarr; <a href="keywords.html#SHAKESET">FMCSC_SHAKESET</a> is 4 or 5), or ii)
the simulation is a <a href="keywords.html#DYNAMICS">hybrid run</a> in
<a href="keywords.html#CARTINT">mixed Cartesian/internal coordinate space</a> and the
Monte Carlo move set is able to alter the constrained coordinate. For the latter case,
CAMPARI will print a preemptive warning (note that the vast majority of default constraints selectable via
<a href="keywords.html#SHAKESET">FMCSC_SHAKESET</a> are not accessible to the various MC move types).
It is an important restriction that it should not be possible to constrain the atoms defining
the constraint group more than toward complete internal rigidity. Everything
else would be redundant constraints. These explicitly or implicitly redundant entries are to be avoided
at all cost (they occur by accidentally and explicitly providing the same line twice or by overconstraining
a subset of atoms locally). CAMPARI will not detect those and assume - from an ensemble point of view -
there to be more constraints than actually present potentially leading to temperature
artifacts.<br/>
<br/>
<b>Mode 'T':</b>
&nbsp;
This works analogously to the previous mode only that intramolecular entries
have to be provided for the atom indices for the first molecule of a given type only.
CAMPARI will then replicate such an entry for all molecules of the same type.
Use <a href="keywords.html#SEQREPORT">FMCSC_SEQREPORT</a> to get an overview of the molecule types
in the system. This is a helpful simplification of the purely index-based mode if simulations with custom constraints
are attempted in the presence of a bath of solute molecules which all need to be identically constrained. There
is no support for intermolecular constraints in this mode unless there is exactly one molecule
of each participating type.<br/>
<br/>
Note that entries belonging to ineligible indices will be ignored or may cause a fatal termination of
CAMPARI depending on what exactly the issue is. The correct maintaining of holonomic constraints
can be analyzed by studying  <a href="outputfiles.html#Trajectory_output">trajectory output</a>
with suitable software or - in most cases at least - by consulting
output file <a href="outputfiles.html#INTHISTS_BL.dat">INTHISTS_BL.dat</a>.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_PSWFILE">Preferential sampling weights input file:</a></h4>
Keyword: <b><a href="keywords.html#PSWFILE">FMCSC_PSWFILE</a></b><br/>
<br/>
If the simulation is a Monte Carlo simulation in torsional space (see <a href="keywords.html#DYNAMICS">FMCSC_DYNAMICS</a>
and <a href="keywords.html#CARTINT">FMCSC_CARTINT</a>), this keywords allows the user to alter
the default picking probabilities for entities eligible to a given move type. This function can be overlapping
with the constraint requests described <a href="inputfiles.html#FMCSC_FRZFILE">elsewhere</a>.
The file is a sequence of blocks with each block indicated by a starting letter (separate line):<br/>
<br/>
'T' or 't':<br/>
(molecule type-based adjustments for <a href="keywords.html#PARTICLEFLUCFREQ">particle insertion/deletion moves</a>):<br/>
In the simplest case, the input is of the type:<br/>
-----------------<br/>
<span style="font-style: italic;">i&nbsp;&nbsp; w1<br/>
-----------------<br/>
</span>Here, <i>i</i> denotes the number of the molecule type (as
defined by sequence input), and <i>w1</i> is the target sampling weight (floating point number) for this molecule type in particle
insertion and deletion moves. Weights are set such that any entries not listed explicitly in this file
have a weight of 1.0 (normalization happens after processing all entries). Therefore, it is redundant to choose <i>w1</i>
as unity. A value of zero is possible, but hard to justify in this case given that the molecule types that are allowed
to fluctuate are set via user input as well. Note that the weights for particle permutation moves in the
<a href="keywords.html#ENSEMBLE">semi-grand ensemble</a> can <b>not</b> be altered.<br/>
<br/>
'M' or 'm':<br/>
(molecule-based adjustments for <a href="keywords.html#RIGIDFREQ">rigid-body moves</a>):<br/>
For rigid-body moves it is permissible to control picking probabilities on a per-molecule level. Therefore,
input is of the type:<br/>
-----------------<br/>
<span style="font-style: italic;">i&nbsp;&nbsp; w1&nbsp;w2<br/>
-----------------<br/>
</span>Here, <i>i</i> denotes the number of the molecule (as
defined by sequence input), <i>w1</i> is the target sampling weight for single molecule rigid-body moves,
and <i>w2</i> is the target sampling weight for cluster rigid-body moves. Note that it can indeed be useful
to control these separately, for example when sampling macromolecular assemblies in the presence of small
molecules. In such a case, preferential picking of the macromolecules for cluster moves may prove beneficial.
Some restrictions on cluster rigid-body moves are outlined <a href="keywords.html#CLURBFREQ">elsewhere</a>.<br/>
<br/>
'A' or 'a':<br/>
(degree of freedom-based adjustments for <a href="keywords.html#OTHERFREQ">certain dihedral angle pivot moves</a>):<br/>
Specifically, for the various branches of single dihedral angle pivot moves (
<a href="keywords.html#OTHERFREQ">OTHER moves</a>) that are meant to primarily operate on those dihedral
angles that cannot be sampled by the specialized move sets within CAMPARI
(&rarr; <a href="keywords.html#OTHERUNKFREQ">FMCSC_OTHERUNKFREQ</a> and <a href="keywords.html#OTHERNATFREQ">FMCSC_OTHERNATFREQ</a>),
CAMPARI allows adjustment of the picking frequencies at the level of individual degrees of freedom. Input is as follows:<br/>
-----------------<br/>
<span style="font-style: italic;">i&nbsp;&nbsp; w1&nbsp;<br/>
-----------------<br/>
</span>Here, <i>i</i> denotes the atom number corresponding to the Z-matrix line associated with the 
dihedral angle for which the sampling weight should be changed. Since every dihedral angle can only occur
in at most one of the three subcategories sampled by OTHER moves (<i>viz</i>., degrees of freedom in 
residues that are not natively supported; degrees of freedom in 
native CAMPARI residues that cannot be sampled by any of the other move types;
degrees of freedom that can be sampled by other CAMPARI move types), only a single weight (<i>w1</i>)
is sufficient. Note that this means that the sets of weights supplied in a section of this type
may not all act on the same list of degrees of freedom.<br/>
<br/>
'R' or 'r':<br/>
(residue-based adjustments for all other moves):<br/>
For all other moves, CAMPARI allows adjustment of the picking frequencies at the residue level. Input is as follows:<br/>
-----------------<br/>
<span style="font-style: italic;">i&nbsp;&nbsp; w1&nbsp;w2&nbsp;w3&nbsp;w4&nbsp;w5&nbsp;w6&nbsp;w7&nbsp;w8&nbsp;w9&nbsp;w10&nbsp;w11<br/>
-----------------<br/>
</span>Here, <i>i</i> denotes the number of the residue (as
defined by sequence input). The <i>wi</i> weights refer to
<a href="keywords.html#PIVOTMODE">polypeptide &phi;/&psi;-pivot moves</a> (<i>w1</i>),
<a href="keywords.html#OMEGAFREQ">polypeptide &omega;-pivot moves</a> (<i>w2</i>),
<a href="keywords.html#CHIFREQ">sidechain moves</a> (<i>w3</i>),
<a href="keywords.html#NUCFREQ">polynucleotide (and other polymer) backbone pivot moves</a> (<i>w4</i>),
<a href="keywords.html#PKRFREQ">polypeptide pucker backbone moves</a> (<i>w5</i>),
<a href="keywords.html#SUGARFREQ">sugar pucker backbone moves</a> (<i>w6</i>),
<a href="keywords.html#TORCROFREQ">polypeptide Dinner-Ulmschneider exact concerted rotation moves with &omega;-sampling</a> (<i>w7</i>),
<a href="keywords.html#TORCRFREQ">polypeptide Dinner-Ulmschneider exact concerted rotation moves without &omega;-sampling</a> (<i>w8</i>),
<a href="keywords.html#NUCCRFREQ">polynucleotide Dinner-Ulmschneider exact concerted rotation moves</a> (<i>w9</i>),
<a href="keywords.html#ANGCRFREQ">polypeptide Ulmschneider-Jorgensen exact bond angle-based concerted rotation moves</a> (<i>w10</i>), and
<a href="keywords.html#CRMODE">polypeptide Favrin <i>et al.</i> inexact concerted rotation moves</a> (<i>w11</i>).
This is tricky
for moves sampling stretches of residues, and the complications are discussed <a href="keywords.html#CRFREQ">elsewhere</a>.<br/>
Before presenting an example, a few remarks are in order. First, it is permissible to truncate each line after <i>w1</i>,
if there is no need to alter the picking probabilities for move types controlled by any of the omitted parameters.
It is, however, not possible to alter - for example - the picking weight for a specific residue for sugar pucker moves
without also specifying <i>w1-w5</i>.<br/>
In general, entries that request weight alterations for ineligible entities produce 
a warning. As mentioned above, missing entries are all assumed to be 1.0, which allows the file to remain small in most cases.
Weights must be zero or positive numbers, otherwise the behavior may be undefined. Note that it is <b>not</b> possible
to utilize the molecule type- or molecule-based modes to control picking frequencies for move types listed in the residue-based
section (also holds for any other combination). Furthermore, CAMPARI does not tolerate if sampling weights deplete all entities
for a given move type and will exit with an error in such a case.<br/>
<br/>
<u>Example</u>:<br/>
----------<br/>
<big><code>
R<br/>
12 1.0 1.0 10.0<br/>
<br/>
M<br/>
1 1.0 10.0<br/>
31 0.1<br/>
<br/>
R<br/>
14 1.0 1.0 1.0 5.0 1.0 5.0 1.0 1.0 5.0<br/>
<br/>
A<br/>
71 3.0<br/>
<br/>
</code></big>
----------<br/>
<br/>
This input file would lead to the following changes to the default picking probabilities (in each case, the alteration
is contingent upon the presence of the corresponding move type in the move set and the eligibility/presence of the chosen entity):
<ul>
<li>Residue #12 would be preferentially picked for sidechain moves (factor 10 to all other eligible residues).</li>
<li>Molecule #1 would be preferentially picked for cluster rigid body moves (factor 10 to all other
eligible molecules).</li>
<li>Molecule #31 would be less frequently picked for single molecule rigid body moves (factor 0.1 to
all other eligible molecules).</li>
<li>Residue #14 would be preferentially sampled in all nucleotide-specific moves (pivot, puckering,
and concerted rotation). For pivot and puckering, this is a factor of 5.0 to all other residues. For
concerted rotation, this is a factor of 5.0 for residue #14 being the 3'-terminal residue in a polynucleotide 
concerted rotation stretch compared to all other eligible 3'-terminal residues in a stretch.</li>
<li>The dihedral angle in Z-matrix entry #71 (corresponds with atom number) would be preferentially sampled in 
whatever subcategory of OTHER moves this degree of freedom belongs to. The factor of 3.0 is with respect to all other degrees of
freedom in the same subcategory.</li>
</ul>
Note that multiple blocks of a type are possible. Also note the skipping of unwanted entries on 3 out of 5 relevant lines.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_BIOTYPEPATCHFILE">Biotype patch input file:</a></h4>
Keyword: <b><a href="keywords.html#BIOTYPEPATCHFILE">FMCSC_BIOTYPEPATCHFILE</a></b><br/>
<br/>
The highest level patch that can be applied in CAMPARI is to change the biotypes specific atoms are assigned to.
In the parameter file, <a href="parameters.html#S1_Biotypes">biotypes</a> are the key assignment for every 
possible atom in residues natively supported by CAMPARI, because they indirectly set all other, atom-specific
parameter assignments handled by the parameter file, <i>i.e.</i>, <a href="parameters.html#S2_LJ-types">(LJ) atom types</a>,
<a href="parameters.html#S3_Charge_types">charge types</a>, and <a href="parameters.html#S9_Bonded_types">bonded types</a>.
<a href="parameters.html#S2_LJ-types">Atom types</a> in turn determine other parameters such as mass or valency.
This means that (re)assigning the biotype of a given atom changes all those properties at once.<br/>
There are atom-specific parameters that cannot be changed by a biotype patch, <i>viz.</i>, certain parameters
of the solvation model (see <a href="inputfiles.html#FMCSC_SAVPATCHFILE">below</a>). Similarly, parameters
that are based on groups of atoms are unaffected by biotype patches (see
<a href="parameters.html#S4_Free_energies_of_solvation">free energy of solvation parameters</a> in the parameter file
and corresponding <a href="inputfiles.html#FMCSC_FOSPATCHFILE">patches</a>). Biotype patches are processed
before any other parameter patch is considered. This means that it is possible to override aspects of 
a biotype patch by subsequent application of <a href="inputfiles.html#FMCSC_LJPATCHFILE">atom type patches</a>,
<a href="inputfiles.html#FMCSC_MPATCHFILE">mass patches</a>, <i>etc.</i><br/>
Biotype patches can be applied in two variants; the first is atom-specific, and the second applies patches
consistently across all residues of identical type.<br/> 
<br/>
<u>Atom-specific example:</u><br/>
---------------------------------------<br/>
<big><code>
A<br/>
113&nbsp;&nbsp; 118<br/>
114&nbsp;&nbsp; 119<br/>
115&nbsp;&nbsp; 120</code></big><br/>
---------------------------------------<br/>
<br/>
This input file selects the atom-specific mode ('A'), and would (re)assign biotypes for atoms 113-115 (and only those atoms)
in the system. The chosen biotypes correspond to the N, C, and
CA atoms of tyrosine. Such a patch (incomplete obviously) would be appropriate for example when simulating or analyzing a system
containing phosphorylated tyrosine. Then, it is often convenient to set a number of basic parameters by
taking advantage of the similarity of the two entities and the corresponding overlap in parameter space.
Support for residues not natively supported by CAMPARI is the main use
of biotype patches. Another example is as follows:<br/>
<br/>
<u>Residue-level example:</u><br/>
---------------------------------------<br/>
<big><code>
R<br/>
1&nbsp;&nbsp; 1100<br/>
2&nbsp;&nbsp; 1101<br/>
3&nbsp;&nbsp; 1102<br/>
4&nbsp;&nbsp; 1103<br/>
5&nbsp;&nbsp; 1104<br/>
6&nbsp;&nbsp; 1105</code></big><br/>
---------------------------------------<br/>
<br/>
Suppose the system is a dense phase of identical small molecules with 6 atoms each. Then, this patch could be used
to very efficiently provide complete parameter support for this system. For this, we assume that biotypes 1100-1105 
have been created by the user, and that the relevant information for these biotypes has been provided. By virtue
of selecting option 'R', this patch would be automatically propagated to all residues of identical type and identical
terminus state (the latter is relevant for polymers only, <i>i.e.</i>, the patch would (and most likely could) not be propagated 
from an alanine residue in the middle of a polypeptide to a C- or N-terminal alanine residue in the same or a different
polypeptide). If the residue in question is not natively supported by CAMPARI, only those unsupported residues of identical name
are grouped that have exactly the same number and names of atoms and that are in the same chain position (C-terminal <i>vs</i>.
N-terminal <i>vs</i>. nonterminal).
Two possible applications for a patch like the second example above could be simulations of an unsupported small molecule
liquid or gas or the desire to keep multiple parameterizations for the same small molecule in the same parameter file
and to switch between them via patch (this in general will be less error-prone and easier to maintain than creating
divergent parameter files). When using the 'R' option for a biotype patch, the atom indices in use should correspond
to the first residue of that type (otherwise the patch request is ignored). Keep in mind that atom names
in <a href="outputfiles.html#basename_START.pdb">pdb output files</a> will also be altered by biotype patches.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_MPATCHFILE">Atomic mass patch input file:</a></h4>
Keyword: <b><a href="keywords.html#MPATCHFILE">FMCSC_MPATCHFILE</a></b><br/>
<br/>
Atomic masses are always read and assigned to each atom in the system, and this input file allows changing the
value assumed based on the <a href="parameters.html#S2_LJ-types">assigned atom type</a>.
In general, masses are not explicit components of <a href="keywords.html#DYNAMICS">Monte Carlo sampling</a>, and are not
expected to change thermodynamic properties of the system provided that all other parameters remain fixed.
Instead, masses influence system dynamics. An exception are potentials explicitly computing the mass, an example
being <a href="keywords.html#SC_EMICRO">spatial mass density restraints</a>.<br/>
<br/>
<u>Example:</u><br/>
---------------------------------------<br/>
<big><code>
666&nbsp;&nbsp; 14.<br/>
1444&nbsp;&nbsp;  40.<br/>
2211&nbsp;&nbsp;  2.</code></big><br/>
---------------------------------------<br/>
<br/>
This input file would change the masses for atoms 666, 1444, and 2211 to those (approximately) of nitrogen, argon, and deuterium, respectively.
Chosen values are required to be at least 1.0, since massless sites require constraints for stable integration,
but cannot use standard <a href="keywords.html#SHAKEMETHOD">constraint solvers</a>. The most common 
application for mass patches would be in improving <a href="keywords.html#DYNAMICS">integrator</a> stability by artificially increasing the
mass of those atoms responsible for the fastest degrees of freedom in the system, <i>i.e.</i>, hydrogen.
Another application are otherwise unsupported residues, for which the guessed atom types are inappropriate, or
which contain elements that the parameter file does not even have theoretical support for.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_RPATCHFILE">Atomic radius patch input file:</a></h4>
Keyword: <b><a href="keywords.html#RPATCHFILE">FMCSC_RPATCHFILE</a></b><br/>
<br/>
Similar to <a href="keywords.html#MPATCHFILE">atomic masses</a>, every atom is always assigned an atomic radius, and this input file allows changing the
value assumed derived from parameters, <i>i.e.</i>, either based on the <a href="parameters.html#S2_LJ-types">assigned atom type</a>
or on the <a href="parameters.html#S2_LJ-types">radius override</a> in the parameter file.
Atomic radii are generally supposed to correspond to the exclusion radius of an atom, and are elements of energy terms
such as the <a href="keywords.html#SC_IMPSOLV">ABSINTH implicit solvation model</a> and all 
<a href="keywords.html#SC_EMICRO">spatial density restraints</a>, but also of <a href="keywords.html#SAVCALC">solvation</a>
or <a href="keywords.html#HOLESCALC">void space</a> analysis.<br/>
<br/>
<u>Example:</u><br/>
---------------------------------------<br/>
<big><code>
6&nbsp;&nbsp; 1.<br/>
17&nbsp;&nbsp;  0.1<br/>
22&nbsp;&nbsp;  2.5</code></big><br/>
---------------------------------------<br/>
<br/>
This input file would change the radii for atoms 6, 17, and 22 to 1.0, 0.1, and 2.5 &#8491;, respectively.
Chosen values are required to be at least 0.0, <i>i.e.</i> sizeless atoms are explicitly allowed.
Note that atomic radii and topology in polyatomic molecules have to be matched to one another. This
is because parameters derived from atomic radii, such as <a href="keywords.html#SAVPATCHFILE">maximum SAV fractions</a>
 or <a href="keywords.html#ASRPATCHFILE">volume reduction factors</a>, which are also patchable, neglect
overlap terms corresponding to three or more covalently bound atoms. Also note that there should be no performance
gain associated with reducing the radii of atoms to zero for the <a href="keywords.html#SC_IMPSOLV">ABSINTH implicit solvation model</a>.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_TSTAT_FILE">Group-based thermostat coupling input file:</a></h4>
Keyword: <b><a href="keywords.html#TSTAT_FILE">FMCSC_TSTAT_FILE</a></b><br/>
<br/>
This is a very simple input file allowing the (obsolete) adjustment
of the <a href="references.html#ref12_1">Berendsen weak-coupling</a> and the
<a href="references.html#ref12_3">Bussi et al.</a> velocity rescaling thermostats to couple parts of the system
to independent thermostats, <i>i.e.</i>, to apply velocity adjustments based on
instantaneous temperatures for the subsystem and to instantaneous
velocities for just that subsystem. There is
no physical justification for doing this in general as it effectively
restricts energy transfer between the subsystems and can easily lead to
artifactual results.
It was used predominantly as a trick to circumvent subsystem freezing
which can
sometimes occur if one subsystem has much faster kinetics and different
levels
of integrator noise than others (see <a href="keywords.html#TSTAT">FMCSC_TSTAT</a>). The input file is
very simple. There are two
modes, a molecule-based one (first line 'M') and a molecule type-based
one (first line 'T'):<br/>
<br/>
<b>Mode 'M':</b>
&nbsp;
One number per line after the first line ('M'), where the number
specifies the temperature
coupling group the molecule corresponding to the line is supposed to be
part of. For
a system with 100 molecules, there are 100 lines <i>after</i> the
first one, each one
containing an integer with a coupling group.<br/>
<br/>
<b>Mode 'T':</b>
&nbsp;
One number per line after the first line ('T'), where the number
specifies the temperature
coupling group the molecule type corresponding to the line is supposed
to be part of. As usual,
molecule types are numbered intrinsically by occurrence in the sequence
input file. For
a system with two molecules types (say a protein in water), there would
only be two lines <i>after</i> the first one, both containing an
integer with a coupling group.<br/>
<br/>
Note that coupling groups have to be positive integers, and that
non-consecutive integers are re-numbered
accordingly to provide a consecutive set.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_WL_GINITFILE">Wang-Landau sampling initial guess input file:</a></h4>
Keyword: <b><a href="keywords.html#WL_GINITFILE">FMCSC_WL_GINITFILE</a></b><br/>
<br/>
Required for: Overriding of the flat initial guess for the (logarithmic) target distribution in a <a href="keywords.html#MC_ACCEPT">Wang-Landau run</a>. Naturally, this is is irrelevant for any simulation not (at least partially) using the WL method.<br/>
<br/>
The (logarithmic) <a href="keywords.html#WL_MODE">target distribution</a> of a Wang-Landau run is not generally 
expected to be flat, neither globally nor locally. This means that a flat initial guess (the default) may 
slow down the construction of the distribution's coarse features (especially at boundaries). In these cases,
it may be helpful to supply a nonflat initial guess.<br/>
The format of this input for one-dimensional cases is simply two columns, the first giving the center of the bin in question,
and the second giving the initial guess for the target distribution. Because all distributions are handled in logarithmic 
space, the only criterion of relevance are differences between bins and the relation of these differences to 
the <a href="keywords.html#WL_F0">starting value for the f parameter</a>. The bin spacing has to be constant. Range
settings provided via keywords <a href="keywords.html#WL_MAX">FMCSC_WL_MAX</a> and <a href="keywords.html#WL_BINSZ">FMCSC_WL_BINSZ</a>
are overwritten. Note that the assessment of convergence (f parameter reductions) does not rely on the estimate of the target 
distribution (<i>e.g.</i>, ln g(E)), but rather on the simple visitation histogram. This means that it may be necessary
to choose a relatively  <a href="keywords.html#WL_FREEZE">large number of buffer steps</a> with a nonflat initial guess
to ensure that all relevant bins are populated before the actual iteration begins.<br/>
<br/>
<u>Example:</u><br/>
---------------------------------------<br/>
<big><code>
0.1 10.2<br/>
0.3 4.1<br/>
0.5 3.4<br/>
0.7 3.2<br/>
0.9 6.8</code></big><br/>
---------------------------------------<br/>
<br/>
This would initialize the histogram to 5 bins with a spacing of 0.2 and a maximum value of 1.0. This example could correspond
to a normalized reaction coordinate (such as <a href="keywords.html#WL_RC">Z<sub>&alpha;</sub></a>) and would give an extremely
coarse result (the bin spacing will never vary throughout a WL run).<br/>
For the two-dimensional case, CAMPARI expects N<sub>d2</sub>+1 columns per row and an additional header line. N<sub>d2</sub> is the
number of bins in the initial guess for the second dimension (always a <a href="keywords.html#WL_RC">geometric reaction coordinate</a>).
In each non-header row, the values listed should be the bin center coordinate in the first dimension followed by the N<sub>d2</sub>
initial estimates for the target distribution. Note that all N<sub>d2</sub> entries must be listed on every data row. The header line
itself must contain an integer identical to N<sub>d2</sub> followed by the N<sub>d2</sub> increasing bin center coordinates in the second
dimension.<br/>
<br/>
<u>Example:</u><br/>
-----------------------------------------------------<br/>
<big><code>
6    3.4 3.8 4.2 4.6 5.0 5.4<br/>
0.1  1.2 1.7 2.5 2.8 2.1 1.2<br/>
0.3  0.1 0.2 0.4 0.6 0.4 0.1<br/>
0.5  0.1 0.1 0.4 0.5 0.3 0.1<br/>
0.7  0.2 0.7 1.0 0.8 0.4 0.0<br/>
0.9  0.5 1.8 2.1 1.2 0.2 0.0</code></big><br/>
-----------------------------------------------------<br/>
<br/>
This would initialize the histogram to 5x6 bins with spacings of 0.2 and 0.4 maximum values of 1.0 and 5.6.
This example could correspond to a two geometric reaction coordinates, <i>e.g.</i>, Z<sub>&alpha;</sub> and the radius
of gyration for a very short peptide.<br/> 
<br/>
<br/>
<br/>
<h3><a class="NoHi" id="Sec3">Section 3: Files redefining standard terms of the energy function</a></h3>
<hr style="width: 100%; height: 2px;"></hr> (<a href="inputfiles.html#TOP">back to top</a>)<br/>
<h4><a class="NoHi" id="FMCSC_REFILE">MPI replica exchange input file:</a></h4>
Keyword: <b><a href="keywords.html#REFILE">FMCSC_REFILE</a></b><br/>
<br/>
Required for: Use of Hamiltonian (or temperature) replica-exchange methodology &rarr; <a href="keywords.html#REMC">FMCSC_REMC</a><br/><br/>
Note that the code needs to be compiled with MPI support (see INSTALL)
for this option be available. On the first line a set of integers indicates
different(!) "dimensions" of the requested set of replicas. Such dimensions are environmental
terms and/or parameters of the Hamiltonian which are supposed to differ between the individual replicas.
The by far most common application is to specify different temperatures. However, in theory many more exchange parameters
are possible and may be useful for certain applications (for examples <a href="keywords.html#FEG_IPP">FMCSC_FEG_IPP</a> and similar
parameters in the case of free energy calculations). In CAMPARI, the following are available in REMC/D runs:<br/>
<ol>
<li>Temperature</li>
<li>Scaling factor for repulsive inverse power potential (see <a href="keywords.html#SC_IPP">FMCSC_SC_IPP</a>)</li>
<li>Scaling factor for dispersive 6<sup>th</sup> power potential (see <a href="keywords.html#SC_ATTLJ">FMCSC_SC_ATTLJ</a>)</li>
<li>Scaling factor for the Weeks-Chandler-Andersen (WCA) potential (see <a href="keywords.html#SC_WCA">FMCSC_SC_WCA</a>) </li>
<li>Scaling factor for polar (Coulombic) potential (see <a href="keywords.html#SC_POLAR">FMCSC_SC_POLAR</a>)</li>
<li>Scaling factor for free energy of solvation term for ABSINTH model (see <a href="keywords.html#SC_IMPSOLV">FMCSC_SC_IMPSOLV</a>)</li>
<li>Implicit solvent dielectric constant (see <a href="keywords.html#IMPDIEL">FMCSC_IMPDIEL</a>, note that this is currently only supported as the implicit
dielectric for the <a href="keywords.html#SC_IMPSOLV">ABSINTH model</a>, and does <b>not</b> work when interpreted as the (generalized) reaction-field dielectric)</li>
<li>Scaling factor for torsional bias potential (see <a href="keywords.html#SC_TOR">FMCSC_SC_TOR</a>)</li>
<li>Scaling factor for secondary structure bias potential (see <a href="keywords.html#SC_ZSEC">FMCSC_SC_ZSEC</a>)</li>
<li>Global secondary structure bias potential: target &#945;-content (see <a href="keywords.html#ZS_FR_A">FMCSC_ZS_FR_A</a>)</li>
<li>Global secondary structure bias potential: target &#946;-content (see <a href="keywords.html#ZS_FR_B">FMCSC_ZS_FR_B</a>)</li>
<li>Screening model for implicit solvent (see <a href="keywords.html#SCRMODEL">FMCSC_SCRMODEL</a>)</li>
<li>Sigmoidal steepness for free energy of solvation term (&#964;<sub>F</sub>) in ABSINTH model (see <a href="keywords.html#FOSTAU">FMCSC_FOSTAU</a>)</li>
<li>Sigmoidal steepness for dielectric screening (&#964;<sub>S</sub>) in ABSINTH model (see <a href="keywords.html#SCRTAU">FMCSC_SCRTAU</a>)</li>
<li>Sigmoidal midpoint for free energy of solvation term (&#967;<sub>F</sub>) in ABSINTH model (see <a href="keywords.html#FOSMID">FMCSC_FOSMID</a>)</li>
<li>Sigmoidal midpoint for dielectric screening (&#967;<sub>S</sub>) in ABSINTH model (see <a href="keywords.html#SCRMID">FMCSC_SCRMID</a>)</li>
<li>Contact dielectric for screening models 3, 4, or 7-9 (see <a href="keywords.html#CONTACTDIEL">FMCSC_CONTACTDIEL</a> and <a href="keywords.html#SCRMODEL">FMCSC_SCRMODEL</a>)</li>
<li>Generalized mean for screening models 5-8 (see <a href="keywords.html#ISQM">FMCSC_ISQM</a> and <a href="keywords.html#SCRMODEL">FMCSC_SCRMODEL</a>)</li>
<li>Mixing contribution from distance dependence for screening models 3, and 7-9 (see <a href="keywords.html#SCRMIX">FMCSC_SCRMIX</a> and <a href="keywords.html#SCRMODEL">FMCSC_SCRMODEL</a>)</li>
<li>Scaling factor for ghosted repulsive inverse power potential (see <a href="keywords.html#FEG_IPP">FMCSC_FEG_IPP</a>)</li>
<li>Scaling factor for ghosted dispersive 6<sup>th</sup> power potential (see <a href="keywords.html#FEG_ATTLJ">FMCSC_FEG_ATTLJ</a>)</li>
<li>Scaling factor for ghosted Coulombic interactions (see <a href="keywords.html#FEG_POLAR">FMCSC_FEG_POLAR</a>)</li>
<li>Scaling factor for tabulated potentials (see <a href="keywords.html#SC_TABUL">FMCSC_SC_TABUL</a>)</li>
<li>Scaling factor for polymeric biasing potential (see <a href="keywords.html#SC_POLY">FMCSC_SC_POLY</a>)</li>
<li>Scaling factor for distance restraint potential (see <a href="keywords.html#SC_DREST">FMCSC_SC_DREST</a>)</li>
<li>Scaling factor for ghosted bond length potentials (see <a href="keywords.html#FEG_BONDED_B">FMCSC_FEG_BONDED_B</a>)</li>
<li>Scaling factor for ghosted bond angle potentials (see <a href="keywords.html#FEG_BONDED_A">FMCSC_FEG_BONDED_A</a>)</li>
<li>Scaling factor for ghosted improper dihedral potentials (see <a href="keywords.html#FEG_BONDED_I">FMCSC_FEG_BONDED_I</a>)</li>
<li>Scaling factor for ghosted torsional potentials (see <a href="keywords.html#FEG_BONDED_T">FMCSC_FEG_BONDED_T</a>)</li>
<li>Global E-score for DSSP biasing potential (see <a href="keywords.html#DSSP_ESC">FMCSC_DSSP_ESC</a>)</li>
<li>Global H-score for DSSP biasing potential (see <a href="keywords.html#DSSP_HSC">FMCSC_DSSP_HSC</a>)</li>
<li>Time step for dynamics calculation &rarr; this acts as a dummy in pure Monte Carlo runs (see <a href="keywords.html#TIMESTEP">FMCSC_TIMESTEP</a>)</li>
<li>Scaling factor for density restraint potential (see <a href="keywords.html#SC_EMICRO">FMCSC_SC_EMICRO</a>)</li>
<li>Threshold setting for density restraint potential (see <a href="keywords.html#EMTHRESHOLD">FMCSC_EMTHRESHOLD</a>)</li>
</ol>
This header line is followed by vectors of individual conditions (one condition per
line) in the n-dimensional space defined before.<br/>
<br/>
<u>Example:</u><br/>
-------------------<br/>
<big><code>1&nbsp;&nbsp; 4<br/>
298.0&nbsp; 0.9<br/>
350.0&nbsp; 0.8</code></big><br/>
-------------------<br/>
Such an input file would set up two replicas; the first one at a temperature (code #1) of 298K and with a global scaling factor for the 
WCA potential of  0.9 (code #4) and the second one at a temperature of 350K with a WCA scaling factor of 0.8.<br/>
<br/>
Note that the normal requests in the key-file for any dimension which is
part of the replica space are going to be ignored (in the above
example, putting a value for FMCSC_<a href="keywords.html#SC_WCA">SC_WCA</a>
would have no effect).<br/>
<br/>
Also note that setting scaling factors for energy terms to zero in this
input file will be slightly different than when this is done in any
other calculation. As a result, there might be no speed-up for
individual replicas (this is usually not desired anyway). Most other changes will be hidden from the user and relate to
the necessity to be able to calculate cross-energies for which such
scaling factors might not be zero. Lastly, performance may be degraded for certain choices requiring extensive recalculation
of expensive terms, <i>e.g.</i>, the choice of screening model for the ABSINTH model or the threshold setting for the density
restraint potential. In such cases, very small values for <a href="keywords.html#REFREQ">FMCSC_REFREQ</a> may be 
counterproductive.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_FEGFILE">Free energy growth (ghosting) input file:</a></h4>
Keyword: <b><a href="keywords.html#FEGFILE">FMCSC_FEGFILE</a></b><br/>
<br/>
Required for: Use of specific, energetic decoupling of parts of the system from the rest via ghosting of interactions &rarr; <a href="keywords.html#GHOST">FMCSC_GHOST</a><br/><br/>
The input file for specifically scaling interactions of certain
molecules or residues with the rest of the system. This is only
supported for a limited range of Hamiltonians at the moment (see <a
 href="keywords.html#GHOST">FMCSC_GHOST</a> and related keywords in <a
 href="keywords.html#KEYWORDS">KEYWORDS</a>):<br/>
(the first line gives the mode)<br/>
<br/>
'T' or 't':<br/>
The input is then simply:<br/>
----------<br/>
<span style="font-style: italic;">i&nbsp;&nbsp;0</span><br/>
----------<br/>
The integer <i>i</i> denotes the molecule type, for which the interactions with the rest
of the system are to be ghosted. The <b>optional</b> <i>0</i> can be used to override
the scaling settings in order to decouple an individual group from the system entirely.
This is tailored to rather specific applications, for instance the stepwise calculation
of free energies of solvation on a residue-by-residue basis. Note that full de-coupling
requires all the bonded FEG-parameters to be set to unity (see <a href="keywords.html#FEG_BONDED_B">FMCSC_FEG_BONDED_B</a>, <i>etc</i>). This has primarily technical reasons.<br/>
<br/>
'M' or 'm':<br/>
Analogous but with molecule number instead of molecule type number<br/>
<br/>
'R' or 'r':<br/>
Analogous but with residue number instead of molecule type number<br/>
<br/>
<u>Example:</u><br/>
----------<br/>
<big><code>r<br/>
1&nbsp;&nbsp;0<br/>
2</code></big><br/>
----------<br/>
This will ghost the first residue in the sequence entirely (<i>i.e.</i>, it contributes nothing to the
simulation). The second residue in the sequence will have scaled interactions according to the
setting of the FEG-scaling parameters (see <a
 href="keywords.html#KEYWORDS">KEYWORDS</a>, <i>e.g.</i>  <a href="keywords.html#FEG_IPP">FMCSC_FEG_IPP</a>). Always keep in mind that
interactions between ghosted entities as well as within ghosted
entities are computed according to the selection in <a
 href="keywords.html#FEG_MODE">FMCSC_FEG_MODE</a> and that this may sometimes
lead to unexpected or inconsistent behavior. For example, a water box full of ghosted
water molecules will behave like any other water box (with absolutely no ghosting) if 
<a href="keywords.html#FEG_MODE">FMCSC_FEG_MODE</a> is set to 1.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_BPATCHFILE">Bonded potential terms patch input file:</a></h4>
Keyword: <b><a href="keywords.html#BPATCHFILE">FMCSC_BPATCHFILE</a></b><br/>
<br/>
Prerequisite: This file is only relevant if at least one of the bonded potential terms
(see <a href="keywords.html#SC_BONDED_B">bond length potentials</a>, <a href="keywords.html#SC_BONDED_A">bond angle potentials</a>,
<a href="keywords.html#SC_BONDED_I">improper dihedral angle potentials</a>, <a href="keywords.html#SC_BONDED_T">torsional potentials</a>, and <a href="keywords.html#SC_BONDED_M">CMAP potentials</a>) is in use.<br/><br/>
This file can be employed to override default assignments for bonded interactions or to 
add bonded interactions on internal coordinates that were previously not assigned a bonded interaction.
A fundamental limitation is that such new bonded interactions can only be applied to internal coordinates that could
have been assigned a term by default, <i>e.g.</i>, it is <b>not</b>
possible to apply a bond length potential to atoms a-b if  a is not in fact covalently bound to b.<br/>
Default assignments are made based on the parameter 
file (see <a href="keywords.html#PARAMETERS">PARAMETERS</a> and <a href="parameters.html">dedicated
documentation</a>), and the user can force CAMPARI to print those to log-output via
<a href="keywords.html#BONDREPORT">FMCSC_BONDREPORT</a>. In the patch file, each line has to contain as its first entry
a keyword that is either "PATCH_BOND", "PATCH_ANGLE", "PATCH_IMPROPER", "PATCH_TORSION", or "PATCH_CMAP".
Take glycine dipeptide (<a href="inputfiles.html#FMCSC_SEQFILE">acetyl and N-methylamide caps</a>) in 
<a href="keywords.html#UAMODEL">united-atom representation</a> as an example. This molecule has the CH<sub>3</sub> and C atoms
of the acetyl group, N, C<sub>&alpha;</sub>, and C of glycine, and the N and CH<sub>3</sub> atoms
of the C-terminal capping group as indices 1, 2, 4, 5, 6, 9, and 10, respectively.<br/>
<br/>
<u>Example:</u><br/>
---------------------------------------<br/>
<big><code>
PATCH_BOND 1 2 23<br/>
PATCH_ANGLE 4 5 6 11<br/>
PATCH_IMPROPER 4 2 5 8 44<br/>
PATCH_TORSION 5 6 9 10 112<br/>
PATCH_CMAP 9 6 5 4 2 3
</code></big><br/>
---------------------------------------<br/>
<br/>
The first line of this input file would replace the existing bond length potential between the CH<sub>3</sub> and C atoms 
of the acetyl group with the one in the <a href="keywords.html#PARAMETERS">parameter file</a> in use that has index 23.
If no potential was previously assigned (fatal in <a href="keywords.html#CARTINT">Cartesian dynamics</a>),
it would be added by this patch. If no potential with index 23 is found in the parameter file, this line of the 
patch file is ignored (and a warning is produced).
It is important to note that the patch file exclusively handles the assignment, but not the parameters
themselves. The reason is that appending the list of bonded potentials in the parameter
file allows reusing an entry multiple times. If the parameters themselves were also part of the patch file,
they would have to be repeated over and over again in such a case, which would increase the amount of work
and the likelihood of errors creeping in.<br/>
Similarly, the second line would change (or add) the bond angle potential acting across the N&rarr;C<sub>&alpha;</sub>&rarr;C
angle in glycine to the one with index 11 in the parameter file in use. The same notes and caveats as before apply.
The third line would alter the assignment for the improper dihedral angle formed by the (central) N of glycine
along with the C<sub>&alpha;</sub> and polar hydrogen (index 8) atoms of glycine, and the carbonyl carbon of the acetyl group.
The potential type would be the torsional potential with index 44. The fourth line would assign the 112<sup>th</sup> 
torsional potential in the parameter file to the &omega;-torsion of the peptide bond connecting glycine and the
C-terminal N-methylamide. Lastly, the fifth line would change (or add) a CMAP potential acting on the consecutive 
&psi;- and &phi;-angles of glycine. Note that the atom order is reversed compared to a standard CMAP assignment
as found in the CHARMM force field.<br/>
A few more comments are in order. First, redundant permutations in the lists of atoms are understood by
CAMPARI (see <a href="parameters.html#S9_Bonded_types">here</a> for definitions of what redundant permutations
are for the individual cases). Second, CAMPARI will only process patches that relate to an energy term
that is actually turned on. For instance, if in the above example <a href="keywords.html#SC_BONDED_M">FMCSC_SC_BONDED_M</a>
were zero, the fifth line would not be processed at all. Third, for processed patches that fail due to the atoms
not forming an internal coordinate of the specified type and/or due to the potential choice being out of range, 
a warning is produced. Similarly, every successful patch will be summarized to log-output as well. Fourth, patch keywords
that CAMPARI fails to understand will cause a fatal exit of the program. Fifth, multiple redefinitions of the 
same internal coordinate are considered according to input order, <i>i.e.</i>, only the last definition will end
up being relevant. This implies that is not possible to "stack" multiple potentials on an individual internal
coordinate. Sixth and last,
atom numbering will always correspond to CAMPARI's internal order, which can usually be extracted
from <a href="outputfiles.html#basename_START.pdb">pdb files</a> produced at the beginning or end of a run.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_LJPATCHFILE">Size exclusion and dispersion parameters patch input file:</a></h4>
Keyword: <b><a href="keywords.html#LJPATCHFILE">FMCSC_LJPATCHFILE</a></b><br/>
<br/>
Prerequisite: This file is always read, but the assignments are only relevant if one of the short-range
interaction potentials is active (<a href="keywords.html#SC_IPP">IPP</a>, <a href="keywords.html#SC_ATTLJ">ATTLJ</a>, 
or <a href="keywords.html#SC_WCA">WCA</a>) or any functionality or potential depending on derived
parameters (such as atomic radii or volumes used, for example in the <a href="keywords.html#SC_IMPSOLV">ABSINTH implicit
solvation model</a>) is in use.<br/><br/>
This file can be employed to override the Lennard-Jones &sigma; and &epsilon; parameters by changing
the default <a href="parameters.html#S2_LJ-types">atom type</a> assignment for a specific instance of a
<a href="parameters.html#S1_Biotypes">biotype</a>.
Each line has to contain two integers, the first giving the atom number (CAMPARI-internal numbering), and 
the second the reassigned atom type number in reference to the <a href="parameters.html">parameter file</a> in use.<br/>
<br/>
<u>Example:</u><br/>
---------------------------------------<br/>
<big><code>
15&nbsp;&nbsp; 11<br/>
1443&nbsp;&nbsp;  1</code></big><br/>
---------------------------------------<br/>
<br/>
This input file would attempt to reassign atom 15 to atom type 11 and atom 1443 to atom type 1.
Note that the success depends on the existence of the appropriate number of <a href="parameters.html#S2_LJ-types">atom types</a>
in the parameter file. As long as they are continuously numbered and all parameters are provided, it is feasible to
create arbitrary amounts of atom types in the parameter file with the limiting case that - theoretically -
every actual atom in the system could be assigned unique parameters with unique pairwise and 14-exceptions through
the use of this patch facility.<br/>
An important caveat lies in the fact that masses, atomic numbers, valencies, and other quantities specified
in the <i>atom</i> lines of the parameter file are not altered. This is particularly relevant when dealing
with unsupported residues (see <a href="inputfiles.html#FMCSC_SEQFILE">sequence input</a> and
<a href="keywords.html#PDBANALYZE">trajectory analysis</a>), since CAMPARI will 
make an automatic guess for a biotype assignment, from which the default assignment is gleaned. If this yields
unsatisfactory results regarding mass or valency, this patch file is not helpful. Instead, the input pdb should
be altered to aid CAMPARI in guessing an appropriate type.
As usual, the index in the first column corresponds strictly to the internal numbering used by CAMPARI, which
can, for instance, be read out from the pdb file produced at the beginning of the simulation
(&rarr; <a href="outputfiles.html#basename_START.pdb">here</a>) <b>
unless</b> nucleotides are part of the system and are requested to conform to pdb convention (&rarr;
<a href="keywords.html#PDB_NUCMODE">FMCSC_PDB_NUCMODE</a>).<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_CPATCHFILE">Charge patch input file:</a></h4>
Keyword: <b><a href="keywords.html#CPATCHFILE">FMCSC_CPATCHFILE</a></b><br/>
<br/>
Prerequisite: This file is only relevant if the <a href="keywords.html#SC_POLAR">polar potential</a> is used.<br/><br/>
This file can be employed to override partial charge assignment extracted from the parameter 
file (see <a href="keywords.html#PARAMETERS">PARAMETERS</a> and <a href="parameters.html">dedicated
documentation</a>). Each line has to contain two entries, an integer giving the atom number
and a floating point number giving the new partial charge.<br/>
<br/>
<u>Example:</u><br/>
---------------------------------------<br/>
<big><code>
33&nbsp;&nbsp; -1.76<br/>
34&nbsp;&nbsp;  0.88<br/>
35&nbsp;&nbsp;  0.88</code></big><br/>
---------------------------------------<br/>
<br/>
This input file would change the atomic partial charges on atoms 34 and 35 to 0.88e each, and that on atom 33 
to -1.76e. Note that it is up to the user to ensure that the resultant charges are sane and self-consistent,
although CAMPARI will perform its usual tests and provide warnings/exits in case they are not (keywords
<a href="keywords.html#ELECREPORT">FMCSC_ELECREPORT</a> and <a href="keywords.html#UNSAFE">FMCSC_UNSAFE</a>
may become important or useful here). Also note that the numbering, which the index in the first column refers
to, is strictly the internal numbering used by CAMPARI. This can generally be read out from the pdb file
produced at the beginning of the simulation (&rarr; <a href="outputfiles.html#basename_START.pdb">here</a>) <b>
unless</b> nucleotides are part of the system and are requested to conform to pdb convention (&rarr;
<a href="keywords.html#PDB_NUCMODE">FMCSC_PDB_NUCMODE</a>).<br/>
This functionality is merely thought to provide any corrections to partial charges which would break the identity
of biotypes set up in the parameter files and maintained within CAMPARI itself. The most common example of this
would be partial charges for terminal polypeptide residues in the AMBER-class of force fields.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_NCPATCHFILE">Residue-level net charge flag patch input file:</a></h4>
Keyword: <b><a href="keywords.html#NCPATCHFILE">FMCSC_NCPATCHFILE</a></b><br/>
<br/>
Prerequisite: This file is only relevant if the <a href="keywords.html#SC_POLAR">polar potential</a> is used. Additional 
requirements are that the <a href="keywords.html#ELECMODEL">electrostatic interaction model</a> is group-based 
and/or that the cutoff treatment for long-range Coulombic interactions is such that <a href="keywords.html#LREL_MC">LREL_MC</a>
is 1, 2, or 3 and/or that <a href="keywords.html#LREL_MD">LREL_MD</a> is either 4 or 5.<br/><br/>
This file can be employed for two purposes. First, it can disable the flag CAMPARI assigns to a residue that has at least one charge group
in it with a total charge exceeding a <a href="keywords.html#POLTOL">tolerance setting</a> in absolute magnitude.
It is currently impossible to turn the residue-level flag on (since it would not in general be clear
which charge group to do this for and why).
Second, it can aid the automatic determination of charge groups by defining a series of target values
(presumably different from 0.0). Charge groups are generally determined by an algorithm that finds
sets of atoms of a desired (integer or zero) net charge, which are also topologically close, do 
not isolate other polar atoms (topologically speaking), and do not introduce distinctions between atoms of identical biotype.
For built-in residues, CAMPARI has a specific understanding of what to expect (<i>e.g.</i>, it would look first for
two charge groups with net charge of 1.0 in an N-terminal lysine residue). However, this understanding may not
always be appropriate, and no such understanding is available for unsupported entities. This is the 
primary purpose of patching the charge group targets.<br/>
Each line has to contain two or more entries. The first is an integer giving the residue number, the second
is either 0 or 1 (with only a value of 0 specified for a residue that contains charge groups matching the criteria
outlines above being relevant), and a number of floating point values.<br/>
<br/>
<u>Example:</u><br/>
---------------------------------------<br/>
<big><code>
3&nbsp;&nbsp;  1 -1.0 -1.0 -1.0 0.0<br/>
7&nbsp;&nbsp;  0</code></big><br/>
---------------------------------------<br/>
<br/>
This input file would flag the 7th residue in the system as no longer carrying a net charge irrespective of  
the partial charges on its atoms and irrespective of how they were grouped into charge groups.
For the 3rd residue, it would instruct CAMPARI to look for 3 separate charge groups with net charges
of 1.0 each, <i>e.g.</i>, residue 3 could be an unsupported small molecule such as fully deprotonated citric acid (3-).
The 0.0 specified at the end is optional, since the program will always look for as many net neutral groups as possible
after processing of special groups. Note that each entry for a charge group target will be processed in exactly the order
provided. Once a suitable group is found, the code proceeds to the next target. Also note that the sum of target values
must match exactly the total charge of the residue in question. If any given target cannot be reached by any acceptable
grouping (there are restrictions as mentioned above in addition to limitations of the search size in terms of numbers of bonds),
the entry in question will simply be skipped. It is therefore possible to exercise a reasonable amount of control
over the charge group parsing, but it is likely that some trial-and-error is involved.<br/>
As alluded to before, the main application domain for changing the charge flag via this patch (residue 7 in the example above) would be the 
suppression of the computation of long-range interactions between residues flagged as charged simply because
their partial charges do not group cleanly into neutral groups. Specifically, polarization across neighboring
residues in a polymer without any formal charges would generally flag both residues as charged. This is because
in such a case a net neutral group is split arbitrarily because of the requirement for charge groups to be confined to residues
(common with charges fit from QM data without constraints). The main application for the second functionality (residue 3) is
to deal with unsupported residues, correctly reflect manual <a href="inputifles.html#FMCSC_CPATCHFILE">charge patches</a> or
altered parameters, to deal with unusual species such as zwitterions, or to achieve an approximate grouping for 
charges that do not group exactly (which can be highly relevant for certain
<a href="keywords.html#SCRMODEL">screening models within the ABSINTH framework</a>).<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_SAVPATCHFILE">Atomic solvation parameters patch input file:</a></h4>
Keyword: <b><a href="keywords.html#SAVPATCHFILE">FMCSC_SAVPATCHFILE</a> and <a href="keywords.html#ASRPATCHFILE">FMCSC_ASRPATCHFILE</a></b><br/>
<br/>
Prerequisite: There are two atomic parameters that can be patched from their default values, which are primarily relevant
in calculations using the <a href="keywords.html#SC_IMPSOLV">ABSINTH implicit solvation model</a>. Both use an identical file format
and identical restrictions on the possible values (interval from 0.0 to 1.0), and are therefore described together.<br/><br/>
Specifically, the maximum value for the fraction of the solvent-accessible volume (&eta;<sub>i,max</sub>) and
the atomic volume reduction factor used in most computations based on atomic volumes can be altered. Both quantities are 
derived from default, hard-coded molecular topologies and not from input structures (unless an unsupported residue 
is in use, where the former is of course missing). They are both reported in output file
<a href="outputfiles.html#SAV_BY_ATOM.dat">SAV_BY_ATOM.dat</a> as columns 4 and 7, respectively.
A patch file supplied via <a href="keywords.html#SAVPATCHFILE">FMCSC_SAVPATCHFILE</a> will alter the &eta;<sub>i,max</sub> values,
and a patch file supplied via <a href="keywords.html#ASRPATCHFILE">FMCSC_ASRPATCHFILE</a> will alter the volume reduction
factors. In either case, each line has to contain two entries, an integer giving the atom number
and a floating point number giving the new parameter constrained to the interval [0:1].<br/>
<br/>
<u>Example for <a href="keywords.html#SAVPATCHFILE">FMCSC_SAVPATCHFILE</a>:</u><br/>
---------------------------------------<br/>
<big><code>
16&nbsp;&nbsp; 0.86<br/>
4&nbsp;&nbsp;  0.92<br/>
22&nbsp;&nbsp;  0.99</code></big><br/>
---------------------------------------<br/>
<br/>
This input file would change the values &eta;<sub>i,max</sub> for atoms 4, 16, and 22 to 0.92, 0.86, and 0.99, respectively.
CAMPARI will not perform any noteworthy tests whether the patched values are meaningful.
In this context, it is important to note that very small values for the &eta;<sub>i,max</sub> may lead
to large gradients due to the compression of the interpolation regime between solvated and desolvated states.
Aside from printing a report for all successful changes, the results of the patch can also be assessed
by inspecting the aforementioned <a href="outputfiles.html#SAV_BY_ATOM.dat">SAV_BY_ATOM.dat</a> (assuming
<a href="keywords.html#SAVCALC">SAVCALC</a> is appropriate).<br/>
In general, neither class of parameters is considered free, so patches of this type would seem most likely to be
useful when implementing Hamiltonian support for otherwise unsupported residues. It may also come in useful
for cases, where either the atomic size parameters and/or the molecular topology mean that the pairwise volume
reduction factors are inappropriate (better approximation required and/or triple overlaps significant), although
it should be mentioned that the <i>radius</i> functionality of the <a href="parameters.html#S2_LJ-types">parameter file</a>
and <a href="keywords.html#RPATCHFILE">radius patches for specific atoms</a> are probably the more appropriate tools in such a case.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_FOSPATCHFILE">Free energy of solvation patch input file:</a></h4>
Keyword: <b><a href="keywords.html#FOSPATCHFILE">FMCSC_FOSPATCHFILE</a></b><br/>
<br/>
Prerequisite: This file is only relevant if the direct mean file interaction (DMFI) of the ABSINTH implicit solvation
model is used (&rarr; <a href="keywords.html#SC_IMPSOLV">FMCSC_SC_IMPSOLV</a>).<br/><br/>
This file provides the input facility to specifically override the details of the DMFI model,
<i>i.e.</i>, the parsing of the molecule into solvation groups (by atoms), the weight factors for individual
atoms constituting solvation groups, and the reference free energies of solvation themselves. Depending
on the choice for <a href="keywords.html#FOSMODE">FMCSC_FOSMODE</a>, it may also be relevant to 
supply patched values for solvation enthalpies and heat capacities.
For global changes of the latter type, it should be pointed out that changes directly in the parameter file
 are simpler (see <a href="keywords.html#PARAMETERS">PARAMETERS</a> and
<a href="parameters.html#S4_Free_energies_of_solvation">dedicated documentation</a>).
Each line has to contain between three and six entries. The first is an integer giving
the atom number using the default CAMPARI atom order (that can usually be extracted from 
<a href="outputfiles.html#basename_START.pdb">basic output files</a> &rarr; see
<a href="inputfiles.html#FMCSC_CPATCHFILE">above</a>). The second is another integer identifying by a code
the target solvation group. The chosen code is arbitrary as long as all separate solvation groups are referred
to by a different integer. The third number is a floating point value giving the weight for the indicated atom.
The sum of weights for all atoms using the same integer code in column 2 have to add up to unity exactly (if
fractions occur that cannot be specified with arbitrary precision in decimal floating point representation, it is
advised to provide enough digits to emulate double precision).
The remaining numbers are floating point numbers giving (in this order) the reference free energy of solvation for a group,
the corresponding solvation enthalpy, and the heat capacity
(in kcal/mol, kcal/mol, and cal mol<sup>-1</sup> K<sup>-1</sup>, respectively).
In general, only the first number is required for a successful patch, and it needs to be specified only
with the first instance of the corresponding integer code.
Numbers given for further atoms belonging to the same group are ignored. Note that missing values
for enthalpy and heat capacity would override values given in the parameter file (if any) by instead
setting the heat capacity to zero and the enthalpy to be equal to the free energy.<br/>
A single character in the first line lets the user choose between three different modes. All three modes
have identical restrictions beyond the ones already stated. First, the resultant solvation groups overlap with 
existing solvation groups (that are most conveniently specified in optional output file 
<a href="outputfiles.html#FOS_GROUPS.vmd">FOS_GROUPS.vmd</a> &rarr; <a href="keywords.html#FOSREPORT">FMCSC_FOSREPORT</a>).
None of the atoms in one of those original groups must remain unaccounted for by the patch. Second, newly defined groups
must not cross residue boundaries unless they represent trivial changes to existing groups that do already
cross residue boundaries (peptide unit). Here, "trivial" means that the constituting atoms are exactly the same,
and that only the (nonzero) weights and/or the reference free energy of solvation are altered. Third, all weights
have to be positive.<br/>
<br/>
<b>Mode 'A':</b>
&nbsp;
The specifications after the first line ('A') are assumed to apply only to those atoms exactly specified.
This is the simplest input mode that allows maximal control (each atom can be controlled individually).
Suppose the simulation encompasses two formamide molecules (6 atoms each). The standard group parsing is to have all
atoms except the aldeyhde hydrogen (indices 4 and 10) constitute a single group with equal weights (0.2).
<br/>
<br/><u>Example:</u><br/>
---------------------------------------<br/>
<big><code>
A<br/>
7&nbsp;&nbsp; 33 0.5 -8.0<br/>
8&nbsp;&nbsp; 33 0.25<br/>
9&nbsp;&nbsp; 33 0.25<br/>
11&nbsp; 33 0.0<br/>
12&nbsp; 33 0.0</code></big><br/>
---------------------------------------<br/>
<br/>
The above input would change the solvation group setup for the second formamide molecule such that the group
now only consists of the heavy atoms (7-9) with asymmetric weights (0.5 for N, 0.25 for C and O), and an
altered reference free energy of solvation. The code "33" is arbitrary as explained above.
Omitting the entries for atoms 11 and 12 would cause an error
because these atoms were formerly part of an existing solvation group, and would be left unaccounted for.
<br/>
<br/><u>Another Example:</u><br/>
---------------------------------------<br/>
<big><code>
A<br/>
9&nbsp;&nbsp; 14 0.5 -5.0<br/>
8&nbsp;&nbsp; -1 0.5 -5.0<br/>
7&nbsp;&nbsp; -1 0.5<br/>
10&nbsp; 15 1.0 -0.1<br/>
11&nbsp; 14 0.25<br/>
12&nbsp; 14 0.25</code></big><br/>
---------------------------------------<br/>
<br/>
This example would define three solvation groups for the second formamide molecule. The CO unit
(code "-1"), the NH<sub>2</sub> unit (code "14"), and the lone aldehyde hydrogen (code "15"). Because
the aldehyde hydrogen was previously not part of a solvation group, it can be included freely into
new solvation groups.<br/>
<br/>
<b>Mode 'T':</b>
This mode works similarly to mode 'A' with the major differences that all requests are
generalized in molecule type-dependent fashion. The specified indices have to refer to the 
first instance of a given target molecule type. The two examples listed above would both lead
to premature termination of CAMPARI, because both reference the second of two formamide 
molecules.
<br/>
<br/><u>Example:</u><br/>
----------------------------------------------------------------------------<br/>
<big><code>
T<br/>
3&nbsp;&nbsp; 1 0.3333333333333 -5.5 -10.0 50.0<br/>
1&nbsp;&nbsp; 2 0.5 -5.5 -6.5<br/>
2&nbsp;&nbsp; 2 0.5<br/>
5&nbsp;&nbsp; 1 0.3333333333333<br/>
6&nbsp;&nbsp; 1 0.3333333333333</code></big><br/>
----------------------------------------------------------------------------<br/>
<br/>
As a result of this input file, the solvation groups on <i>both</i> formamide molecules would be altered.
They would be changed to separate groups covering the CO and NH<sub>2</sub> units, respectively, with
both groups having reference free energies of solvation of -5.5kcal/mol and equal atomic weights. In addition,
the CO unit would receive a solvation enthalpy of -6.5kcal/mol, and the NH<sub>2</sub> unit would receive an 
enthalpy of -10.0kcal/mol and a heat capacity of 50.0 cal mol<sup>-1</sup> K<sup>-1</sup>. These latter
modifications would only be relevant if a <a href="keywords.html#FOSMODE">temperature-dependent ABSINTH DMFI</a>
were in use.<br/>
<br/>
<b>Mode 'R':</b>
This mode allows selection and generalization by residue type, but works otherwise identically to mode 'T'.
For polymeric systems, it will sometimes be of interest to selectively change details of the definition
of solvation groups for a specific type of residue. For example, how does polyglutamine respond to
global alterations in the sidechain solvation groups. The numbering should refer to the atomic 
indices of the first instance of the targeted residue type. However, care must be taken, since CAMPARI -
by necessity - distinguishes terminal from nonterminal residues with regards to its type. Consider
the sequence (Gln)<sub>5</sub> with charged termini. To change the assignments for the internal
glutamine residues, one would have to find the atomic indices of the <i>second</i> glutamine residue,
and enter the appropriate modifications. Both the N-terminal and the C-terminal glutamine residue
would require an <i>additional</i> specification each, since they are technically of different type.
The same is true for crosslinked residues (if any).<br/><br/>
As with all patch functionality, it is recommended that users check meticulously for possible
errors in the parsing of patch input files. By enabling <a href="keywords.html#FOSREPORT">FMCSC_FOSREPORT</a>
in the key-file, CAMPARI will provide a report of all modified solvation groups (before <i>vs.</i> after),
and users are encouraged to use this output in helping them diagnose potential errors beforehand.<br/>
<br/>
<br/>
<br/>
<h3><a class="NoHi" id="Sec4">Section 4: Files defining auxiliary terms to the energy function</a></h3>
<hr style="width: 100%; height: 2px;"></hr> (<a href="inputfiles.html#TOP">back to top</a>)<br/>
<h4><a class="NoHi" id="FMCSC_CMAPDIR">CMAP input files:</a></h4>
Keyword: <b><a href="keywords.html#CMAPDIR">FMCSC_CMAPDIR</a></b><br/>
<br/>
Required for: Use of CMAP or similar corrections &rarr; <a href="keywords.html#SC_BONDED_M">FMCSC_SC_BONDED_M</a><br/><br/>
CMAP corrections are two-dimensional correction energy surfaces introduced in the CHARMM force field that operate
on the &phi;/&psi;-surfaces of polypeptide residues. From input files, a 2D map is constructed and serves as an additional
energy term that is made smooth and continuously differentiable via suitable interpolation. The names of the files
containing these maps have be put directly into the parameter file (see <a href="../params/charmm.prm">charmm.prm</a> for an example). This is detailed
<a href="parameters.html#S8_CMAP_potential_types">elsewhere</a>. With appropriate choices for the filenames and the directory they reside
in (&rarr; <a href="keywords.html#CMAPDIR">FMCSC_CMAPDIR</a>), it is possible to utilize this functionality as a general
bias or correction potential even for force fields (parameter files) that do not natively contain such corrections. <br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_TORFILE">Torsional bias potentials input file:</a></h4>
Keyword: <b><a href="keywords.html#TORFILE">FMCSC_TORFILE</a></b><br/>
<br/>
Required for: Biasing potentials acting on individual polypeptide or polynucleotide, rotatable dihedral angles &rarr; <a href="keywords.html#SC_TOR">FMCSC_SC_TOR</a><br/><br/>
The specification of simple torsional bias potentials is possible through either global or residue-specific input. The point of these potentials is usually to restrain the polymer to some specific conformation with the most common
application being the restraint to the conformation provided by a structural input file.
The first line has to contain the character 'G' (Global) or 'R' (Residue-specific) to distinguish
between the two.<br/>
<br/>
The actual input starts in the second line and is interpreted as
follows:<br/>
<br/>
<u>'G'-mode:</u><br/>
An integer code with a defined number of parameters. Modes 1 and 3 specify
harmonic restraint potentials and modes 2 and 4 are used for Gaussian well potentials.<br/>
<ol>
<li>
V<sub>TOR</sub> = &#931;<sub>i</sub> k<sub>i</sub>&#183;(&#977;<sub>i</sub>-&#977;<sub>i</sub><sup>0</sup>)<sup>2</sup><br/>
<br/>
Here, "i" runs over most of the torsions which are rotatable in Monte Carlo or torsional dynamics calculations. Dihedral angles in unsupported residues
are supported (up to a maximum number of 12), while dihedral angles in supported residues, which are not sampled natively by CAMPARI, are excluded.
The sugar bond in nucleotides and the &#966;-angle in proline are included, whereas any other pucker degrees of freedom are not. For details,
see <a href="keywords.html#CARTINT">FMCSC CARTINT</a>, <a href="inputfiles.html#FMCSC_SEQFILE">FMCSC_SEQFILE</a>, and 
<a href="keywords.html#DYNAMICS">FMCSC DYNAMICS</a>.<br/>
To make the file format general, 12 equilibrium positions (&#977;<sub>i</sub><sup>0</sup>) and 12 force constants (k<sub>i</sub>) have to be provided.
Entries 7-12 will address &#967;<sub>1-6</sub> for all supported residues. For any unsupported residues of recognized polymer type, these entries
will be reserved for the unclassified dihedral angles, which - by definition - are usually equivalent to the &#967;-angles. Nucleotides will
have the (up to) five freely rotatable backbone angles (see <a href="inputfiles.html#FMCSC_SEQFILE">FMCSC_SEQFILE</a>) in entries 1-5 and the dihedral around
the sugar bond (C4*-C3*) in entry 6. Polypeptides will have &#969;-, &#966;-, and &#968;-angles in entries 1-3 (entries 4-6 are unused). Supported small
molecules will generally not use anything in entries 1-6 except secondary amides which have &#969;-angles in entry 1. Unsupported residues of unrecognized
polymer type (including small molecules) utilize all 12 entries for their respective dihedral angles. The counting is by Z-matrix line in ascending order.
This means that degrees of freedom in unsupported residues of unsupported polymer type going beyond the 12<sup>th</sup> dihedral angle or in unsupported
residues of supported polymer type going beyond the 6<sup>th</sup> dihedral angle cannot be biased this way. Using this feature for unsupported residues
requires a good understanding of the content provided for <a href="inputfiles.html#FMCSC_PDBFILE">FMCSC PDBFILE</a> and
<a href="keywords.html#PDB_TEMPLATE">FMCSC PDB_TEMPLATE</a> and knowledge of the Z-matrix format.<br/>
Lastly, dihedral angles defined by sulfide bridges are supported even though they are not native CAMPARI dihedral angles.
The first cysteine (in sequence) will use entries 7-9 for the &#967;<sub>1</sub>-angle, the C<sub>&alpha;</sub>-C<sub>&beta;</sub>-S-S angle, and the
C<sub>&beta;</sub>-S-S-C<sub>&beta;</sub> angle, respectively. The second cysteine will use entries 7-8 for its &#967;<sub>1</sub>- and
C<sub>&alpha;</sub>-C<sub>&beta;</sub>-S-S angles. The 12 equilibrium positions have to be listed first followed by the 12 force constants for a
total of 25 columns (the first one being occupied by the mode identifier).<br/>
<br/>
</li>
<li>
V<sub>TOR</sub> = -exp(-&#931;<sub>i</sub> (&#977;<sub>i</sub>-&#977;<sub>i</sub><sup>0</sup>)<sup>2</sup> / (2.0&#963;<sub>i</sub><sup>2</sup>) )<br/>
<br/>
This potential uses an identical file format except that the standard widths &#963;<sub>i</sub> are read in the latter 12 entries rather than the harmonic force constants. V<sub>TOR</sub> creates Gaussian wells (in multiple dimensions) for each residue. It can be used to subtly favor certain regions of phase space without introducing any stringent restraints. The formula implies a unit factor such that the conformation perfectly matching the &#977;<sub>i</sub><sup>0</sup> will give a favorable energy of <a href="keywords.html#SC_TOR">FMCSC_SC_TOR</a> kcal/mol.<br/>
<br/>
</li>
<li>
V<sub>TOR</sub> = &#931;<sub>i</sub> k<sub>i</sub>&#183;(&#977;<sub>i</sub>-&#977;<sub>i,struc</sub><sup>0</sup>)<sup>2</sup><br/>
<br/>
The only difference to mode 1 is that the &#977;<sub>i,struc</sub><sup>0</sup> are obtained directly from the simulations initial configuration (<i>e.g.</i> from structural input) and not from this input file. Hence, only 12 entries are needed with the necessary force constants k<sub>i</sub>. Note that this does not work in 
<a href="keywords.html#PDBANALYZE">trajectory analysis mode</a>, <i>i.e.</i>, the wrong minimum positions will be used.<br/>
<br/>
</li>
<li>
V<sub>TOR</sub> = -exp( -&#931;<sub>i</sub> (&#977;<sub>i</sub>-&#977;<sub>i,struc</sub><sup>0</sup>)<sup>2</sup> / (2.0&#963;<sub>i</sub><sup>2</sup>) )<br/>
<br/>
To mode 2 what mode 3 is to  mode 1. 
<br/>
</li>
</ol>
<br/>
Example:<br/>
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------<br/>
<code>&nbsp;G<br/>
&nbsp;1&nbsp;0.0&nbsp;-57.0&nbsp;-47.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.02&nbsp;0.02&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0</code><br/>
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------<br/>
<br/>
This input file would turn on a biasing potential (stiffness: 0.02 kcal/(mol&#183; deg.<sup>2</sup>)) biasing the backbone torsional angles of all polypeptide residues toward the ideal &#945;<sub>R</sub>-geometry. It would also bias all present
polynucleotide residues to the same conformation in their respective backbone degrees of freedom #2 and #3.
It would not affect any polypeptide termini or small molecules.<br/>
<br/>
<u>'R'-mode:</u><br/>
Largely, this format agrees with the global specifications. The only differences are that:
<ol type='i'>
<li>An extra column is needed giving the residue number for which the torsional bias subsequently specified on the same line is to be turned on.<br/></li>
<li>Multiple entries can be provided, each for an individual residue. Note that it is also possible to override (not(!) append) the settings for an
individual residue within the same file.<br/></li>
</ol>
<br/>
Example (let us assume residue 3 is a lysine and residue 12 a uracil ribonucleotide (RPU):<br/>
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------<br/>
<code>R<br/>
3&nbsp;&nbsp;1&nbsp;&nbsp;0.0&nbsp;-57.0&nbsp;-47.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;120.0&nbsp;70.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.02&nbsp;0.02&nbsp;0.0&nbsp;0.0&nbsp;0.05&nbsp;0.05&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0<br/>
12&nbsp;&nbsp;4&nbsp;&nbsp;30.0&nbsp;30.0&nbsp;30.0&nbsp;30.0&nbsp;30.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0&nbsp;0.0</code><br/>
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------<br/>
<br/>
This input file would turn on biasing potentials for two residues - #3
and #12. There would be a harmonic restraint biasing the backbone conformation of LYS<sub>3</sub> to that of an ideal &#945;<sub>R</sub>-helix. A stronger bias (k<sub>i</sub> = kcal/(mol&#183; deg.<sup>2</sup>)) would bias the value of &#967;<sub>1</sub> and &#967;<sub>2</sub> to values of 120.0&deg; and 70.0&deg;, respectively. In contrast, RPU<sub>12</sub> would experience a Gaussian bias favoring the backbone conformation it initially assumes (excluding the sugar bond) with a general width of 30.0&deg;.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_POLYFILE">Polymeric biasing potentials input file:</a></h4>
Keyword: <b><a href="keywords.html#POLYFILE">FMCSC_POLYFILE</a></b><br/>
<br/>
Required for: Biasing potentials on polymeric properties of molecules &rarr; <a href="keywords.html#SC_POLY">FMCSC_SC_POLY</a><br/><br/>
Polymeric biasing potentials (see <a href="keywords.html#SC_POLY">FMCSC_SC_POLY</a>) are 
a feature in CAMPARI that requires some expertise on the user's end, because the
resultant energy landscape for a restrained molecule can be very rugged. The potentials
apply a harmonic restraint to at most two polymeric descriptors: 1) a transform of the radius of gyration
meant to map values to the unit interval over a wide range of chain lengths and conditions; 2) 
a measure of asphericity. This input file allows the user to request polymeric restraints on a per-molecule basis or on a per molecule-type
basis. The first line has to contain the character 'M' or 'T' to distinguish between the two.<br/>
<br/>
The actual input starts in the second line and is interpreted as
follows:<br/>
<br/>
'M'-mode:<br/>
An integer code corresponding to the target molecule (numbered as in
the <a href="inputfiles.html#FMCSC_SEQFILE">sequence file</a>) with four mandatory parameters, specifying the harmonic
restraints on the two parameters:<br/>
<br/>
&nbsp;t = &lt;f<sub>1</sub>&#183;(f<sub>2</sub>&#183;(R<sub>g</sub>/L<sub>c</sub>))<sup>f<sub>3</sub>/N<sup>0.33</sup></sup>&gt;<br/>
&#948; = 1.0 - 3.0&#183;(&#955;<sub>1</sub>&#183;&#955;<sub>2</sub> + &#955;<sub>2</sub>&#183;&#955;<sub>3</sub> + &#955;<sub>1</sub>&#183;&#955;<sub>3</sub>)/(&#955;<sub>1</sub> + &#955;<sub>2</sub> + &#955;<sub>3</sub>)<sup>2</sup><br/>
<br/>
Here, the f<sub>i</sub> are arbitrary factors meant to achieve the desired mapping to the interval [0,1] (values are 2.5, 1.75 and 4.0).
The radius of gyration is denoted by R<sub>g</sub>, L<sub>c</sub> is an estimate of the contour length of the chain, and N is its sequence length.
The &#955;<sub>i</sub> are the eigenvalues of the gyration tensor (&rarr; <a href="outputfiles.html#POLYAVG.dat">elsewhere</a> for additional
information).<br/>
<br/>
V<sub>POLY</sub> = k<sub>t</sub>&#183;(t - t<sub>0</sub>)<sup>2</sup> + k<sub>&#948;</sub>&#183;(&#948; - &#948;<sub>0</sub>)<sup>2</sup><br/>
<br/>
The four floating point numbers required are: t<sub>0</sub>, &#948;<sub>0</sub>, k<sub>t</sub>,
k<sub>&#948;</sub>. Both t<sub>0</sub> and &#948;<sub>0</sub> have to lie within the interval [0:1] as they 
are (pseudo-)normalized quantities.<br/>
<br/>
The stiffnesses are simply (twice) the spring constants of the two
harmonic terms and should be given in kcal/mol as both order
parameters are unitless.<br/>
<br/>
<u>Example:</u><br/>
---------------------------------------------------------------<br/>
<big><code>M<br/>
1&nbsp;&nbsp; 0.8 &nbsp; 0.5 &nbsp; 10.0&nbsp;&nbsp; 100.0<br/>
13&nbsp;&nbsp; 0.2&nbsp;&nbsp; 0.1&nbsp;&nbsp; 100.0&nbsp;&nbsp; 0.0</code></big><br/>
---------------------------------------------------------------<br/>
<br/>
This input file would turn on a biasing potential driving molecule #1
toward a value of t of 0.8 (stiffness 10 kcal/mol and toward an
asphericity of 0.5 (stiffness 100 kcal/mol). Conversely, molecule #13
would be driven toward a value of t of 0.2 with stiffness of 100
kcal/mol with no restraint on &#948;. It is very important to note
that asphericity and t are <b>not independent</b>. Hence, restraining
both of them to a pair of target values that are mutually exclusive will simply lead to a competition
between the two force constants (aided by the underlying Hamiltonian and entropy); a scenario
of little practical use. Furthermore, it should be emphasized that t is a
<i>pseudo></i>normalized quantity and that its accessible range will be somewhat ill-defined
for systems of extreme size (very small, very large) and/or of extreme conformation
(fully extended rods, collapsed points). To a lesser extent this is even
true for &#948;, which by virtue of polymer geometry (finite width) will prohibit values
rigorously approaching zero or unity for finite length chains.<br/>
<br/>
'T'-mode:<br/>
An integer indicating corresponding to the target molecule <i>type</i> as
determined by the first occurrence in the sequence file. Otherwise the input
is treated the same.<br/>
<br/>
Note that these potentials will never be disabled even if the molecule in question is 
practically rigid. In conjunction with the energy function and degrees of freedom
this can easily lead to extreme strain (<i>e.g.</i> when using <a href="keywords.html#CARTINT">Cartesian dynamics</a>),
which may compromise simulation stability.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_TABCODEFILE">Tabulated potential code requests:</a></h4>
Keyword: <b><a href="keywords.html#TABCODEFILE">FMCSC_TABCODEFILE</a></b><br/>
<br/>
Required for: Use of tabulated interactions on interatomic distances &rarr; <a href="keywords.html#SC_TABUL">FMCSC_SC_TABUL</a><br/><br/>
This file allows the specification of specific interactions for tabulated
potentials to apply to. It works in almost exactly the same way as the one requesting
general distance distribution analysis (see
<a href="inputfiles.html#FMCSC_PCCODEFILE">detailed format description</a>). The major difference
is that the tabulated potential mappings ignore analysis groups and only partition by molecule types 
(relevant for input modes 2/3).<br/>
The two functionalities naturally use the same framework. The requests are in fact
identical because they efficiently allow the user to specify atom pairs or sets of
pairs to become associated with an integer code corresponding to either a specific
interaction potential (here) or to a separate distance distribution
analysis request (when the request is made by <a href="keywords.html#PCCODEFILE">FMCSC_PCCODEFILE</a>).
The only other major difference is that inconsistent numbering of the
tabulated potentials will not be tolerated, as it would require a
nonsensical file for the actual potential input (see
<a href="inputfiles.html#FMCSC_TABPOTFILE">FMCSC_TABPOTFILE</a>). It is important to
keep in mind that mistakes are very easily made in an input file
this flexible. Hence, there is an output file summarizing all the terms
requested with additional information (see documentation on <a href="outputfiles.html#TABULATED_POT.idx">TABULATED_POT.idx</a>,
which is enabled by keyword <a href="keywords.html#TABREPORT">FMCSC_TABREPORT</a>).
Finally, from an implementation point of view, it must be mentioned that the type of
input specification (modes 1-3 are available, see
<a href="inputfiles.html#FMCSC_PCCODEFILE">detailed format description</a>)
has no impact on the speed of the calculation. The
requests are always transformed into identical data structures, which
are queried efficiently during energy and force evaluations.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_TABPOTFILE">Tabulated potential input file:</a></h4>
Keyword: <b><a href="keywords.html#TABPOTFILE">FMCSC_TABPOTFILE</a></b><br/>
<br/>
Required for: Use of tabulated interactions on interatomic distances &rarr; <a href="keywords.html#SC_TABUL">FMCSC_SC_TABUL</a><br/><br/>
A file with n+1 columns, where n is the highest integer code of a
unique potential requested in <a
 href="inputfiles.html#FMCSC_TABCODEFILE">FMCSC_TABCODEFILE</a>. The
number of rows is arbitrary and will be determined by the program.
Note, however, that the potentials have to be stored in memory.<br/>
<br/>
The first column contains the distance information (only regularly spaced bins
are allowed) and should cover the distance range available to the
specific atom pairs (only one distance spectrum possible for all
potentials).<br/>
<br/>
The i+1th to i+(n+1)th columns then contain the n individual potentials
according to their integer code in units of kcal/mol (<i>i.e.</i>, if the index file indicates
that atoms 3 and 25 interact through potential 8, then the 9th column
of the potential file gives the tabulated potential used for this
atom-atom pair). <br/>
<br/>
<u>Example:</u><br/>
----------------------------------------------<br/>
<big><code>
2.5&nbsp;&nbsp; 10.0&nbsp; 10.0<br/>
3.5&nbsp;&nbsp; 10.0&nbsp; 10.0<br/>
4.5&nbsp;&nbsp; 10.0&nbsp;&nbsp; 0.0<br/>
5.5&nbsp;&nbsp; 0.0 &nbsp; 0.0<br/>
6.5&nbsp;&nbsp; 10.0&nbsp; 10.0<br/>
7.5&nbsp;&nbsp; 10.0&nbsp; 10.0</code></big><br/>
----------------------------------------------<br/>
<br/>
Note the regular distance spacing.<br/>
The possible integer codes to be used in the index file would hence be
1 or 2, anything else will create an error. Also note that the code
does cubic Hermite spline interpolation on the potentials, <i>i.e.</i> provides a continuous
and smooth function. The use of a cubic spline is the reason that it is recommended
to explicitly taper off the potential to a constant value at the limits of the distance
range considered. The inclusion of the two bins at 2.5 and 7.5&#8491; is made for this purpose,
since otherwise the tangent at 3.5 and 6.5&#8491; would have been set to 0.0. 
Correspondingly, distance values outside of the limits are
treated just like the last available bin (<i>i.e.</i>, in the example above
distance values of 2.0 or 50.0 would both give a potential value of 10.0 kcal/mol).
Tangent data can be supplied in a separate file (see <a href="inputfiles.html#FMCSC_TABTANGFILE">below</a>).<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_TABTANGFILE">Tabulated potential derivatives (tangents) input file:</a></h4>
Keyword: <b><a href="keywords.html#TABTANGFILE">FMCSC_TABTANGFILE</a></b><br/>
<br/>
Required for: Use of tabulated interactions on interatomic distances &rarr; <a href="keywords.html#SC_TABUL">FMCSC_SC_TABUL</a>
if it is desired that the derivatives at the tabulated points are not estimated numerically<br/><br/>
A file with n columns, where n is the highest integer code of a
unique potential requested in <a
 href="inputfiles.html#FMCSC_TABCODEFILE">FMCSC_TABCODEFILE</a>. The
number of rows is arbitrary and will be determined by the program.
Note, however, that the potentials and tangents have to be stored in memory.<br/>
<br/>
The columns contain the n individual potential derivatives
according to their integer code in units of kcal&#183;mol<sup>-1</sup>&#183;&#8491;<sup>-1</sup>
(<i>i.e.</i>, if the index file indicates
that atoms 3 and 25 interact through potential 8, then the 8th column
of the potential derivative file gives the tangents to be assumed for this tabulated potential and atom-atom pair).
Note that the required derivative is that with respect to the (increasing) distance coordinate.
The potentials themselves are given in a <a href="inputfiles.html#FMCSC_TABPOTFILE">different input file</a>.<br/>
<br/>
<u>Example:</u><br/>
----------------------------------------------<br/>
<big><code>
0.0&nbsp;&nbsp; 0.0<br/>
0.0&nbsp; -5.0<br/>
-10.0&nbsp;-2.0<br/>
0.0 &nbsp;&nbsp; 2.0<br/>
10.0&nbsp;&nbsp; 5.0<br/>
0.0&nbsp;&nbsp; 0.0<br/></code></big>
----------------------------------------------<br/>
<br/>
Note that CAMPARI takes the information about how many tabulated potentials are present
exclusively from the <a href="inputfiles.html#FMCSC_TABPOTFILE">potential input</a>.
It is therefore up to the user to ensure that the appropriate number of columns is present
in this input file. If this file is incomplete or absent, the missing tangent values
are constructed numerically from the potentials themselves. It is not possible to 
provide values starting from a distance value other than the first one in the
<a href="inputfiles.html#FMCSC_TABPOTFILE">tabulated potential input file</a>. The provided
derivatives are not checked in any way and taken strictly <i>as is</i>.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_DRESTFILE">Distance restraint potential input file:</a></h4>
Keyword: <b><a href="keywords.html#DRESTFILE">FMCSC_DRESTFILE</a></b><br/>
<br/>
Required for: Biasing potential to introduce distance restraints between pairs of atoms &rarr; <a href="keywords.html#SC_DREST">FMCSC_SC_DREST</a><br/><br/>
The first line has to contain only a single integer, <i>viz</i>, the number of restraints in
the following list.<br/>
<br/>
The actual input starts in the second line and is interpreted as
follows:<br/>
<br/>
Five mandatory parameters: two integers (i,j) specifying the atom pair for
which the restraint potential is to be applied; a type indicator specifying the type of
restraint potential to act on the distance of these two atoms; the parameters of the potential,
specifically, the target distance R<sub>0</sub> in &#8491;, and the parameter k<sub>ij</sub>
in kcal&#183;mol<sup>-1</sup>&#183;&#8491;<sup>-2</sup> (half the force constant for the implied Hookean spring).
The available types are all harmonic in nature and differ as follows:<br/>
<ol>
<li>V<sub>DREST</sub>(i,j) = k<sub>ij</sub>&#183;(R<sub>ij</sub> - R<sub>0</sub>)<sup>2</sup></li>
<li>V<sub>DREST</sub>(i,j) = k<sub>ij</sub>&#183;H(R<sub>ij</sub> - R<sub>0</sub>) (R<sub>ij</sub> - R<sub>0</sub>)<sup>2</sup></li>
<li>V<sub>DREST</sub>(i,j) = k<sub>ij</sub>&#183;H(R<sub>0</sub> - R<sub>ij</sub>) (R<sub>ij</sub> - R<sub>0</sub>)<sup>2</sup></li>
</ol>
Here, H(x) is the Heaviside step function. The above potential would be added to the overall Hamiltonian.<br/>
<br/>
<u>Example:</u><br/>
---------------------------------------<br/>
<big><code>2<br/>
11&nbsp;&nbsp; 45&nbsp;&nbsp; 1&nbsp;&nbsp; 10.0&nbsp;&nbsp; 5.0<br/>
22&nbsp;&nbsp; 56&nbsp;&nbsp; 2&nbsp;&nbsp; 15.0&nbsp;&nbsp; 10.0</code></big><br/>
---------------------------------------<br/>
<br/>
This would add two distance restraints to the Hamiltonian. The first would act
between atoms 11 and 45 and restraint their distance to a minimum position of 10.0&#8491; with a k<sub>ij</sub>
of 5.0 kcal&#183;mol<sup>-1</sup>&#183;&#8491;<sup>-2</sup>. The second would act between atoms 22 and 56 
and harmonically restrain their distance if it exceeds a lower bound of 15.0&#8491; with a k<sub>ij</sub>
of 10.0 kcal&#183;mol<sup>-1</sup>&#183;&#8491;<sup>-2</sup>. Negative force constants
are converted to positive ones (the negative potential is not allowed). Note
that negative distances are disallowed for obvious reasons. Furthermore, positive distances
can result in a frustrated energy landscape if they are small enough to place the minimum restraint
energy in a distance regime leading to steric clashes  in the presence of
<a href="keywords.html#SC_IPP">excluded volume terms</a>. This is particularly harmful
in <a href="keywords.html#DYNAMICS">gradient-based sampling approaches</a>.
Users should keep in mind that it is possible to specify multiple restraint terms for the same atom pair
and that distance restraints are never subjected to cutoffs (unlike the more general
functionality of <a href="keywords.html#SC_TABUL">tabulated potentials</a>).<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_EMMAPFILE">Density restraint input map:</a></h4>
Keyword: <b><a href="keywords.html#EMMAPFILE">FMCSC_EMMAPFILE</a></b><br/>
<br/>
Required for: Biasing potential to introduce a global, spatial density restraint on a simulation density
derived from an atomic property such as mass &rarr; <a href="keywords.html#SC_EMICRO">FMCSC_SC_EMICRO</a><br/><br/>
This input file has to be provided in a specific file format in binary form. The <a href="references.html#ref16_1">NetCDF library</a>
allows a reasonably efficient storage of array data in binary form while maintaining a flexible and adaptable interface.
For instance, <a href="keywords.html#XYZPDB">trajectory output</a> can also be written in NetCDF format.
The file itself is binary, but the library comes with tools to convert appropriately formatted files
from human-readable versions. For further details, users are referred to the NetCDF documentation.<br/>
The format itself is similar to that used by <a href="references.html#ref23_10">UCSF Chimera</a>
when writing density information in NetCDF format. Specifically, it should have the following fields defined:<br/>
<ul>
<li>Dimensions of <b>spatial</b> (the dimensionality itself) and <b>X</b>, <b>Y</b>, and <b>Z</b> (for the case of a three-dimensional 
input density, which is the only case currently supported), where the latter three dimensions give the numbers of
grid cells in the respective dimensions</li>
<li>A variable <b>deltas</b> with an attached attribute of <b>units</b>; this is an array of size <b>spatial</b> and
type <i>float</i>, and the default unit is &#8491;</li>
<li>A variable <b>densities</b> with an attached attribute of <b>units</b>; this is a three-dimensional array of size <b>Z,Y,X</b> and
type <i>float</i>, and the default unit is g/cm<sup>3</sup></li>
<li>Optional, global attributes <b>xyz_origin</b> and <b>xyz_step</b>, both (currently) requiring the specification of three
floating point numbers in double precision (<b>xyz_origin</b> is completely optional and <b>xyz_step</b> can substitute for
variable <b>deltas</b> (and vice versa))</li>
<li>An optional variable <b>origin</b> with an attached attribute of <b>units</b> that is also an array of size <b>spatial</b> and
type <i>float</i> with a default unit of &#8491;</li>
<li>The data section has to contain the appropriate number of entries for <b>densities</b> and also for <b>deltas</b> in case
<b>xyz_step</b> is not defined</li>
</ul>
Note that spatial density files written by CAMPARI define some additional, global attributes (<b>title</b>, <b>program</b>,
and <b>programVersion</b>), none of which are read, however. Note as well that the format described here is also used for
the spatial density files CAMPARI writes itself, specifically <a href="outputfiles.html#DENSITY.nc">DENSITY.nc</a> and
<a href="outputfiles.html#DENSITY_INPUT_PHYS.nc">DENSITY_INPUT_PHYS.nc</a>.<br/>
Once the input density map is read, it is processed, <i>i.e.</i>, interpreted quantitatively, to yield the physical
input density &Xi;<sub>ijk</sub> for a given lattice cell with indices i, j, and k. The details of this linear transformation
are described <a href="keywords.html#EMMAPFILE">elsewhere</a>, and rely on several keywords associated with the density
restraint potential, most notably <a href="keywords.html#EMTHRESHOLD">EMTHRESHOLD</a> and <a href="keywords.html#EMTOTMASS">EMTOTMASS</a>.
The physical input density (the interpreted map is written to output file <a href="outputfiles.html#DENSITY_INPUT_PHYS.nc">DENSITY_INPUT_PHYS.nc</a>)
is compared to the simulation density to yield the density restraint potential, E<sub>EMICRO</sub> as described
<a href="keywords.html#SC_EMICRO">elsewhere</a>. The computation of the simulation density is 
desribed in detail for keyword <a href="keywords.html#EMCALC">EMCALC</a>.<br/>
<br/>
<br/>
<br/>
<h3><a class="NoHi" id="Sec5">Section 5: Files relevant for global behavior of analysis functionality</a></h3>
<hr style="width: 100%; height: 2px;"></hr> (<a href="inputfiles.html#TOP">back to top</a>)<br/>
<h4><a class="NoHi" id="FMCSC_ANGRPFILE">Analysis group input file:</a></h4>
Keyword: <b><a href="keywords.html#ANGRPFILE">FMCSC_ANGRPFILE</a></b><br/>
<br/>
This input file allows the user to override the default parsing the software does for
analysis purposes where data for molecules of identical type are often pooled
(see <a href="outputfiles.html">documentation of output files</a>). In addition, it allows
tagging of analysis groups as solvent which has consequences for some of the analyses.
The first letter in the file indicates the mode:<br/>
<br/>
'T' or 't':<br/>
(molecule type-based selection):<br/>
Obviously, it is impossible to parse finer than by molecule type using this mode.
The sole purpose here is solvent-tagging (by default, all single-residue molecules
are tagged as solvent molecules):<br/>
----------------<br/>
<span style="font-style: italic;">XXX<br/>
----------------<br/></span>
XXX is a positive integer (solute) or negative integer (solvent) and the row number corresponds to the molecule type number
by sequence of occurrence (see <a href="inputfiles.html#FMCSC_SEQFILE">SEQFILE</a>). Note that it is not 
possible to pool molecules of different type into the same analysis group, and that the
actual group numbers provided in this file are irrelevant (beyond their sign).<br/>
<br/>
'M' or 'm':<br/>
The format is identical but row numbers now indicate the molecule number instead of molecule
type number (again, by sequence of occurrence). This format allows breaking of molecules of the same type into separate analysis
groups which is useful (and required from a statistical mechanics point of view) anytime one or more
of the molecules of a specific type are made unique by the application of restraint or other bias
potentials, by an asymmetric system setup, <i>etc</i> ... Furthermore, this can be useful for 
error analyses even if the molecules are rigorously identical.
The number provided by XXX will - as before - indicate solute/solvent-identity via its sign but
each desired unique group now requires also a unique integer (at least as many as there are molecule types).<br/>
<br/>
<br/>
<h4><a class="NoHi" id="Frameindexinput">Input files to provide snapshot index sets for trajectory analysis mode:<br/></a></h4>
Keyword: <b><a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a></b><br/>
<br/>
Prerequisite: Trajectory analysis mode &rarr; <a href="keywords.html#PDBANALYZE">FMCSC_PDBANALYZE</a><br/><br/>
This input file allows analysis runs to operate only on an arbitrary subset of an input trajectory. In addition,
it allows the user to provide floating-point weights for each snapshot in the trajectory. This can be useful
if a trajectory is analyzed, which was obtained using methodology that does not yield a well-defined statistical 
ensemble. In such a case, it is common to apply reweighting techniques of the weighted histogram variety that 
may yield per-snapshot weights. In order for the analysis features of CAMPARI to still be useful, it will then
be necessary to deviate from the assumed unweighted (Boltzmann) averaging.<br/>
Therefore, this file accepts two alternative formats (per row). If a single integer is specified, the corresponding frame
(numbered sequentially in the input trajectory and starting at 1) is added to the list of frames to analyze with a weight of unity. If 
an integer and a real number are provided, the frame specified by the integer is added to the list of frames to analyze with
a weight given by the real number. If a file format is used that allows nonsequential access of trajectory snapshots,
and if floating point weights are absent, the list thus obtained is processed "as is", which allows reordering of trajectories and
the use of duplicate frames. In all other cases, the list is first ordered, and duplicate frames are removed 
(see <a href="keywords.html#FRAMESFILE">elsewhere</a> for details on this). <br/>
<br/>
<br/>
<br/>
<h3><a class="NoHi" id="Sec6">Section 6: Files relevant to specific analysis routines</a></h3>
<hr style="width: 100%; height: 2px;"></hr> (<a href="inputfiles.html#TOP">back to top</a>)<br/>
<h4><a class="NoHi" id="FMCSC_BBSEGFILE">Backbone segment distribution datafile:</a></h4>
Keyword: <b><a href="keywords.html#BBSEGFILE">FMCSC_BBSEGFILE</a></b><br/>
<br/>
Required for: Analysis of &phi;/&psi;-based backbone segment distributions &rarr; <a href="keywords.html#SEGCALC">FMCSC_SEGCALC</a><br/><br/>
A36x36 (<i>i.e.</i>, 10 degree-spacing) &#934;/&#936;-map indicating regions by
integer codes that (can) form regular polypeptide secondary structure.<br/>
<br/>
&nbsp;&nbsp;1 = &#946; <br/>
&nbsp;&nbsp;2 = P<sub>II</sub> (Polyproline Type II Helix)<br/>
&nbsp;&nbsp;3 = Unusual Region ("Pass")<br/>
&nbsp;&nbsp;4 = &#945;<sub>R</sub> (Right-Handed &#945;-Helix)<br/>
&nbsp;&nbsp;5 = Inverse C<sub>7</sub> Equatorial (&#947;'-Turn)<br/>
&nbsp;&nbsp;6 = Classic C<sub>7</sub> Equatorial (&#947;-Turn)<br/>
&nbsp;&nbsp;7 = Unusual Region (Helix with 7 Residues per Turn)<br/>
&nbsp;&nbsp;8 = &#945;<sub>L</sub> (Left-Handed &#945;-Helix)<br/>
<br/>
The file reproduces the visual picture a Ramachandran map gives for L-polypeptides and is
read in accordingly. There are two default versions ("bbseg.dat" and
"bbseg2.dat") in the data directory. The first one is a little tighter
in its definitions (in particular P<sub>II</sub> and &#945;<sub>R</sub>) while the second one
is more lenient.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_TRACEFILE">Input file with trace of parallel simulation run for analysis in post-processing:</a></h4>
Keyword: <b><a href="keywords.html#TRACEFILE">FMCSC_TRACEFILE</a></b><br/>
<br/>
Required for: (Un)scrambling of trajectories in parallel <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a>
on sets of trajectories generated for example by replica exchange simulations &rarr; <a href="keywords.html#REMC">FMCSC_REMC</a><br/>
Alternatively required for: Automatic adjustment of conformational space network in analyses relying on 
<a href="keywords.html#CCOLLECT">structural clustering</a> to reflect reseeding history of a <a href="keywords.html#MPI_PIGS">PIGS calculation</a><br/><br/>
Any number of lines giving a total of N<sub>nodes</sub>+1 integers, where N<sub>nodes</sub> is the number of different 
<a href="keywords.html#REPLICAS">replicas</a>. The numbers in the first column have to be increasing monotonously, and the
number in the remaining columns have to be such that integers are bound to the interval from 1 to N<sub>nodes</sub>.
In addition, if the run is <a href="keywords.html#REMC">meant to unscramble a replica exchange trajectory</a>,
every integer in the interval from 1 to N<sub>nodes</sub> has to occur exactly once on every line. A failure to comply
with this requirement will cause CAMPARI (MPI) to crash or hang.<br/>
<br/>
In parallel trajectory analysis runs, this file is interpreted as a running map of some initial conformation with a set of conditions based on the assumption
that swaps have occurred in the process (simulation) generating the <a href="inputfiles.html#FMCSC_PDBFILE">input trajectories</a>. 
Then, the desired outcome of using this functionality is to reorganize the set of trajectory data such that 
we obtain trajectories that are continuous in geometry. The most common application for this would be to "unscramble"
replica exchange trajectories.<br/>
The provided step numbers (first column) are related to the frames in 
the input trajectories via keywords <a href="keywords.html#RE_TRAJSKIP">FMCSC_RE_TRAJSKIP</a> and <a href="keywords.html#RE_TRAJOUT">FMCSC_RE_TRAJOUT</a>.
The assumption is that for a given step, swaps have occurred first, then trajectory files were appended. In order for the 
procedure to work as intended, the current map for every snapshot in the input trajectories has to be present. It does not matter if 
there is more information, or whether the step numbers are exactly matched with <a href="keywords.html#RE_TRAJOUT">FMCSC_RE_TRAJOUT</a>.<br/>
<br/>
<u>Example:</u><br/>
<br/>
<big><code>
30  4 3 2 1<br/>
60  1 3 2 4<br/>
90  1 2 3 4<br/>
120 1 4 2 3<br/>
</code></big><br/><br/>
Let us furthermore assume that <a href="keywords.html#RE_TRAJSKIP">FMCSC_RE_TRAJSKIP</a> is 0, <a href="keywords.html#RE_TRAJOUT">FMCSC_RE_TRAJOUT</a> is 50,
and there are 10 frames in the trajectories. The <a href="keywords.html#REPLICAS">number of replicas</a> obviously has to be 4.
Then, the above file implies that for the first snapshot, CAMPARI assumes a mapping of "4 3 2 1" meaning that the first snapshot in the 
trajectory starting with "N_000_" (corresponding to the first replica), will end up being transferred to (and eventually analyzed and/or written by) 
the fourth replica. This is because the mapping current at the 30th step is, based on the file, the only one CAMPARI can assume current
also for the 50th step, <i>i.e.</i>, the first snapshot in the trajectories. By the same logic, the second snapshot would use the mapping "1 2 3 4"
(from step 90), and all further snapshots would use the mapping "1 4 2 3" (from step 120).
This implies that the second line in the input file is redundant (but does no harm). If a snapshot is computed to correspond to a step
number smaller than any step number listed in this file, the default mapping is assumed (here,"1 2 3 4").<br/>
In serial trajectory analysis runs or parallel trajectory analysis runs in <a href="keywords.html#MPIAVG">MPI averaging mode</a>,
this input file is relevant only if <a href="keywords.html#CCOLLECT">structural clustering</a> is used. It is then used 
to correct the snapshot-to-snapshot connectivity map based on the knowledge of the 
history of reseeding protocol, <i>e.g.</i> from a <a href="keywords.html#MPI_PIGS">PIGS calculation</a>.
Thus, the interpretation of the input is exactly as described for the <a href="outputfiles.html#N_000_PIGSTRACE.dat">PIGS output trace</a>. This
differs from the previous example in that the map in columns 2 to N<sub>nodes</sub>+1 simply indicates a reseeding by reference to the 
index of the replica (the same integer can occur multiple times).<br/>
<br/>
<u>Example:</u><br/>
<br/>
<big><code>
30  1 2 3 4<br/>
60  1 2 3 4<br/>
90  3 3 3 4<br/>
120 1 2 3 4<br/>
</code></big><br/><br/>
If we assume that <a href="keywords.html#RE_TRAJSKIP">FMCSC_RE_TRAJSKIP</a> is 0, <a href="keywords.html#RE_TRAJOUT">FMCSC_RE_TRAJOUT</a> is 1, 
<a href="keywords.html#RE_TRAJTOTAL">FMCSC_RE_TRAJTOTAL</a> is 125, and <a href="keywords.html#NRSTEPS">FMCSC_NRSTEPS</a> is 500
(4 concatenated trajectories of 125 snapshots each &rarr; <a href="keywords.html#TRACEFILE">FMCSC_TRACEFILE</a>
for details), then the only two reseeding events are seen at step 90 of the parallel simulation. Specifically, after completion of step 90 and writing to
trajectories, the snapshots in the first and second replicas at replaced with that of the third replica at step 90
(see <a href="outputfiles.html#N_000_PIGSTRACE.dat">elsewhere</a> for details). Unlike in the prior example, the line at step #120 does <i>not</i>
imply any change in structure. Thus, the input file in this example will modify the extracted transition network of structural clusters as follows.<br/>
<ol>
<li> The (directional) transition counts between clusters connected by snapshots 125 and 126, snapshots 250 and 251, and snapshots 
375 and 376 will be decremented by 1 (concatenation points).</li>
<li> The (directional) transition counts between clusters connected by snapshots 90 and 91 and snapshots 215 and 216 will be decremented by 1 (reseedings).</li>
<li> The (directional) transition counts between clusters connected by snapshots 340 and 91 and snapshots 340 and 216 will be incremented by 1 (reseedings).</li>
</ol>
While the above example deals with the case where all snapshots are preserved in the trajectory, this is not usually the case. Subsampled 
or otherwise mismatched settings for keywords <a href="keywords.html#REFREQ">FMCSC_REFREQ</a>, <a href="keywords.html#XYZOUT">FMCSC_XYZOUT</a>, and
<a href="keywords.html#EQUIL">FMCSC_EQUIL</a> can easily create a scenario where the reseeding information and the concatenated 
snapshots are not overlapping at all. In this situation, CAMPARI will use the trace and the derived map of simulation snapshots
(that also relies on <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a> and <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>)
to identify suitable alternatives for the addition and removal of links, which is considerably more difficult for the former. These links
should in all case be geometrically continuous albeit with a time spacing that is not necessarily the expected one.
Particularly opaque cases arise if a <a href="keywords.html#FRAMESFILE">file with selected trajectory frames</a> is used, or if a value larger
than 1 is chosen for <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>. Both scenarios pose additional restrictions but are expected to be
resolved correctly whenever possible. The above strategy will work for single continuous PIGS (or similar) type of simulation but not for
concatenations of arbitrary trajectories or multiple PIGS runs. In this case, keywords <a href="keywords.html#TRAJBREAKSFILE">FMCSC_TRAJBREAKSFILE</a>
and <a href="keywords.html#TRAJLINKSFILE">FMCSC_TRAJLINKSFILE</a> can be used to achieve - albeit in a more tedious manner - the same effect.<br/>
Suitable input files for this keyword (FMCSC_TRACEFILE) are created by CAMPARI itself during both replica exchange runs
(keyword <a href="keywords.html#RETRACE">FMCSC_RETRACE</a>
enables the generation of output file <a href="outputfiles.html#N_000_REXTRACE.dat">N_000_REXTRACE.dat</a>) and PIGS runs (the same keyword enables the
generation of output file <a href="outputfiles.html#N_000_PIGSRACE.dat">N_000_PIGSTRACE.dat</a>). Note that it is quite tedious to map step 
numbers of an original parallel simulation run to the analysis of a single concatenated trajectory. For the read-in of the trace to function
correctly, keywords <a href="keywords.html#RE_TRAJTOTAL">FMCSC_RE_TRAJTOTAL</a> (only for PIGS), <a href="keywords.html#RE_TRAJSKIP">FMCSC_RE_TRAJSKIP</a>, and
<a href="keywords.html#RE_TRAJOUT">FMCSC_RE_TRAJOUT</a> are all required. Note that this mapping to original step numbers
is lost if a <a href="keywords.html#FRAMESFILE">file with trajectory frames</a> in random access mode is used (here, instead
the step numbers become the line numbers of this file). This option (or the use of <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a>
in general) is only supported in serial analysis run (<i>i.e.</i> not using <a href="keywords.html#MPIAVG">MPI parallel analysis</a>).<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_PCCODEFILE">Requests for general PC analysis input file:</a></h4>
Keyword: <b><a href="keywords.html#PCCODEFILE">FMCSC_PCCODEFILE</a></b><br/>
<br/>
Required for: Specific interatomic distance distribution (or pair correlation) functions or instantaneous output thereof &rarr; <a href="keywords.html#PCCALC">FMCSC_PCCALC</a> and <a href="keywords.html#INSTGPC">FMCSC_INSTGPC</a><br/><br/>
Generalized pair correlations / distance distributions (or arbitrary application of tabulated
potentials &rarr; <a href="inputfiles.html#FMCSC_TABCODEFILE">TABCODEFILE</a>) in systems with many atoms can be
quite complicated and this input file tries to minimize the effort
needed on the user-end to reward as diverse requests as possible. The
file works in either of currently three modes listed below. Note that
the input format also somewhat corresponds to the memory structure used
to store the request.<br/>
<br/>
<b>Mode 1:</b><br/>
<br/>
The first line specifies the mode and the number of listed specific
atom-atom entries. These entries consist of the numbers of two atoms and
with the integer code for the GPC component to which this atom-atom
distance distribution should contribute. This specific list format is
trivial to master but very limited when lots of analogous requests are
wanted. It is probably best used in highly specific systems (such as a
single polypeptide).<br/>
<br/>
<u>Example:</u><br/>
<br/>
<i>Suppose you study a peptide with three lysine residues and want to know
the distribution of the three neutralizing counterions (atoms 301, 302,
303 in the system). Suppose the lysine sidechain nitrogens are atoms
27, 48, and 155, then the appropriate input file is as follows:</i><br/>
<br/>
------------------------<br/>
<big><code>1&nbsp; 9<br/>
27&nbsp; 301&nbsp; 1<br/>
27&nbsp; 302&nbsp; 1<br/>
27&nbsp; 303&nbsp; 1<br/>
48&nbsp; 301&nbsp; 1<br/>
48&nbsp; 302&nbsp; 1<br/>
48&nbsp; 303&nbsp; 1<br/>
155&nbsp; 301&nbsp; 1<br/>
155&nbsp; 302&nbsp; 1<br/>
155&nbsp; 303&nbsp; 1</code></big><br/>
------------------------<br/>
<br/>
<b>Mode 2:</b><br/>
<br/>
Similar to mode 1, only that the list-format is interpreted to be a
request to treat all molecules in the <a href="inputfiles.html#FMCSC_ANGRPFILE">same analysis group</a> analogously. Atom
numbering must now refer to the first molecule in its group if the two molecules
are different. If the atoms belong to the same molecule and the
molecule is the first in its group, then the request is honored as an
intramolecular distance distribution. To request intermolecular
distance distributions for molecules in the same analysis group, the numbers
should correspond to the first and second instance of this molecule type
in the group. All other requests are ignored.<br/>
<br/>
<u>Example:</u><br/>
<br/>
<i>Take the same example, only that there is a high concentration of
background salt. Using mode 1, the file with the same request would
become fairly long. Also suppose that you want to know the
chloride-chloride distribution function, and the sodium-chloride and
sodium-sodium distribution functions. Let us assume the first two sodium
atoms are 401 and 402 and that analysis groups correspond to molecule types (the default).
Then, this is the appropriate input file (note
it is actually shorter and now requests four unique distance
distributions):</i><br/>
<br/>
-----------------------<br/>
<big><code>2&nbsp; 6<br/>
27&nbsp; 301&nbsp; 1<br/>
48&nbsp; 301&nbsp; 1<br/>
155&nbsp; 301&nbsp; 1<br/>
301&nbsp; 302&nbsp; 2<br/>
301&nbsp; 401&nbsp; 3<br/>
401&nbsp; 402&nbsp; 4</code></big><br/>
-----------------------<br/>
<br/>
<b>Mode 3:</b><br/>
<br/>
This is a slightly different mode which honors requests in form of
atom-atom matrices. The main disadvantages of this input format are that
its sanity can easily be corrupted and that large molecules
are difficult to handle. The idea is for the user to supply a list of
analysis group-to-analysis group requests using header lines of "<i>T i j</i>"
(intermolecular between types <i>i</i> and <i>j</i>) or "<i>t i</i>" (intramolecular
within type <i>i</i>. The main advantage is that now explicit referencing to
atom numbers is needed and that it can accomplish fairly complicated
requests in a very compact format. This is best explained in an example.<br/>
<br/>
<u>Example:</u><br/>
<br/>
<i>Suppose you have a mix of SPC water (analysis group #1: 3 atoms) and urea (analysis group
#2: 8 atoms) and want to know a variety of possible site-site
distribution functions to characterize solution structure. Then an appropriate
input file would look as follows:</i><br/>
<br/>
--------------------------------------------------<br/>
<big><code>3<br/>
T&nbsp; 1&nbsp; 2<br/>
1&nbsp; 0&nbsp; 2&nbsp; 2&nbsp; 3&nbsp; 3&nbsp; 3&nbsp; 3<br/>
0&nbsp; 4&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0<br/>
0&nbsp; 4&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0<br/>
T&nbsp; 1&nbsp; 1<br/>
5&nbsp; 6&nbsp; 6<br/>
0&nbsp; 7&nbsp; 7<br/>
0&nbsp; 0&nbsp; 7<br/>
T&nbsp; 2&nbsp; 2<br/>
0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0<br/>
0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 8&nbsp; 8&nbsp; 8&nbsp; 8<br/>
0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 9&nbsp; 9&nbsp; 9&nbsp; 9<br/>
0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 9&nbsp; 9&nbsp; 9&nbsp; 9<br/>
0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0<br/>
0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0<br/>
0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0<br/>
0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0&nbsp; 0</code></big><br/>
--------------------------------------------------<br/>
<br/>
<i>Note that the first matrix is asymmetric (first type always determines
number of rows). Here we request to measure the water O to urea C (#1),
N (#2), and H (#3), and the water H to urea O (#4) distribution
functions. The other two matrices are intermolecular for molecules in the same
group and are by definition symmetric. Note that only the upper
right half is read; the numbers must all be there, however. We request all
possible water-water correlation functions (O-O, O-H, and H-H in #5,
#6, and #7) and the O-H (#8) and N-H (#9) distribution functions for
urea-urea. Of course it is still necessary to know the order of atoms
within a molecule type.</i><br/>
<br/>
<b>Notes:</b><br/>
<br/>
Inconsistent numbering in the components is removed and
announced through log-output (<i>i.e.</i>, components might be renumbered).
Nothing stops a user from supplying illogical requests (<i>i.e.</i>, for
example to omit some of the 9's in the urea-urea matrix in the
last example) since the code makes no assumptions about molecular symmetry.
Furthermore, in particular for modes 1 and 2, nothing stops the user
form supplying an atom pair more than once which can give rise to
arbitrary weights when several pairs are pooled together. This is
potentially very misleading and should be kept in mind when
encountering surprising results. It is worthwhile to keep in mind that modes 2 and 3 can easily encompass a 
very large number of distances, which can make the analysis costly 
and <a href="outputfiles.html#GENERAL_DIS.dat">instantaneous output</a> very large.
Also remember that <a href="inputfiles.html#FMCSC_ANGRPFILE">analysis groups</a>
can split a molecule type into multiple groups but never combine multiple
types into a single group, which is why the above input modes 2 and 3 continue to work 
in all circumstances. 
Lastly, CAMPARI will prohibit pooling
data from inter- and intramolecular distances into the same component. This
is because intermolecular distances are directly transformed to pair correlation
functions whereas intramolecular ones are not.
In any event, it is always recommended to double-check this input file.<br/>
The related functionality of <a href="inputfiles.html#FMCSC_TABCODEFILE">reading in a map for tabulated potentials</a>
is almost identical to the one for generalized distance distribution analyses. The major differences
lies in the fact that tabulated potentials do not respect analysis groups, <i>i.e.</i>,
they will always assume molecule type partitioning in modes 2 and 3.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_BESSELFILE">Input file for tabulated Bessel functions:<br/></a></h4>
Keyword: <b><a href="keywords.html#BESSELFILE">FMCSC_BESSELFILE</a></b><br/>
<br/>
Required for: Computation of predicted fiber diffraction patterns (preliminary implementation) &rarr; <a href="keywords.html#DIFFRCALC">FMCSC_DIFFRCALC</a><br/>
<br/>
This file should have a header line with two integers and a floating
point number specifying the number of Bessel functions (order, "N"),
the number of distance bins ("M"), and the distance resolution the
actual file then has N columns and M rows of floating point numbers to
supply the tabulated Bessel functions. Note that the first distance
bin is assumed to be exactly zero. Also note that this file is very
easily corrupted and that its sanity has to be ensured by the user. The
file delivered with CAMPARI is called "BesselFunctions.dat" in the
data/ directory and tabulates Bessel functions up to an order of 500
for 2001 distance bins. The latter corresponds to a maximum argument
value of 500.0, as the distance resolution is set to 0.25. Finally,
note that the whole file is read into memory. The file size will
therefore be a good indicator of what the memory demand imposed by
diffraction calculations in addition to the underlying calculation is.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="Atomindexinput">Input files to provide atomic index sets:<br/></a></h4>
Keywords: <b><a href="keywords.html#ALIGNFILE">FMCSC_ALIGNFILE</a></b>, <b><a href="keywords.html#CFILE">FMCSC_CFILE</a></b> (see <a href="keywords.html#CDISTANCE">FMCSC_CDISTANCE</a> for details), <b><a href="keywords.html#SAVATOMFILE">FMCSC_SAVATOMFILE</a></b>, and <b><a href="keywords.html#TRAJIDXFILE">FMCSC_TRAJIDXFILE</a></b><br/>
<br/>
This is the simplest input file format CAMPARI uses and consists of nothing
but a number of lines specifying a single atomic index (assuming CAMPARI's
intrinsic ordering &rarr; be careful with <a href="keywords.html#PDBNUCMODE">PDBNUCMODE</a>)
on each line. N lines will therefore constitute an index set of size N
assuming all entries are valid (legal indices). In some cases, such an index file
may be obtained by running an appropriate script over a CAMPARI-generated
pdb file. Atom index sets are used in a variety of contexts.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="Atompairindexinput">Input files to provide atom pair index sets:<br/></a></h4>
Keyword: <b><a href="keywords.html#CFILE">FMCSC_CFILE</a></b><br/>
<br/>
This input file type is very similar to <a href="inputfiles.html#Atomindexinput">atomic index input files</a>.
Here, instead of providing a single index entry on each line denoting an individual atom, instead a pair
of indices needs to provided to specify a pair of atoms. This is currently only used in certain
modes of structural clustering (see <a href="keywords.html#CDISTANCE">FMCSC_CDISTANCE</a> and
<a href="keywords.html#CFILE">FMCSC_CFILE</a> for details).
<br/>
<br/>
<u>Example:</u><br/>
----------------<br/>
<big><code>
2 &nbsp; 4<br/>
3 &nbsp; 12</code></big><br/>
----------------<br/>
<br/>
The above input file would define two pairs of atoms with indices of 2/4 and 3/12, respectively.
This could for example be a set of (two) interatomic distances to be used for structural clustering.
Note that double entries (entry order is ignored) are removed.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FYCindexinput">Input files to provide torsional index sets:<br/></a></h4>
Keyword: <b><a href="keywords.html#CFILE">FMCSC_CFILE</a></b><br/>
<br/>
This input file type is technically identical to <a href="inputfiles.html#Atomindexinput">atomic index input files</a>.
The only difference is that here a list of system dihedral angle indices is provided. This is currently
only used in structural clustering (see <a href="keywords.html#CCOLLECT">elsewhere</a> for details). The intrinsic
numbering of dihedral angles is most easily extracted as follows:</br>
It is recommended to write a temporary input file contain all integers from 1 to <i>N<sub>dih</sub></i>, where
<i>N<sub>dih</sub></i> is the total number of eligible dihedral angles in the system. If <i>N<sub>dih</sub></i> is not
known, any large enough number will suffice. By supplying this temporary file as input to a CAMPARI run doing
structural clustering with a <a href="keywords.html#CDISTANCE">distance criterion</a> based on dihedral angles on
the system of interest, CAMPARI will upon execution print the list of all selected dihedral angles (<i>i.e.</i>, all dihedral angles)
at the beginning of the run. This list can be saved and used to select the dihedral angles much more easily than by
counting them off.<br/>
In general, the order of residues follows the <a href="inputfiles.html#FMCSC_SEQFILE">sequence input file</a>. For each
residue, possible &omega;-angles come first and are followed by the &phi;-, &psi;-, nucleic acid backbone and sidechain
&chi;-angles. The first dihedral angle of any 5-membered ring pucker (the &phi;-angle
in proline and similar polypeptide residues and the dihedral angle along the C3*-C4* bond in polynucleotides) is the
only pucker degree of freedom to be included. In special cases, the sidechain &chi;-angles are followed by additional entries.
If the system contains disulfide bridges, the C<sub>&alpha;</sub>-C<sub>&beta;</sub>-S-S angle is next. For the first
(in terms of sequence) of the two residues of a disulfide bond, and additional entry is reserved for the C<sub>&beta;</sub>-S-S-C<sub>&beta;</sub> angle.
Lastly, the user may request unsupported (nonnative) dihedral angles to be part of the set. This requires setting
keyword <a href="keywords.html#TMD_UNKMODE">FMCSC_TMD_UNKMODE</a> to values larger than 0. The interpretation of which
dihedral angles to make eligible is exactly as described for that keyword. The unsupported dihedral angles are always listed last
and follow the order of Z matrix lines (atoms). The corresponding Z matrix lines are printed out in the initial summary 
and are also shown in output file <a href="outputfiles.html#FYC.dat">FYC.dat</a> when using the alternative output mode
by supplying a negative value for keyword <a href="keywords.html#TOROUT">FMCSC_TOROUT</a>.
<br/>
<br/>
<br/>
<u>Example:</u><br/>
----------------<br/>
Suppose the sequence input file specifies proline dipeptide (Ace-Pro-Nme) as the first and
only molecule in the system. We assume <a href="keywords.html#TMD_UNKMODE">FMCSC_TMD_UNKMODE</a> is
set to 3. Following the procedure outlined above for understanding the numbering
requires all integers from 1 to 6 to be present in the input file, <i>e.g.</i>:<br/><br/>
<big><code>
1<br/>
2<br/>
3<br/>
4<br/>
5<br/>
6</code></big><br/>
----------------<br/>
<br/>
This will give the following output:<br/><br/>
<big><code>
#     1 is     UNS (         4) angle of residue #       1 (ACE)<br/>
#     2 is   OMEGA   angle of residue #       2 (PRO)<br/>
#     3 is     PHI   angle of residue #       2 (PRO)<br/>
#     4 is     PSI   angle of residue #       2 (PRO)<br/>
#     5 is   OMEGA   angle of residue #       3 (NME)<br/>
#     6 is     UNS (        24) angle of residue #       3 (NME)<br/>
</code></big><br/>
----------------<br/>
<br/>
The two "UNS" dihedral angles are the (generally uninteresting) methyl spins in both capping groups.
As an example, an input file targeting only the &phi;- and &psi;-angles of the proline residue
would require just the two entries "3" and "4" to be present. Note that the numbering will generally change
on account of changes to the sequence, to <a href="keywords.html#TMD_UNKMODE">FMCSC_TMD_UNKMODE</a>, or to the 
presence of disulfide bonds. It is therefore recommended to always inspect the initial portion of what CAMPARI 
prints out in order to avoid errors caused by files that were not updated.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_CLUFILE">Input file to provide a coarse-grained (symbolic) rajectory:<br/></a></h4>
Keyword: <b><a href="keywords.html#CLUFILE">FMCSC_CLUFILE</a></b><br/>
<br/>
Required for: Using file-based clustering, <i>i.e.</i>, setting keyword <a href="keywords.html#CMODE">FMCSC_CMODE</a>
to either 6 or 7. This input file provides the symbolic trajectory itself as a list of integers.<br/>
<br/>
Symbolic trajectories as written by CAMPARI itself are an appropriate input file format
(see documentation on <a href="outputfiles.html#STRUCT_CLUSTERING.clu">STRUCT_CLUSTERING.clu</a>). The column to
be used is selected by keyword <a href="keywords.html#CLUFILECOL">FMCSC_CLUFILECOL</a>. Note that clusters
will be renumbered automatically if CAMPARI detects that the numbering in the input file is not
continuous. All lines of the input file are read regardless of the settings for <a href="keywords.html#NEQUIL">FMCSC_NEQUIL</a>
or <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>. In general, the input will be processed correctly
if it is analyzed with the same settings as it was created. This is because the snapshot connectivity
map (and all related input files) and the data read-in (if any) continue to work exactly as before in
file-based clustering.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_TRAJBREAKSFILE">Input file to provide trajectory breaks:<br/></a></h4>
Keyword: <b><a href="keywords.html#TRAJBREAKSFILE">FMCSC_TRAJBREAKSFILE</a></b><br/>
<br/>
Required for: Avoiding that every microstate transition in the trajectory is counted toward output and analysis in
the inferred graph (transition matrix) printed to output file
<a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a>. By extension,
this is relevant for all functionalities relying on this information (as summarized
<a href="keywords.html#TMAT_MD">elsewhere</a>) and their corresponding outputs. It also affects (independently)
<a href="outputfiles.html#PROGIDX_xxxxxxxxxxxx.dat">the results from the progress index method</a>.
Naturally, this is is irrelevant for any simulation not using <a href="keywords.html#CCOLLECT">clustering and related methods</a>.<br/>
<br/>
A trajectory usually is a series of continuous snapshots of a system. The implied continuity is used to infer
the system's properties in terms of transitions, pathways, and dynamics. This is often done using network-based
methods (most prominently in Markov state models). The connectivity of such a network or graph is defined by the edges
and their probabilities, which are most often inferred from the trajectory itself. However, not all transitions may be
qualitatively similar, <i>e.g.</i>, the swaps in <a href="keywords.html#REMC">replica exchange calculations</a> differ
from the incremental evolution produced by the basic sampling engine. As another example, trajectories may be concatenated
in <a href="keywords.html#MPIAVG">certain parallel runs</a> or in <a href="keywords.html#PDB_ANALYZE">trajectory analysis mode</a>.
This input file can be used to eliminate such apparent transitions before the corresponding analyses are performed.
Input file <a href="inputfiles.html#FMCSC_TRAJLINKSFILE">FMCSC_TRAJLINKSFILE</a> provides the complementary facility of
adding missing transitions.<br/>
<br/>
<br/>
<u>Example:</u><br/>
----------------<br/>
<big><code>
10000<br/>
20000<br/>
30000<br/>
40000<br/>
50000</code></big><br/>
----------------<br/>
<br/>
This file would tell CAMPARI that the transitions from the 10000<sup>th</sup> to the 10001<sup>st</sup> step, from
the 20000<sup>th</sup> to the 20001<sup>st</sup> step, and so on, are all to be excluded from relevant analyses.
Note that this input <b>always</b> refers to the basic and complete simulation length defined via <a href="keywords.html#NRSTEPS">FMCSC_NRSTEPS</a>.
This means that the numbers must <b>not</b> be adjusted to reflect settings for <a href="keywords.html#EQUIL">FMCSC_EQUIL</a>,
<a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>, <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a>, or similar keywords.
The use of a frames file should be handled with particular care. If the file accesses the trajectory in random access ("as is") mode,
all step numbers are assumed to refer to the line number in the frames file rather than the index of the frame on that line. This is 
a general change of interpretation inherent to <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a> with certain 
<a href="keywords.html#PDB_FORMAT">input file formats</a>. Conversely, if the file accesses the trajectory in strictly sequential
mode, step numbers continue to refer to the <i>original</i> trajectory. In a parallel setting, this can apply separately and 
identically to the individual trajectories (<a href="keywords.html#REMC">RE mode</a>) or globally to the concatenated
trajectory (<a href="keywords.html#MPIAVG">MPI averaging mode</a>).<br/>
The actual coarse-grained trajectory reported in
<a href="outputfiles.html#STRUCT_CLUSTERING.clu">STRUCT_CLUSTERING.clu</a> is naturally only printed for snapshots
that ended up in the set to be analyzed, which
means that any effective transition spanning one or more excluded step-to-step transitions is also excluded.
Note that the removal of transitions can of course fracture the network, which may render some analyses
impossible to perform or difficult to interpret. Lastly, users should be warned that a large value for
<a href="keywords.html#NRSTEPS">NRSTEPS</a> will lead to a large memory allocation when reading the file
(irrespective of the file's actual size).<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_TRAJLINKSFILE">Input file to provide additional trajectory links:<br/></a></h4>
Keyword: <b><a href="keywords.html#TRAJLINKSFILE">FMCSC_TRAJLINKSFILE</a></b><br/>
<br/>
Required for: Adding snapshot-based microstate transitions to an input trajectory to modify the
inferred graph (transition matrix) printed to output file
<a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a>. By extension,
this is relevant for all functionalities relying on this information (as summarized
<a href="keywords.html#TMAT_MD">elsewhere</a>) and their corresponding outputs. It also affects (independently)
<a href="outputfiles.html#PROGIDX_xxxxxxxxxxxx.dat">the results from the progress index method</a>.
Naturally, this is is irrelevant for any simulation not using <a href="keywords.html#CCOLLECT">clustering and related methods</a>.<br/>
<br/>
A single trajectory is a series of continuous snapshots of a system. The implied continuity is used to infer
the system's properties in terms of transitions, pathways, and dynamics. This is often done using network-based
methods (most prominently in Markov state models). The connectivity of such a network or graph is defined by the edges
and their probabilities. Networks are often constructed from sets of many trajectories
whose start and end points may or may not be related. For analysis in CAMPARI, they are generally concatenated.
It is impossible to infer the correct connectivity purely from the sequence of snapshots in the concatenated trajectory.
As a complement to <a href="inputfiles.html#FMCSC_TRAJBREAKSFILE">FMCSC_TRAJBREAKSFILE</a>, this input file
allows the user to manually specify missing transitions between snapshots. These occur, for example, in 
cases where several trajectories are branched off the same point and subsequently concatenated (as in
the analysis of a <a href="keywords.html#MPI_PIGS">PIGS simulation</a>).<br/>
The first letter in the file indicates the mode:<br/><br/>
<b>'Y' or 'y'</b>: (attempted remapping of links requested)<br/>
<b>'N' or 'n'</b>: (only exactly matched links are kept)
<br/>
<br/>
<u>Example:</u><br/>
----------------<br/>
<big><code>
N<br/>
1000 2001<br/>
1000 3001<br/>
1000 4001<br/>
1000 5001</code></big><br/>
----------------<br/>
<br/>
This file would tell CAMPARI that the 1000<sup>th</sup> snapshot in the (concatenated) input trajectory not
only directly precedes the 1001<sup>st</sup> snapshot but also to the 2001<sup>st</sup>, 3001<sup>st</sup>,
4001<sup>st</sup>, and 5001<sup>st</sup> snapshots. Duplicates are not allowed. For every entry, a single
integer count is added to the transition count matrix element corresponding to the transition from the cluster
containing the snapshot in the first column to the cluster containing the snapshot in the second column
(order therefore matters). 
Note that this input <b>always</b> refers to the basic and complete simulation length defined via <a href="keywords.html#NRSTEPS">FMCSC_NRSTEPS</a>.
This means that the numbers must <b>not</b> be adjusted to reflect settings for <a href="keywords.html#EQUIL">FMCSC_EQUIL</a>,
<a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>, <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a>, or similar keywords.
The use of a frames file should be handled with particular care. If the file accesses the trajectory in random access ("as is") mode,
all step numbers are assumed to refer to the line number in the frames file rather than the index of the frame on that line. This is 
a general change of interpretation inherent to <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a> with certain 
<a href="keywords.html#PDB_FORMAT">input file formats</a>. Conversely, if the file accesses the trajectory in strictly sequential
mode, step numbers continue to refer to the <i>original</i> trajectory.<br/>
In the example, the mode of not remapping snapshots was chosen. This means that, if any of the snapshots in question is missing
because of the aforementioned keywords, that particular link will not be used (a warning is provided).
If the mode is 'Y' instead, CAMPARI will try to remap these snapshots to those that are available and nearest in time (sequence)
in the original trajectory. Specifically, the first ones before and after are used for the first and second columns of the input
file, respectively. In addition, there will be a check against added and destroyed links based on
<a href="keywords.html#TRACEFILE">FMCSC_TRACEFILE</a> and <a href="inputfiles.html#FMCSC_TRAJBREAKSFILE">FMCSC_TRAJBREAKSFILE</a>.
If CAMPARI detects that there is no longer a "safe" connection between snapshots surrounding the unmapped entry,
it will discard the link in question. Because the stacked use of all these modifications gets confusing quite easily,
a snapshot-based link map will be created (including warnings) before the actual execution begins.
Keyword <a href="keywords.html#CADDLINKMODE">FMCSC_CADDLINKMODE</a> also provides somewhat overlapping functionality,
but this is not at the snapshot level. Lastly, users should be warned that a large value for
<a href="keywords.html#NRSTEPS">NRSTEPS</a> will lead to a large memory allocation when reading the file
(irrespective of the file's actual size).<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_CLUFOLDFILE">Input file to provide the target set for committor probabilities:<br/></a></h4>
Keyword: <b><a href="keywords.html#CLUFOLDFILE">FMCSC_CLUFOLDFILE</a></b><br/>
<br/>
Required for: Defining a target set for committor probabilities (as reported in output files
<a href="outputfiles.html#PFOLD_MINUS_xxx.dat">PFOLD_MINUS_xxx.dat</a>, 
<a href="outputfiles.html#PFOLD_PLUS_xxx.dat">PFOLD_PLUS_xxx.dat</a>,
<a href="outputfiles.html#PFOLD_MINUS_yyy_CFEP_xxxxxxxx.dat">PFOLD_MINUS_yyy_CFEP_xxxxxxxx.dat</a>, and
<a href="outputfiles.html#PFOLD_PLUS_yyy_CFEP_xxxxxxxx.dat">PFOLD_PLUS_yyy_CFEP_xxxxxxxx.dat</a>)
comprised of one or more clusters based on multiple input snapshots.
This is is irrelevant for any simulation not using <a href="keywords.html#CCOLLECT">clustering and related methods</a>,
not <a href="keywords.html#DOPFOLD">requesting the committor analysis</a>, or for <a href="install.html">executables not linked against HSL</a>.<br/>
<br/>
This file uses a trivial single column input format where the user indicates the integers that correspond to the snapshots that
are used to identify the clusters that compose the target (folded) set <i>B</i> in all committor
analyses (see <a href="keywords.html#DOPFOLD">DOPFOLD</a> and <a href="keywords.html#DOPFOLD_MINUS">DOPFOLD_MINUS</a>).
The cluster or clusters have to be in the same strongly connected component as those for the alternative set (<i>A</i>)
requested via <a href="inputfiles.html#FMCSC_CLUUNFOLDFILE">FMCSC_CLUUNFOLDFILE</a>.<br/>
Note that this input <b>always</b> refers to the basic and complete simulation length defined via <a href="keywords.html#NRSTEPS">FMCSC_NRSTEPS</a>.
This means that the numbers must <b>not</b> be adjusted to reflect settings for <a href="keywords.html#EQUIL">FMCSC_EQUIL</a>,
<a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>, <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a>, or similar keywords.
The use of a frames file should be handled with particular care. If the file accesses the trajectory in random access ("as is") mode,
all step numbers are assumed to refer to the line number in the frames file rather than the index of the frame on that line. This is 
a general change of interpretation inherent to <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a> with certain 
<a href="keywords.html#PDB_FORMAT">input file formats</a>. Conversely, if the file accesses the trajectory in strictly sequential
mode, step numbers continue to refer to the <i>original</i> trajectory.<br/>
In case this file is not specified or not found, CAMPARI will use the value associated with keyword
<a href="keywords.html#INISYNSNAP">INISYNSNAP</a>
for this purpose, which defaults to using the cluster that has the largest number of snapshots.<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_CLUUNFOLDFILE">Input file to provide the alternative set for committor probabilities:<br/></a></h4>
Keyword: <b><a href="keywords.html#CLUUNFOLDFILE">FMCSC_CLUUNFOLDFILE</a></b><br/>
<br/>
Required for: Defining an alternative set for committor probabilities (as reported in output files
<a href="outputfiles.html#PFOLD_MINUS_xxx.dat">PFOLD_MINUS_xxx.dat</a>, 
<a href="outputfiles.html#PFOLD_PLUS_xxx.dat">PFOLD_PLUS_xxx.dat</a>,
<a href="outputfiles.html#PFOLD_MINUS_yyy_CFEP_xxxxxxxx.dat">PFOLD_MINUS_yyy_CFEP_xxxxxxxx.dat</a>, and
<a href="outputfiles.html#PFOLD_PLUS_yyy_CFEP_xxxxxxxx.dat">PFOLD_PLUS_yyy_CFEP_xxxxxxxx.dat</a>)
comprised of one or more clusters based on multiple input snapshots.
This is is irrelevant for any simulation not using <a href="keywords.html#CCOLLECT">clustering and related methods</a>,
not <a href="keywords.html#DOPFOLD">requesting the committor analysis</a>, or for <a href="install.html">executables not linked against HSL</a>.<br/>
<br/>
This file is exactly identical to  <a href="inputfiles.html#FMCSC_CLUFOLDFILE">FMCSC_CLUFOLDFILE</a> only that it lets
the user specify the alternative (<i>A</i>) rather than the target (<i>B</i>) set.<br/>
<br/>
<br/>
<br/>
<h3><a class="NoHi" id="Sec7">Section 7: Files relevant to the NetCDF analysis mode:</a></h3>
<hr style="width: 100%; height: 2px;"></hr> (<a href="inputfiles.html#TOP">back to top</a>)<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_NCDM_NCFILE">Input file to provide binary data for the NetCDF analysis mode:<br/></a></h4>
Keyword: <b><a href="keywords.html#NCDM_NCFILE">FMCSC_NCDM_NCFILE</a></b><br/>
<br/>
Required for: Providing the data in NetCDF binary format <a href="keywords.html#netcdf_anal">to be mined</a>.
<br/>
<br/>
<b>Variables</b><br/>
<br/>
The only <a href="references.html#ref16_1">variable</a> that must be present in the NetCDF input 
data base is the one that contains the values of the degrees of freedom (features) per frame that the user 
wants to submit to <a href="keywords.html#CMODE">structural clustering</a> and downstream analyses. CAMPARI expects this 
variable to be named "featuresvals" in the data base, but this assumption can be changed with keyword <a href="keywords.html#NCDM_NMFV">NCDM_NMFV</a>.
This variable has to have two dimensions, the first one reflecting the number of frames (to which it can be 
given the UNLIMITED size) and the second one the number of features per frame. The name of these dimensions are 
inferred directly from the input NetCDF data base, but can not be changed when converting an <a href="keywords.html#NCDM_ASFILE">ASCII file</a>
(defaul names are "nframes" for the first (possibly unlimited) dimension (number of frames) and "nfeats" for 
the second dimension (number of features)). One optional attribute to this variable can be specified and must 
be named "periodic_range" (see below). According to the <a href="references.html#ref16_1">NetCDF documentation</a> 
names of attributes can be changed easily from the user side.
<br/>
<br/>
Two optional variables for feature weights are recognized by CAMPARI if present in the input NetCDF file. 
Static weights can be specified with variable "static_weights" and dynamic ones with variable "dynamic_weights". 
This naming is fixed, and in case the input data base does not comply with this standard, there are tools 
to also rename variables from the user side. The actual use of those weights depends on the requested analysis. 
For example, if dynamic weights are present in the data base but a <a href="keywords.html#CDISTANCE">distance criterion</a>
is selected such that dynamic weights are not supported, they will be completely discarded 
(the only exception to this rule being when converting an <a href="inputfiles.html#FMCSC_NCDM_ASFILE">ASCII file</a>, 
in that case weights are reflected in the converted file, but obviusly not used in analysis). As a final note on the 
use of feature weights, the <a href="keywords.html#netcdf_anal">NetCDF data mining mode</a> supports inferring the 
weights directly from the data (see <a href="keywords.html#CDISTANCE">CDISTANCE</a> and <a href="keywords.html#CMODWEIGHTS">CMODWEIGHTS</a>), 
regardless whether those have been specified in the input data base or not. This also means that input 
weights can be overwritten if this is desired (see <a href="keywords.html#CMODWEIGHTS">CMODWEIGHTS</a>).
<br/>
<br/>
<b>Dimensions</b><br/>
<br/>
As already mentioned above, there are two <a href="references.html#ref16_1">dimensions</a> 
that characterize all the variables recognized by CAMPARI. The first dimension is the unlimited dimension 
(number of frames), which is the first dimension for the feature-containing variable <a href="keywords.html#NCDM_NMFV">NCDM_NMFV</a> 
and for the dynamic weights (static weights obviously do not have this dimension as there is only one 
value per degree of freedom). The second one is the number of features per frame and is relevant to all the 
variables, <i>viz.</i>, to the one that holds the values of the features <a href="keywords.html#NCDM_NMFV">NCDM_NMFV</a> 
and to the static and dynamic weights, if specified.
<br/>
<br/>
<b>Attributes</b><br/>
<br/>
There is only one <a href="references.html#ref16_1">attribute</a> that can influence 
the behavior of CAMPARI in <a href="keywords.html#netcdf_anal">NetCDF data mining mode</a>.
If the input data are periodic, the user can specify the periodicity range through 
the attribute "periodic_range" assigned to the feature-containing variable <a href="keywords.html#NCDM_NMFV">NCDM_NMFV</a>.
This attribute should contain two values, one for the left boundary and one for the right boundary of the 
periodic interval. In case a <a href="keywords.html#CDISTANCE">distance criterion</a> that implies periodicity 
is specified but no periodicity attribute is assigned to the feature-containing variable, CAMPARI assumes it is 
analyzing angle data in degrees and uses  the default periodic interval with boudaries at -180&deg; and 180&deg;. 
In general, periodic data are adjusted to be within the periodic interval in use (default or specified by the user). 
This means that, for example, the user does not have to adjust his/her periodic data to the reference periodic 
interval if the data acquisition has been done on different interavals, CAMPARI will adjust the values automatically, 
and this adjustment is NOT reflected in ANY possible output file (<a href="keywords.html#NCDM_WRTINPUT">NCDM_WRTINPUT</a>). 
However, the user has to be keep in mind that these acquisitions have to be consistent with the reference periodic interval.
It is not possible to have, say, angles measured once in the interval [-180&deg;180&deg;] and once in [0&deg;360&deg;] 
as these intervals overlap. It is instead correct to mix measurements coming from, for instance, 
independent angle readouts once in the interval [-180&deg;180&deg;] and once in [180&deg;540&deg;].
It is worth noting also that, differently from what happens for <a href="inputfiles.html#FMCSC_NCDM_ASFILE">ASCII conversion</a>, 
when a NetCDF file is used as input, the keywork <a href="keywords.html#NCDM_PRDCRNG">NCDM_PRDCRNG</a> 
is completely ignored.
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_NCDM_ASFILE">Input file to provide ASCII data to be converted in NetCDF standard and/or analyzed:<br/></a></h4>
Keyword: <b><a href="keywords.html#NCDM_ASFILE">FMCSC_NCDM_ASFILE</a></b><br/>
<br/>
Required for: Providing the data in ASCII format to be analysed with <a href="keywords.html#CMODE">structural clustering</a> 
and downstream routines and/or to be converted to the <a href="inputfiles.html#FMCSC_NCDM_NCFILE">NetCDF standard</a>
required by CAMPARI for <a href="keywords.html#netcdf_anal">data mining</a>. 
<br/>
<br/>
CAMPARI expects to find an ASCII file with as many columns (features) as specified with keyword <a href="keywords.html#NCDM_NRFEATS">NCDM_NRFEATS</a>.
To subsample the features to be used in analysis, the only option is to provide an additional input file (see <a href="inputfiles.html#FMCSC_NCDM_CFILE">NCDM_CFILE</a>).
In case <a href="keywords.html#NCDM_CHECKAS">checking of the input ASCII file</a> is explicitly requested, CAMPARI checks that 
every line of the input file contains the exact number of entries expected from <a href="keywords.html#NCDM_NRFEATS">NCDM_NRFEATS</a>. 
This processing slows down considerably the reading-in of the data, and is disabled by default. The number of rows that correspond 
to the frames that are contained in the ASCII file should be specified with keyword <a href="keywords.html#NCDM_NRFRMS">NCDM_NRFRMS</a>. 
This specification is important. Any other number than the exact number of frames contained in the ASCII file will produce unexpected and most likely 
wrong behavior. To subsample the frames of the input ASCII in analysis, the user can rely on the mutually exclusive usage of keywords <a href="keywords.html#CCOLLECT">CCOLLECT</a> 
or <a href="inputfiles.html#FMCSC_NCDM_FRAMESFILE">NCDM_FRAMESFILE</a>, which however are ignored in conversion (the entire ASCII file is always 
converted if conversion is requested, see <a href="keywords.html#NCDM_WRTINPUT">NCDM_WRTINPUT</a>). Keyword <a href="keywords.html#NCDM_NRFRMS">NCDM_NRFRMS</a> 
must not account for the possible presence of features weights, which is checked upon by CAMPARI automatically. 
Indeed, in case the user wants to specify weights, those can be appended to the input ASCII file. If CAMPARI finds 2&times;<a href="keywords.html#NCDM_NRFRMS">NCDM_NRFRMS</a> 
lines, it assumes that the input ASCII file contains dynamic weights, with the first line of weights (line <a href="keywords.html#NCDM_NRFRMS">NCDM_NRFRMS</a> + 1) 
containing the weights corresponding to the degrees of freedom of the first snapshot and so on. If CAMPARI detects 2&times;<a href="keywords.html#NCDM_NRFRMS">NCDM_NRFRMS</a> 
+ 1, it assumes there are both dynamic and static weights, with the latter at the end of the file (last line). If the 
ASCII file is formed by <a href="keywords.html#NCDM_NRFRMS">NCDM_NRFRMS</a> + 1 lines, the last one is interpreted as static weights.
Any other combination will produce an error. The actual use of feature weights depends on the choice of <a href="keywords.html#CDISTANCE">CDISTANCE</a> 
and keyword <a href="keywords.html#CMODWEIGHTS">CMODWEIGHTS</a> (and associated ones) may become relevant.
<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_NCDM_CFILE">Input file to select specific degrees of freedom in <a href="keywords.html#netcdf_anal">NetCDF data mining mode</a>:<br/></a></h4>
Keyword: <b><a href="keywords.html#NCDM_CFILE">FMCSC_NCDM_CFILE</a></b><br/>
<br/>
Required for: Selecting specific features in analysis.<br/>
<br/>
This simple single-column list of integers lets the user select (subsample) the features to be used for <a href="keywords.html#CMODE">structural clustering</a> 
and downstream analyses with respect to the set of degrees of freedom present in the input file (<a href="inputfiles.html#FMCSC_NCDM_NCFILE">NCDM_NCFILE</a> 
or <a href="inputfiles.html#FMCSC_NCDM_ASFILE">NCDM_ASFILE</a>).
<br/>
<br/>
<u>Example:</u><br/>
----------------<br/>
<big><code>
1 <br/>
3 <br/>
4 <br/>
</code></big>
----------------<br/>
<br/>
The above exemplified NCDM_CFILE file instructs CAMPARI to select only the first, 
third and fourth degrees of freedom of the original input file (<a href="inputfiles.html#FMCSC_NCDM_NCFILE">NCDM_NCFILE</a> 
or <a href="inputfiles.html#FMCSC_NCDM_ASFILE">NCDM_ASFILE</a>) for <a href="keywords.html#CMODE">structural clustering</a> 
and downstream routines. Duplicates, nulls and negatives are removed automatically. 
The features subsamplig is not reflected in the possible converted ASCII file (<a href="keywords.html#NCDM_WRTINPUT">NCDM_WRTINPUT</a>), 
but does reflect in the output summary database in case it is required when providing as input a NetCDF file (<a href="inputfiles.html#FMCSC_NCDM_NCFILE">NCDM_NCFILE</a>, 
mainly for debug, see <a href="keywords.html#NCDM_WRTINPUT">NCDM_WRTINPUT</a>).
<br/>
<br/>
<br/>
<h4><a class="NoHi" id="FMCSC_NCDM_FRAMESFILE">Input file to select specific frames in <a href="keywords.html#netcdf_anal">NetCDF data mining mode</a>:<br/></a></h4>
Keyword: <b><a href="keywords.html#NCDM_FRAMESFILE">FMCSC_NCDM_FRAMESFILE</a></b><br/>
<br/>
Required for: Selecting specific frames in analysis.<br/>
<br/>
With this single-column list of integers, the user can specify a set of frames from the input file (<a href="inputfiles.html#FMCSC_NCDM_NCFILE">NCDM_NCFILE</a> 
or <a href="inputfiles.html#FMCSC_NCDM_ASFILE">NCDM_ASFILE</a>) to be analyzed in <a href="keywords.html#CMODE">structural clustering</a> 
and downstream analyses. 
<br/>
<br/>
<u>Example:</u><br/>
----------------<br/>
<big><code>
1 <br/>
3 <br/>
4 <br/>
3 <br/>
107 <br/>
</code></big>
----------------<br/>
<br/>
As the example above documents, duplicates are legal and the frames file does not need to be sorted, it will be processed 
exactly the way it is provided, consistently with the underlying NetCDF file format (see also <a href="keywords.html#FRAMESFILE">FRAMESFILE</a>).
Nulls and negatives will be removed. However, a frame value that exceeds the number of
frames present in the input file (<a href="inputfiles.html#FMCSC_NCDM_NCFILE">NCDM_NCFILE</a> or <a href="inputfiles.html#FMCSC_NCDM_ASFILE">NCDM_ASFILE</a>)
will trigger an error. In case feature weights are present, they will be processed 
(and possibly reordered/duplicated) in order to match the frames specified with this input 
file. It is important to note that, differently from the standard execution mode of CAMPARI, 
the specification of a FRAMESFILE in the <a href="keywords.html#netcdf_anal">NetCDF data mining mode</a>
madates setting <a href="keywords.html#CCOLLECT">CCOLLECT</a> to 1, <i>viz.</i>, it is explicitly 
disabled the option to subsample a frames file with <a href="keywords.html#CCOLLECT">CCOLLECT</a>.
The action of the frames file is not reflected in any way in the NetCDF obtained from an input ASCII file, but 
it is reflected in the debug output when the input file is already in NetCDF format (see <a href="keywords.html#NCDM_WRTINPUT">NCDM_WRTINPUT</a>).
<br/>
<br/>
<br/>
<!--
<h4>Coefficient matrix for linear combinations of torsions (LCTs):</a></h4>
keyword: <b>FMCSC_TORLCFILE</b><a id="FMCSC_TORLCFILE"></a><br/>
<br/>
If keyword <a href="inputfiles.html#FMCSC_TORLCMODE">FMCSC_TORLCMODE</a>
is set to 1:<br/>
<br/>
This file should have (2*n_dof+1) columns (where n_dof is the number of
degrees of freedom = torsions in the system) and a number of rows
consistent with the spec. given for FMCSC_<a
 href="keywords.html#NRTORLC">NRTORLC</a> (see <a
 href="keywords.html#KEYWORDS">KEYWORDS</a>).<br/>
<br/>
The first column holds a weight-factor only relevant for the LCT energy
term.<br/>
<br/>
The remaining columns hold the coefficients for this particular LCT
interpreted as follows:<br/>
<br/>
Let a_i denote torsion i in the system with i = 1 ... n_dof:<br/>
Then the linear combination f_k (kth row if this file) is given as:<br/>
<br/>
<img style="width: 285px; height: 17px;"
 alt="f_k = SUM_i( g_k|2*i-1 * cos(a_i) + g_k|2*i * sin(a_i) )"
 src="website_equations/inputfiles_torlcfile.gif"><br/>
<br/>
The columns 2:(2*n_dof+1) essentially represent the coefficient matrix
G with entries g_k|l<br/>
The order of torsions is always by residue, then as phi, psi, chi1,
chi2, and then as component (cos, sin)<br/>
<br/>
example alanine dipeptide (two torsions, phi and psi), let's assume <a
 href="keywords.html#NRTORLC">NRTORLC</a> was set to 2:<br/>
<br/>
<big><big><code>0.1&nbsp; -1.0&nbsp; 1.0 0.5&nbsp; 0.5<br/>
0.01&nbsp; 1.0&nbsp; 0.8&nbsp; 0.4&nbsp; 0.3</code></big></big><br/>
<br/>
<br/>
then f_1 = -1.0*cos(phi) + 1.0*sin(phi) + 0.5*cos(psi) + 0.5*sin(psi)
with a weight factor of 0.1<br/>
and&nbsp;f_2 =&nbsp;1.0*cos(phi) + 0.8*sin(phi) + 0.4*cos(psi) +
0.3*sin(psi) with a weight factor of 0.01<br/>
<br/>
If keyword <a href="inputfiles.html#FMCSC_TORLCMODE">FMCSC_TORLCMODE</a>
is set to 2:<br/>
<br/>
This file should have (n_dof+1) columns (where n_dof is the number of
degrees of freedom = torsions in the system) and a number of rows
consistent with the spec. given for FMCSC_<a
 href="keywords.html#NRTORLC">NRTORLC</a> (see <a
 href="keywords.html#KEYWORDS">KEYWORDS</a>).<br/>
<br/>
The first column holds a weight-factor only relevant for the LCT energy
term.<br/>
<br/>
the remaining columns hold the coefficients for this particular LCT
interpreted as follows:<br/>
<br/>
Let a_i denote torsion i in the system with i = 1 ... n_dof:<br/>
Then the linear combination f_k (kth row if this file) is given as:<br/>
<br/>
<img style="width: 117px; height: 17px;"
 alt="f_k = SUM_i( g_k|i * a_i )"
 src="website_equations/inputfiles_torlcfile2.gif"><br/>
<br/>
The columns 2:(n_dof+1) essentially represent the coefficient matrix G
with entries g_k|l<br/>
The order of torsions is always by residue, then as phi, psi, chi1, chi2<br/>
<br/>
example alanine dipeptide (two torsions, phi and psi), let's assume <a
 href="keywords.html#NRTORLC">NRTORLC</a> was set to 2:<br/>
<br/>
<big><big><code>0.1&nbsp; 1.0&nbsp; 0.5<br/>
0.01&nbsp; 0.4&nbsp; 0.3</code></big></big><br/>
<br/>
<br/>
then f_1 = 1.0*phi + 0.5*psi&nbsp;with a weight factor of 0.1<br/>
and&nbsp;f_2 = 0.4*phi + 0.3*psi&nbsp;with a weight factor of 0.01<br/>
<br/>
<br/>
<h4>Coefficient matrix for reverse transform (linear combinations
of torsions (LCTs) TO native torsions):</a></h4>
keyword: <b>FMCSC_TORLCFILE2</b><a id="FMCSC_TORLCFILE2"></a><br/>
<br/>
Analogous to the coefficient matrix for the forward transform except
for the missing first column (<i>i.e.</i>, no weights in this file). The
matrix should of course always be the inverse of the matrix spec.d in
FMCSC_TORLCFILE.<br/>
<br/>
<br/>
<h4>Potential file for linear combinations of torsions (LCTs):</a></h4>
keyword: <b>FMCSC_LCTPOTFILE</b><a id="FMCSC_LCTPOTFILE"></a><br/>
<br/>
This is only used in conjunction with the LCT potential (see keyword
FMCSC_<a href="keywords.html#SC_LCTOR">SC_LCTOR</a>). To apply a
Hamiltonian over LCTs (see above for how to specify these LCTs) the
potential is given in tabulated form through this file. the program
then does linear interpolation over the provided discreet dataset.<br/>
<br/>
Outside of this, the file is very analogous to that used with
FMSCS_TABPOTFILE.<br/>
<br/>
There are n+1 columns, where n is the number of LCTs to use (through
FMCSC_<a href="keywords.html#NRTORLC">NRTORLC</a>). The first column
contains the bin centers for the values of the various LCTs. The
remaining columns then hold the actual potential data. There have to be
at least as many bins as specified by FMCSC_LCTBINS. The sanity of this
file is completely up to the user. only constant bin sizes are
supported, and the program determines it automatically (<i>i.e.</i>, FMCSC_<a
 href="keywords.html#TORLCRES">TORLCRES</a> is irrelevant for this
file).<br/>
-->
<br/>
<br/>
<br/>
</div>
                    
                </div>
                </div>
                </div>
                
            <!-- Side Navigation Bar -->
            <!--<div class="art-sidebar1">-->                                    
                <!-- Contributors Block -->
                <!--<div class="art-Block">
                    <div class="art-Block-tl"></div>
                    <div class="art-Block-tr"></div>
                    <div class="art-Block-bl"></div>
                    <div class="art-Block-br"></div>
                    <div class="art-Block-tc"></div>
                    <div class="art-Block-bc"></div>
                    <div class="art-Block-cl"></div>
                    <div class="art-Block-cr"></div>
                    <div class="art-Block-cc"></div>
                    <div class="art-Block-body">
                        <div class="art-BlockHeader">
                            <div class="l"></div>
                            <div class="r"></div>
                            <div class="art-header-tag-icon">
                                <div class="t">Contributors - Links</div>
                            </div>
                        </div><div class="art-BlockContent">
                            <div class="art-BlockContent-body">
                                <div>
                                <b>Pappu Lab</b><br />
                                Washington University in St. Louis, MO<br />
                                Website: <a href="lima.wustl.edu">lima.wustl.edu</a><br />
                                Phone: (999) 999-9999 <br/>
                                <br/>
                                <b>Andreas Vitalis</b><br />
                                Unistat de Zurich<br />
                                Website: <a href="papajohns.com">vitalis.net</a><br />
                                Phone: (111) 999-9999 <br/>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>-->
                
                <!-- Search Block -->
                <!--
                <div class="art-Block">
                    <div class="art-Block-tl"></div>
                    <div class="art-Block-tr"></div>
                    <div class="art-Block-bl"></div>
                    <div class="art-Block-br"></div>
                    <div class="art-Block-tc"></div>
                    <div class="art-Block-bc"></div>
                    <div class="art-Block-cl"></div>
                    <div class="art-Block-cr"></div>
                    <div class="art-Block-cc"></div>
                    <div class="art-Block-body">
                        <div class="art-BlockHeader">
                            <div class="l"></div>
                            <div class="r"></div>
                            <div class="art-header-tag-icon">
                                <div class="t">Find Text</div>
                            </div>
                        </div><div class="art-BlockContent">
                            <div class="art-BlockContent-body">
                                <div><form method="get" id="newsletterform" action="javascript:void(0)">
                                <input type="text" value="" id="email" id="s" style="width: 95%;" />
                                <span class="art-button-wrapper">
                                    <span class="l"> </span>
                                    <span class="r"> </span>
                                    <input class="art-button" type="submit" id="search" value="Search"/>
                                </span>
                                </form></div>
                            </div>
                        </div>
                    </div>
                </div> 
                -->
                
                <!-- News/Updates Block -->
                <!--
                <div class="art-Block">
                    <div class="art-Block-tl"></div>
                    <div class="art-Block-tr"></div>
                    <div class="art-Block-bl"></div>
                    <div class="art-Block-br"></div>
                    <div class="art-Block-tc"></div>
                    <div class="art-Block-bc"></div>
                    <div class="art-Block-cl"></div>
                    <div class="art-Block-cr"></div>
                    <div class="art-Block-cc"></div>
                    <div class="art-Block-body">
                        <div class="art-BlockHeader">
                            <div class="l"></div>
                            <div class="r"></div>
                            <div class="art-header-tag-icon">
                                <div class="t">Recent News</div>
                            </div>
                        </div><div class="art-BlockContent">
                            <div class="art-BlockContent-body">
                                <div>
                                    <p><b>Jun 14, 2008</b><br/>
                                    Aliquam sit amet felis. Mauris semper, 
                                    velit semper laoreet dictum, quam 
                                    diam dictum urna, nec placerat elit 
                                    nisl in quam. Etiam augue pede, 
                                    molestie eget, rhoncus at, convallis 
                                    ut, eros. Aliquam pharetra.<br/>
                                    <a href="javascript:void(0)">Read more...</a></p>
                                                          
                                    <p><b>Aug 24, 2008</b><br/>
                                    Aliquam sit amet felis. Mauris semper, 
                                    velit semper laoreet dictum, quam 
                                    diam dictum urna, nec placerat elit 
                                    nisl in quam. Etiam augue pede, 
                                    molestie eget, rhoncus at, convallis 
                                    ut, eros. Aliquam pharetra.<br/>
                                    <a href="javascript:void(0)">Read more...</a></p>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div> --><!-- End Sidebar -->

            </div>
        </div>
    </div> <!-- End main page contents -->

            
    <!-- Footer -->
    <div class="art-Footer">
    <div class="art-Footer-background">
    <div class="art-Footer-inner">
        <a href="#" class="art-rss-tag-icon" title="RSS"></a>
        <div class="art-Footer-text">
        <p><a href="#">Contact Us</a> | <a href="#">Terms of Use</a>
        | <a href="#">Lincense Information</a><br />
        Copyright &copy; 2010 . All Rights Reserved.</p>
        </div>
    </div>
    </div>
    </div>
    <p class="art-page-footer">XHTML and CSS valid. Powered by <a href="http://www.flashmint.com/">FlashMint</a> - flash templates provider.</p>
    <div style="text-align: center; font-size: 0.75em;">Design downloaded from <a href="http://www.freewebtemplates.com/">free website templates</a>.</div>
                
</div>
</body>
</html>
