<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<!--
############################################################################
# LICENSE INFO:                                                            #
############################################################################
#    This file is part of CAMPARI.                                         #
#                                                                          #
#    Version 3.0                                                           #
#                                                                          #
#    Copyright (C) 2017, The CAMPARI development team (current and former  #
#                        contributors)                                     #
#                        Andreas Vitalis, Adam Steffen, Rohit Pappu, Hoang #
#                        Tran, Albert Mao, Xiaoling Wang, Jose Pulido,     #
#                        Nicholas Lyle, Nicolas Bloechliger, Marco Bacci,  #
#                        Davide Garolini, Jiri Vymetal                     #
#                                                                          #
#    Website: http://sourceforge.net/projects/campari/                     #
#                                                                          #
#    CAMPARI is free software: you can redistribute it and/or modify       #
#    it under the terms of the GNU General Public License as published by  #
#    the Free Software Foundation, either version 3 of the License, or     #
#    (at your option) any later version.                                   #
#                                                                          #
#    CAMPARI is distributed in the hope that it will be useful,            #
#    but WITHOUT ANY WARRANTY; without even the implied warranty of        #
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
#    GNU General Public License for more details.                          #
#                                                                          #
#    You should have received a copy of the GNU General Public License     #
#    along with CAMPARI.  If not, see <http://www.gnu.org/licenses/>.      #
############################################################################
# AUTHORSHIP INFO:                                                         #
############################################################################
#                                                                          #
# MAIN AUTHOR:   Andreas Vitalis                                           #
# CONTRIBUTIONS: Adam Steffen, Marco Bacci                                 #
#                                                                          #
############################################################################
-->
<html xmlns="http://www.w3.org/1999/xhtml" dir="ltr" lang="en-US" xml:lang="en">
<head>
    <!--
    Designed by CAMPARI Development Group
    Base template (without user's data) checked by http://validator.w3.org : "This page is valid XHTML 1.0 Transitional"
    -->
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"></meta>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge"></meta>
    <title>CAMPARI Output Files</title>
    <link rel="stylesheet" href="style_red.css" type="text/css" media="screen"></link>
</head>
<body>
<div id="art-page-background-simple-gradient"></div>
<div id="art-main">
    <!-- Header -->
    <div class="art-Sheet2">
        
        <!-- Banner Art -->
        <div class="art-Header">
            <div class="art-Header-png"></div>
        </div>
                
        <!-- Top Navigation Bar -->
        <div class="art-nav">
            <div class="l"></div>
            <div class="r"></div>
            <ul class="art-menu">
                <li><a href="index.html"><span class="l"></span><span class="r"></span><span class="t">Home</span></a></li>
                <li><a href="documentation.html" class=" active"><span class="l"></span><span class="r"></span><span class="t">Documentation</span></a></li>
                <li><a href="references.html"><span class="l"></span><span class="r"></span><span class="t">References</span></a></li>
                <li><a href="contact.html"><span class="l"></span><span class="r"></span><span class="t">Support</span></a></li>
                <li><a href="download.html"><span class="l"></span><span class="r"></span><span class="t">Download</span></a></li>
            </ul>
        </div>
            
        <!-- Top Sub Navigation Bar -->
        <div class="nav-menu">
            <ul class="navlist">
                <li><a href="documentation.html">Overview</a></li>
                <li><a href="install.html">Install</a></li>
                <li><a href="tutorials.html">Tutorials</a></li>
                <li><a href="inputfiles.html">Inputfiles</a></li>
                <li><a href="keywords.html">Keywords</a></li>
                <li><a href="outputfiles.html" class=" active">Outputfiles</a></li>
                <li><a href="parameters.html">Parameters</a></li>
                <li><a href="development.html">Development</a></li>
                <li></li>
            </ul>
        </div> 
    </div>
        
    <!-- Main Page Contents -->
    <div class="art-Sheet">
        <div class="art-Sheet-bl"></div>
        <div class="art-Sheet-br"></div>
        <div class="art-Sheet-bc"></div>
        <div class="art-Sheet-cl"></div>
        <div class="art-Sheet-cr"></div>
        <div class="art-Sheet-cc"></div>
        <div class="art-Sheet-body">
            
            <div class="art-contentLayout">
                <!-- Page Contents -->
                <div class="art-content">
                <div class="art-Post">
                <div class="art-Post-body">
<div class="art-PostContent">
<br/>
<h2><a class="NoHi" id="TOP">Overview of Analysis Features and Output Files in CAMPARI</a></h2>
<hr/>
Please also note the <a href="outputfiles.html#COMMENTS">general and other comments</a> below.<br/>
<ol>
  <li><b>Polymeric Analyses</b>
    <ol type='a'>
      <li>Average radius of gyration (and t), asphericity and acylindricity, end-to-end distance, hydrodynamic radius, ordered Eigenvalues of gyration tensor<br/>
          <i>Available per analysis group for all non-rigid molecules (see <a href="outputfiles.html#POLYAVG.dat">POLYAVG.dat</a>, end-to-end distance also requires at least two residues in molecule)</i></li>
      <li>Average internal scaling behavior<br/>
          <i>Available per analysis group for all non-rigid molecules (see <a href="outputfiles.html#INTSCAL.dat">INTSCAL.dat</a>)</i></li>
      <li>Average radial density profile<br/>
          <i>Available per analysis group for all non-rigid molecules (see <a href="outputfiles.html#DENSPROF.dat">DENSPROF.dat</a>)</i></li>
      <li>Average Kratky profile<br/>
          <i>Available per analysis group for all non-rigid molecules (see <a href="outputfiles.html#KRATKY.dat">KRATKY.dat</a>)</i></li>
      <li>Simulated averaged fiber diffraction data (preliminary version)<br/>
          <i>Available system-wide (see <a href="outputfiles.html#DIFFRACTION.dat">DIFFRACTION.dat</a>)</i></li>
      <li>Radius of gyration histogram<br/>
          <i>Available per analysis group for all non-rigid molecules (see <a href="outputfiles.html#RGHIST.dat">RGHIST.dat</a>)</i></li>
      <li>Histogram of end-to-end distances<br/>
          <i>Available per analysis group for all non-rigid molecules (see <a href="outputfiles.html#RETEHIST.dat">RETEHIST.dat</a>)</i></li>
      <li>2D-Histogram of radius of gyration and instantaneous asphericity &#948;<br/>
          <i>Available per analysis group for all non-rigid molecules (see <a href="outputfiles.html#RDHIST.dat">RDHIST.dat</a>)</i></li>
      <li>Distribution of void spaces<br/>
          <i>Available system-wide for single polymer simulations (see <a href="outputfiles.html#HOLES.dat">HOLES.dat</a>)</i></li>
     <li>Instantaneous (running output) of system-averaged polymeric properties<br/>
          <i>Available system-wide (see <a href="outputfiles.html#POLYMER.dat">POLYMER.dat</a>)</i></li>
    </ol>
  </li>
  <li><b>Topological Analyses</b>
    <ol type='a'>
     <li>Bond length distributions<br/>
          <i>Available per bond (two atoms directly bonded) length (see <a href="outputfiles.html#INTHISTS_BL.dat">INTHISTS_BL.dat</a>)</i></li>
     <li>Bond angle distributions<br/>
          <i>Available per bond angle (across three atoms bonded in direct sequence) (see <a href="outputfiles.html#INTHISTS_BA.dat">INTHISTS_BA.dat</a>)</i></li>
     <li>Dihedral angle distributions<br/>
          <i>Available per dihedral angle (along four atoms bonded in direct sequence) (see <a href="outputfiles.html#INTHISTS_DI.dat">INTHISTS_DI.dat</a>)</i></li>
     <li>Improper torsional angle distributions<br/>
          <i>Available per improper dihedral (central carbon with three atoms directly bonded to it) (see <a href="outputfiles.html#INTHISTS_IM.dat">INTHISTS_IM.dat</a>)</i></li>
     <li>Angular correlation function<br/>
          <i>Available per analysis group for all polypeptide or polynucleotide molecules with at least two complete residues (see <a href="outputfiles.html#PERSISTENCE.dat">PERSISTENCE.dat</a>)</i></li>
     <li>Average turn angles (kinked-ness and handed-ness) along the main chain<br/>
          <i>Available per analysis group for all polypeptide or polynucleotide molecules with at least two complete residues (see <a href="outputfiles.html#TURNS_RES.dat">TURNS_RES.dat</a>)</i></li>
     <li>Contact histograms<br/>
          <i>Available system-wide for all residues of solute molecules (see <a href="outputfiles.html#CONTACT_HISTOGRAMS.dat">CONTACT_HISTOGRAMS.dat</a>)</i></li>
     <li>Average contact maps<br/>
          <i>Available system-wide for all residues of solute molecules (see <a href="outputfiles.html#CONTACTMAP.dat">CONTACTMAP.dat</a>)</i></li>
     <li>Intramolecular distance distributions of amide sites<br/>
          <i>Available system-wide for polypeptides (see <a href="outputfiles.html#AMIDES_PC.dat">AMIDES_PC.dat</a>)</i></li>
    </ol>
  </li>
  <li><b>Solution, Molecular Structure, and Relative and Global Spatial Analyses</b>
    <ol type='a'>
     <li>Pair correlation functions for molecular centers of mass<br/>
          <i>Available per pairs of analysis groups for all solute molecules (see <a href="outputfiles.html#RBC_PC.dat">RBC_PC.dat</a>)</i></li>
     <li>Distance distributions and pair correlation functions of arbitrary sites, both inter- and intramolecularly<br/>
          <i>Available system-wide per custom request (see <a href="outputfiles.html#GENERAL_PC.dat">GENERAL_PC.dat</a>)</i></li>
     <li>Net molecular cluster statistics<br/>
          <i>Available system-wide for all solute molecules (molecules tagged as solvent are completely ignored for all cluster analyses; see <a href="outputfiles.html#CLUSTERS.dat">CLUSTERS.dat</a>)</i></li>
     <li>Molecular cluster statistics<br/>
          <i>Available per solute molecule (see <a href="outputfiles.html#MOLCLUSTERS.dat">MOLCLUSTERS.dat</a>)</i></li>
     <li>Histogram of contact order in clusters<br/>
          <i>Available per cluster size (see <a href="outputfiles.html#COOCLUSTERS.dat">COOCLUSTERS.dat</a>)</i></li>
     <li>Global solvent-accessible volume analysis<br/>
          <i>Available system-wide as instantaneous output along with a running average (see <a href="outputfiles.html#SAV.dat">SAV.dat</a>)</i></li>
     <li>Average site-specific solvent-accessible volumes<br/>
          <i>Available system-wide per atom (see <a href="outputfiles.html#SAV_BY_ATOM.dat">SAV_BY_ATOM.dat</a>)</i></li>
     <li>Distribution functions for site-specific solvent-accessible volumes<br/>
          <i>Available via user-selection per atom (see <a href="outputfiles.html#SAV_HISTS.dat">SAV_HISTS.dat</a> and <a href="outputfiles.html#SAV.dat">SAV.dat</a>)</i></li>
     <li>Average absolute spatial distributions of atomic properties<br/>
           <i>Available system-wide (see <a href="outputfiles.html#DENSITY.nc">DENSITY.nc</a>)</i></li>
    </ol>
  </li>
  <li><b>Polypeptide Secondary Structure Analyses</b>
    <ol type='a'>
     <li>Secondary structure segment statistics according to dihedral criterion<br/>
          <i>Available per residue per analysis group for all polypeptides with at least one complete (non-terminal) peptide residue (see <a href="outputfiles.html#BB_SEGMENTS_NORM.dat">BB_SEGMENTS_NORM.dat</a>, <a href="outputfiles.html#BB_SEGMENTS.dat">BB_SEGMENTS.dat</a>, <a href="outputfiles.html#BB_SEGMENTS_NORM_RES.dat">BB_SEGMENTS_NORM_RES.dat</a> and <a href="outputfiles.html#BB_SEGMENTS_RES.dat">BB_SEGMENTS_RES.dat</a>)</i></li>
     <li>Secondary structure segment statistics according to hydrogen bonding criterion<br/>
          <i>Available system-wide if intra- and/or intermolecular hydrogen bonds may be formed (see <a href="outputfiles.html#DSSP.dat">DSSP.dat</a>, <a href="outputfiles.html#DSSP_NORM.dat">DSSP_NORM.dat</a>, <a href="outputfiles.html#DSSP_NORM_RES.dat">DSSP_NORM_RES.dat</a> and <a href="outputfiles.html#DSSP_RES.dat">DSSP_RES.dat</a>)</i></li>
     <li>2D-histogram of global, hydrogen bonding-determined &#945;,&#946;-content<br/>
          <i>Available system-wide if intra- and/or intermolecular hydrogen bonds may be formed (see <a href="outputfiles.html#DSSP_EH_HIST.dat">DSSP_EH_HIST.dat</a>)</i></li>
     <li>Histograms of global secondary structure (&#945; and &#946;) content via hydrogen bonding criterion and "good"ness of those hydrogen bonds<br/>
          <i>Available system-wide if intra- and/or intermolecular hydrogen bonds may be formed (see <a href="outputfiles.html#DSSP_HIST.dat">DSSP_HIST.dat</a>)</i></li>
     <li>Running output of DSSP annotation<br/>
          <i>Available system-wide if intra- and/or intermolecular hydrogen bonds may be formed (see <a href="outputfiles.html#DSSP_RUNNING.dat">DSSP_RUNNING.dat</a>)</i></li>
     <li>Histograms of global secondary structure (&#945; and &#946;) content order parameters<br/>
          <i>Available per analysis group for all polypeptides with at least one complete (non-terminal) peptide residue (see <a href="outputfiles.html#ZSEC_HIST.dat">ZSEC_HIST.dat</a>)</i></li>
     <li>2D-Histogram of global &#945;,&#946;-content<br/>
          <i>Available per analysis group for all polypeptides with at least one complete (non-terminal) peptide residue (see <a href="outputfiles.html#ZAB_2DHIST.dat">ZAB_2DHIST.dat</a>)</i></li>
     <li>2D-Histogram of global &#946;-content and t (a proxy for the radius of gyration)<br/>
          <i>Available per analysis group for all polypeptides with at least one complete (non-terminal) peptide residue (see <a href="outputfiles.html#ZBETA_RG.dat">ZBETA_RG.dat</a>)</i></li>
     <li>Global Ramachandran analysis<br/>
          <i>Available system-wide for all polypeptides with at least one complete (non-terminal) peptide residue (see <a href="outputfiles.html#RAMACHANDRAN.dat">RAMACHANDRAN.dat</a>)</i></li>
     <li>Ramachandran analysis<br/>
          <i>Available by request per residue or per analysis group for all polypeptides with at least one complete (non-terminal) peptide residue (see <a href="outputfiles.html#RESRAMA_xxxxx.dat">RESRAMA_xxxxx.dat</a> and <a href="outputfiles.html#MOLRAMA_xxxxx.dat">MOLRAMA_xxxxx.dat</a>)</i></li>
     <li>Average J(H<sub>&#945;</sub>,H<sub>N</sub>) NMR coupling constants<br/>
          <i>Available per polypeptide residue excluding glycine, proline, and &#945;-aminoisobutyric acid (see <a href="outputfiles.html#JCOUPLING.dat">JCOUPLING.dat</a>)</i></li>
    </ol>
  </li>
  <li><b>Other Analyses and Simulation Output</b>
    <ol type='a'>
     <li>Trajectory output<br/>
          <i>Available system-wide or per <a href="keywords.html#TRAJIDXFILE">user control</a> (see <a href="outputfiles.html#Trajectory_output">below</a>)</i></li>
     <li>Instantaneous values for the coordinate root mean square deviation to a reference after alignment<br/>
          <i>Available per <a href="keywords.html#ALIGNFILE">user control</a> (see <a href="outputfiles.html#RMSD.dat">RMSD.dat</a>)</i></li>
     <li>Histograms of particle numbers in grand canonical ensemble simulations<br/>
          <i>Available per molecule type whose numbers are allowed to fluctuate (see <a href="outputfiles.html#PARTICLENUMHIST.dat">PARTICLENUMHIST.dat</a>)</i></li>
     <li>Average dipole moments<br/>
          <i>Available per analysis group (molecular) and per residue for all net neutral moieties (see <a href="outputfiles.html#MOLDIPOLES.dat">MOLDIPOLES.dat</a> and <a href="outputfiles.html#RESDIPOLES.dat">RESDIPOLES.dat</a>)</i></li>
     <li>Average and instantaneous statistics on energetic overlaps in parallel runs with multiple Hamiltonians<br/>
          <i>Available system-wide (see <a href="outputfiles.html#N_XXX_OVERLAP.dat">N_XXX_OVERLAP.dat</a> and <a href="outputfiles.html#N_XXX_EVEC.dat">N_XXX_EVEC.dat</a>)</i></li>
     <li>Average swap probabilities and swapping history for parallel runs with replica exchange<br/>
          <i>Available system-wide (see <a href="outputfiles.html#N_000_PROBABILITIES.dat">N_000_PROBABILITIES.dat</a> and <a href="outputfiles.html#N_000_REXTRACE.dat">N_000_REXTRACE.dat</a>)</i></li>
     <li>Reseeding history for parallel runs with PIGS<br/>
          <i>Available system-wide (see <a href="outputfiles.html#N_000_PIGSTRACE.dat">N_000_PIGSTRACE.dat</a>)</i></li>
     <li>Instantaneous (running output) of titration state of titratable residues<br/>
          <i>Available per titratable polypeptide residue (see <a href="outputfiles.html#PHTIT.dat">PHTIT.dat</a>)</i></li>
     <li>Instantaneous energies resolved by individual contributions to the Hamiltonian<br/>
          <i>Available system-wide (see <a href="outputfiles.html#ENERGY.dat">ENERGY.dat</a>)</i></li>
     <li>Instantaneous ensemble variables in dynamics calculations<br/>
          <i>Available system-wide in dynamics or hybrid calculations (see <a href="outputfiles.html#ENSEMBLE.dat">ENSEMBLE.dat</a>)</i></li>
     <li>Instantaneous totals for accepted Monte Carlo moves resolved by move type<br/>
          <i>Available system-wide in Monte Carlo calculations (see <a href="outputfiles.html#ACCEPTANCE.dat">ACCEPTANCE.dat</a>)</i></li>
     <li>Instantaneous values or Fourier terms for dihedral angles around rotatable bonds<br/>
          <i>Available system-wide or resolved by analysis group (see <a href="outputfiles.html#FYC.dat">FYC.dat</a> and <a href="outputfiles.html#TRCV_xxx.tmp">TRCV_xxx.tmp</a>)</i></li>
     <li>Extracted features for <a href="keywords.html#CMODE">structural clustering</a><br/>
          <i>Available system-wide (see <a href="outputfiles.html#CLUSTERING_FEATURES.nc">CLUSTERING_FEATURES.nc</a>)</i></li>
     <li>Results from <a href="keywords.html#CMODE">structural clustering</a> (post-processing)<br/>
          <i>Available system-wide (see <a href="outputfiles.html#STRUCT_CLUSTERING.clu">STRUCT_CLUSTERING.clu</a>, <a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a>, and <a href="outputfiles.html#TMAT_xxxxxx_yyy.dat">TMAT_xxxxxx_yyy.dat</a>)</i></li>
     <li>Results from the <a href="keywords.html#CPROGINDMODE">progress index method</a> (post-processing)<br/>
          <i>Available system-wide (see  <a href="outputfiles.html#PROGIDX_xxxxxxxxxxxx.dat">PROGIDX_xxxxxxxxxxxx.dat</a> and
<a href="outputfiles.html#FRAMES_NBL.nc">FRAMES_NBL.nc</a>)</i></li>
     <li>Network analyses derived from structural clustering (post-processing)<br/>
          <i>Available system-wide (see <a href="outputfiles.html#basename_yyy.fwts">{basename}_yyy.fwts</a>,
<a href="outputfiles.html#EIGEN_xxx.dat">EIGEN_xxx.dat</a>, <a href="outputfiles.html#PFOLD_PLUS_xxx.dat">PFOLD_PLUS_xxx.dat</a>,
<a href="outputfiles.html#PFOLD_MINUS_xxx.dat">PFOLD_MINUS_xxx.dat</a>,
<a href="outputfiles.html#MFPT_yyy_CFEP_xxxxxxxx.dat">MFPT_yyy_CFEP_xxxxxxxx.dat</a>, <a href="outputfiles.html#PFOLD_PLUS_yyy_CFEP_xxxxxxxx.dat">PFOLD_PLUS_yyy_CFEP_xxxxxxxx.dat</a>,
<a href="outputfiles.html#PFOLD_MINUS_yyy_CFEP_xxxxxxxx.dat">PFOLD_MINUS_yyy_CFEP_xxxxxxxx.dat</a>, and
<a href="outputfiles.html#MSM_SYN_TRAJ_xxxxx_yyy.frames">MSM_SYN_TRAJ_xxxxx_yyy.frames</a>)</i></li>
     <li>Principal component (PCA) analysis (post-processing)<br/>
          <i>Available system-wide (see <a href="outputfiles.html#PRINCIPAL_COMPONENTS.evs">PRINCIPAL_COMPONENTS.evs</a> and
<a href="outputfiles.html#PRINCIPAL_COMPONENTS.dat">PRINCIPAL_COMPONENTS.dat</a>)</i></li>
     <li>Histogram outputs for <a href="keywords.html#MC_ACCEPT">Wang-Landau sampling</a> runs<br/>
          <i>Available system-wide (see <a href="outputfiles.html#WANGLANDAU_G.dat">WANGLANDAU_G.dat</a>,
 <a href="outputfiles.html#WANGLANDAU_GH.dat">WANGLANDAU_GH.dat</a>, and
<a href="outputfiles.html#WL_TMP">temporary histogram output</a>)</i></li>
     <li>Weights for snapshot-based reweighting for <a href="keywords.html#SCULPT">energy landscape sculpting</a> runs<br/>
          <i>Available system-wide (see <a href="outputfiles.html#ELS_WFRAMES.dat">ELS_WFRAMES.dat</a>)</i></li>
    </ol>
  </li>
  <li><b>Other Files</b>
    <ol type='a'>
     <li>Visualization scripts (VMD)<br/>
          <i>Available system-wide (see <a href="outputfiles.html#basename_VIS.vmd">{basename}_VIS.vmd</a>, 
<a href="outputfiles.html#DIPOLE_GROUPS.vmd">DIPOLE_GROUPS.vmd</a>,
<a href="outputfiles.html#MONOPOLES.vmd">MONOPOLES.vmd</a>, <a href="outputfiles.html#FOS_GROUPS.vmd">FOS_GROUPS.vmd</a>, and
<a href="outputfiles.html#STRUCT_CLUSTERING.vmd">STRUCT_CLUSTERING.vmd</a>)</i></li>
     <li>Index lists<br/>
          <i>Available system-wide (see <a href="outputfiles.html#INTERNAL_COORDS.idx">INTERNAL_COORDS.idx</a>,
<a href="outputfiles.html#TABULATED_POT.idx">TABULATED_POT.idx</a>, and <a href="outputfiles.html#GENERAL_PC.idx">GENERAL_PC.idx</a>)</i></li>
     <li>Complete input file for <a href="keywords.html#SC_TOR">torsional bias potential</a> based on simplified input mode<br/>
          <i>Available system-wide (see <a href="outputfiles.html#SAMPLE_TORFILE.dat">SAMPLE_TORFILE.dat</a>)</i></li>
     <li>Transform of input spatial density<br/>
          <i>Available system-wide (see <a href="outputfiles.html#DENSITY_INPUT_PHYS.nc">DENSITY_INPUT_PHYS.nc</a>)</i></li>
     <li>Wisdom file for DDFTs by FFTW<br/>
          <i>Available system-wide (not documented further, see <a href="keywords.html#EWWISDOMFILE">FMCSC_EWWISDOMFILE</a>)</i></li>
     <li>Log output for shared memory parallelization<br/>
          <i>Available system-wide (see <a href="outputfiles.html#THREADS.log">THREADS.log</a>)</i></li>
     <li>High precision simulation restart files<br/>
          <i>Available system-wide (see <a href="outputfiles.html#basename_1.rst">{basename}_1.rst</a> and
<a href="outputfiles.html#basename_2.rst">{basename}_2.rst</a>)</i></li>
     <li>Initial and final structure files<br/>
          <i>Available system-wide (see <a href="outputfiles.html#basename_START.int">{basename}_START.int</a>,
<a href="outputfiles.html#basename_START.pdb">{basename}_START.pdb</a>,
<a href="outputfiles.html#basename_END.int">{basename}_END.int</a>, and
<a href="outputfiles.html#basename_END.pdb">{basename}_END.pdb</a>)</i></li>
    </ol>
  </li>
</ol>
<h3><br/></h3>
<h3>Individual Descriptions of Output Files:<br/></h3>
<hr/>
<h4><a class="NoHi" id="COMMENTS"><b>GENERAL COMMENTS:</b></a></h4>
The output that CAMPARI generates is fit into different hierarchical constructs. For instance,
<a href="outputfiles.html#Trajectory_output">trajectory information</a> is instantaneous and provided for each atom in the system individually.
Analyses which are performed on-the-fly (<i>i.e.</i>, concurrently with a running simulation) will often pool
data for cases in which the tagging of individual particles of identical type would be statistically meaningless
due to them being freely permutable. The user may control this behavior primarily through the concept of <a href="inputfiles.html#FMCSC_ANGRPFILE">analysis groups</a>.
Furthermore, not all output is meaningful (and may be expensive to obtain) for small molecules which may be considered solvent rather than solutes. Some analyses respect tagging specific analysis groups as solvent and others as solutes. Below, the descriptions of output
files is given in no particular order except that closely related output files will appear immediately after one
another. Is is recommended to navigate back to the top of the page and use the links in the
<a href="outputfiles.html#TOP">overview section</a>.<br/>
<br/>
<h4><b>NOTE ON UNITS:</b></h4>
The implied units should be stated explicitly for most output files. To minimize errors in human porting
and adapting of parameters, and to comply with established standards (such as the PDB-standard),
simulation software will often try to work in units that yield numbers that do not require very large or
very small exponents. CAMPARI's internal units are kcal/mol for energy, &#8491; for length, degrees
for angles (although occasionally radian are used), the elementary charge for charges, K for temperature,
g/mol for mass, and ps for time.<br/>
<br/>  
<h4><b>IMPORTANT COMMENT FOR PARALLEL RUNS:</b></h4>
The inner level of parallelism in CAMPARI, <i>i.e.</i>,
the <a href="keywords.html#NRTHREADS">shared memory (OpenMP) parallelization</a> produces
only one <a href="outputfiles.html#THREADS.log">output file that is specific to it</a>. Everything
else should look exactly the same as it would in a run with OpenMP disabled.
The outer layer of parallelism in CAMPARI is the option to simulate or analyze several copies
of the system, which is driven by MPI. In those multi-replica <a href="keywords.html#MPI_settings">parallel computations</a>,
in which the information for each node is
<a href="keywords.html#REMC">collected separately</a>, all output files will have the node specific
prefix "N_xxx_", where xxx is the node's rank (integer with leading
zeros). This allows a single working directory for the whole job
(presumably then running on a network-mounted directory, although this is
of course not necessary). In addition "N_xxx.log" is an automatic
log-file (created only in MPI runs) to avoid congestion of
the standard output stream used in serial runs. In multi-replica parallel computations, in which the information coming
from the nodes is <a href="keywords.html#MPIAVG">centrally collected</a>, only a few files have the same
prefix (the log-file, the restart file, as well as the initial and final
.pdb/.int-files). Note that the prefix always uses at least 3 digits to represent replica numbers
but that more digits are used if necessary. For all other files see comments below.<br/>
<br/>
<h4><b>IMPORTANT COMMENT FOR DYNAMICS RUNS:</b></h4>
In runs using a <a href="keywords.html#DYNAMICS">dynamics integrator</a> for sampling, some of the output is
shifted by one step relative to other quantities. The reason is the
following: In a single integration step, the energy of the
current structure is computed first. The structure is then advanced as
guided by the acting accelerations and velocities for individual
degrees of freedom. At the end of the step all analysis routines are
performed on the new structural information. However, potential
energies pertaining to the previous structure are printed out leading
to a mismatch of one step (this is in addition to the standard leapfrog mismatch of velocity- and structure-dependent
quantities).
<br/>
<br/>
<hr/>
<br/>
<br/>
<a class="NoHi" id="THREADS.log"><b>THREADS.log</b></a><br/>
<br/>
If the <a href="keywords.html#NRTHREADS">shared memory (OpenMP) parallelization</a>
of CAMPARI is in use, this file can contain information about how the total work load is distributed across
threads. The amount of information provided is controlled by keyword
<a href="keywords.html#THREADS_VERBOSE">FMCSC_THREADS_VERBOSE</a>. In general,
the output can contain the following pieces of information:
<ol>
<li>Timing information and expected finishing time (this is usually misleading in <a href="keywords.html#DYNAMICS">hybrid runs</a>). The setting for <a href="keywords.html#FLUSHTIME">FMCSC_FLUSHTIME</a> determines how often this is reported.</li>
<li>Fixed bounds (atoms, residues, molecules) for work load distribution across threads.</li>
<li>Final dynamic bounds after load balancing for work load distribution across threads. Settings for keywords <a href="keywords.html#THREADS_DLB_FREQ">FMCSC_THREADS_DLB_FREQ</a>, <a href="keywords.html#THREADS_DLB_STOP">FMCSC_THREADS_DLB_STOP</a>, and
<a href="keywords.html#THREADS_DLB_EXT">FMCSC_THREADS_DLB_EXT</a> matter for this output.</li>
<li>Imbalance measures for dynamic load balancing (see previous item for associated keywords).</li>
</ol>
Most of the detailed information is only parsable by and meant for developers.<br/>
<br/>
In MPI calculations, this file is produced independently by all replicas using the standard prefix if the calculation
uses the hybrid OpenMP/MPI-parallel executable. Note that timing information and load balancing are obtained independently
for all MPI processes and do not account for possible communication or synchronization delays incurred by MPI.<br/>
<br/>
<br/>
<a class="NoHi" id="basename_1.rst"><b>{basename}_1.rst</b></a><br/>
<a class="NoHi" id="basename_2.rst"><b>{basename}_2.rst</b></a><br/>
<br/>
The restart file: it is an ASCII-file containing high precision
information on the system. To be properly read, it requires the exact
same sequence file the original run was run with (see <a
 href="keywords.html#RESTART">FMCSC_RESTART</a>). The format assumptions
the code makes are very strict, so segmentation faults due to formatted
read are not uncommon in case the user creates a mismatch between
restart file and attempted calculation (<i>i.e.</i>, changes the sequence).<br/>
Note that the two files are (over)written in alternating fashion. This is to avoid 
cases where the restart file is ultimately destroyed due to a system error
killing the simulation during the writing step. For read-in purposes, the appropriate
version of the file needs to be renamed to {basename}.rst. The units are the CAMPARI internal ones (see above).<br/>
&nbsp;<br/>
<br/>
<a class="NoHi" id="basename_START.int"><b>{basename}_START.int</b></a><br/>
<a class="NoHi" id="basename_START.pdb"><b>{basename}_START.pdb</b></a><br/>
<br/>
A Cartesian coordinate file in pdb-format and a Z-matrix file in
TINKER-format of the initial
configuration as built by CAMPARI (random, read-in or
regular-extended),
mostly for validation purposes. Note that the output pdb file can be adjusted 
in formatting by keywords <a href="keywords.html#PDB_NUCMODE">FMCSC_PDB_NUCMODE</a> (for
atom partitioning in DNA/RNA),
<a href="keywords.html#PDB_W_CONV">FMCSC_PDB_W_CONV</a> (for
atom and residue names and treatment of polypeptide caps), and 
<a href="keywords.html#PDB_OUTPUTSTRING">FMCSC_PDB_OUTPUTSTRING</a>
(for field widths). Modifications to the latter will generally make the resultant
pdb files incompatible with visualization software or other tools.<br/>
A few notes on the Z-matrix format:
<ol type='a'>
<li>The header line gives the number of atoms found subsequently.</li>
<li>The first column runs through the atom indices.</li>
<li>The last seven columns give reference indices (always backward) and Z-matrix variables (bond length,
bond angle, dihedral angle or second bond angle), and the chirality indicator.</li>
<li>The data in between (two or three columns) are an element identifier, an (optional)
naming aid for annotated output, and the atom type as defined in the 
<a href="parameters.html#S2_LJ-types">parameter file</a> in use (see keyword <a href="keywords.html#PARAMETERS">PARAMETERS</a>).</li>
<li>If the system contains multiple molecules, the Z-matrix is broken into pieces,
<i>i.e.</i>, the first three atoms of every(!) molecule do not have the full
number of reference atoms. Note that this means that these .int-files
do not fully describe the system (as the rigid-body coordinates of the
individual molecules are not contained).</li>
</ol><br/>
<br/>
<a class="NoHi" id="basename_END.int"><b>{basename}_END.int</b></a><br/>
<a class="NoHi" id="basename_END.pdb"><b>{basename}_END.pdb</b></a><br/>
<br/>
Both a Z-matrix file and a Cartesian file in pdb-format for the final
configuration obtained
through the simulation. This is again mainly for validation purposes.
Note, however
that both {basename}_END.pdb and <a href="outputfiles.html#basename_START.pdb">{basename}_START.pdb</a> are used in
<a href="outputfiles.html#basename_VIS.vmd">{basename}_VIS.vmd</a> and are expected to be
pdb files conforming to the standard, <i>i.e.</i>,
the script file will not work properly if both of these files are
absent or if the pdb <a href="keywords.html#FMCSC_PDB_OUTPUTSTRING">output format</a> was altered. The
pdb files provide the template for reading in the trajectory file. Of course,
<a href="outputfiles.html#basename_VIS.vmd">{basename}_VIS.vmd</a>
can be manually edited to use an alternative template file provided
independently. See <a href="outputfiles.html#basename_START.int">above</a>
for notes on the Z-matrix format.<br/>
<br/>
<br/>
<a class="NoHi" id="Trajectory_output"><b>Trajectory output</b></a><br/><br/>
Note that all trajectory output (whichever format it is obtained in) can be comfortably read in
and visualized by the (free) visualization software VMD utilizing the automatically
generated scripts the code provides &rarr; <a href="outputfiles.html#basename_VIS.vmd">{basename}_VIS.vmd</a>.
In all cases, the keyword <a href="keywords.html#XYZOUT">FMCSC_XYZOUT</a> controls writing
frequency and the assumed unit of length is &#8491;.<br/>
<br/>
Also note that in <a href="keywords.html#MPIAVG">MPI averaging</a> mode, trajectory output may cycle through the
nodes, <i>i.e.</i>, if the frequency is set to 10, and the universe is of size
4, then after steps 10, 50, 90, ... a file or entry is written based on the
structure the first node (master) has, after 20, 60, 100, ... steps a
file or entry is written based on the second node (first slave) and so on. This mode
is selected by setting keyword <a href="keywords.html#MPIAVG_XYZ">FMCSC_MPIAVG_XYZ</a> to zero.
The advantage is that data spanning the entire parallel collection of individual simulations are obtained
in a single file with the downside of some logistic overhead associated with re-extracting
individual and continuous trajectories from said file. If <a href="keywords.html#MPIAVG_XYZ">FMCSC_MPIAVG_XYZ</a>
is not set to zero explicitly, each replica will produce its own trajectory file as is always the
case in <a href="keywords.html#REMC">replica exchange</a> (RE) runs. The cyclical output described above applies
universally to most other output files producing instantaneous output (such as <a href="outputfiles.html#ENERGY.dat">ENERGY.dat</a>) in MPI averaging calculations.<br/>
<br/>
The general rule for trajectory output is that the entirety of the system is stored. In case of
<a href="keywords.html#BOUNDARY">periodic boundary conditions</a>, the convention is that the geometric centers (or centers of mass)
of all molecules always reside inside the central unit cell. Most of this behavior can be controlled by auxiliary keywords 
<a href="keywords.html#XYZ_SOLVENT">FMCSC_XYZ_SOLVENT</a>, <a href="keywords.html#XYZ_FORCEBOX">FMCSC_XYZ_FORCEBOX</a>,
<a href="keywords.html#XYZ_REFMOL">FMCSC_XYZ_REFMOL</a>, <a href="keywords.html#ALIGNCALC">FMCSC_ALIGNCALC</a>, 
and <a href="keywords.html#TRAJIDXFILE">FMCSC_TRAJIDXFILE</a>.<br/>
<br/>
Furthermore, in <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>, trajectories that are being rewritten (for example to change format, or to extract specific frames) do not represent the supplied weights in any way.<br/>
<br/>
Lastly note that in (S)GMC calculations physical presence is indicated by coordinate offsets in the z-direction (for output only). Since the reserve particle bath molecules exist throughout and since very few visualization programs deal with changing topologies from snapshot to snapshot, CAMPARI provides the (mostly meaningless) coordinates of bath molecules misplaced by one or several box sizes in the positive z-direction. Such trajectories may be visualized using the automatic scripts and allow for easy and intuitive visual analysis of (S)GMC runs (see <a href="outputfiles.html#basename_VIS.vmd">{basename}_VIS.vmd</a>).<br/>
<br/>
<a class="NoHi" id="basename_xxxxx.arc"><b>{basename}_xxxxx.arc</b></a><br/>
<br/>
These are TINKER format snapshots continuously numbered in xxxxx which
are readable by appropriate visualization software like VMD.<br/>
<br/>
<a class="NoHi" id="basename_xxxxx.pdb"><b>{basename}_xxxxx.pdb</b></a><br/>
<br/>
These are pdb-snapshots just like xyz-ones. PDB-files have a rigid (and unfortunately somewhat flawed)
format but are heavily annotated, <i>i.e.</i>, allow for successful parsing even if 
atoms (such as hydrogens) are missing or if atom orders are different between different software
packages. For larger software cross-compatibility, the keywords <a href="keywords.html#PDB_W_CONV">FMCSC_PDB_W_CONV</a> and
<a href="keywords.html#PDB_NUCMODE">FMCSC_PDB_NUCMODE</a> can be used to determine some details of how pdb-files are annotated.
Conversely, keyword <a href="keywords.html#PDB_OUTPUTSTRING">FMCSC_PDB_OUTPUTSTRING</a> can be used to create
a non-standard pdb format that corrects the aforementioned flaws such as the limited field widths for 
coordinate precision or atom index.<br/>
<br/>
<a class="NoHi" id="basename_traj.pdb"><b>{basename}_traj.pdb</b></a><br/>
<br/>
<a href="outputfiles.html#basename_xxxxx.pdb">Single files</a> are tedious but the pdb-format does not possess a natural facility to represent
trajectories. The most common workaround is to annotate individual snapshots as different models (using MODEL/ENDMDL syntax) as is done for structural ensembles determined by NMR. This is also the solution chosen here. Note that the file size of
pdb- and xyz-files is generally large due to the ASCII-format and the annotations.<br/>
<br/>
<a class="NoHi" id="basename_traj.xtc"><b>{basename}_traj.xtc</b></a><br/>
<br/>
The code allows to output the coordinates in three different binary formats, .xtc (employed for example by GROMACS), .dcd (employed for example by CHARMM and NAMD) and the general binary NetCDF format (.nc) developed by Unidata and employed
for example by AMBER since version 9 &rarr; <a href="keywords.html#XYZPDB">FMCSC_XYZPDB</a>. The xtc-format is most compact but requires linking to an external library. Similarly, the NetCDF format is most versatile but also requires linking against a (much more) extensive library (see <a href="install.html">INSTALL</a>).<br/>
<br/>
<a class="NoHi" id="basename_traj.dcd"><b>{basename}_traj.dcd</b></a><br/>
<br/>
See <a href="outputfiles.html#basename_traj.xtc">{basename}_traj.xtc</a>. The binary .dcd trajectory file.<br/>
<br/>
<a class="NoHi" id="basename_traj.nc"><b>{basename}_traj.nc</b></a><br/>
<br/>
See <a href="outputfiles.html#basename_traj.xtc">{basename}_traj.xtc</a>. The binary .nc trajectory file.<br/>
<br/>
<br/>
<a class="NoHi" id="basename_VIS.vmd"><b>{basename}_VIS.vmd</b></a><br/>
<br/>
This output file is a TCL script for the freely available and widely
used <a href="references.html#ref23_9">molecular visualization software VMD</a>. It allows a
quick visualization of the trajectory and can be easily
customized to allow variations on the defaults. On a LINUX command line
it is run as "vmd -e {basename}_VIS.vmd". It offers the following
functionality:<br/>
<ol>
<li>Automatic
loading of the trajectory (whichever format) and setup of typical
representations<br/></li>
<li>Visualization of boundary graphics (a
transparent sphere/cylinder or lines for rectangular cuboids, available
dependent on setting for keyword <a href="keywords.html#FMCSC_SHAP">FMCSC_SHAPE</a>):<br/>
 &rarr; droplets: "see_box radius o1 o2 o3 res."
Here, radius is the sphere's radius, the vector o1,o2,o3 defines the
center, and res is the resolution of display (proportional to the
number of polygons used for rendering).<br/>
 &rarr; boxes: "see_box l1 l2 l3 a1 a2 a3 o1 o2 o3."
Here, l1,l2,l3 is the vector of unit cell side lengths, a1,a2,a3 is the
vector of unit cell angles, and o1,o2,o3 is the vector pointing to the
lower left corner.<br/>
 &rarr; cylinders: "see_box radius height o1 o2 o3 res."
Here, radius and height are the cylinder's radius and height, the 
vector o1,o2,o3 defines the vector pointing to the center of the bottom
circular cross-section, and res is the resolution of display (proportional to the
number of polygons used for rendering).<br/>
</li>
<li>Automatic visualization of simulation boundary:<br/>
 &rarr; NVT/NVE simulations: "see_system molecule",
"hide_system molecule." Here, molecule is the ID number of the molecule
(typically 0). Note that the dimensions of the system are written to
the script-file directly, and not pulled from the trajectory file
itself.<br/>
 &rarr; NPT/NPH simulations: "see_system molecule",
"hide_system molecule" Analogous, only that the dimensions of the
system are pulled from the trajectory file itself and vary from frame
to frame (not that this is not available for xtc-trajectories as the
VMD plugin currently does not re-read unit cell variables for every
frame).<br/>
 &rarr; &#956;VT simulations: "see_system molecule",
"see_bath molecule", "see_all molecule", hide_system molecule", ... In
case of grand canonical or semi-grand canonical simulations, the bath
particles are output into the trajectory file, but shifted in the
positive z-direction by two periodic images (two diameters in case of
droplets). The functions above will selectively activate or deactivate
representations associated with either the bath or the (real) system.
In contrast to fixed N-ensembles, the visualization of system
boundaries is turned on by default. This is to make the concurrent
output of bath and real system less confusing.<br/></li>
<li>Simplified alignment procedure:<br/>
 &rarr; "align_it selection_string reference_molecule
reference_frame molecule." Here, the selection_string defines a
VMD-style selection of atoms (as you would use for a representation),
reference_molecule and reference_frame are the reference IDs for the
molecule and frame to serve as the alignment target, and molecule is
the ID number of the molecule to be aligned (and which can of course be
identical to reference_molecule). Note that the alignment procedure
shifts and rotates the actual coordinates thereby rendering the display
of system boundaries meaningless (as those are defined in absolute
coordinates).<br/></li>
<li>Automated RMSD calculations:<br/>
 &rarr; "rmsd_it filename selection_string
reference_molecule reference_frame molecule." This is very similar to
the alignment routine. The resultant RMSD for all frames of the chosen
molecule will be written to the file "filename". Note that RMSDs are
extensive quantities which scale with the size of the chosen reference.
see above for compatibility with system boundary display. All the above
functions are TCL procedures defined in this output file, and therefore
available upon parsing the script. They are used by simply typing
the desired command in the VMD console window. All options to the
procedures have defaults, so it might not always be necessary to supply
every one of them.<br/></li>
</ol>
<br/>
<br/>
<a class="NoHi" id="RMSD.dat"><b>RMSD.dat</b></a><br/>
<br/>
Instantaneous values of the coordinate root mean square deviation (RMSD) after alignment in &#8491;<br/>
<br/>
Format: f13.5 or f13.5,1x,f13.5<br/>
<br/>
There are either one ot two columns, which hold coordinate RMSD values computed after alignment. The alignment set  
is a <a href="keywords.html#ALIGNFILE">user-defined subset</a>. The first column is always the RMSD computed only over the
set of atoms used for alignment. As by standard convention, normalization of the MSD is with respect to the number of atoms 
in the set. The reference coordinates are either a <a href="keywords.html#PDB_TEMPLATE">user-supplied reference structure</a>
or the preceding trajectory snapshot <a href="keywords.html#ALIGNCALC">being considered for this purpose</a>. The second
column is optional and written only if the user also requested an <a href="keywords.html#CFILE">additional set of atoms</a>.
In this case, the value in the second column is the RMSD across this additional set of atoms given the same alignment. 
A separate distance set, which should of course differ from the alignment set, can be used, for example, to understand the extent of
relativ motion between a structurally well-defined reference (<i>e.g.</i>, a protein) and a coupled entity
(<i>e.g.</i>, a bound ligand).<br/>
This functionality is only available in <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a>.
Note that the functions defined in the <a href="outputfiles.html#basename_VIS.vmd">visualization scripts</a>
offer easier control over the coordinate subsets to use.<br/>
<br/>
When using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this instantaneous output file does not utilize the weights in any way.<br/>
<br/>
<br/>
<a class="NoHi" id="ENERGY.dat"><b>ENERGY.dat</b></a><br/>
<br/>
Instantaneous values of individual energy terms in kcal/mol<br/>
<br/>
Format: i12,1x,22(g14.7,1x)<br/>
<br/>
Columns are as follows:
<ol>
<li>Step number</li>
<li>Total energy</li>
<li><a href="keywords.html#SC_IPP">Excluded volume potential</a></li>
<li><a href="keywords.html#SC_EXTRA">Structural correction terms</a></li>
<li><a href="keywords.html#SC_ATTLJ">Dispersive interaction term</a></li>
<li><a href="keywords.html#SC_IMPSOLV">ABSINTH direct mean-field interaction</a></li>
<li><a href="keywords.html#SC_WCA">Weeks-Chandler-Andersen potential</a></li>
<li><a href="keywords.html#SC_POLAR">Polar (electrostatic) potential terms</a></li>
<li><a href="keywords.html#SC_TOR">Torsional bias terms</a></li>
<li><a href="keywords.html#SC_ZSEC">Global secondary structure bias terms</a></li>
<li><a href="keywords.html#SC_TABUL">Tabulated potentials</a></li>
<li><a href="keywords.html#SC_DREST">Harmonic distance restraint terms</a></li>
<li><a href="keywords.html#SC_OSMO">Compartmentalization potential terms</a></li>
<li><a href="keywords.html#SOFTWALL">Soft-wall boundary potential</a></li>
<li><a href="keywords.html#SCULPT">Sum of all bias terms incurred by energy landscape sculpting</a></li>
<li><a href="keywords.html#SC_POLY">Polymeric biasing term</a></li>
<li><a href="keywords.html#SC_BONDED_B">Bond stretching term</a></li>
<li><a href="keywords.html#SC_BONDED_A">Angle bending term</a></li>
<li><a href="keywords.html#SC_BONDED_I">Improper dihedral term</a></li>
<li><a href="keywords.html#SC_BONDED_T">Torsional term</a></li>
<li><a href="keywords.html#SC_DSSP">DSSP restraint term</a></li>
<li><a href="keywords.html#SC_BONDED_M">CMAP term</a></li>
<li><a href="keywords.html#SC_EMICRO">Spatial density restraint term</a></li>
</ol>
In MPI averaging mode these are always cycling instantaneous values (as
described for <a href="outputfiles.html#Trajectory_output">structural output</a>). It is also important to know that
this file gives values that no longer reflect the correct energy surface if the 
<a href="keywords.html#SCULPT">energy landscape sculpting</a> method is in use. Instead, the 
values in the file will always be the native ones (as if sculpting were turned off). The reason is that
the mapping from native to sculpted values is always possible (meaning that sculpted energies and
instantaneous weights can be easily calculated from the output in ENERGY.dat), but the reverse
is not. For added simplicity, the total bias terms are printed out in a dedicated column (#15). This number
is <b>not</b> included in the total reported in column #2. It is also directly related to the output in
<a href="outputfiles.html#ELS_WFRAMES.dat">ELS_WFRAMES.dat</a>.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this instantaneous output file does not utilize the weights in any way.<br/>
<br/>
In (S)GMC calculations, energy outputs represent combined totals from both present and non-present molecules.<br/>
<br/>
<br/>
<a class="NoHi" id="ENSEMBLE.dat"><b>ENSEMBLE.dat</b></a><br/>
<br/>
Instantaneous values of various global ensemble metrics <br/>
<br/>
Format: i12,1x,X(g14.7,1x)<br/>
<br/>
What is written out depends on the ensemble that is simulated (see <a
 href="keywords.html#ENSEMBLE">FMCSC_ENSEMBLE</a>). Currently, only the NVT
and NVE ensembles are supported, for which the following instantaneous
quantities are printed (the last two columns are restricted to
<a href="keywords.html#CARTINT">internal coordinate space</a> dynamics):<br/>
<br/>
Step number, potential energy, kinetic energy, total energy (kinetic +
potential), temperature, Gauss measure, (Cartesian kinetic energy, sum of potential energy and
Cartesian kinetic energy)<br/>
<br/>
In general, leap-frog integrators as used in CAMPARI obtain velocities (kinetic energy,
temperature) and positions (potential energy) at points shifted by half the
integration time step. Therefore, in ENSEMBLE.dat, potential energies are usually lagging relative to all
other output, <i>i.e.</i>, they are for the previous structure (not the one written to the
trajectory-file) given the reported step number. The only exception to this are
<a href="keywords.html#REMC">replica exchange moves</a>. Here, when a new structure is swapped in, both the potential energy is updated to
the new structure, with its corresponding (now lagging) velocities. This leads to the
signature that consecutive steps have the exact same potential energy, since for the
next normal integration step the same (now lagging) potential energy is reported.
All energies are given in kcal/mol, and temperature is reported in K. The Gauss measure
is the sum over all atoms of the squares of all forces acting on that atom that are not due to
interaction potentials divided by the atom's mass:<br/>
Z = &Sigma; m<sub>i</sub> ( d<sup>2</sup>r<sub>i</sub> / dt<sup>2</sup> - f<sub>i</sub>/ m<sub>i</sub>)<sup>2</sup><br/>
Here, the r<sub>i</sub> and f<sub>i</sub> are the atom's position and "proper" force vectors.
Z measures a least squares difference between the unconstrained motion under the influence
of interaction forces and the actual motion constrained by ensemble and/or geometric considerations. Consequently,
it should only be zero for constraint-free simulations in the microcanonical ensemble. For cases with
any type of constraint the quantity is typically not easily interpretable since the value for Z scales
with the net strength of the "proper" forces acting on all particles. To reiterate, "proper" forces
are those accounted for by the potential energy, whereas constraint forces are introduced by
geometric constraints, thermostats, <i>etc.</i> and do not appear elsewhere. The value for Z is reported in 
gmol<sup>-1</sup>&#8491;<sup>2</sup>ps<sup>-4</sup>.<br/>
Aside from these general comments, dynamics runs in <a href="keywords.html#CARTINT">internal coordinate space</a>
offer two additional columns using the Cartesian kinetic energy. Predominantly, these runs report the kinetic
energy (column 3) in the assumed diagonal mass matrix approximation, <i>i.e.</i>, T = 0.5 &omega; M &omega;, where
M is the matrix with rotational inertia and net molecular masses for angular and linear degrees of freedom on its diagonal,
respectively, and &omega; is the composite vector of angular and linear velocities. For cases with a nondiagonal
mass matrix (<i>i.e.</i>, everything except rigid molecules with spherical symmetry), this is an approximation to the
kinetic energy computed as T = &Sigma; 0.5 m<sub>i</sub> v<sub>i</sub><sup>2</sup>. Therefore, the last two columns
give complementary information using these numbers. Note that the Cartesian velocities are not available directly,
but must either be obtained via finite differencing or are (as done here) computed by a recursive formalism
from the angular and linear velocities of the internal coordinate description.<br/>
<br/>
In MPI averaging mode these are always cycling instantaneous values (as
described for <a href="outputfiles.html#Trajectory_output">structural output</a>).<br/>
<br/>
Note that this output file is not written in any type of <a href="keywords.html#PDBANALYZE">trajectory analysis run</a>
or for runs that exclusively use <a href="keywords.html#DYNAMICS">Monte Carlo sampling</a>.<br/>
<br/>
<br/>
<a class="NoHi" id="ACCEPTANCE.dat"><b>ACCEPTANCE.dat</b></a><br/>
<br/>
Current totals of accepted Monte Carlo (MC) moves of different types<br/>
<br/>
Format: i12,1x,20(i11)<br/>
<br/>
Step number,
accepted <a href="keywords.html#COUPLE">FYC-moves</a>,
accepted <a href="keywords.html#PIVOTMODE">FY moves</a>,
accepted <a href="keywords.html#OMEGAFREQ">&#969;-moves</a>,
accepted <a href="keywords.html#TORCRFREQ">exact torsional CR moves without &#969;-sampling</a>,
accepted <a href="keywords.html#TORCROFREQ">exact torsional CR moves with &#969;-sampling</a>,
accepted <a href="keywords.html#CRFREQ">inexact torsional CR moves</a>,
accepted <a href="keywords.html#UJCRFREQ">exact torsional CR moves with bond angle sampling</a>,
accepted <a href="keywords.html#CHIFREQ">sidechain moves</a>,
accepted <a href="keywords.html#NUCFREQ">nucleic acid pivot moves</a>,
accepted five-ring pucker moves (<a href="keywords.html#SUGARFREQ">polynucleotides</a> and <a href="keywords.html#PKRFREQ">polypeptides</a>),
accepted <a href="keywords.html#NUCCRFREQ">exact torsional CR moves for polynucleotides</a>, 
accepted <a href="keywords.html#RIGIDFREQ">single molecule rigid-body moves</a>,
accepted <a href="keywords.html#ROTFREQ">single molecule rotations</a>,
accepted <a href="keywords.html#ROTFREQ">single molecule translations</a>,
accepted <a href="keywords.html#CLURBFREQ">cluster rigid-body moves</a>,
accepted molecule insertion moves, accepted molecule deletion moves, accepted molecule
type permutation moves (&rarr; <a href="keywords.html#PARTICLEFLUCFREQ">GC moves</a>),
accepted <a href="keywords.html#REFREQ">replica exchange moves</a>,
accepted <a href="keywords.html#OTHERFREQ">single torsion pivot moves (any type)</a><br/>
<br/>
The replica exchange moves are counted such that if the single replica
is getting a new structure at the end of the swap cycle, the counter is
incremented by one (<i>i.e.</i>, one is the maximum number of accepted RE moves
per RE cycle (whose frequency is determined by <a
 href="keywords.html#REFREQ">FMCSC_REFREQ</a>)). Note that for REMD
calculations
(see <a href="keywords.html#DYNAMICS">FMCSC_DYNAMICS</a>), accepted
replica exchange moves are only reported at the end in the log-file for
each node.<br/>
<br/>
In MPI averaging mode, these are always net totals over all nodes. This
is because the statistics for accepted moves do not observe well-defined fluctuations. Information
for individual nodes is partially available through the residue- and
molecule-resolved acceptance counts written to the log file.<br/>
<br/>
In (S)GMC calculations, information stemming from both present and nonpresent molecules is provided.<br/>
<br/>
Note that this output file is not written in any type of <a href="keywords.html#PDBANALYZE">trajectory analysis run</a> or
in runs that do not employ <a href="keywords.html#DYNAMICS">Monte Carlo sampling</a> at all.<br/>
<br/>
<br/>
<a class="NoHi" id="PARTICLENUMHIST.dat"><b>PARTICLENUMHIST.dat</b></a><br/>
<br/>
Histograms of particle counts for physically present molecules during runs in
<a href="keywords.html#ENSEMBLE">grand canonical or related ensembles</a><br/>
<br/>
Format: (N<sub>t</sub>+1)(i8)  with N<sub>t</sub> being the number of fluctuating particle types<br/>
<br/>
This output file lists number histograms in (S)GMC calculations for particle types allowed to fluctuate (<i>i.e.</i>, those coupled to the chemostat).
The first line is a header line with the integer indices of the molecule types reported upon in the respective columns (and "N" for the number column).
It has N<sub>max</sub>+2 rows where N<sub>max</sub> is the largest number of physically present molecules of any given type during the simulation.
The second row corresponds to the histogram count for no particles of that type being present and so on up to N<sub>max</sub> molecules being present. The actual particle number for each row is given in the first column.
Note that for a sane (S)GMC calculation, the limits N<sub>present</sub>&rarr;0 or N<sub>present</sub>&rarr;N<sub>reserve</sub> should not be encountered. Aside from log-outputs and eventual simulation terminations, this output file can help determine such exceptions.
These data are necessary to extract the sampled bulk concentration for each fluctuating type.<br/>
<br/>
<br/>

<a class="NoHi" id="SAMPLE_TORFILE.dat"><b>SAMPLE_TORFILE.dat</b></a><br/>
<br/>
Sample input file for <a href="keywords.html#SC_TOR">torsional bias potential</a><br/>
<br/>
Format: See <a href="inputfiles.html#FMCSC_TORFILE">description of input file format</a><br/>
<br/>
This output file is created if the <a href="keywords.html#SC_TOR">torsional bias potential</a> is in use and if
keyword <a href="keywords.html#TORREPORT">FMCSC_TORREPORT</a> is enabled. It then proceeds to write its 
processed input in a residue-based format to this output file. If the original input itself was also residue-based
using only the explicit modes (1 and 2), in which all parameters are specified explicitly, this is strictly redundant.
The use of this output file is to preserve the settings derived from a simplified input format for later use.
Specifically, if torsional bias minimum positions were inferred from <a href="keywords.html#PDBFILE">structural input</a>,
this information is typically lost later on, for example upon a restart of the simulation of upon changing the structural
input file. It is therefore <i>always</i> recommended to use this option to preserve the exact settings for the
torsional bias potential.<br/>
<br/>
In MPI calculations, this file is produced independently by all replicas using the standard prefix. This is normally redundant
but can be important when using keyword <a href="keywords.html#PDB_MPIMANY">FMCSC_PDB_MPIMANY</a>.<br/>
<br/>
<br/>

<a class="NoHi" id="POLYMER.dat"><b>POLYMER.dat</b></a><br/>
<br/>
Instantaneous measures of system size and shape<br/>
<br/>
Format: i12,2x,g14.7,1x,f12.5,1x,f12.5,1x,f12.5,1x,f12.5,1x,g14.7,1x,g14.7,1x,g14.7<br/>
<br/>
This file contains instantaneous system-wide polymer properties that
are primarily useful as an estimator for the quality of sampling: step
number, radius of gyration (R<sub>g</sub>) in &#8491;, t, asphericity,
acylindricity, &nbsp;&#948;<sup>inst</sup>, &#955;<sub>1</sub>, &#955;<sub>2</sub>, &#955;<sub>3</sub>. The &#955;<sub>i</sub> are the sorted (largest first) eigenvalues of the system's gyration tensor (in &#8491;<sup>2</sup>), for the rest see below. Note that values obtained in periodic boundary conditions may be misleading.<br/>
<br/>
Definitions:<br/>
&nbsp;t = &lt;f<sub>1</sub>&#183;(f<sub>2</sub>&#183;(R<sub>g</sub>/L<sub>c</sub>))<sup>f<sub>3</sub>/N<sup>0.33</sup></sup>&gt;<br/>
Here, the f<sub>i</sub> are arbitrary factors meant to keep the parameter within the interval [0,1] (values are 2.5, 1.75 and 4.0). N is the system's total number of residues and L<sub>c</sub> its estimated contour length in &#8491;.<br/>
&nbsp;Asphericity = (1/R<sub>g</sub><sup>2</sup>) &#183; [&#955;<sub>1</sub> - 0.5&#183;(&#955;<sub>2</sub> + &#955;<sub>3</sub>)]<br/>
&nbsp;Acylindricity = (1/R<sub>g</sub><sup>2</sup>) &#183; [&#955;<sub>2</sub> - &#955;<sub>3</sub>]<br/>
&nbsp;&#948;<sup>inst</sup> = 1.0 - 3.0&#183;(&#955;<sub>1</sub>&#183;&#955;<sub>2</sub> + &#955;<sub>2</sub>&#183;&#955;<sub>3</sub> + &#955;<sub>1</sub>&#183;&#955;<sub>3</sub>)/(&#955;<sub>1</sub> + &#955;<sub>2</sub> +
&#955;<sub>3</sub>)<sup>2</sup><br/>
<br/>
In MPI averaging these values come from a different node at each
step producing output, and they cycle in just the same fashion as described for
<a href="outputfiles.html#Trajectory_output">structural output</a> (<i>i.e.</i>, node 3 in an universe of size 5 could be found
in rows 3,8,13,18,....).<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this instantaneous output file does not utilize the weights in any way.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="POLYAVG.dat"><b>POLYAVG.dat</b></a><br/>
<br/>
Various averages quantifying size and shape of the molecules in each analysis group<br/>
<br/>
Format: 12(g14.7,1x)<br/>
<br/>
This file is resolved by analysis group and preceding the actual data
line there is a descriptive line giving the number of the analysis group
(as initially reported by CAMPARI), the reference molecule, and the
residues comprising that reference molecule.<br/>
<br/>
The data line gives the ensemble-averaged polymer properties for all the
molecules belonging to that analysis group. They are all locally (molecule-wise) defined
and hence true descriptors of sizes and shapes of the types of polymer
chains in the system: radius of gyration (&lt;R<sub>g</sub>&gt;) in &#8491;, &lt;R<sub>g</sub><sup>2</sup>&gt; in &#8491;<sup>2</sup>, t, asphericity,
acylindricity, &nbsp;&#948;<sup>*</sup>, &nbsp;&#948;, end-to-end distance in &#8491;, (estimated) hydrodynamic radius &lt;R<sub>h</sub>&gt; in &#8491;, &lt;&#955;<sub>1</sub>&gt;, &lt;&#955;<sub>2</sub>&gt;, &lt;&#955;<sub>3</sub>&gt;. For end-to-end distances see description in <a href="outputfiles.html#RETEHIST.dat">RETEHIST.dat</a>, the &#955;<sub>i</sub> are the sorted (largest first) eigenvalues of the gyration tensor (in &#8491;<sup>2</sup>), for the rest see below.
<br/>
Definitions:<br/>
&nbsp;t = &lt;f<sub>1</sub>&#183;(f<sub>2</sub>&#183;(R<sub>g</sub>/L<sub>c</sub>))<sup>f<sub>3</sub>/N<sup>0.33</sup></sup>&gt;<br/>
Here, the f<sub>i</sub> are arbitrary factors meant to keep the parameter within the interval [0,1] for a large variety of molecular sizes and shapes (values are 2.5, 1.75 and 4.0). N is the molecule's chain length in number of residues and L<sub>c</sub> its estimated contour length in &#8491;. The parameter t is meant to provide a crude "normalization" for R<sub>g</sub> which works across multiple polymeric domains.<br/>
&nbsp;Asphericity = &lt;[&#955;<sub>1</sub> - 0.5&#183;(&#955;<sub>2</sub> + &#955;<sub>3</sub>)]/&lt;R<sub>g</sub><sup>2</sup>&gt;&gt; = (1/&lt;R<sub>g</sub><sup>2</sup>&gt;)
&#183; [&#955;<sub>1</sub> - 0.5&#183;(&lt;&#955;<sub>2</sub>&gt; + &lt;&#955;<sub>3</sub>&gt;)]<br/>

&nbsp;Acylindricity = (1/&lt;R<sub>g</sub><sup>2</sup>&gt;) &#183; [&lt;&#955;<sub>2</sub>&gt; - &lt;&#955;<sub>3</sub>&gt;]<br/>
&nbsp;&#948; = 1.0 - 3.0&#183;&lt;&#955;<sub>1</sub>&#183;&#955;<sub>2</sub> + &#955;<sub>2</sub>&#183;&#955;<sub>3</sub> + &#955;<sub>1</sub>&#183;&#955;<sub>3</sub>&gt;/&lt;(&#955;<sub>1</sub> + &#955;<sub>2</sub> + &#955;<sub>3</sub>)<sup>2</sup>&gt;<br/>
&nbsp;&#948;<sup>*</sup> = 1.0 - 3.0&#183;&lt;(&#955;<sub>1</sub>&#183;&#955;<sub>2</sub> + &#955;<sub>2</sub>&#183;&#955;<sub>3</sub> + &#955;<sub>1</sub>&#183;&#955;<sub>3</sub>)/(&#955;<sub>1</sub> + &#955;<sub>2</sub> +
&#955;<sub>3</sub>)<sup>2</sup>&gt;<br/>
&nbsp;R<sub>h</sub> = (1.0/N<sub>d</sub>) &#183; &lt;1.0/&#931;<sub>i&lt;j</sub> r<sub>ij </sub>&gt;<sup>-1</sup><br/>
Here, N<sub>d</sub> is the total number of pairwise atom-atom distances in the molecule of interest.<br/> 
<br/>
In MPI averaging runs, these data are averaged over all nodes.<br/>
<br/>
In (S)GC ensemble calculations, the data are only obtained over physically present molecules.<br/>
<br/>
<br/>
<a class="NoHi" id="RETEHIST.dat"><b>RETEHIST.dat</b></a><br/>
<br/>
Histograms of end-to-end distances in &#8491; for different analysis groups<br/>
<br/>
Format: f12.5,1x,N(g14.7,1x), where N is the number of relevant
analysis groups in the system, <i>i.e.</i>, those which are composed of at
least two residues and have proper internal degrees of freedom.<br/>
<br/>
Row 1 is a header line, which has all the analysis group numbers (as initially reported by CAMPARI) in
columns 2 through N+1. The first column gives the bin-center for
the end-to-end to distance in &#8491; and the remaining columns give the
normalized counts for the N analysis groups. The resolution for this is
controlled by the keyword <a href="keywords.html#RGBINSIZE">FMCSC_RGBINSIZE</a>. For polypeptides, the exact
definition of the end is the CA-atom (or analog) on the terminal
residues, whereas for polynucleotides it is the P on the 5'-residue,
and the C3* on the 3'-terminal sugar.<br/>
<br/>
In MPI averaging this is collected over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="RGHIST.dat"><b>RGHIST.dat</b></a><br/>
<br/>
Histograms of absolute values of the radii of gyration in &#8491; for different analysis groups<br/>
<br/>
Format: f12.5,1x,N(g14.7,1x), where N is the number of relevant
analysis groups in the system, <i>i.e.</i>, those which have any internal
degrees of freedom.<br/>
<br/>
Row 1 is a header line, which has all the analysis group numbers (as initially reported by CAMPARI) in
columns 2 through N+1. The first column gives the bin centers in &#8491; for the
radius of gyration axis and the remaining columns give the normalized counts for the N
analysis groups. The resolution for this is controlled by the keyword <a href="keywords.html#RGBINSIZE">FMCSC_RGBINSIZE</a>.<br/>
<br/>
In MPI averaging this is collected over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="RDHIST.dat"><b>RDHIST.dat</b></a><br/>
<br/>
Two-dimensional histograms of the parameter t (related to radius of gyration) versus asphericity &#948;<sup>*</sup>
(see <a href="outputfiles.html#POLYAVG.dat">POLYAVG.dat</a> for details and definitions).
These data are resolved by analysis group.<br/>
<br/>
Due to the (quasi-)normalized nature of both order parameters, these matrices
are of fixed size, <i>i.e.</i>, 100 lines of 100(g14.7,1x), and are separated
by comment lines giving information about the analysis group the matrix
below refers to.<br/>
<br/>
The size parameter t varies with columns, while &#948;<sup>*</sup> changes with rows. Both
axes have values 0.005:0.015,....0.995 as bin centers. The data are
normalized counts over the whole two-dimensional space. The computation frequency
for this is regulated by <a href="keywords.html#POLCALC">FMCSC_POLCALC</a>.<br/>
<br/>
In MPI averaging mode this is averaged automatically over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="DENSPROF.dat"><b>DENSPROF.dat</b></a><br/>
<br/>
The average chain density as a function of distance from the center of
mass (radial density profile)<br/>
<br/>
Format: f12.5,1x,N(g14.7,1x), where N is the number of relevant
analysis groups in the system (<i>i.e.</i>, those which have internal degrees of
freedom).<br/>
<br/>
Row 1 is a header line, which has all the molecule type numbers in
columns 2 through N+1.<br/>
<br/>
The first column gives the bin centers for the distance from the center
of mass and the remaining columns give the averaged densities (in g&#183;cm<sup>-3</sup>)
for the N analysis groups. The resolution for this is currently
controlled by the keyword <a href="keywords.html#RGBINSIZE">FMCSC_RGBINSIZE</a>
(sub-optimal, might change in the future).<br/>
<br/>
In MPI averaging this is collected over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="RAMACHANDRAN.dat"><b>RAMACHANDRAN.dat</b></a><br/>
<br/>
(N+1) rows giving the system-wide, normalized, two-dimensional histogram of polypeptide backbone &phi;/&psi;-angles<br/>
<br/>
Format: (N+1)(g12.5,1x), where N = 360.0&deg;/<a href="keywords.html#ANGRES">FMCSC_ANGRES</a> (finest possible resolution is at
1.0&deg;)<br/>
<br/>
This is the two-dimensional backbone dihedral distribution averaged over all
residues in the system which have peptide backbone torsions (note that
this automatically excludes all terminal residues). Note that if 360.0&deg; is an exact multiple of <a href="keywords.html#ANGRES">FMCSC_ANGRES</a> an extra (unused) bin is produced (hence N+1), and that in case 360.0&deg; is not an exact multiple of 
<a href="keywords.html#ANGRES">FMCSC_ANGRES</a> bin-sizes at the margins may be altered.<br/>
<br/>
In MPI averaging mode this is collected over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<a class="NoHi" id="RESRAMA_xxxxx.dat"><b>RESRAMA_xxxxx.dat</b></a><br/>
<br/>
Just like <a href="outputfiles.html#RAMACHANDRAN.dat">RAMACHANDRAN.dat</a>,
only for the specific residue xxxxx.<br/>
<br/>
<a class="NoHi" id="MOLRAMA_xxxxx.dat"><b>MOLRAMA_xxxxx.dat</b></a><br/>
<br/>
Just like <a href="outputfiles.html#RAMACHANDRAN.dat">RAMACHANDRAN.dat</a>,
only for the specific analysis group xxxxx (as initially reported by CAMPARI).<br/>
<br/>
<br/>
<a class="NoHi" id="INTERNAL_COORDS.idx"><b>INTERNAL_COORDS.idx</b></a><br/>
<br/>
A list of entries in four blocks (although not all blocks will always
be written: see <a href="keywords.html#WHICHINT">FMCSC_WHICHINT</a>). If all four are present,
the first block corresponds to bond lengths, the second one to bond
angles, the third to improper dihedral angles, and the fourth to proper dihedral angles. <br/>
<br/>
First block (format: i6,':',1x,2(a4,' '),1x,'(',2(i7,' '),'): ',g12.5).
<br/>
The first integer is the column this entry can be found in the output
file <a href="outputfiles.html#INTHISTS_BL.dat">INTHISTS_BL.dat</a>.
The two character strings give the biotype names of the two atoms
forming the bond, the two following integers their atom number, and the
last number the starting value in &#8491; (which is highly relevant for
interpreting the output in <a href="outputfiles.html#INTHISTS_BL.dat">INTHISTS_BL.dat</a>).<br/>
<br/>
The second block (format: i6,':',1x,3(a4,' '),1x,'(',3(i7,' '),'):
',g12.5). <br/>
This is analogous to the first block, only that three biotype names and
three atom numbers are given. The central atom is always the central
one in forming the bond angle between the three. The starting (default)
value for the angle is given in degrees ([0:180]).<br/>
<br/>
Third block (format: i6,':',1x,4(a4,' '),1x,'(',4(i7,' '),'):
',g12.5).&nbsp; <br/>
Again analogous, only that four biotype names and atom numbers are
given. The first atom listed is always the one at the planar center
(typically an sp<sup>2</sup> carbon). The second one is the one defining the axis
around which the pseudo-(=improper)torsion is set up. The rules for
defining the hierarchy of the three covalently bound atoms are outlined
elsewhere (see <a href="parameters.html#S9_Bonded_types">corresponding documentation</a>). Improper dihedral angles are at
least conceptually circular quantities defined in CAMPARI over the
interval [-180:180]. In theory they can also be used to describe
chirality at tetrahedral centers as well, but CAMPARI does not
currently support that. Therefore, the starting value for most
improper dihedral angles will typically be 180.0&deg; or -180.0&deg; (depending on
rounding).<br/>
<br/>
Fourth block (format: i6,':',1x,4(a4,' '),1x,'(',4(i7,' '),')') <br/>
The only formal difference to the section on improper dihedral angles is the missing
starting value (see <a href="outputfiles.html#INTHISTS_DI.dat">INTHISTS_DI.dat</a> for details). For proper dihedral angles,
atoms are listed in order, <i>i.e.</i>, atoms 2 and 3 define the rotatable
bond, around which the angle of atom 4 with respect to atom 1 is
evaluated. Torsional angles are circular quantities defined in CAMPARI
over the interval [-180:180]&deg;.<br/>
<br/>
<br/>
<a class="NoHi" id="INTHISTS_BL.dat"><b>INTHISTS_BL.dat</b></a><br/>
<br/>
Bond length histograms (&#8491;).<br/>
<br/>
Format: X(g12.5) where X is the number of columns which can be gathered from <a
 href="outputfiles.html#INTERNAL_COORDS.idx">INTERNAL_COORDS.idx</a> and each column corresponds
to a topologically defined bond length<br/>
<br/>
Each histogram is setup in the following way. There are 101 total bins (rows)
with a (currently fixed and hard-coded) bin-size of 0.01&#8491;. The 51<sup>st</sup> bin
extends from (r<sub>0</sub> - 0.005)&#8491; to (r<sub>0</sub> + 0.005)&#8491;. Here, r<sub>0</sub> is the starting
value for that particular bond length term (as found in <a
 href="outputfiles.html#INTERNAL_COORDS.idx">INTERNAL_COORDS.idx</a>).
This setup is to ensure that different bond lengths (generally ranging
from 1-2&#8491; for biomolecules) are adequately supported without
sacrificing detailed resolution, as fluctuations in bond lengths are
generally expected to be relatively small. In case of out-of-range
values, terminal bins (1 or 101) are overstocked. This allows to at
least recognize that range exceptions have occurred.<br/>
<br/>
In MPI averaging mode this is collected over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present molecules are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="INTHISTS_BA.dat"><b>INTHISTS_BA.dat</b></a><br/>
<br/>
Bond angle histograms (&deg;). <br/>
<br/>
Format: X(g12.5) where X is the number of columns which can be gathered from <a
 href="outputfiles.html#INTERNAL_COORDS.idx">INTERNAL_COORDS.idx</a>  and each column corresponds
to a topologically defined bond angle<br/>
<br/>
This output file is highly analogous to <a
 href="outputfiles.html#INTHISTS_BL.dat">INTHISTS_BL.dat</a>.
For each histogram, there are 101 total bins with a (currently fixed
and hard-coded) bin-size of 1&deg;. The 51<sup>st</sup> bin extends from
(a<sub>0</sub> - 0.5)&deg; to (a<sub>0</sub> + 0.5)&deg;. Here, a<sub>0</sub> is the starting value for
that particular bond angle term (as found in <a
 href="outputfiles.html#INTERNAL_COORDS.idx">INTERNAL_COORDS.idx</a>).
This setup is to ensure that different bond angles (generally ranging
from 90-150&deg; for biomolecules) are adequately supported without
sacrificing detailed resolution. Note that angles are restricted to the
interval [0:180]&deg; and that the program might become unstable
anyway in case sufficiently collinear atoms, <i>i.e.</i>, bond angles near
those limits, are encountered. Range exceptions are dealt with
analogously to <a href="outputfiles.html#INTHISTS_BL.dat">INTHISTS_BL.dat</a>.<br/>
<br/>
In MPI averaging mode this is collected over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present molecules are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="INTHISTS_IM.dat"><b>INTHISTS_IM.dat</b></a><br/>
<br/>
Improper dihedral angle histograms (&deg;). <br/>
<br/>
Format: X(g12.5) where X is the number of columns which can be gathered from <a
 href="outputfiles.html#INTERNAL_COORDS.idx">INTERNAL_COORDS.idx</a> and each column corresponds
to a topologically defined improper dihedral angle<br/>
<br/>
This output file is analogous to <a href="outputfiles.html#INTHISTS_BL.dat">INTHISTS_BL.dat</a>.
For each histogram, there are 101 total bins with a (currently fixed
and hard-coded) bin-size of 1&deg;. The 51<sup>st</sup> bin extends from
(a<sub>0</sub> - 0.5)&deg; to (a<sub>0</sub> + 0.5)&deg;. Here, a<sub>0</sub> is the starting value for
that particular improper term (as found in <a
 href="outputfiles.html#INTERNAL_COORDS.idx">INTERNAL_COORDS.idx</a>).
Note that due to the circular nature ([-180:180]&deg;) of improper dihedral
angles the histogram is wrapped around. assuming for example that the
starting value is 180.0, then the 51st bin seems to span the values of
[179.5:180.5]&deg;. Internally, this of course is
represented as -179.5&deg; but creating a histogram which is fully
faithful to the internal representation would be highly impractical.
Improper dihedral angles are not expected to change significantly during a simulation
and hence range exceptions should be rare. If they occur, they are
dealt with as before (see <a href="outputfiles.html#INTHISTS_BL.dat">INTHISTS_BL.dat</a>).<br/>
<br/>
In MPI averaging mode this is collected over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present molecules are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="INTHISTS_DI.dat"><b>INTHISTS_DI.dat</b></a><br/>
<br/>
Proper dihedral angle histograms (&deg;).<br/>
<br/>
Format: X(g12.5) where X is the number of columns which can be gathered from <a
 href="outputfiles.html#INTERNAL_COORDS.idx">INTERNAL_COORDS.idx</a> and each column corresponds
to a topologically defined dihedral angle<br/>
<br/>
For proper dihedral angles, there are exactly 100 bins (rows) and no
"starting" value (unlike for <a href="outputfiles.html#INTHISTS_BL.dat">INTHISTS_BL.dat</a>).
The reason is that torsions around freely rotatable
bonds typically have neither a well-defined range nor a unique minimum.
As circular quantities, they are defined over the interval [-180:180]&deg;.
Since many torsions in rigid systems have well-defined minima near 180&deg;,
however, the bins (3.6&deg; bin-width) are chosen as follows:
The first bin spans the interval from [-178.2:-174.6]&deg;, the bins are then evenly
spaced up to the 99th bin which spans the interval [174.6:178.2]&deg;. The
100th bin collects data from the wrap-around area, <i>i.e.</i>, from the two
continuous half-bins [-180:-178.2]&deg; and [178.2:180]&deg;. Via this method,
torsions with a stable minimum position of 180.0&deg;. are not right on
a bin boundary, but rather on a bin center. There can be no range
exceptions for proper dihedral angles.<br/>
<br/>
In MPI averaging mode this is collected over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present molecules are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="BB_SEGMENTS.dat"><b>BB_SEGMENTS.dat</b></a><br/>
<br/>
Backbone segment statistics, <i>i.e.</i>, statistics as to how probable it is
to find consecutive, non-terminal polypeptide residues in stretches with the same 
backbone conformation (area with finite size in &#934;/&#936;-space).<br/>
<br/>
Format: (8i10)<br/>
<br/>
The data are always organized in 8 columns corresponding to 8 different segment types, but
there is a variable number of X rows printed for each analysis group (and these blocks are separated by informative
comment lines). Here, X is the longest segment observed for that
analysis group for any segment type.<br/>
<br/>
Overview of segment types (&phi;/&psi;-regions):<br/>
&nbsp;&nbsp;1 = &#946; <br/>
&nbsp;&nbsp;2 = P<sub>II</sub> (Polyproline Type II Helix)<br/>
&nbsp;&nbsp;3 = Unusual Region ("Pass")<br/>
&nbsp;&nbsp;4 = &#945;<sub>R</sub> (Right-Handed &#945;-Helix)<br/>
&nbsp;&nbsp;5 = Inverse C<sub>7</sub> Equatorial (&#947;'-Turn)<br/>
&nbsp;&nbsp;6 = Classic C<sub>7</sub> Equatorial (&#947;-Turn)<br/>
&nbsp;&nbsp;7 = Unusual Region (Helix with 7 Residues per Turn)<br/>
&nbsp;&nbsp;8 = &#945;<sub>L</sub> (Left-Handed &#945;-Helix)<br/>
<br/>
These are the default definitions as provided by the files <a href="../data/bbseg.dat">bbseg.dat</a>
and <a href="../data/bbseg2.dat">bbseg2.dat</a> in the data/ directory (the latter being a little more
lenient on the definitions of some regions). It is quite
straightforward to create your own file (see <a
 href="inputfiles.html#FMCSC_BBSEGFILE">elsewhere</a>) with modified
definitions (as the default ones are useless for D-amino acids, for
example). The rows then give the absolute observed numbers of segments
with increasing length of segments. Counts are strictly exclusive (<i>i.e.</i>,
a residue part of a segment of length 4 is NOT also member of a segment
of length 3, 2, or 1). Also note that glycine will often render this
analysis less meaningful because of the accessibility on the right-hand
side of the map which is incongruent with standard L-amino acids. These unnormalized results are printed primarily for backwards
compatibility, and the normalized variant (see <a href="outputfiles.html#BB_SEGMENTS_NORM.dat">below</a>)
is generally to be preferred. If frame weights (&rarr; <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a>)
are in use, only the normalized version is printed.<br/>
<br/>
In MPI averaging mode, net totals for all nodes are printed out.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="BB_SEGMENTS_RES.dat"><b>BB_SEGMENTS_RES.dat</b></a><br/>
<br/>
Backbone segment statistics, <i>i.e.</i>, statistics as to how probable to find consecutive polypeptide residues
in similar conformation<br/>
<br/>
Basic Format:i8,X*8(1x,i10) with X being the longest segment encountered globally<br/>
<br/>
This file is a superset of the data presented in <a href="outputfiles.html#BB_SEGMENTS.dat">BB_SEGMENTS.dat</a>
since the values are resolved by individual residues (although data for molecules combined in analysis 
groups are still pooled). 
There are NR rows and (1 + X*8) columns, where NR is the total number of polypeptide residues
summed over each example molecule for all analysis groups, and X is the longest segment for any residue position
and in any of the 8 classes of backbone segments (see above description of <a
 href="outputfiles.html#BB_SEGMENTS.dat">BB_SEGMENTS.dat</a>).<br/>
<br/>
To illustrate the meaning and usefulness of this output, the following
example is provided. Assume that a calculation on a single polypeptide yields the following in
 <a href="outputfiles.html#BB_SEGMENTS.dat">BB_SEGMENTS.dat</a>:<br/>
<br/>
34 counts in P<sub>II</sub> (#2) and length 1, and 4 counts in &#945;<sub>R</sub> (#4) and length
3.<br/>
<br/>
The data in BB_SEGMENTS_RES.dat then could look as follows: First of all, 25 columns are
obtained the first one being the residue number.<br/>
Columns 2 through 9 would be the segments of length 1 and types 1
through 8 mapped onto the sequence. If we only had a dipeptide with just one relevant residue and
a maximum segment length of one, this would be exactly the same information as in
<a href="outputfiles.html#BB_SEGMENTS.dat">BB_SEGMENTS.dat</a>. Continuing with the example above,
one would find a total of 34 counts in column 3 (the
second segment type, which is P<sub>II</sub>), distributed over the residues. This
gives an indication (for heterogeneous or folded sequences) what individual residues prefer
structurally. Columns 2, 4, 5, 6, 7, 8, and 9 would all be full of
zeros.
<br/>
Columns 10 through 17 would be the segments of length 2, and hence - in
this example - all be strictly zero.
<br/>
Columns 18 through 25 would be the segments of length 3. In column 21
(which corresponds to the fourth segment type which is &#945;<sub>R</sub>), there
would be a total of 12 (4&#183;3) counts distributed over the residues.
Clearly, it takes three consecutive residues to make a three-residue
&#945;<sub>R</sub>-segment, and hence one would expect to see those 12 counts in
neighboring (or groups of neighboring) residues. All other columns here
would be full of zeros.&nbsp;<br/>
<br/>
Note as well that should an analysis group comprise multiple molecules (of the same type), the 
data are simply pooled together for identical positions. The first column will always give the 
residue numbering consistent with the first (reference) molecule of that analysis group but the
data in subsequent columns will be sums over all molecules belonging to the analysis group parsed 
simply by position in the chain. These unnormalized results are printed primarily for backwards
compatibility, and the normalized variant (see <a href="outputfiles.html#BB_SEGMENTS_NORM_RES.dat">below</a>)
is generally to be preferred. If frame weights (&rarr; <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a>)
are in use, only the normalized version is printed.<br/>
<br/>
In MPI averaging mode, net totals for all nodes are printed out.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="BB_SEGMENTS_NORM.dat"><b>BB_SEGMENTS_NORM.dat</b></a><br/>
<br/>
This input file is the normalized variant of <a href="outputfiles.html#BB_SEGMENTS.dat">BB_SEGMENTS.dat</a>
above. The normalized values can be interpreted as expected average occurrences <i>per molecule and snapshot</i> 
for the molecules comprising a given analysis group.<br/>
<br/>
Format: 8(g14.6,1x)<br/>
<br/>
Note that the underlying frequencies are still mutually exclusive, but that for a longer molecule, there can of course
be multiple short segments of various types <i>per molecule and snapshot</i>.<br/>
<br/>
In MPI averaging mode, this is automatically averaged over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="BB_SEGMENTS_NORM_RES.dat"><b>BB_SEGMENTS_NORM_RES.dat</b></a><br/>
<br/>
This input file is the normalized variant of <a href="outputfiles.html#BB_SEGMENTS_RES.dat">BB_SEGMENTS_RES.dat</a>
above. The normalized values can be interpreted as expected average participation in segments of the indicated
type and length <i>per residue and snapshot</i>. Since data for identical molecules unified in a single analysis group
are still pooled together, the reported numbers in such a case are in addition averaged over those molecules.<br/>
<br/>
Basic Format: i8,8*X(1x,g12.5) with X again being the longest segment encountered globally<br/>
<br/>
Note that the frequencies are still mutually exclusive, <i>i.e.</i> a residue can never be simultaneously
part of multiple segments. This means that the sum of the values in each row should never
exceed unity (due to unaccounted regions of &phi;/&psi;-space, it can be less than unity, however).<br/>
<br/>
In MPI averaging mode, this is automatically averaged over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="DSSP_RES.dat"><b>DSSP_RES.dat</b></a><br/>
<br/>
This output file gives the per-residue statistics for participation of
that residue in DSSP segments of a certain length and type (for the <b>D</b>efine <b>S</b>econdary <b>S</b>tructure of <b>P</b>roteins method see <a href="references.html#ref20_1">reference</a>).
The DSSP integer codes used are as follows:<br/>
<br/>
&nbsp;1) Extended (E) in a parallel &#946;-strand with only a single
partnering strand<br/>
&nbsp;2) Extended (E) in an antiparallel &#946;-strand with only a single
partnering strand<br/>
&nbsp;3) Extended (E) in a sandwiched strand with both partners being
parallel<br/>
&nbsp;4) Extended (E) in a sandwiched strand with one partner being
parallel, the other antiparallel<br/>
&nbsp;5) Extended (E) in a sandwiched strand with both partners being
antiparallel<br/>
&nbsp;6) &#945;-helical (H)<br/>
&nbsp;7) 3<sub>10</sub>-helical (G), but not &#945;-helical<br/>
&nbsp;8) &#960;-helical (I), but not 3<sub>10</sub>- or &#945;-helical<br/>
&nbsp;9) Any 3,4, or 5-residue-spanning turn but not anything above (T)<br/>
10) In a bend but not anything above (S)<br/>
11) Extended (E) in any single- or double-paired beta-strand<br/>
<br/>
Aside from this, the format is roughly analogous to <a href="outputfiles.html#BB_SEGMENTS_RES.dat">BB_SEGMENTS_RES.dat</a>.<br/>
<br/>
Basic Format: i8,(X*11)(1x,i10) with X being the longest DSSP segment length encountered globally<br/>
<br/>
The first of the 11*X+1 columns is always the residue number for the list of all
eligible residues (terminal polypeptide residues and polypeptide caps are
considered for DSSP analysis inasmuch as they provide at least one hydrogen bond
donor or acceptor; however, terminal residues can typically not be part of actual
annotated segments due to the context dependencies inherent to DSSP assignments).
In columns 2-12, the per-residue statistics for segments of length 1 for the 11 types of segments are
listed. This - in analogy to <a href="outputfiles.html#BB_SEGMENTS_RES.dat">BB_SEGMENTS_RES.dat</a> -
is continued in columns 13-23 for segments of length 2, in columns
14-24 for segments of length 3 and so on. The number of columns written
out depends on the longest segment of any type recorded. The most important 
difference to <a href="outputfiles.html#BB_SEGMENTS_RES.dat">BB_SEGMENTS_RES.dat</a> is that the
data here are not pooled by analysis group but instead an entry for every
eligible polypeptide residue exists. The reason is the much larger complexity of the assignment
which can involve intermolecular contacts which are easier to parse when provided
with the full output. These unnormalized results are printed primarily for backwards
compatibility, and the normalized variant (see <a href="outputfiles.html#DSSP_NORM_RES.dat">below</a>) 
is generally to be preferred. If frame weights (&rarr; <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a>)
are in use, only the normalized version is printed.<br/>
<br/>
In MPI averaging mode, net totals for all nodes are printed out.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis. This refers (of course) both
to intra- and intermolecular hydrogen bonds (which a bath molecule can never be part of). Note that this is not necessarily
congruent with the bath reference state in case restraint potentials on DSSP scores are used.<br/>
<br/>
<br/>
<a class="NoHi" id="DSSP.dat"><b>DSSP.dat</b></a><br/>
<br/>
This output file gives the global segment statistics for
DSSP assignments (see <a href="references.html#ref20_1">reference</a>).<br/>
<br/>
Format: 11(i10)<br/>
<br/>
The file is simply the integral over all residues of the data in <a href="outputfiles.html#DSSP_RES.dat">DSSP_RES.dat</a> and therefore
analogous to <a href="outputfiles.html#BB_SEGMENTS.dat">BB_SEGMENTS.dat</a>.
Different columns (1-11) correspond to the different DSSP-assignments,
and different rows to different segment lengths. There are two things
to consider:<br/>
<ol type="i">
<li>The data here are not resolved by analysis group,
these data can always be recovered from <a
 href="outputfiles.html#DSSP_RES.dat">DSSP_RES.dat</a>.<br/></li>
<li>A segment of length i&gt;1 gives rise to a single count in <a
 href="outputfiles.html#DSSP.dat">DSSP.dat</a>, but to multiple counts
(one to each participating residue) in <a
 href="outputfiles.html#DSSP_RES.dat">DSSP_RES.dat</a>. This is
important when normalizing these statistics (&rarr; <a href="outputfiles.html#DSSP_NORM.dat">DSSP_NORM.dat</a>).<br/></li>
</ol>
These unnormalized results are printed primarily for backwards
compatibility, and the normalized variant (see <a href="outputfiles.html#DSSP_NORM.dat">below</a>)
is generally to be preferred. If frame weights (&rarr; <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a>)
are in use, only the normalized version is printed.<br/>
<br/>
In MPI averaging mode, net totals for all nodes are printed out.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis. This refers (of course) both
to intra- and intermolecular hydrogen bonds (which a bath molecule can never be part of). Note that this is not necessarily
congruent with the bath reference state in case restraint potentials on DSSP scores are used.<br/>
<br/>
<br/>
<a class="NoHi" id="DSSP_NORM.dat"><b>DSSP_NORM.dat</b></a><br/>
<br/>
This input file is the normalized variant of <a href="outputfiles.html#DSSP.dat">DSSP.dat</a>
above. The normalized values can be interpreted as expected average occurences <i>per snapshot</i> 
for the entire system. Note that this normalization is different from the one for
<a href="outputfiles.html#BB_SEGMENTS_NORM.dat">BB_SEGMENTS_NORM.dat</a> because DSSP results are
obtained system-wide, and are never averaged on a per molecule basis due to the possibility
of intermolecular hydrogen bonds.<br/>
<br/>
Format: 11(g14.6,1x)<br/>
<br/>
Note that the frequencies are still mutually exclusive, but that for a system with many and/or long chains,
there can of course be multiple short segments of various types <i>per snapshot</i>.<br/>
<br/>
In MPI averaging mode, this is automatically averaged over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="DSSP_NORM_RES.dat"><b>DSSP_NORM_RES.dat</b></a><br/>
<br/>
This input file is the normalized variant of <a href="outputfiles.html#DSSP_RES.dat">DSSP_RES.dat</a>
above. The normalized values can be interpreted as expected average participation in segments of the indicated
type and length <i>per residue and snapshot</i>. Unlike for <a href="outputfiles.html#BB_SEGMENTS_NORM_RES.dat">BB_SEGMENTS_NORM_RES.dat</a> , there is no pooling of data for DSSP analysis, so
each residue is resolved and normalized individually.<br/>
<br/>
Basic Format: i8,11*X(1x,g12.5) with X again being the longest segment encountered globally<br/>
<br/>
Note that the frequencies are still mutually exclusive, <i>i.e.</i> a residue can never be simultaneously
part of multiple segments. This means that the sum of the values in each row should never
exceed unity (due to unclassified states, it can be less than unity, however). <br/>
<br/>
In MPI averaging mode, this is automatically averaged over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="DSSP_RUNNING.dat"><b>DSSP_RUNNING.dat</b></a><br/>
<br/>
This output file gives a detailed overview for a given snapshot of the
DSSP-assignments (see <a href="references.html#ref20_1">reference</a>).
This instantaneous output is created every time the DSSP analysis is performed
(see <a href="keywords.html#DSSPCALC">FMCSC_DSSPCALC</a>).&nbsp; After a header
section which gives the current snapshot number, the following output
is given:<br/>
<br/>
Format: i8,1x,'(',a3,'):',2x,a1,1x,5(a1),1x,'(',i2,')',1x,4(i5,f9.2,1x)<br/>
<br/>
The first integer gives the actual residue number, and the first
character string its 3-letter code. This is followed by the global
DSSP-assignment character (E, H, G, I, T, S, or blank). The five
following characters are positions 2-6 in the DSSP analysis string:<br/>
<ol start="2">
<li>Hydrogen bond: position i &rarr; i &#177; 3 (&lt;, &gt;, X, 3, or blank)<br/></li>
<li>Hydrogen bond: position i &rarr; i &#177; 4 (&lt;, &gt;, X, 4, or blank)<br/></li>
<li>Hydrogen bond: position i &rarr; i &#177; 5 (&lt;, &gt;, X, 5, or blank)<br/></li>
<li>Chain bending (S, or blank)<br/></li>
<li>Chain chirality (+, -, or blank)<br/></li>
</ol>
This is followed by the integer code used to represent the global
DSSP-assignment (see <a href="outputfiles.html#DSSP_RES.dat">DSSP_RES.dat</a> for those). After that, the strongest
hydrogen bond partner for which residue i is the donor is listed in
terms of its relative position in the chain (e.g., -4 for a canonical
&#945;-helix) along with the DSSP energy of that hydrogen bond (in kcal/mol). If no
H-bond was found (the <a href="keywords.html#DSSP_MAXHB">default energy cutoff in DSSP</a> is -0.5kcal/mol), nothing
is listed. The following block is the strongest hydrogen bond partner
for which residue i is the acceptor (e.g., +4 for an &#945;-helix) along
with the energy. This is followed by the second strongest donor H-bond
and the second strongest acceptor H-bond. At the very end, a summary
line gives the instantaneous values for the fractions of residues in E
or H and the corresponding H-bond energy weight factors (see <a
 href="keywords.html#DSSP_MODE">FMCSC_DSSP_MODE</a> for details). <br/>
<br/>
In MPI averaging mode, this feature is currently not supported.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this instantaneous output file does not utilize the weights in any way.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis. This means
that output is suppressed for molecules which are (currently) part of the reserve particle bath, <i>i.e.,</i>
the lengths of entries for each snapshot is variable if eligible polypeptides are amongst the molecule types
allowed to fluctuate in number.<br/>
<br/>
<br/>
<a class="NoHi" id="DSSP_HIST.dat"><b>DSSP_HIST.dat</b></a><br/>
<br/>
One-dimensional secondary structure and hydrogen bond energy histograms (related to &alpha;- and &beta;-content of
polypeptides) as estimated by DSSP<br/>
<br/>
Format: 6(g15.7,1x)<br/>
<br/>
The file contains four different, one-dimensional histograms as follows:<br/>
<ul style="list-style-type: none;">
<li style="background-image: none; padding-left:1px;">Column 1: Bin centers for columns 3 and 4</li>
<li style="background-image: none; padding-left:1px;">Column 2: Bin centers for columns 5 and 6</li>
<li style="background-image: none; padding-left:1px;">Column 3: H-fraction, <i>i.e.</i>, simply the fraction of residues
(counting only eligible peptide residues) with a DSSP-assignment of 'H'</li>
<li style="background-image: none; padding-left:1px;">Column 4: E-fraction, <i>i.e.</i>, the fraction of residues in 'E'</li>
<li style="background-image: none; padding-left:1px;">Column 5: H-bond-Score_H according to the equation given for <a
 href="keywords.html#DSSP_MODE">FMCSC_DSSP_MODE</a>, <i>i.e.</i>, a quantity measuring
the goodness of all the hydrogen bonds responsible for the
DSSP assignments of residues into 'H'. Other H-bonds do NOT contribute
to this score.</li>
<li style="background-image: none; padding-left:1px;">Column 6: H-bond-Score_E, <i>i.e.</i>, the analogous quantity for
&#946;-sheet assignment ('E').</li>
</ul>
<br/>
In MPI averaging mode, this is automatically averaged over all the nodes.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis. This refers (of course) both
to intra- and intermolecular hydrogen bonds (which a bath molecule can never be part of). Note that this is not necessarily
congruent with the bath reference state in case restraint potentials on DSSP scores are used.<br/>
<br/>
<br/>
<a class="NoHi" id="DSSP_EH_HIST.dat"><b>DSSP_EH_HIST.dat</b></a><br/>
<br/>
Two-dimensional secondary structure histograms (related to &alpha;- and &beta;-content of polypeptides) as estimated
by DSSP<br/>
<br/>
Format: 100(g14.6,1x)<br/>
<br/>
This is the two-dimensional histogram of the two order parameters
E-score and H-score (see <a href="keywords.html#DSSP_MODE">FMCSC_DSSP_MODE</a>). Both quantities are
binned into 100 bins with unitless bin centers of 0.005:0.01:0.995. Increasing
column number corresponds to increasing values for the H-score and
increasing row number corresponds to increasing values for the E-score. Note that it is possible (although
very unlikely) for the E- or H-score
to exceed a value of unity (if <a href="keywords.html#DSSP_MODE">DSSP_MODE</a> is set to 3). In this case,
terminal bins will be overstocked.<br/>
<br/>
In MPI averaging mode, this is automatically averaged over all the nodes.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis. This refers (of course) both
to intra- and intermolecular hydrogen bonds (which a bath molecule can never be part of). Note that this is not necessarily
congruent with the bath reference state in case restraint potentials on DSSP scores are used.<br/>
<br/>
<br/>
<a class="NoHi" id="ZSEC_HIST.dat"><b>ZSEC_HIST.dat</b></a><br/>
<br/>
Secondary structure content histograms (torsional definition) for polypeptides<br/>
<br/>
Format: f7.3,1x,g14.5,1x,g14.5<br/>
<br/>
This file contains the normalized histograms for two order parameters,
Z<sub>&#945;</sub> and Z<sub>&#946;</sub> (see <a href="keywords.html#SC_ZSEC">FMCSC_SC_ZSEC</a>).
These are sequence-averaged, smoothed &#945;- and &#946;-propensities
(according to a smooth top hat (Gaussian around a circular plateau) distance metric). This file is
resolved by analysis group and the individual histograms are separated
by descriptive comment lines (analogous to <a
 href="outputfiles.html#POLYAVG.dat">POLYAVG.dat</a>). The first column
simply is the value of the (unitless) order parameter ([0.0:1.0]), the second
column are the normalized counts for Z<sub>&#945;</sub>, and the third are the
normalized counts for Z<sub>&#946;</sub>. This analysis will only be performed if
there are non-terminal polypeptide residues in the system.<br/>
<br/>
In MPI averaging mode, net distributions over all nodes are printed out.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="ZAB_2DHIST.dat"><b>ZAB_2DHIST.dat</b></a><br/>
<br/>
Two-dimensional secondary structure content histograms (torsional definition) for polypeptides<br/>
<br/>
Format: 100(g14.7,1x)<br/>
<br/>
This file offers the full coupling information about the two order
parameters Z<sub>&#945;</sub> and Z<sub>&#946;</sub>, <i>i.e.</i>, the data are a two-dimensional
histogram of those quantities and hence a superset of the data in <a
 href="outputfiles.html#ZSEC_HIST.dat">ZSEC_HIST.dat</a>. The latter can be
recovered via integration from <a
 href="outputfiles.html#ZAB_2DHIST.dat">ZAB_2DHIST.dat</a>. The data
are arranged in a simple 100x100 matrix with columns to the right
corresponding to increasing values of Z<sub>&#945;</sub> and rows down to
increasing values of Z<sub>&#946;</sub>. The assumed bin-sizes are 0.01 in both
directions making the bin centers identical to [0.005:0.01:0.995] for
both axes. Note that that the upper right half (toward the corner of
both parameters being unity) should always be empty if the two basins
are chosen in non-overlapping fashion (see <a href="keywords.html#SC_ZSEC">FMCSC_SC_ZSEC</a> for details). Also note that
this file is - just like <a
 href="outputfiles.html#ZSEC_HIST.dat">ZSEC_HIST.dat</a> - resolved by analysis group,
and that comment lines separate the individual matrices.<br/>
<br/>
In MPI averaging mode, net distributions over all nodes are printed out.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="ZBETA_RG.dat"><b>ZBETA_RG.dat</b></a><br/>
<br/>
Two-dimensional histograms correlating size with &beta;-secondary structure content (torsional definition) for polypeptides<br/>
<br/>
Format: 100(g14.7,1x)<br/>
<br/>
This file is identical to <a href="outputfiles.html#RDHIST.dat">RDHIST.dat</a>,
only that Z<sub>&#946;</sub> replaces &#948;<sup>*</sup> as the order parameter along the
different rows of the output matrix (the axis scaling is identical).<br/>
<br/>
In MPI averaging mode, net distributions over all nodes are printed out.<br/>
<br/>
In (S)GMC calculations, only physically present polypeptides are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="FYC.dat"><b>FYC.dat</b></a><br/>
<br/>
Instantaneous values for system dihedral angles and some bond angles<br/>
<br/>
Format: i12,1x,N(f9.3,1x) where N is a number of system internal coordinates considered (these are the either the
ones that are default (native CAMPARI) degrees of freedom in <a href="keywords.html#CARTINT">rigid-body/torsional space</a> runs, or
all torsional degrees of freedom that are effectively rotatable given the current sampler settings &larr; choice is via
sign of <a href="keywords.html#TOROUT">FMCSC_TOROUT</a>)<br/>
<br/>
Each data line in the file corresponds to a single simulation or analysis snapshot and contains step number and the
selected degrees of freedom.<br/>
In <b>mode 1</b> (<a href="keywords.html#TOROUT">FMCSC_TOROUT</a> is specified as a positive integer), these are all the &#969;-,
&#966;-, &#968;-, nucleic acid backbone, &#967;-angles, and pucker degrees of freedom. For eligible residues, the latter
will always be written regardless of whether they are sampled (but they will not be written if they
do not formally exist &rarr; <a href="inputfiles.html#FMCSC_SEQFILE">SEQFILE</a> for details). Ordering is by residue 
and for each residue as given. To assist in deciphering the output, a header line is provided containing 
residue names, indicators for which degree of freedom is printed out in each column
(OME, PHI, PSI, PUC, NUC, CHI) and eventually numbering for degrees of freedom of which
several may exist (CHI, PUC, or NUC). All values are provided in units of degrees. For disulfide bridges, the C<sub>&alpha;</sub>-C<sub>&beta;</sub>-S-S angles
are written for both residues as "CHI 02" and the C<sub>&beta;</sub>-S-S-C<sub>&beta;</sub> is printed once for the first residue in terms of
sequence numbering as "CSSC".
The order and nature of the pucker degrees of freedom provided
follows CAMPARI's internal structure. Let A-D denote the main chain atoms in the natural building direction (N&rarr;C for polypeptides,
5'&rarr;3' for polynucleotides), <i>i.e.</i>, A-B-C-D. Rotation around the bond B-C is hindered due the presence of the five-membered ring
B-E-F-G-C. Then, the seven degrees of freedom to be written out are:<br/>
<ol>
<li>Torsion A-B-C-D</li>
<li>Torsion A-B-C-G</li>
<li>Torsion B-C-G-F</li>
<li>Torsion G-C-B-E</li>
<li>Angle B-C-G</li>
<li>Angle C-G-F</li>
<li>Angle C-B-E</li>
</ol>
Note that the three torsions around the bond B-C are formally decoupled and will not be linearly related (albeit still correlated).
The total number of ring degrees of freedom given the backbone A-B-C is twelve (remaining positions D to G).
Subtracting five bond length constraints (B-E, E-F, F-G, G-C, C-D), we obtain seven actual degrees of freedom for which the above set
is chosen explicitly by CAMPARI. However, additional dependent variables are likely to exist such as bond angles D-C-G, G-F-E, A-B-E, B-E-F
(but not A-B-C and B-C-D &rarr; backbone geometry is kept rigid). For trigonal sites, the improper dihedral angles may vary
as well. If additional atoms are attached to the ring atoms, a majority of their geometrical parameters will be exactly constrained.
Nonetheless, additional dependent bond angle variables will appear, for example the H-C-H bond angle on the C<sub>&beta;</sub>,
C<sub>&gamma;</sub>, or C<sub>&delta;</sub> atoms of the proline sidechain (consult <a
 href="outputfiles.html#INTHISTS_BA.dat">INTHISTS_BA.dat</a> to see which exact set of bond angles varied during a simulation).
For a nucleotide example, torsion G-C-B-E would be C2*-C3*-C4*-O4*.<br/>
When sampling unsupported residues that are recognized as eligible polypeptide or polynucleotide residues, the corresponding
native CAMPARI degrees of freedom (recognized as such, which excludes all &chi;-angles, for example) are included in this output.
This is in contrast to all other degrees of freedom in those or other unsupported residues (use mode 2 to obtain information about
those).
<br/><br/>
In <b>mode 2</b> (<a href="keywords.html#TOROUT">FMCSC_TOROUT</a> is specified as a negative integer), the values printed to FYC.dat
are just those dihedral angles (never any bond angles) that are elements of the Z-matrix (see
<a href="outputfiles.html#basename_START.int">Z-matrix file written initially</a>) and are altered by the current sampler. The only exception
to this are dihedral angles resulting from chemical crosslinks, which are generally not elements of the Z-matrix. This mode is 
useful for applications with <a href="keywords.html#FRZFILE">custom constraints</a> or restricted Monte Carlo move sets. It is required
for obtaining instantaneous information about dihedral angles that are not sampled natively by CAMPARI in
internal coordinate space. Here, the header is altered and rather than providing residue-level
information simply prints the atomic index corresponding to the Z-matrix angle listed. Ordering is by index. Because the reference atoms
of dihedral angles created by crosslinks can be the same as those of native dihedral angles, their index is reported as its negative (which
implies that they are listed first). For disulfides, the two C<sub>&alpha;</sub>-C<sub>&beta;</sub>-S-S angles are read in this direction making
the sulfur atom of the crosslinked residue the relevant index. For the C<sub>&beta;</sub>-S-S-C<sub>&beta;</sub> angle,
the index is unique and refers to the C<sub>&beta;</sub> atom of the residue with higher sequence number.
Naturally, this output is more tedious to interpret in terms of finding specific angles. 
For pucker sampling, only all torsions (1-4) listed for mode 1 satisfy the criteria. Some caveats apply in <a href="keywords.html#CARTINT">Cartesian
space simulations</a>, <i>viz.</i>, <a href="keywords.html#SHAKESET">holonomic constraints</a> are always ignored, and 
dihedral angles in flexible rings are only considered for the supported 5-membered ring systems.
It is a considerable advantage of mode 2 that in <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> it  
can be used in conjunction with arbitrary <a href="keywords.html#FRZFILE">constraints</a> to isolate output for small subparts of a large system. 
<br/>
Users should keep in mind that this file can easily become quite large,
and <a href="outputfiles.html#Trajectory_output">trajectory output in binary format</a> is clearly the preferred way of storing
the system's configuration in most cases.<br/>
<br/>
In MPI averaging mode, these cycle just like energies (see <a
 href="outputfiles.html#POLYMER.dat">POLYMER.dat</a> for example).<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this instantaneous output file does not utilize the weights in any way.<br/>
<br/>
This file is unaffected by (S)GMC calculations and is one of the few output
files including information from non-present (reserve particle bath) molecules. To indicate
presence, values for reserve particles are offset by 720&deg;.<br/>
<br/>
<br/>
<a class="NoHi" id="PERSISTENCE.dat"><b>PERSISTENCE.dat</b></a><br/>
<br/>
This file provides estimates of an angular correlation function of the peptide or nucleic acid
backbone as a function of sequence separation. For peptides, this uses
the (artificial) N&rarr;C vector on each residue, so only residues which
have an amide nitrogen and a carbonyl carbon can contribute. For
nucleic acids, it uses the (artificial) P&rarr;C3* vector, so again, only
residues which have both the phosphorus and the sugar C3* carbon can
contribute. This measure is also resolved by analysis group.<br/>
<br/>
Format: i10,1x,g14.7,1x,g14.7<br/>
<br/>
There are 3 columns, and for each molecule type there are as many rows as there can
be sequence separations realized. The first column holds sequence separation
(|i-j|), the second column gives the angular correlation function (as &lt;cos(&#920;(N<sub>i</sub>&rarr;C<sub>i</sub>,N<sub>j</sub>&rarr;C<sub>j</sub>))&gt;),
and the third column gives a "normalized" version as |i-j|<sup>3/2</sup> &#183; &lt;cos(&#920;(N<sub>i</sub>&rarr;C<sub>i</sub>,N<sub>j</sub>&rarr;C<sub>j</sub>))&gt;. The notation uses a polypeptide example.<br/>
<br/>
In MPI averaging mode this is automatically averaged over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present solute molecules are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="TURNS_RES.dat"><b>TURNS_RES.dat</b></a><br/>
<br/>
Analysis group-resolved turn propensities along polypeptide and polynucleotide chains in degrees<br/>
<br/>
Format: (i10,6(1x,g14.7))<br/>
<br/>
These data are based on the same vectors that are used to calculate angular correlation
functions (see <a href="outputfiles.html#PERSISTENCE.dat">PERSISTENCE.dat</a>). 
The reported values correspond to simulation-averaged turn and dihedral angles,
and for each analysis group, N<sub>i</sub>-1 rows are provided, where N<sub>i</sub> is the number of 
eligible residues for the example molecule for that analysis group.
Column two then gives the average angle between residues i and i+1 (for
peptides: N&rarr;C vectors, for nucleotides: P&rarr;C3* vectors), <i>i.e.</i>,
the angle is mapped to the first of the two residues. Column three
gives the average dihedral (atoms N-C-N-C or P-C3*-P-C3*) for residues
i and i+1. Columns four and five do the same for residues i and i+2,
and columns six and seven for residues i and i+3. Consequently, for
each molecule type the last entry in columns four and five (would
correspond to the angles for the second-last possible residue) as well
as the two last entries in columns six and seven (would correspond to
the angles for the third-last possible residue) are meaningless and
written out as zeros. All values are interpreted to be in degrees.<br/>
<br/>
Example: let there be only molecule type, say FOR-GLY-ASN-NH2:<br/>
The output would look something like:<br/>
------------------------------------------------------------------------------------------------------------<br/>
<big><code># descriptive comment line<br/>
2&nbsp;&nbsp; 45.6&nbsp;&nbsp;&nbsp;125.5&nbsp;&nbsp;
0.0000&nbsp;&nbsp;0.0000&nbsp;&nbsp;0.0000&nbsp;&nbsp;0.0000</code></big><br/>
------------------------------------------------------------------------------------------------------------<br/>
Output in columns 4-7 is truncated since only the data for the angles between GLY and ASN are meaningful.
No row is written for the asparagine residue because there are no other eligible residue further
towards the C-terminus.<br/>
<br/>
In MPI averaging mode this is automatically averaged over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present solute molecules are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="CONTACTMAP.dat"><b>CONTACTMAP.dat</b></a><br/>
<br/>
Average contact frequencies resolved by residues<br/>
<br/>
Format (data lines): NR(g14.7,1x) where NR is the total number of residues that are part of solute molecules<br/>
<br/>
This output file provides a header line which gives the column indices by stating residue name and number.
This is followed by a matrix of size NRxNR where NR is the total number of residues in the system
which are part of solute molecules (see <a href="inputfiles.html#FMCSC_ANGRPFILE">ANGRPFILE</a>).
Depending on the setting for <a href="keywords.html#CONTACTOFF">CONTACTOFF</a> the main diagonal
and the first few off-diagonals may be excluded and hence report only zero.
The file contains the probabilities of observing a contact between the indicated residues (column index is identical
to row index) throughout the simulation. Two different definitions of a contact are considered; the first via
minimum atom-atom distances, the second via distance of residue centers-of-mass (the definitions are equivalent
if both residues possess a single atom). The lower half (column index smaller than row index) contains the map
according to minimal interatomic distances, while the upper half (column index larger than row index) contains the map
according to center-of-mass distances.
Note that analysis and output which would pertain to solvent molecules are simply skipped.<br/>
<br/>
In MPI averaging mode this is automatically averaged over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present solute molecules are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="CONTACT_HISTOGRAMS.dat"><b>CONTACT_HISTOGRAMS.dat</b></a><br/>
<br/>
Histograms of net contact numbers<br/>
<br/>
Format: i10,1x,g14.7,1x,g14.7<br/>
<br/>
Since average contact maps do not report on fluctuation properties of the system, this additional
output provides normalized histograms of net contact numbers observed throughout the simulation.
There are three columns, the first giving the number of total solute residue contacts, the second the 
histogram when using minimum atom-atom distances to define a contact, and the third the histogram when using 
the distance of residues' centers-of-mass to define a contact. This information is probably most
useful with a single solute macromolecule in the system. The variance of the histograms can report on the magnitude of conformational
fluctuations at the contact level and will - for example - go through a sharp peak for a polymer undergoing a
coil-to-globule transition as the function of some control parameter. This peak will be located located in the transition
region.<br/>
<br/>
In MPI averaging mode, net totals over all nodes are printed out.<br/>
<br/>
In (S)GMC calculations, only physically present solute molecules are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="CLUSTERS.dat"><b>CLUSTERS.dat</b></a><br/>
<br/>
Global statistics pertaining to contact formation between different molecules (molecular clusters)<br/>
<br/>
Format: i10,1x,g14.7<br/>
<br/>
The first column gives the number of molecules in the cluster, and the
second column the probability of observing such a cluster (this
probability is normalized in a per-molecule sense, <i>i.e.</i>, if 3 molecules
form a cluster and 3 more molecules are isolated, then the probability
for clusters of size 1 and 3 is 0.5 each. The clusters are extracted
based on a solute-solute contact matrix, where a contact is defined by the
smallest distance between any pair of atoms in any pair of residues
belonging to the two solutes (<i>i.e.</i>, just like contact definition #1). Contact
and cluster algorithms are computationally rather expensive which is the 
reason they are restricted to molecules tagged as solutes (see
<a href="inputfiles.html#FMCSC_ANGRPFILE">ANGRPFILE</a>). When simulating macromolecules in a large bath of 
solvent, the algorithm could otherwise slow down the simulation substantially since
the contact-contact determination does not employ grid-based neighbor lists.
If solvent contacts are excluded, this of course implies that two solute molecules bridged by a solvent molecule will most likely
be counted as separate (in contrast, all of the molecules would form a single cluster if solvent
were interpreted as solute as well).
It is also noteworthy that the algorithm is completely blind to molecular identity which can lead
to unwanted artifacts. For example, in a mix of water and ions, water molecules have - independent
of solution structure and Hamiltonian - a higher likelihood of forming clusters due to a higher
likelihood of forming contacts. The latter arises since minimum atom-atom distances are considered and the water
oxygen and hydrogens span a finite base radius per molecule which is not true for monoatomic molecules.
Hence, cluster output is interpreted most easily for solutions of identical solute molecules.<br/>
<br/>
In MPI averaging mode this is automatically averaged over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present molecules are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="MOLCLUSTERS.dat"><b>MOLCLUSTERS.dat</b></a><br/>
<br/>
Molecule-resolved cluster statistics (see <a href="outputfiles.html#CLUSTERS.dat">CLUSTERS.dat</a> for comparison)<br/>
<br/>
Format: i10,M(1x,g14.7), where M is the size of the largest molecular cluster encountered<br/>
<br/>
The first column holds the solute molecule number (according to sequence
input). Each row corresponds to the cluster statistics for an
individual solute molecule, <i>i.e.</i>, if a molecule is half of the time by itself
and half of the time part of a dimer, then the numbers in columns 2 and
3 would be 0.5 each. This file can be integrated to extract the global
cluster statistics (<a href="outputfiles.html#CLUSTERS.dat">CLUSTERS.dat</a>). It is useful if different molecule types or
different phases (with slow exchange) are present in the system, since
it reveals propensities to form cluster for individual molecules.<br/>
<br/>
In MPI averaging mode this is automatically averaged over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present molecules are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="COOCLUSTERS.dat"><b>COOCLUSTERS.dat</b></a><br/>
<br/>
Contact order-resolved cluster statistics (see <a href="outputfiles.html#CLUSTERS.dat">CLUSTERS.dat</a> for comparison)<br/>
<br/>
Format: (g12.5,M(1x,g14.7)), where M is the size of the largest molecular cluster encountered<br/>
<br/>
The first column holds the values for the contact order, which is
defined as:<br/>
<br/>
COO = #(inter-solute links within clusters)/(size(cluster)-1)<br/>
<br/>
This means that a linearly (minimally) connected cluster always has a
contact order of 1, while more strongly connected clusters might have
larger values. The contact order histograms are listed in the following
columns of this file with increasing cluster size.<br/>
Note that the individual contact order histograms for each cluster size no longer
reflect the likelihood of finding clusters of that size.
Note as well that the contact order by definition is a discrete function (the
theoretical number of discrete values goes as [size&#183;(size-1)/2 - size +
2], which means the distribution for a cluster size of 2 is strictly a
delta-function at COO=1.0).<br/>
<br/>
In MPI averaging mode this is automatically averaged over all nodes.<br/>
<br/>
In (S)GMC calculations, only physically present molecules are considered for analysis.<br/>
<br/>
<br/>
<a class="NoHi" id="INTSCAL.dat"><b>INTSCAL.dat</b></a><br/>
<br/>
Scaling of internal distances resolved by analysis group<br/>
<br/>
Format: i6,1x,g14.7<br/>
<br/>
As a function of sequence separation, this function measures a mean Euclidean distance
between pairs of atoms in polymer residues (units of &#8491;). 
The first column gives the value for the sequence separation (residue number
difference), and the second column gives the average distance obtained for all
possible atom-atom pairs in all molecules belonging to the analysis group satisfying that intramolecular sequence separation.
These data can be used to adjudicate the gross polymeric properties of chains (for example,
to distinguish collapsed from expanded chains).<br/>
<br/>
In MPI averaging calculations, this is automatically averaged over all nodes.<br/>
<br/>
In (S)GC ensemble calculations, the data are only obtained over physically present molecules.<br/>
<br/>
<br/>
<a class="NoHi" id="KRATKY.dat"><b>KRATKY.dat</b></a><br/>
<br/>
Kratky scattering profiles resolved by analysis group<br/>
<br/>
The predicted scattering probability is defined computationally as:<br/>
<br/>
p(q) = (1/N<sub>ij</sub>) &#183; &lt; &#931;<sub>i&lt;j</sub>sin(q&#183;r<sub>ij</sub>)/(q*r<sub>ij</sub>) &gt;<br/>
<br/>
Here, q is the wave vector length and the r<sub>ij</sub> are the N<sub>ij</sub> unique (and purely
intramolecular) distances.<br/>
<br/>
Format: f12.6,1x,g14.7,1x,g14.7<br/>
<br/>
For each analysis group, the file has X rows, one for each wave vector
(see keywords <a href="keywords.html#SCATTERRES">FMCSC_SCATTERRES</a> and <a
 href="keywords.html#SCATTERVECS">FMCSC_SCATTERVECS</a>), with the first column
giving the value for the wave vector, the second giving the scattering
probability, and the third giving the product q<sup>2</sup>&#183;p(q) in &#8491;<sup>2</sup>. The data
in the second column can be used in a double-logarithmic plot to try to estimate
the scaling exponent for fractal objects (or apply Porod's law), but identifying
the proper data range to consider can be tricky.<br/>
<br/>
In MPI averaging calculations, this is automatically averaged over all nodes.<br/>
<br/>
In (S)GC ensemble calculations, the data are only obtained over physically present molecules.<br/>
<br/>
<br/>
<a class="NoHi" id="AMIDES_PC.dat"><b>AMIDES_PC.dat</b></a><br/>
<br/>
Normalized distance counts for a set of different distances to
characterize hydrogen bonding patterns, amide stacks and the like for
protein backbone groups (bb) as well as amide sidechains of glutamine and asparagine residues (sc). In
order to obtain pair correlation functions from these data, all
normalized distributions have to be divided by an appropriate
(normalized) prior. The most natural choice to obtain such a prior may be from a
simulation for which all of the energy scaling factors (like
<a href="keywords.html#SC_IPP">FMCSC_SC_IPP</a>, ....) are set to zero (ideal chain). This analysis
routine works purely on intramolecular distances, but does <i>not</i>
distinguish by molecule type (and hence is probably most useful for
systems with only one relevant molecule type).<br/>
<br/>
Format: f14.7,1x,9(g14.7,1x)<br/>
<br/>
The data are organized in 10 columns as follows:<br/>
<ul style="list-style-type: none;">
<li style="background-image: none; padding-left:0px;"><b>Column 1</b>: Absolute value for considered distance</li>
<li style="background-image: none; padding-left:0px;"><b>Column 2</b>: Normalized distance histogram for pairs of points corresponding to the centers of two distinct N&rarr;C amide bonds located on the polypeptide backbone (<b>bb-bb</b>)</li>
<li style="background-image: none; padding-left:0px;"><b>Column 3</b>: Normalized distance histogram for pairs of backbone amide hydrogen atoms and backbone amide carbonyl oxygen atoms (<b>bb-bb</b>)</li>
<li style="background-image: none; padding-left:0px;"><b>Column 4</b>: Normalized distance histogram for pairs of points corresponding to the centers of two distinct N&rarr;C amide bonds both located within Gln/Asn sidechains (<b>sc-sc</b>)</li>
<li style="background-image: none; padding-left:0px;"><b>Column 5</b>: Normalized distance histogram for pairs of Gln/Asn sidechain amide <i>trans</i> hydrogen atoms (opposite side of oxygen atom of same amide group) and Gln/Asn sidechain amide carbonyl oxygen atoms (<b>sc-sc</b>)</li>
<li style="background-image: none; padding-left:0px;"><b>Column 6</b>: Normalized distance histogram for pairs of Gln/Asn sidechain amide <i>cis</i> hydrogen atoms (same side as oxygen atom of same amide group) and Gln/Asn sidechain amide carbonyl oxygen atoms (<b>sc-sc</b>)</li>
<li style="background-image: none; padding-left:0px;"><b>Column 7</b>: Normalized distance histogram for pairs of points corresponding to the centers of two distinct N&rarr;C amide bonds with on located within Gln/Asn sidechains and the other located on the polypeptide backbone (<b>sc-bb</b>)</li>
<li style="background-image: none; padding-left:0px;"><b>Column 8</b>: Normalized distance histogram for pairs of Gln/Asn sidechain amide <i>trans</i> hydrogen atoms (opposite side of oxygen atom of same amide group) and backbone amide carbonyl oxygen atoms (<b>sc-bb</b>)</li>
<li style="background-image: none; padding-left:0px;"><b>Column 9</b>: Normalized distance histogram for pairs of Gln/Asn sidechain amide <i>cis</i> hydrogen atoms (same side as oxygen atom of same amide group) and backbone amide carbonyl oxygen atoms (<b>sc-bb</b>)</li>
<li style="background-image: none; padding-left:0px;"><b>Column 10</b>: Normalized distance histogram for pairs of backbone amide hydrogen atoms and Gln/Asn sidechain amide carbonyl oxygen atoms (<b>bb-sc</b>)</li>
</ul>
<br/>
In MPI averaging mode averages over all nodes are printed out.<br/>
<br/>
In (S)GC ensemble calculations, the data are only obtained over physically present molecules.<br/>
<br/>
<br/>
<a class="NoHi" id="RBC_PC.dat"><b>RBC_PC.dat</b></a><br/>
<br/>
Pair correlation functions based upon the centroid-to-centroid (or center of mass-to-center of mass) distances for
the solute molecules comprising the various analysis groups 
(see <a href="inputfiles.html#FMCSC_ANGRPFILE">ANGRPFILE</a> in <a
 href="inputfiles.html#INPUTFILES">INPUTFILES</a>) in the system<br/>
<br/>
Format: f14.7,1x,N+1(g14.7,1x), where N is the number of unique combinations of solute analysis groups<br/>
<br/>
The header (comment line) gives the pairs of analysis groups, for which
the pair correlation function is listed in that
particular column. The first column gives the center of the distance
bins and the second column gives an analytically obtained estimate for the volume element
for the chosen system (<i>i.e.</i>, the (approximate) probability distribution p<sub>ideal</sub>(r)). These volume elements are non-trivial to compute and currently unavailable for <a href="keywords.html#SHAPE">cylinders</a>. If
obtained numerically (which is trivial to do by switching off all terms to the Hamiltonian for the same system),
the resultant pair correlation functions will suffer from shot noise in p<sub>ideal</sub>(r) whenever the system often occupies low entropy regions,
hence the analytical approach. This itself brings forth some complications if the simulation
container has a soft wall (see <a href="keywords.html#BOUNDARY">BOUNDARY</a>). Here, the analytical volume element is computed
assuming a hard-wall (<i>i.e.</i>, using the radius provided via <a href="keywords.html#SIZE">SIZE</a>).
The softness of the wall allows the particles, however, to exceed distances of the sphere diameter
for which the volume element is zero, which would lead to a value of infinity in g(r). To make
the data for such large distances accessible, CAMPARI will silently use the same volume element it has for the
last bin with finite volume element. This allows reconstruction of the raw counts in most cases.
Exceptions are simulations with fluctuating particle numbers or fluctuating volumes
(see <a href="keywords.html#ENSEMBLE">ENSEMBLE</a>). The simple distance histograms are always
printed for cylindrical system because the volume element is missing.<br/>
<br/>
These purely intermolecular pair correlation functions can be used to assess solution structure, <i>i.e.</i>, (in
particular for small molecules) whether some species in the system like
to preferentially interact with other species (<i>e.g.</i>, in a solution of
cat- and anions one would of course expect the pair
distribution for ions with opposite charge to show preferential association). For larger, flexible
molecules the centroid is only moderately meaningful and hence the
distributions become more difficult to interpret. Note that the geometric center is used in pure MC applications
and the center of mass in any other case. This has reasons of computational efficiency.
<br/>
<br/>
In MPI averaging calculations, the data are collected over all nodes.<br/>
<br/>
In (S)GC ensemble calculations, the data are only obtained over physically present molecules but
assume no bulk concentration (the effective bulk concentration will be simply given by the total
number of particles seen for each fluctuating type throughout the simulation).<br/> 
<br/>
<br/>
<a class="NoHi" id="GENERAL_PC.idx"><b>GENERAL_PC.idx</b></a><br/>
<br/>
This is an index file for the analysis of generalized atom-atom distance distributions (see <a
 href="outputfiles.html#GENERAL_PC.dat">GENERAL_PC.dat</a>). This file
will list the atom pairs (whether specific or analysis group-wise) contributing
to each column in <a href="outputfiles.html#GENERAL_PC.dat">GENERAL_PC.dat</a>.
The residue numbering provided is relative to the molecule in question (starting at 1) and not absolute.
The numbering of analysis groups is absolute and does not generally correspond to the (arbitrary) integers
used in the <a href="inputfiles.html#FMCSC_ANGRPFILE">analysis group input file</a> but rather
to those printed to log output by means of <a href="keywords.html#SEQREPORT">FMCSC_SEQREPORT</a>.
Note that the numbering of codes might
not be preserved exactly with respect to the user-based input file due to continuous numbering being enforced.
These adjustments are always printed to log output. <br/>
<br/>
<br/>
<a class="NoHi" id="GENERAL_PC.dat"><b>GENERAL_PC.dat</b></a><br/>
<br/>
Distance distributions for arbitrary atom-atom pairs<br/>
<br/>
N+1 columns: f11.4,1x,N(g11.5,1x) where N is the number of unique components requested<br/>
<br/>
The first column gives the center of each distance bin. The N following
columns give either the normalized histograms for requested intramolecular 
distance distributions or the pair correlation functions for requested intermolecular distances (requests through
<a href="keywords.html#PCCODEFILE">FMCSC_PCCODEFILE</a>). Note that the volume element underlying normalization for
intermolecular distances is printed out in column 2 of <a href="outputfiles.html#RBC_PC.dat">RBC_PC.dat</a>,
and that the same caveats apply, in particular the fact that only distance histograms are available
in <a href="keywords.html#SHAPE">cylindrical systems</a>. In addition, the two sites for intermolecular
distributions may be located far away from each of the two molecules' reference points leading to a 
discrepancy between distances possible to be sampled by those two sites versus distances possible to be sampled
by the two reference points (the latter being the only case for which the analytical volume element applies
rigorously for all possible distances). In periodic boundary conditions, this manifests itself as the sampling
of "illegal" distances (larger than half the longest box diagonal), in soft-wall boundaries a similar effect
may persist unless an atom-based convention is used (see <a href="keywords.html#BOUNDARY">BOUNDARY</a>).<br/>
<br/>
In MPI averaging calculations this is automatically averaged over all
nodes.<br/>
<br/>
In (S)GC ensemble calculations, data are only obtained over physically present molecules. Intermolecular
pair correlation functions assume no bulk concentration (the effective bulk concentration will be simply given by the total
number of particles seen for each fluctuating type contributing to a particular general PC component throughout the simulation).<br/>
<br/>
<br/>
<a class="NoHi" id="GENERAL_DIS.dat"><b>GENERAL_DIS.dat</b></a><br/>
<br/>
Instantaneous data traces for interatomic distances for arbitrary pairs of atoms<br/>
<br/>
Following two header lines, there are N+1 columns: i11,1x,N(g11.5,1x) where N is the number of total unique distances
underlying all components<br/>
<br/>
In the first header line, CAMPARI will print the index of the first atom in each pair contributing in to the
analysis of interatomic distances (N integers). The second header line contains the indices of the corresponding second atoms.
For the actual data lines, the first column gives the step number and the remaining N columns correspond to the instantaneous,
interatomic distances corresponding to those pairs of atoms requested through
<a href="keywords.html#PCCODEFILE">FMCSC_PCCODEFILE</a>. These data are (part of) the data
underlying the output in <a href="outputfiles.html#GENERAL_PC.dat">GENERAL_PC.dat</a>.
This output is created by setting keyword <a href="keywords.html#INSTGPC">FMCSC_INSTGPC</a> to 1 or an
appropriate multiple. The output can be used to compute higher-order correlation functions. Note that the\output in 
this file can easily get very large.<br/>
<br/>
For any <a href="keywords.html#MPI_settings">parallel run</a>, this file is produced and printed for all individual replicas
using the standard naming conventions.<br/>
<br/>
In (S)GC ensemble calculations, distances involving one or two atoms that are not physically present are reported
as -1.0.<br/>
<br/>
<br/>
<a class="NoHi" id="SAV.dat"><b>SAV.dat</b></a><br/>
<br/>
This output file reports values for two different types of quantities. First, it reports a quantity vaguely related to the
net solvent-accessible volume (SAV) for the whole simulation system.
This total SAV is not directly related to an actual estimate of molecular solvent-accessible volumes, as it is 
very crudely estimated by summing up the individual SAVs and normalizing them at the residue level such that
they would describe the SAV shell around the net van der Waals volume of the atoms
assuming those had been collapsed into an ideal cylinder of radius 1.5&#8491; (this applies to all residues except
those with only one or very few atoms).
The absolute numbers are not particularly meaningful and overall this quantity is more of an additional tool to monitor sampling /
fluctuations throughout simulations of appropriate systems.<br/>
The second type of quantity or instantaneous values for SAV fractions of user-selected atoms.<br/>
<br/>
Format: i10,1x,g14.7,1x,g14.7,N(1x,g14.7) where N is the number of user-selected atoms (via <a href="keywords.html#SAVATOMFILE">FMCSC_SAVATOMFILE</a>)<br/>
<br/>
The first column provides the step number, while the second and third columns give the cumulative average of the total, normalized heavy-atom SAV,
and instantaneous values for the current snapshot, respectively. From the fourth column onward, the instantaneous SAV fractions of
user-selected atoms for the current snapshot are printed (atom selection via <a href="keywords.html#SAVATOMFILE">FMCSC_SAVATOMFILE</a>).
If no atoms are requested for this type of analysis, the file has at most three columns. The frequency with which this file is
written to depends jointly on the choices for keywords <a href="keywords.html#SAVCALC">FMCSC_SAVCALC</a> and 
<a href="keywords.html#INSTSAV">FMCSC_INSTSAV</a>. This implies that it is possible to perform SAV analysis without producing any
instantaneous output.<br/>
<br/>
In MPI averaging calculations, the instantaneous part cycles through the different nodes (see <a
 href="outputfiles.html#POLYMER.dat">POLYMER.dat</a> for an example) while the average is always given as an average over all the nodes.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
the second column with running averages is missing, and the instantaneous output (now the second column) does not utilize the weights in any way.
Possible atom-specific SAV fractions now start in the third column, and equally do not utilize the weights in any way.<br/>
<br/>
In (S)GMC calculations, only physically present molecules are considered for analysis and the reserve bath particles
do not influence the SAVs of physically present molecules in any way.<br/>
<br/>
<br/>
<a class="NoHi" id="SAV_BY_ATOM.dat"><b>SAV_BY_ATOM.dat</b></a><br/>
<br/>
Analysis of solvent-accessible volumes (SAVs) on a per-atom basis<br/>
<br/>
Format: i10,1x,g14.7,1x,g14.7,1x,g14.7,1x,g14.7,1x,g14.7,1x,g14.7<br/>
<br/>
For details on the computation of SAVs, the reader is referred to the <a href="references.html#ref19_1">original publication</a>,
and <a href="keywords.html#SAVCALC">elsewhere</a> in this documentation.<br/>
In the first column, the file lists the atomic index. Note that this index always refers to CAMPARI's internal
numbering that can usually be gleaned from output files such as the <a href="outputfiles.html#basename_START.pdb">initial pdb-file</a>.
The second and third columns hold the atomic, fractional SAV (symbol &eta;<sub>i</sub><sup>k</sup>), and the atomic solvation state (symbol &upsilon;<sub>i</sub><sup>k</sup>). 
For each snapshot/conformation, the &upsilon;<sub>i</sub><sup>k</sup> are computed from the &eta;<sub>i</sub><sup>k</sup>, and the results are averaged separately. Note
that the solvation state considered here is the one controlled by mapping parameters <a href="keywords.html#FOSTAU">FMCSC_FOSMID</a> and 
<a href="keywords.html#FOSTAU">FMCSC_FOSTAU</a>, and not the one controlled by <a href="keywords.html#SCRTAU">FMCSC_SCRMID</a> and
<a href="keywords.html#SCRTAU">FMCSC_SCRTAU</a>. It is also worth noting that other parameters of the solvation model directly influence the
computed values (most prominently, <a href="keywords.html#SAVPROBE">FMCSC_SAVPROBE</a>).<br/>
The remaining columns give fixed parameters of the solvation model. In the fourth column, the topology-derived maximum &eta;<sub>i</sub><sup>k</sup>
is provided for each atom (<i>i.e.,</i> how much accessible volume does the rigid context in the corresponding model compound always consume?).
The fifth and sixth columns provide atom-specific scaling factors computed from the sigmoidal mapping parameters mentioned before.
This is debugging output that should be ignored. Lastly, the seventh column lists the volume reduction factors used in SAV-computations.
These correspond to the fractions of atomic volumes considered when calculating overlap volumes of atoms with solvation shells of other
atoms. These corrections are necessary on account of covalently bound (overlapping) atoms, and are always split symmetrically between the two participating atoms.<br/>
<br/>
In MPI averaging mode averages over all nodes are printed out.<br/>
<br/>
In (S)GMC calculations, only physically present molecules are considered for analysis and the reserve bath particles
do not influence the SAVs of physically present molecules in any way.<br/>
<br/>
<br/>
<a class="NoHi" id="SAV_HISTS.dat"><b>SAV_HISTS.dat</b></a><br/>
<br/>
Histograms for solvent-accessible volume fractions and solvation states for user-selected atoms<br/>
<br/>
Format: f8.3,1x,2N(g13.6,1x) where N is the number of atoms selected<br/>
<br/>
The header (comment) line simply gives the actual atomic index of the selected atom for each pair of columns after the first one.
The file reports histograms of atomic solvent-accessible volume fractions (symbol &eta;<sub>i</sub><sup>k</sup>; first in each pair of columns)
and atomic solvation states (symbol &upsilon;<sub>i</sub><sup>k</sup>; second in each pair of columns). For details on the computation of
solvent-accessible volumes, the reader is referred to the documentation on output file <a href="outputfiles.html#SAV_BY_ATOM.dat">SAV_BY_ATOM.dat</a>,
to the <a href="references.html#ref19_1">original publication</a>, and <a href="keywords.html#SAVCALC">elsewhere</a>.
In the first column, the file lists the center of each bin. Since both quantities are normalized, the binning is fixed (100 bins
in the interval from zero to unity). The second and third columns hold the histograms for &eta;<sub>i</sub><sup>k</sup> and 
&upsilon;<sub>i</sub><sup>k</sup>, respectively, with the atomic index i being provided in the header line. Each further atom in the list
supplied via <a href="keywords.html#SAVATOMFILE">FMCSC_SAVATOMFILE</a> then gives rise to an additional two columns that are
interpreted analogously. The accumulation frequency for these histograms depends on the setting for <a href="keywords.html#SAVCALC">FMCSC_SAVCALC</a>.<br/>
<br/>
In MPI averaging mode averages over all nodes are printed out.<br/>
<br/>
In (S)GMC calculations, only physically present molecules are considered for analysis and the reserve bath particles
do not influence the SAVs of physically present molecules in any way.<br/>
<br/>
<br/>
<a class="NoHi" id="TABULATED_POT.idx"><b>TABULATED_POT.idx</b></a><br/>
<br/>
This is an index file which is completely analogous to <a
 href="outputfiles.html#GENERAL_PC.idx">GENERAL_PC.idx</a>. Instead of listing
analysis components for atom-atom correlation functions, it summarizes atom-atom components used for
<a href="keywords.html#SC_TABUL">tabulated potentials</a> in the system.<br/>
<br/>
<br/>
<a class="NoHi" id="TRCV_xxx.tmp"><b>TRCV_xxx.tmp</b></a><br/>
<br/>
Signal trains for torsions and relevant bond angles (in radian) or their Fourier terms (see <a
 href="keywords.html#COVMODE">COVMODE</a>) resolved by analysis group (the
xxx (leading zeros with number of digits adjusted to be abloe to accomodate the total number
of analysis groups) denotes the analysis group as written out initially
by CAMPARI).<br/>
<br/>Format (data lines): T(g11.4,1x) with T being (twice) the number of torsions and relevant angles for that molecule type<br/>
<br/>
The header line in each file contains number and example molecule of the given analysis group.
The internal degrees of freedom are strictly ordered by residue and within each residue as &#969;-, &#966;-, &#968;-,
nucleic acid backbone angles, &#967;-angles, pucker degrees of freedom (see first mode for <a href="keywords.html#TOROUT">FMCSC_TOROUT</a>).
Should there be more than one
molecule for that type a single snapshot may provide several rows
(observations) to this file. Note that values are printed out
irrespective of whether the corresponding degree of freedom is actually
sampled. Also, keep in mind that some small molecules have &#969;- (NMF,
NMA, <i>etc.</i>) as well as &#967;-angles (PCR, MOH, <i>etc.</i>) and that
residues on polypeptides and polynucleotides might not have all
backbone angles either (e.g., no &#969;- for N-terminal residues and NH2,
no &#966;/&#968;-angles for NH2, NME, FOR, ACE, no pucker degrees of freedom for anything but nucleotides and proline, <i>etc</i> ...).<br/>
<br/>
This class of instantaneous output files (like <a href="outputfiles.html#FYC.dat">FYC.dat</a>)
was intended to be used to offer and/or allow a variety of external post-processing operations that are based on dihedral angles.
It is nearly obsolete at this point in time, but may be utilized again later.
<a href="outputfiles.html#FYC.dat">FYC.dat</a> provides better control and a generally more convenient format.
Note that <a href="keywords.html#CCOLLECT">structural clustering</a> supports
<a href="keywords.html#CDISTANCE">distance functions defined in dihedral angle space</a>.
<br/>
<br/>
Note that - as in MPI averaging mode values cycling through the nodes
are printed out - this signal by construction contains artificial
periodicity in those calculations. However, even for regular
calculations the file might have to be decomposed to extract signal
trains for individual molecules should there be more than one molecule
belonging to any given analysis group at any time.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this instantaneous output file does not utilize the weights in any way.<br/>
<br/>
In (S)GC ensemble calculations, the data are only written out over physically present molecules, <i>i.e.</i>,
the number of observations per snapshot will fluctuate if numbers for molecule types with torsional
degrees of freedom are allowed to fluctuate.<br/>
<br/>
<br/>
<a class="NoHi" id="MOLDIPOLES.dat"><b>MOLDIPOLES.dat</b></a><br/>
<br/>
Average dipole moments for entire molecules in Debye (pooled by analysis group)<br/>
<br/>
Format (data lines): g14.7,1x,g14.7,1x,g14.7,1x,g14.7,1x,g14.7,1x,g14.7 (6 columns)<br/>
<br/>
This file is resolved by analysis group and preceding the actual data
line there is a descriptive line giving the number of said group 
(as initially printed out by CAMPARI), the reference molecule, and the
residues comprising that reference molecule.<br/>
<br/>
The data line gives the average dipolar properties (base unit for dipole moments is Debye) for all the
molecules of that type. The molecular dipole moment is locally
(molecule-wise) defined:&nbsp; average molecular dipole moment (&lt;|<b>&#956;</b>|&gt;),
average squared molecular dipole moment (&lt;<b>&#956;</b><sup>2</sup>&gt;), standard deviation of
|<b>&#956;</b>| and lastly the individual components &lt;&#956;<sub>x</sub>&gt;, &lt;&#956;<sub>y</sub>&gt;, and
&lt;&#956;<sub>z</sub>&gt;.<br/>
<br/>
Definitions:<br/>
&nbsp;&lt;&#956;&gt; = &lt; | &#931;<sub>i</sub> (q<sub>i</sub>&#183;<b>r</b><sub>i</sub>) | &gt;<br/>
&nbsp;&lt;<b>&#956;</b><sup>2</sup>&gt; = &lt;  &#931;<sub>i</sub> (q<sub>i</sub>&#183;<b>r</b><sub>i</sub>) &#183;  &#931;<sub>i</sub> (q<sub>i</sub>&#183;<b>r</b><sub>i</sub>) &gt;<br/>
&nbsp;&#963;(&#956;) = (&lt;<b>&#956;</b><sup>2</sup>&gt; - &lt;<b>&#956;</b>&gt;<sup>2</sup>)<sup>1/2</sup><br/>
<br/>
The dipole moment is a vectorial quantity. The q<sub>i</sub> and <b>r</b><sub>i</sub> are the
charges and positional vectors of the polar atoms in the molecule. &lt;|<b>&#956;</b>|&gt;
and the square root of &lt;<b>&#956;</b><sup>2</sup>&gt; are identical in the absence of fluctuations (rigid
molecule). Since molecular coordinates are not aligned to a reference frame before computation of dipole
moments, the average vector components are expected to approach zero in the case of free rotational averaging.
Note as well that nonneutral molecule types are omitted.<br/>
<br/>
In MPI averaging calculations, this is automatically averaged over all nodes.<br/>
<br/>
In (S)GC ensemble calculations, the data are only obtained over physically present molecules.<br/>
<br/>
<br/>
<a class="NoHi" id="RESDIPOLES.dat"><b>RESDIPOLES.dat</b></a><br/>
<br/>
Average dipole moments for entire, net-neutral residues in Debye<br/>
<br/>
Format: i10,1x,g14.7,1x,g14.7,1x,g14.7,1x,g14.7,1x,g14.7,1x,g14.7 (7 columns)<br/>
<br/>
This file is largely analogous to <a href="outputfiles.html#MOLDIPOLES.dat">MOLDIPOLES.dat</a>,
only that the data here are resolved by residue with the first column providing residue number. No pooling of data by
analysis group occurs.
Also, all dipole moments (in Debye) over net-neutral residues are computed instead of over entire molecules.
Since residue coordinates are not aligned to a reference frame before computation of dipole
moments, the average vector components are expected to approach zero in the case of free rotational averaging.
Note that while residues might be omitted, the molecules they
belong to still might have a computable dipole moment.<br/>
<br/>
In MPI averaging calculations, this is automatically averaged over all nodes.<br/>
<br/>
In (S)GC ensemble calculations, the data are only obtained over physically present molecules.<br/>
<br/>
<br/>
<a class="NoHi" id="JCOUPLING.dat"><b>JCOUPLING.dat</b></a><br/>
<br/>
This file reports estimates for the vicinal NMR coupling constant (in Hz), J(H<sub>&#945;</sub>,H<sub>N</sub>),
of the alpha- and the amide protons of each eligible polypeptide residue.<br/>
<br/>
Format (data lines): i10,1x,g14.7<br/>
<br/>
Note that proline and &#945;-aminoisobutyric acid are excluded 
because one of the protons is missing, and that for glycine an average over both possible protons is computed and reported.
Column 1 lists residue number and column 2 lists the estimated coupling constant in Hz. The values are estimated by applying
a linear combination of different parameterizations of the Karplus
equation (&rarr; <a href="references.html#ref23_2">reference</a>). Note that estimating NMR parameters for disordered systems
from simulation data carries several caveats which pertain to the parameterization
of the Karplus equation, assumptions about the type of averaging, <i>etc</i> ...<br/>
<br/>
In MPI averaging calculations this is automatically averaged over all
nodes.<br/>
<br/>
In (S)GC ensemble calculations, the data are only obtained over physically present polypeptide residues.<br/>
<br/>
<br/>
<a class="NoHi" id="N_XXX_OVERLAP.dat"><b>N_XXX_OVERLAP.dat</b></a><br/>
<br/>
Overlap measures for parallel runs with different Hamiltonians<br/>
<br/>
Format: i6,1x,k(g14.7,1x)) where k is dependent on the setup of the parallel run (see <a href="keywords.html#REMC">elsewhere</a>)<br/>
<br/>
This output file lists overlap between Hamiltonians in a replica
exchange calculation, and is not written in any other run. For each
Hamiltonian (indexed by XXX) a separate file is written. This file adopts different formats depending on the <a
 href="keywords.html#REMC">replica exchange</a> specifications. In the simple case of pure
temperature replicas, the format for a given Hamiltonian <i>i</i> is just (here, the
Hamiltonian is not explicitly replica-dependent, so there is only one
line):<br/>
<i>Column 1:</i><br/>
&nbsp;<i>i</i><br/>
<i>Column 2:</i><br/>
&nbsp;Temperature of <i>i</i> in K<br/>
<i>Column 3:</i><br/>
&nbsp;&lt; E<sub>i</sub>(x)&gt; in kcal/mol<br/>
<br/>
In the case of arbitrary replicas which include temperature as a
dimension, cross (foreign) energies from node <i>i</i> (self) to node <i>j</i> are
computed and written out in various transformations:<br/>
<i>Column 1:</i><br/>
&nbsp;<i>j</i><br/>
<i>Column 2:</i><br/>
&nbsp;Temperature of <i>j</i> in K<br/>
<i>Column 3:</i><br/>
&nbsp;&lt; E<sub>j</sub>(x) - E<sub>i</sub>(x) &gt; in kcal/mol (the average total work to convert from
<i>i</i> to <i>j</i>)<br/>
<i>Column 4:</i><br/>
&nbsp;-k<sub>b</sub>T<sub>i</sub> &#183; ln( &lt;exp( &#946;<sub>i</sub>&#183;(E<sub>j</sub>(x) - E<sub>i</sub>(x)) )&gt; ) in kcal/mol<br/>
<i>Column 5:</i><br/>
&nbsp;&lt; E<sub>i</sub>(x)&gt; in kcal/mol (only given for line corresponding to self, <i>i.e.</i>,
<i>j</i> = <i>i</i>)<br/>
<br/>
In the case of replicas which allow for computation of derivatives
with respect to the replica parameters (all of the dimensions must come from choices:
2,3,4,5,6,8,9,10,11 &rarr; <a href="inputfiles.html#FMCSC_REFILE">FMCSC_REFILE</a>) the output
includes the derivative:<br/>
<i>Column 1:</i><br/>
&nbsp;<i>j</i><br/>
<i>Column 2:</i><br/>
&nbsp;&lt; E<sub>j</sub>(x) - E<sub>i</sub>(x) &gt; in kcal/mol (the average total work to convert from
<i>i</i> to <i>j</i>)<br/>
<i>Column 3:</i><br/>
&nbsp; -k<sub>b</sub>T<sub>i</sub> &#183; ln( &lt;exp( &#946;&#183;(E<sub>j</sub>(x) - E<sub>i</sub>(x)) )&gt; ) in kcal/mol (the free energy difference between <i>i</i> and <i>j</i> as defined in a free energy perturbation (FEP) sense)<br/>
<i>Column 4:</i><br/>
&nbsp;&lt; &#931;<sub>k&le;<a href="keywords.html#REDIM">REDIM</a></sub> ( dE<sub>i</sub>(x) / d&#955;<sub>k</sub> ) &gt; in kcal/mol (not that the allowed control parameters are all unitless)<br/>
<br/>
A few comments are needed for the reported derivative (required for thermodynamic integration). Since in terms of permissible &#955;<sub>k</sub> the Hamiltonian is a linear combination of the terms, the total derivative is simply obtained by summing over the partials with respect to the various dimensions. As an exception, simultaneous use of choices 9 and either 10 or 11 may render the reported values uninterpretable. Moreover, the subsequent integration may require partial derivatives so that for a multidimensional replica setting, it is usually advantageous to split the run into multiple segments, along which only a single dimension is changing.<br/>
<br/>
In any other case:<br/>
<i>Column 1:</i><br/>
&nbsp;<i>j</i><br/>
<i>Column 2:</i><br/>
&nbsp;&lt; E<sub>j</sub>(x) - E<sub>i</sub>(x) &gt; in kcal/mol (the average total work to convert from
<i>i</i> to <i>j</i>)<br/>
<i>Column 3:</i><br/>
&nbsp; -k<sub>b</sub>T<sub>i</sub> &#183; ln( &lt;exp( &#946;&#183;(E<sub>j</sub>(x) - E<sub>i</sub>(x)) )&gt; ) in kcal/mol (the free energy difference between <i>i</i> and <i>j</i> as defined in a free energy perturbation (FEP) sense)<br/>
<br/>
Note that all the &lt;...&gt; denote ensemble averages for the native Hamiltonian of that node (<i>i</i>). Also note that by user choice (see <a href="keywords.html#REOLALL">FMCSC_REOLALL</a>), the only relevant overlap
information is obtained for the (one or two) neighboring replicas. In this scenario, the
entries in all other rows are strictly meaningless.<br/>
<br/>
<br/>
<a class="NoHi" id="N_XXX_EVEC.dat"><b>N_XXX_EVEC.dat</b></a><br/>
<br/>
Instantaneous cross-energies for parallel runs with different Hamiltonians:<br/>
<br/>
Format: N<sub>nodes</sub>(g18.11,1x) where N<sub>nodes</sub> is the number of different replicas<br/>
<br/>
This output file lists instantaneous cross-energies (in kcal/mol) between
Hamiltonians in a replica exchange calculation and is not written in
any other run. For each Hamiltonian (indexed by XXX) a separate file is written.
Its presence requires the keyword <a href="keywords.html#REOLINST">FMCSC_REOLINST</a> to be set.<br/>
<br/>
The file simply lists energies for the current structure evolved by
Hamiltonian <i>i</i>, for the different Hamiltonians 1 ... <i>i</i> ... N<sub>nodes</sub> (see <a href="keywords.html#REPLICAS">FMCSC_REPLICAS</a>). One
line is written per snapshot analyzed in this way. Note that it is possible to restrict
calculation and output to neighboring replicas exclusively. This is important (and useful) since
cross-energy evaluations are costly and many analysis methods for such data only require 
information from neighboring nodes (see <a href="keywords.html#REOLALL">FMCSC REOLALL</a>).
In this case, the vector for each snapshot contains
at most three relevant entries for a replica <i>i</i>: the left neighbor in the leftmost column (if <i>i</i> is not the first node),
then the current self-energy, and lastly the right neighbor in the rightmost column (if <i>i</i> is not the last node).<br/>
<br/>
This output can be used to determine the free energy difference between
replicas through the Bennett Acceptance Ratio (BAR) method. <br/>
<br/>
<br/>
<a class="NoHi" id="N_000_PROBABILITIES.dat"><b>N_000_PROBABILITIES.dat</b></a><br/>
<br/>
Average swap probabilities for <a href="keywords.html#REMC">replica exchange runs</a>:<br/>
<br/>
Format: N<sub>nodes</sub> lines of N<sub>nodes</sub>(g16.9,1x), where N<sub>nodes</sub> is the number of different replicas<br/>
<br/>
This output file lists average swap probabilities for all pairs of replicas considered for exchange,
for all other pairs it will simply give zero. The matrix is symmetric and printed redundantly.
Statistical weights for individual entries can differ considerably in cases of many <a href="keywords.html#REPLICAS">replicas</a>
and small numbers of swaps (&rarr; <a href="keywords.html#REFREQ">FMCSC_REFREQ</a> and
<a href="keywords.html#RESWAPS">FMCSC_RESWAPS</a>). Each instantaneous value for a swap probability 
is taken in the Metropolis definition (so that values approaching unity indicate maximal
ensemble similarity/compatibility).<br/>
<br/>
This file is only written once for the entire replica exchange run (for the master node).<br/>
<br/>
<br/>
<a class="NoHi" id="N_000_REXTRACE.dat"><b>N_000_REXTRACE.dat</b></a><br/>
<br/>
Replica vs. initial conformation trace for <a href="keywords.html#REMC">replica exchange runs</a>:<br/>
<br/>
Format: N<sub>sc</sub> lines of i12,1x,N<sub>nodes</sub>(i*,1x), where N<sub>nodes</sub> is the number of different replicas,
and N<sub>sc</sub> is the total number of <a href="keywords.html#REFREQ">swap cycles</a> performed<br/>
<br/>
This output file uses an integer code to provide the history of the various starting conformations
(geometrically defined) migrating through the space of different conditions. The first column
holds the step number, at which a given swap cycle was performed, and the remaining columns 
correspond to the different <a href="keywords.html#REPLICAS">conditions</a> defined via
<a href="keywords.html#REFILE">FMCSC_REFILE</a>. In each column, the integer shown indicates the
condition the structure presently associated with it originated from. This means that it is generally possible
to extract a trajectory continuous in geometrical representation from the combined trajectory files
provided by the replica exchange run (see <a href="keywords.html#XYZOUT">FMCSC_XYZOUT</a>).<br/>
<br/>
This file is only written once for the entire replica exchange run (for the master node).<br/>
<br/>
<br/>
<a class="NoHi" id="N_000_PIGSTRACE.dat"><b>N_000_PIGSTRACE.dat</b></a><br/>
<br/>
Replica reseeding trace for <a href="keywords.html#MPI_PIGS">PIGS runs</a>:<br/>
<br/>
Format: N<sub>sc</sub> lines of i12,1x,N<sub>nodes</sub>(i*,1x), where N<sub>nodes</sub> is the number of different replicas,
and N<sub>sc</sub> is the total number of <a href="keywords.html#REFREQ">reseeding cycles</a> performed<br/>
<br/>
This output file uses an integer code to indicate whether a given replica received a new structure
at a given reseeding interval. The first column holds the step number. If the run is a <a href="keywords.html#MPIAVG">PIGS analysis run</a>,
this number is computed from keywords <a href="keywords.html#RE_TRAJTOTAL">FMCSC_RE_TRAJTOTAL</a> and 
<a href="keywords.html#RE_TRAJOUT">FMCSC_RE_TRAJOUT</a>. Otherwise it is the simulation step number.
In PIGS, trajectories are terminated and restarted from points 
corresponding to the current states of other replicas. From column 2, for each of the replicas,
the integer corresponds to the actual replica number (starting at 1) if no new structure was received.
If the number differs, then this is the replica from which the new structure was received.
The PIGS reseeding is performed <i>after</i> analysis operations have been performed. This includes
writing of trajectory output. Therefore, if <a href="keywords.html#XYZOUT">FMCSC_XYZOUT</a> is set to
a factor of <a href="keywords.html#REFREQ">FMCSC_REFREQ</a>, and if <a href="keywords.html#MPIAVG_XYZ">FMCSC_MPIAVG_XYZ</a>
is enabled, trajectory output will contain a time point (corresponding exactly to a reseeding point) that 
contains the final structures of those trajectories being terminated.<br/>
Note that this format is different from that for
<a href="outputfiles.html#N_000_REXTRACE.dat">N_000_REXTRACE.dat</a>.<br/>
<br/>
This file is only written once for the entire PIGS run (for the master node). Note that for a PIGS simulation
that is not <a href="keywords.html#RESTART">restarted</a>, this file is always overwritten whereas for a PIGS analysis 
run it is always appended.<br/>
<br/>
<br/>
<a class="NoHi" id="HOLES.dat"><b>HOLES.dat</b></a><br/>
<br/>
The first two lines of this output file contain the parameters for probe spheres, <i>i.e.</i>, the
sampled the distances from the polymer's center of mass (R) and - secondly - the probe radii (in &#8491;).
Both lines list three items, <i>viz.</i>, starting value, increment, and the number of values.
Subsequent lines hold for each output step (see <a href="keywords.html#HOLESCALC">FMCSC HOLESCALC</a>) series of
ones and zeros:
1 = no overlap (found a spherical void space), 0 = overlap (did not find a spherical void space).
The list is sorted by increasing probe sphere size first,
then each subgroup of sphere sizes is sorted by increasing distance from the center of mass.<br/>
<br/>
Format: (f6.2,f6.2,i4) for the first two rows; the remainder of rows uses N<sub>s</sub>(N<sub>r</sub>(i2),2x)
where N<sub>s</sub> is the number of sphere radii considered, and N<sub>r</sub> is the number of distances
from the center of mass considered<br/>
<br/>
Note that for each snapshot a single random vector from the center of mass is chosen, on which all of the points of increasing distance are situated. This analysis is limited to non-periodic boundaries and simulations with a single molecule which automatically
excludes simulations in the (S)GC-ensembles.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this instantaneous output file does not utilize the weights in any way.<br/>
<br/>
This is currently not supported in MPI averaging calculations.<br/>
<br/>
<br/>
<a class="NoHi" id="PHTIT.dat"><b>PHTIT.dat</b></a><br/>
<br/>
Instantaneous output reporting on estimated titration states of ionizable residues.
The first (header) line lists the ionizable residues. Afterwards, each
subsequent line contains the charge state of these residues, at
intervals specified by the corresponding keyword (see <a
 href="keywords.html#PHOUT">PHOUT</a>).<br/>
<br/>Format (sans header): N<sub>i</sub>(i10), where N<sub>i</sub> is the number of ionizable residues<br/>
<br/>
It is important to note that for this output to
be meaningful, pH-pseudo moves have to be enabled (see <a
 href="keywords.html#PHFREQ">PHFREQ</a>) which implies that it is only available in straight Monte Carlo simulations.
This is in addition to the requirement to set target pH and (continuum) ionic
strength (see <a href="keywords.html#PH">PH</a> and <a href="keywords.html#IONICSTR">IONICSTR</a>).<br/>
<br/>
It is an important limitation of this analysis that only typically occurring charge states and only five canonical amino acid sidechains (Lys, Arg, Asp, Glu, His) are considered, <i>e.g.</i>, it is not possible to estimate pK<sub>a</sub>-values for polynucleotides or under conditions at which unusual (de)protonation events would occur. Furthermore, this output is largely untested
and needs to be considered / analyzed with the utmost care.<br/>
<br/>
This is currently not supported in any form of MPI calculation.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this instantaneous output file does not utilize the weights in any way.<br/>
<br/>
In (S)GC ensemble calculations, output values for non-present molecules are offset by 10 to indicate presence.<br/>
<br/>
<br/>
<a class="NoHi" id="DIPOLE_GROUPS.vmd"><b>DIPOLE_GROUPS.vmd</b></a><br/>
<br/>
If the Coulombic potential term (<a href="keywords.html#SC_POLAR">POLAR</a>) is used and the flag <a
 href="keywords.html#DIPREPORT">DIPREPORT</a> is turned on, this output
file will be written. It is essentially a script for the freely
available and widely used molecular visualization software <a href="references.html#ref23_9">VMD</a>.
On a per-residue basis, it generates transparent
color VdW representations (over a standard, opaque licorice representation) for all
charge groups it has determined. The first charge group in a residue is
always shown in red, the second in gray, the third in orange and so on
(follows ColorID in VMD). The only caveat is that sometimes different
charge groups in adjacent residues might be directly next to one
another <i>and</i> share the same color, so it is important to always keep in
mind that charge groups can <i>never</i> span two residues.<br/>
<br/>On LINUX run by: vmd -e DIPOLE_GROUPS.vmd<br/>
(requires the initial pdb-file <a href="outputfiles.html#basename_START.pdb">{basename}_START.pdb</a>, see above)<br/>
<br/>
This can be useful if partial charges were modified in the parameter file
and/or <a href="keywords.html#CPATCHFILE">patched</a>, and in particular if the 
groups themselves were <a href="keywords.html#NCPATCHFILE">meant to be modified</a>.
Note that, despite its name, this output file visualizes all charge groups irrespective
of whether their total absolute charge is below or above the <a href="keywords.html#POLTOL">tolerance setting</a>.<br/>
<br/>
<br/>
<a class="NoHi" id="FOS_GROUPS.vmd"><b>FOS_GROUPS.vmd</b></a><br/>
<br/>
If the ABSINTH implicit solvent model is used (<a href="keywords.html#SC_IMPSOLV">IMPSOLV</a>) and the flag <a
 href="keywords.html#FOSREPORT">FOSREPORT</a>  is turned on, this output
file is written. It is completely analogous to <a
 href="outputfiles.html#DIPOLE_GROUPS.vmd">DIPOLE_GROUPS.vmd</a>, only
that it documents the solvation (rather than charge) groups in the system. This file is exceptionally
useful if the definitions of solvation groups were altered through a <a href="keywords.html#FOSPATCHFILE">file-based patch</a>.<br/>
<br/>
<br/>
<a class="NoHi" id="MONOPOLES.vmd"><b>MONOPOLES.vmd</b></a><br/>
<br/>
If the Coulombic potential term (<a href="keywords.html#SC_POLAR">POLAR</a>) is used, the flag <a
 href="keywords.html#DIPREPORT">DIPREPORT</a> is turned on, and there
are charge groups with a charge on them, whose absolute magnitude exceeds the
<a href="keywords.html#POLTOL">tolerance setting</a>, this output file is
written. It is similar to <a href="outputfiles.html#DIPOLE_GROUPS.vmd">DIPOLE_GROUPS.vmd</a> and shows the
monopole center determined by the program for each such charge group
 (relevant for energy evaluations: see <a href="keywords.html#LREL_MC">FMCSC_LREL_MC</a> and <a href="keywords.html#LREL_MD">FMCSC_LREL_MD</a>).
It will also reflect changes made to this categorization based on the <a href="keywords.html#NCPATCHFILE">corresponding patch</a>.<br/>
<br/>
<br/>
<a class="NoHi" id="DENSITY.nc"><b>DENSITY.nc</b></a><br/>
<br/>
This is a binary output file in <a href="references.html#ref16_1">NetCDF format</a>.
It contains the ensemble average of the spatial density of an <a href="keywords.html#EMPROPERTY">atomic
property</a> distributed on a grid. The computation of said spatial density is described
for the keyword that this output relies on, <i>viz.</i>, <a href="keywords.html#EMCALC">FMCSC_EMCALC</a>.
The format of the file is also explained <a href="inputfiles.html#FMCSC_EMMAPFILE">elsewhere.</a>
The grid dimensions are generally determined by system <a href="keywords.html#SIZE">size</a>,
<a href="keywords.html#SHAPE">shape</a>, and the choice of <a href="keywords.html#EMDELTAS">formal resolution</a>.
Note that a lattice-represented density has a discretization error. Moreover, the property
in question is assumed to be homogeneously distributed over an estimate of the atomic volume,
which generally corresponds to the exclusion volume of the atom's electrons. The recording
of the density can be "blurred" by changing the point representation function constrained to the
lattice cell size to a <a href="keywords.html#EMBSPLINE">Gaussian-like function with controllable width</a>.
This is preferable over assuming larger, effective volumes for atoms. This file can be visualized and
converted by <a href="references.html#ref23_10">UCSF Chimera</a>.<br/>
<br/>
The computation frequency for this is regulated by <a href="keywords.html#EMCALC">FMCSC_EMCALC</a>.<br/>
<br/>
In MPI averaging mode this is averaged automatically over all nodes.<br/>
<br/>
In (S)GC ensemble calculations, the spatial density data are only obtained using physically present matter.<br/>
<br/>
<br/>
<a class="NoHi" id="DENSITY_INPUT_PHYS.nc"><b>DENSITY_INPUT_PHYS.nc</b></a><br/>
<br/>
This is a binary output file in <a href="references.html#ref16_1">NetCDF format</a> that is
identical in format to <a href="outputfiles.html#DENSITY.nc">DENSITY.nc</a>. It is only produced
if the <a href="keywords.html#SC_EMICRO">spatial density restraint potential</a> is in use, and it
contains the interpreted, <i>i.e.</i>, shifted and scaled 
<a href="keywords.html#EMMAPFILE">input density map</a>. This output is useful because the origin
of the input map may have been shifted slightly by CAMPARI, and because nonlinear modifications controlled
by keywords <a href="keywords.html#EMREDUCE">FMCSC_EMREDUCE</a>, <a href="keywords.html#EMTRUNCATE">FMCSC_EMTRUNCATE</a>, 
and <a href="keywords.html#EMFLATTEN">FMCSC_EMFLATTEN</a> may be in use. Even if the original
input map is kept as is as much as possible, inspection of this output file can give an idea
of what different density levels correspond to in terms of physical density. This file can be visualized and
converted by <a href="references.html#ref23_10">UCSF Chimera</a>.<br/>
<br/>
In MPI averaging calculations, this file is only produced once.<br/>
<br/>
<br/>
<a class="NoHi" id="DIFFRACTION.dat"><b>DIFFRACTION.dat</b></a><br/>
<br/>
2D-histogram of the intensity map for the diffraction pattern in the
reciprocal axial (varies with row) and radial (varies with column)
dimensions.<br/>
<br/>
Format: <a href="keywords.html#DIFFRRMAX">FMCSC_DIFFRRMAX</a>(g14.7,1x) with <a href="keywords.html#DIFFRZMAX">FMCSC_DIFFRZMAX</a> rows<br/>
<br/>
The intensity is arbitrarily scaled, since heavy atoms are
assumed to have mass-dependent atomic scattering factors
(sqrt(intensity)). These diffraction calculations are preliminary and
to be used with caution. They follow the basic formula:<br/>
<br/>
<img style="width: 640px; height: 19px;"
 alt="I(R,Z) = sum_n ( ( sum_i(f*J_n(R*r_i)*cos(Z*z_i - n*phi_i)) ^2 ) + ( sum_i(f*J_n(R*r_i)*sin(Z*z_i - n*phi_i)) ^2 ) )"
 src="website_equations/outputfiles_diffraction.gif"></img><br/>
<br/>
Here, the <i>r<sub>i</sub></i>, <i>z<sub>i</sub></i>, and <i>&#966;<sub>i</sub></i> are cylindrical coordinates of atom <i>i</i>, <i>n</i>
is the index for the order of Bessel functions (finite maximum
determined by keyword <a href="keywords.html#DIFFRJMAX">FMCSC_DIFFRJMAX</a>), <i>f</i> is the atomic scattering factor
(see above) and <i>R</i> and <i>Z</i> are the reciprocal space coordinates on the
diffraction map (in inverse Angstroms). The inner sum is always taken
over all atoms in the system, which have a non-zero scattering factor
<i>f</i>. The cylindrical coordinates are generated by using a user-supplied
constant axis (useful for single-point analyses) or by always using the
maximally long dimension in the system (<i>i.e.</i>, the axis defined by the
maximum atom-atom distance vector). See <a
 href="keywords.html#DIFFRAXIS">FMCSC_DIFFRAXIS</a> and <a
 href="keywords.html#DIFFRAXON">FMCSC_DIFFRAXON</a> for details.<br/>
<br/>
The dimensions of the system are given by the user choices (see
keywords <a href="keywords.html#DIFFRRMAX">FMCSC_DIFFRRMAX</a> and
<a href="keywords.html#DIFFRZMAX">FMCSC_DIFFRZMAX</a>), and 
the computation frequency for this is regulated by <a href="keywords.html#DIFFRCALC">FMCSC_DIFFRCALC</a>.<br/>
<br/>
In MPI averaging mode this is averaged automatically over all nodes.<br/>
<br/>
In (S)GC ensemble calculations, the diffraction data are only obtained using physically present matter.<br/>
<br/>
<br/>
<a class="NoHi" id="CLUSTERING_FEATURES.nc"><b>CLUSTERING_FEATURES.nc</b></a><br/>
<br/>
This is a binary output file containing the input data used for <a href="keywords.html#CCOLLECT">structural clustering</a>.
This file is only available if CAMPARI has been compiled with NetCDF support
(see <a href="install.html">installation instructions</a>).<br/>
<br/>
Format: The format is the NetCDF standard defined for reading/writing data sets in
CAMPARI's <a href="keywords.html#netcdf_analysis">data mining mode</a> (see
<a href="inputfiles.html#FMCSC_NCDM_NCFILE">elsewhere</a>). Briefly, it contains the actual
data in a variable called "featuresvals" of dimension "nframes" by "nfeats" with the former
being unlimited. Possible <a href="keywords.html#CMODWEIGHTS">dynamic weights</a> are in an array
of the same size called "dynamic_weights", and possible static weights are in a vector
called "static_weights" of length "nfeats". <br/>
<br/>
XXXXXXXXXXXXXXXXXXXXXXXXXX<br/>
<br/>In MPI averaging mode, features are extracted independently by the individual replicas from the separate
input files. The extracted features are concatenated in this joint output file. 
Trajectories are arranged by replica number from low  to high numbers.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the data accumulation for clustering does not take note of whether particles
are physically present or not. This ensures coherent calculation of conformational distances. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<a class="NoHi" id="FRAMES_NBL.nc"><b>FRAMES_NBL.nc</b></a><br/>
<br/>
This is a binary output file in NetCDF format. If data for structural clustering were collected (&rarr; <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>), and an algorithm was
used that requires a rigorous snapshot neighbor list (currently only hierarchical clustering &rarr; 
<a href="keywords.html#CMODE">FMCSC_CMODE</a>), this file will be written.
It contains information about the indices and distances of neighboring snapshots within a certain threshold
selected via <a href="keywords.html#CCUTOFF">FMCSC_CCUTOFF</a> for each snapshot.
These neighbor lists will be sorted by ascending distance. Naturally, with a fixed distance
threshold, different snapshots will have different numbers of near neighbors. The file
is partially redundant since for two snapshots i and j meeting the threshold criterion, both i will be
listed as a neighbor of j, and j as a neighbor of i.<br/>
In terms of the internal formatting, the NetCDF-file will contain a dimension ("framepairs")
giving the total number of (redundant) neighbor pairs. Two integer variables of this size
give the indices of the first and second frames for each pair ("snapshots" and "neighbors"),
while the single precision floating point variable "distances" provides the corresponding distance values.
Attributes provide further information about the type of proximity measure used, <i>etc.</i><br/>
This file can easily become very large. Whether it is being written or read (through
use of keyword <a href="keywords.html#NBLFILE">FMCSC_NBLFILE</a>), 1-2 times its physical size will provide
a good indication of the amount of virtual memory needed for the run.
<br/>
<br/>In MPI averaging mode, structural clustering will automatically concatenate the individual
trajectories. The integer indices in the file refer to this joint trajectory with the individual
segments arranged by increasing replica number.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the data accumulation for clustering does not take note of whether particles
are physically present or not. This ensures coherent calculation of conformational distances. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<a class="NoHi" id="STRUCT_CLUSTERING.clu"><b>STRUCT_CLUSTERING.clu</b></a><br/>
<br/>
This is most often a single column file that lists the integer of the determined
cluster (&rarr; <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>) for
each analyzed snapshot. In other words it prints out one or more coarse-grained trajectories
with the integer code referring to individual states. The highest occurring integer in each
column corresponds to the number of clusters for that clustering. 
Structural clustering is performed after the simulation
(usually a <a href="keywords.html#PDBANALYZE">trajectory analysis run</a>) has completed.
Keyword <a href="keywords.html#CMODE">FMCSC_CMODE</a> determines the chosen clustering algorithm.
If the tree-based algorithm is used and <a href="keywords.html#BIRCHMULTI">multi-resolution output</a> is requested,
this output file will have 1+<i>M</i> columns where <i>M</i> is the choice for 
keyword <a href="keywords.html#BIRCHMULTI">FMCSC_BIRCHMULTI</a>. In this case, columns from left to right
will correspond to clusterings with increasingly coarser resolutions (the actual values 
are calculated from keywords <a href="keywords.html#CRADIUS">FMCSC_CRADIUS</a> and <a href="keywords.html#CMAXRAD">FMCSC_CMAXRAD</a> 
and are reported in log output). The numberings for the individual state trajectories are independent
between columns. In all other cases, only a single column with a clustering using
a threshold of <a href="keywords.html#CRADIUS">FMCSC_CRADIUS</a> is reported. 
Clusters are always sorted by decreasing size such that index 1 corresponds to the largest cluster. The order
of clusters with identical sizes must be assumed arbitrary unless keyword
<a href="keywords.html#CRESORT">FMCSC_CRESORT</a> is used (which ensures a tie-free secondary sorting 
by the indices of the centroid representatives).<br/>
To interpret the file and map the output back to a previously generated input trajectory
(<a href="keywords.html#PDBANALYZE">trajectory analysis run</a>), knowledge of keywords
<a href="keywords.html#NRSTEPS">FMCSC_NRSTEPS</a>, <a href="keywords.html#EQUIL">FMCSC_EQUIL</a>,
<a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>, and <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a>
is critical. For example:<br/>
<br/>
<tt>
<a href="keywords.html#PDBANALYZE">FMCSC_PDBANALYZE</a> 1<br/>
<a href="keywords.html#PDBFILE">FMCSC_PDBFILE</a> sample.pdb<br/>
<a href="keywords.html#NRSTEPS">FMCSC_NRSTEPS</a> 1000<br/>
<a href="keywords.html#EQUIL">FMCSC_EQUIL</a> 100<br/>
<a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a> 45<br/>
</tt>
<br/>
Here, we assume <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a> was not specified. Then (assuming
"sample.pdb" does contain at least 1000 snapshots), the first entry in this output file would correspond
to the 135<sup>th</sup> snapshot in "sample.pdb", the second to the 180<sup>th</sup> and so on.<br/>
If clustering is part of an actual production run, an additional complication arises through the setting
for <a href="keywords.html#XYZOUT">FMCSC_XYZOUT</a>, which ideally should be identical to the one for
<a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>. If that is not desired, small multiples or factors are to
be preferred (otherwise it may become impossible to annotate the clustering results structurally as the
structural information <i>per se</i> is lost after termination of CAMPARI). Most of these issues are
addressed automatically when using the <a href="outputfiles.html#STRUCT_CLUSTERING.vmd">dedicated VMD visualization script</a>
that is meant to simplify visualization of clusters.<br/>
<br/>In MPI averaging mode, structural clustering is available and will automatically concatenate the individual
trajectories. Consequently, the resultant time series is longer by a factor corresponding to the number of processes
(&rarr; <a href="keywords.html#REPLICAS">FMCSC_REPLICAS</a>). Trajectories are arranged by replica number from low 
to high numbers. All clustering analysis is performed by a single MPI process.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the data accumulation for clustering does not take note of whether particles
are physically present or not. This ensures coherent calculation of conformational distances. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<a class="NoHi" id="PROGIDX_xxxxxxxxxxxx.dat"><b>PROGIDX_xxxxxxxxxxxx.dat</b></a><br/>
<br/>
This file is produced if the progress-index based algorithm (&rarr; <a href="keywords.html#CMODE">FMCSC_CMODE</a>) within CAMPARI's
structural clustering facility (&rarr; <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>) has been executed successfully.
It contains the progress index itself (columns 1-3), various annotations functions along with auxiliary 
information (columns 4 and 9-20), and information regarding the structure of the underlying (approximate) minimum
spanning tree (MST) in columns 5-8.<br/>
<br/>
Format: There are <i>N<sub>T</sub></i> rows (see below) each formatted as 4(i10,1x),g12.5,1x,15(i10,1x))<br/>
<br/>
The name of the file is dependent on the starting snapshot the progress index was constructed upon. The twelve-digit 
number of said snapshot (with leading zeros) makes up the part of the filename indicated by "xxxxxxxxxxxx". Note that
the numbering refers to the series of snapshots that were present in memory at the time of analysis, <i>i.e.</i>,
it may not correspond to the numbering in a putative <a href="keywords.html#PDBFILE">input trajectory</a> or
to the numbering of elementary simulation steps. The correspondence depends on the settings for 
<a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>, <a href="keywords.html#EQUIL">FMCSC_EQUIL</a>,
and <a href="keywords.html#RESTART">FMCSC_RESTART</a>.<br/>
To understand the output better, the concept of partitions of snapshots is introduced. Suppose you have a 
trajectory in memory composed of <i>N<sub>T</sub></i> snapshots (see comments above regarding relative numbering),
<i>t<sub>i</sub></i>. By selecting successive snapshots according to a well-defined sequence that relies on 
the underlying MST (&rarr; <a href="keywords.html#CMODE">FMCSC_CMODE</a>), two partitions are created, a set A
of all <i>t<sub>i</sub></i> that have already been added and a set B of all <i>t<sub>i</sub></i> that have not yet been added.
The indices of the snapshots themselves with reference to the trajectory in memory (<i>i.e.</i>, the progress index) are printed out in column 3. 
In this simplified annotation (each snapshot is either in state A or B), the <b>original</b> input trajectory
describes a (putative) Markov chain. The number of transitions, <i>t</i>(B&rarr;B), <i>t</i>(A&rarr;B), <i>t</i>(B&rarr;A),
and <i>t</i>(B&rarr;B) yields the two-state Markov model's transition matrix. The mean first passage time (MFPT) from
state A to state B can then (for example) be computed as <i>n</i>(A)/<i>t</i>(A&rarr;B). For each row in the file,
column 1 holds the current <i>n</i>(A), column 2 the current <i>n</i>(B), and column 4 the sum of <i>t</i>(A&rarr;B) and
<i>t</i>(B&rarr;A). We assume boundary conditions of state B, and therefore the number in column 4 is i) always even
(<i>t</i>(A&rarr;B)=<i>t</i>(A&rarr;B)) and ii), the value is "2" for the last snapshot having been added. This is true
unless the user requests <a href="keywords.html#TRAJBREAKSFILE">transitions to be removed</a> (not counted toward the transition matrix).
In this latter case, uneven numbers may be produced.
A plot of the data in column 1 vs. column 4 represents kinetic information. Whenever the transition counts are (locally) minimal, we may
suspect that a barrier between higher density regions has been encountered. This is conditional upon the requirement
that this barrier not be crossed arbitrarily fast at the given time (increment) resolution of the trajectory.<br/>
Alternatively, we consider an auxiliary annotation function, in which the global A/B model is replaced by an A/B/C model.
Essentially, given a specific ordering of the snapshots (progress index), at each given point, we define a partition A 
as the last <i>l<sub>p</sub></i> snapshots having been added, a partition B as the next <i>l<sub>p</sub></i> snapshots to be added,
and a partition C as all the other snapshots. As before, we can presume that the simplified sequence of states (A, B, or C)
defines a (in this case) three-state Markov model. Columns 9-17 specify the transition counts the transition matrix can be 
constructed from in the order of <i>t</i>(A&rarr;A), <i>t</i>(A&rarr;B), <i>t</i>(A&rarr;C), <i>t</i>(B&rarr;A),
<i>t</i>(B&rarr;B), <i>t</i>(B&rarr;C), <i>t</i>(C&rarr;A), <i>t</i>(C&rarr;B), and <i>t</i>(C&rarr;C). Columns 18-20
give the current partition sizes <i>n</i>(A), <i>n</i>(B), and <i>n</i>(C). The parameter <i>l<sub>p</sub></i> is controllable
through keyword <a href="keywords.html#CPROGINDWIDTH">FMCSC_CPROGINDWIDTH</a>. Again,
<a href="keywords.html#TRAJBREAKSFILE">transitions flagged by the user as spurious</a> do not count toward the transition matrix.<br/>
Lastly, there is some information on the underlying (approximate) MST printed out. Column 6 has the index of the snapshot
that the current snapshot was closest to with reference to the trajectory in memory. In terms of the MST, this snapshot 
is the actual neighbor of the current one. Column 5 has the length of the corresponding edge, <i>i.e.</i>, the
<a href="keywords.html#CDISTANCE">distance</a> between the snapshot in column 3 and the snapshot in column 6. Note
that this distance is the correct one irrespective of the setting for <a href="keywords.html#CPROGMSTFOLD">FMCSC_CPROGMSTFOLD</a>.
Column 7 has the position of the snapshot in column 6 within the progress index. Naturally this number must always be smaller than
the number in column 1 (exempting the very first snapshot which has no "parent" in this sense). Finally, column 8 has the 
value of the standard annotation function for the snapshot in column 6. A dot plot of columns 1 vs. 7 is a useful annotation
function as well.<br/>
As a final comment, the data in column 3 are primarily useful in conjunction with standard instantaneous output. For example,
the progress index determined for folding simulations of small proteins could be annotated by residue-level DSSP assignments
available through output file <a href="outputfiles.html#DSSP_RUNNING.dat">DSSP_RUNNING.dat</a>.<br/>
To summarize, the output in the various columns is:<br/>
<ol>
<li>Cumulative partition function (<i>n</i>(A) in two-state model)</li>
<li>"Inverse" partition function (<i>n</i>(B) in two-state model)</li>
<li>Index of current snapshot in original data (usually related to time of occurrence)</li>
<li>Global cut function (<i>t</i>(A&rarr;B) + <i>t</i>(B&rarr;A) in the two-state model)</li>
<li>Length of the spanning tree edge connecting current snapshot and its parent (variable units)</li>
<li>Index of parent of current snapshot in original data</li>
<li>Position of parent (column 6) in progress index</li>
<li>Global cut function (<i>t</i>(A&rarr;B) + <i>t</i>(B&rarr;A) in the two-state model) at the position of the parent of the current snapshot</li>
<li><i>t</i>(A&rarr;A) in the three-state model</li>
<li><i>t</i>(A&rarr;B) in the three-state model (this together with column 12 constitutes the "localized" cut function)</li>
<li><i>t</i>(A&rarr;C) in the three-state model</li>
<li><i>t</i>(B&rarr;A) in the three-state model (this together with column 10 constitutes the "localized" cut function)</li>
<li><i>t</i>(B&rarr;B) in the three-state model</li>
<li><i>t</i>(B&rarr;C) in the three-state model</li>
<li><i>t</i>(C&rarr;A) in the three-state model</li>
<li><i>t</i>(C&rarr;B) in the three-state model</li>
<li><i>t</i>(C&rarr;C) in the three-state model</li>
<li><i>n</i>(A) in the three-state model</li>
<li><i>n</i>(B) in the three-state model</li>
<li><i>n</i>(C) in the three-state model</li>
</ol>
Note that <a href="tutorial11.html">Tutorial 11</a> contains an R-script that reads this file and creates
a relevant plot from it.<br/>
<br/>In MPI averaging mode, the progress index will automatically be constructed from the joint set of 
trajectories. Consequently, the resultant sequence of snapshots is longer by a factor corresponding to the number of processes
(&rarr; <a href="keywords.html#REPLICAS">FMCSC_REPLICAS</a>). Trajectories are arranged by replica number from low 
to high numbers, and this is what the index in column 3 refers to. The entire analysis is performed on a single computing core.
By default, the resultant spurious transitions at the trajectory breaks will count toward the reported transition numbers
(corrigible by user-defined requests for <a href="keywords.html#FMCSC_TRAJBREAKSFILE">removing such spurious transitions</a>).<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the data accumulation for clustering does not take note of whether particles
are physically present or not. This ensures coherent calculation of conformational distances. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<a class="NoHi" id="MFPT_yyy_CFEP_xxxxxxxx.dat"><b>MFPT_yyy_CFEP_xxxxxxxx.dat</b></a><br/>
<br/>
Cut-based free energy profiles based on the mean first passage time to a reference cluster.<br/>
<br/>
This file is produced if the user requested a cut-based pseudo free energy profile based on the mean first passage time to an eligible
reference cluster (<a href="keywords.html#INISYNSNAP">FMCSC_INISYNSNAP</a>). It is available if a clustering produced by any of CAMPARI's built-in clustering
algorithms (&rarr; <a href="keywords.html#CMODE">FMCSC_CMODE</a> and <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>) has
been performed.
Specifically, it corresponds to choice 1 for <a href="keywords.html#CMSMCFEP">FMCSC_CMSMCFEP</a>.
The file contains a cumulative probability as progress variable (column 3), annotation functions (columns 4, 5, 7),
the quantity used to order clusters (mean first passage time in column 6), and information useful in mapping
clusters (columns 1 and 2).<br/>
<br/>
Format: There are <i>N<sub>C</sub></i> rows each formatted as either (i8,1x,i8,1x,g15.8,1x,i8,1x,g11.4,1x,g13.6,1x,g13.6)
or (i8,1x,i8,1x,g15.8,1x,g13.6,1x,g11.4,1x,g13.6,1x,g13.6)
with <i>N<sub>C</sub></i> being the number of appropriate clusters produced by the clustering that belong to the same
strongly connected component as the first one.<br/>
<br/>
The reader is referred to the <a href="references.html#ref23_12">literature</a> for the theory behind
cut-based pseudo free energy profiles. Briefly speaking, the data are ordered to a criterion measuring
kinetic distance, <i>viz.</i>, here the mean first passage time to the reference cluster obtained by treating
the clustering as a Markov state model (these values are in column 6). In principle, we can now define a series of evenly spaced
cutting planes (in terms of the mean first passage time) and enumerate all the edges in the graph representing
the clustering which cross this plane (the total cut weight, which is usually but not always an integer value
&rarr; <a href="keywords.html#CADDLINKMODE">FMCSC_CADDLINKMODE</a> and <a href="keywords.html#CLINKWEIGHT">FMCSC_CLINKWEIGHT</a>).
To better utilize data, the cutting
planes are not spaced evenly but coincide with the various mean first passage time values for all the other
clusters. The resultant cut weight is printed in column 4. For small cut weights, the 
Ford-Fulkerson theorem (maximum (probability) flow between states is equivalent to the cut with minimal weight separating them)
is used to rationalize that such a cut can be converted to the free energy of a barrier.
These data are printed in column 5 as &Delta;G<sub>AB</sub> = -k<sub>b</sub>T ln N<sub>AB</sub>/NE,
where N<sub>AB</sub> is the cut weight and NE is the total edge weight (including all link modifications such 
as those <a href="keywords.html#CADDLINKMODE">FMCSC_CADDLINKMODE</a> or <a href="keywords.html#TRAJBREAKSFILE">FMCSC_TRAJBREAKSFILE</a>, see
<a href="keywords.html#TMAT_MD">FMCSC_TMAT_MD</a> for details). 
This idea holds approximately for the first barrier surrounding the reference cluster. The primary usefulness of the data
in columns 4 and 5 is to summarize information about the kinetic partitioning of the system in compact form. Note 
that the model underlying the cut function agrees exactly with the two-state one in the
<a href="outputfiles.html#PROGIDX_xxxxxxxxxxxx.dat">output file generated by the progress index method</a>
(but the ordering is different).<br/>
A plot of column 6 vs. column 5 has the disadvantage that the x-axis is heavily contracted for free energy basins
and expanded for barrier regions. To obtain a plot with a more intuitive progress variable, we can for each cut enumerate
the total probability of one of the partitions (here, the one containing the reference node). These data are printed
in column 3, and a plot of column 3 vs. column 5 corresponds to the standard cut-based pseudo free energy profile. 
If <a href="keywords.html#CREWEIGHT">FMCSC_CREWEIGHT</a> is not zero, or if <a href="keywords.html#EIGVAL_MD">spectral decomposition</a>
has been used to obtain the steady state of the underlying transition matrix, these equilibrium probabilities are
used in column 3. Otherwise the raw sampling weights, which are generally different, are taken. Note that the mean first
passage time is a function of the transition matrix and not the sampling weights, so the latter choice is inconsistent.<br/>
The file can be produced for different time directions (based on <a href="keywords.html#TMAT_MD">FMCSC_TMAT_MD</a>) indicated by
"yyy" being either "FWD" (forward time) or "BWD" (backward time). The string "xxxxxxx" (leading zeroes) gives the absolute
index of the target cluster when ordered by decreasing sampling weight (<i>i.e.</i>, it corresponds exactly to the numbering
in output files <a href="outputfiles.html#STRUCT_CLUSTERING.clu">STRUCT_CLUSTERING.clu</a> and
<a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a>). Because the mean first passage
time across distinct strongly connected components can be inifinity, only clusters belonging to the same component are printed. 
If <a href="keywords.html#INISYNSNAP">FMCSC_INISYNSNAP</a> was chosen to be -1, multiple files may be produced
differing in "xxxxxxx", one for each eligible strongly connected component with the reference cluster being the largest 
(by sampling weight) it contains.<br/>
To summarize, the output in the various columns is:<br/>
<ol>
<li>Cluster count (primitive partition function)</li>
<li>Index of the centroid of the current cluster in the original data</li>
<li>Cumulative probability of clusters with a lower value of the ordering parameter (usually after <a href="keywords.html#CREWEIGHT">Markov state model reweighting</a>)</li>
<li>Global cut function (<i>t</i>(A&rarr;B) + <i>t</i>(B&rarr;A) in the implied two-state model)</li>
<li>Negative logarithm of the cut function (pseudo free energy in kcal/mol)</li>
<li>Mean first passage times (in units of input snapshot spacing times <a href="keywords.html#CLAGTIME">FMCSC_CLAGTIME</a>)</li>
<li>Mean square displacement of the (meaning across the) cutting surface in purely geometric terms (variable units)</li>
</ol>
Note that the use of an ordering parameter relative to a reference state raises the possibility of finding
clusters with virtually identical distances, which however would be mutually far away from each other by the same ordering parameter.<br/>
<br/>In MPI averaging mode, structural clustering will automatically concatenate the individual
trajectories (ordered by increasing replica number). By default, the resultant graph will contain spurious transitions 
(at least) at the trajectory breaks that also affect the output of other functionalities.
These can be corrected by the aforementioned modifications to the link structure.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the data accumulation for clustering does not take note of whether particles
are physically present or not. This ensures coherent calculation of conformational distances. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<a class="NoHi" id="PFOLD_PLUS_yyy_CFEP_xxxxxxxx.dat"><b>PFOLD_PLUS_yyy_CFEP_xxxxxxxx.dat</b></a><br/>
<br/>
Cut-based free energy profiles based on (+) committor probabilities.<br/>
<br/>
This file is produced if the user requested a cut-based pseudo free energy profile based on the (+) committor probabilities based on
eligible sets of clusters defining a <a href="keywords.html#CLUFOLDFILE">folded set</a> and an
<a href="keywords.html#CLUUNFOLDFILE">unfolded set</a>. It is available if a clustering produced by any of CAMPARI's built-in clustering
algorithms (&rarr; <a href="keywords.html#CMODE">FMCSC_CMODE</a> and <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>) has
been performed, the code was linked to the HSL libraries, and forward committor probabilities have been
computed (<a href="keywords.html#DOPFOLD">FMCSC_DOPFOLD</a>).
If all conditions are met, it is written if the choice for <a href="keywords.html#CMSMCFEP">FMCSC_CMSMCFEP</a> was either 8 or 10.<br/>
The format and interpretation of the file are identical to output file <a href="outputfiles.html#MFPT_yyy_CFEP_xxxxxxxx.dat">MFPT_yyy_CFEP_xxxxxxxx.dat</a>,
and the user is referred there for documentation.
The only difference in the content of the file is that column 6 now holds the (+) committor probabilities sorted in decreasing order. These should be strictly
bound to the interval [0:1] (deviations indicate an erroneous calculation). The only other difference to the mean first passage time-based
output is that these files are always written for at most a single strongly connected component (<i>i.e.</i>, option
-1 for <a href="keywords.html#INISYNSNAP">FMCSC_INISYNSNAP</a> is ignored). The committor
probabilities themselves are also printed to output file <a href="outputfiles.html#PFOLD_PLUS_xxx.dat">PFOLD_PLUS_xxx.dat</a>.<br/>
<br/>
<br/>
<a class="NoHi" id="PFOLD_MINUS_yyy_CFEP_xxxxxxxx.dat"><b>PFOLD_MINUS_yyy_CFEP_xxxxxxxx.dat</b></a><br/>
<br/>
Cut-based free energy profiles based on (-) committor probabilities.<br/>
<br/>
These files are identical to output files <a href="outputfiles.html#PFOLD_PLUS_yyy_CFEP_xxxxxxxx.dat">PFOLD_PLUS_yyy_CFEP_xxxxxxxx.dat</a>
only that they rely on the (-) committor sorted in increasing order. Computation of the (-) committors additionally requires keyword
<a href="keywords.html#DOPFOLD_MINUS">FMCSC_DOPFOLD_MINUS</a>. The values themselves are also printed to output file
<a href="outputfiles.html#PFOLD_MINUS_xxx.dat">PFOLD_MINUS_xxx.dat</a>.<br/>
<br/>
<br/>
<a class="NoHi" id="STRUCT_CLUSTERING.vmd"><b>STRUCT_CLUSTERING.vmd</b></a><br/>
<br/>
This is a dedicated visualization script that is meant to allow the user to use <a href="references.html#ref23_9">VMD</a>
to quickly generate pictures of the results of structural clustering analysis 
(&rarr; <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>).
It essentially sets up a series of representations corresponding to structural clusters
exceeding a certain threshold size by picking out the appropriate frames from the
trajectory file (either input or output).
Hints and explanations are as follows:<br/>
<ol type="a">
<li>If the run was a <a href="keywords.html#PDBANALYZE">trajectory analysis run</a>, the script
will automatically point to the trajectory file (and either the specified <a href="keywords.html#PDB_TEMPLATE">template file</a>
or the <a href="outputfiles.html#basename_START.pdb">standard reference pdb-files</a>). If
<a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a> was set to something larger than 1, only those frames 
corresponding to snapshots considered for clustering will actually be loaded.</li>
<li>Conversely, if the run was a standard production run, the script
will automatically point to the standard <a href="outputfiles.html#Trajectory_output">trajectory output file</a>
and the <a href="outputfiles.html#basename_START.pdb">standard reference pdb-files</a>. The entire
output trajectory will be loaded, but clustering frame numbers will be adjusted such that
only those snapshots also present in the trajectory will be displayed (as the other ones
are missing). This means that it is possible for the display to be misleading in terms of cluster
spread and size unless <a href="keywords.html#XYZOUT">FMCSC_XYZOUT</a> was identical to
or a factor of <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>.</li>
<li>Currently, only those clusters with at least 50 members are included for visualization in this script.
This is to prevent complete clutter with thousands of tiny "clusters" and resultant
performance degradation within VMD.</li>
<li>The script includes an alignment process which aligns all loaded frames to the same
reference structure, <i>i.e.</i>, either the specified <a href="keywords.html#PDB_TEMPLATE">template file</a>
or on of the <a href="outputfiles.html#basename_START.pdb">standard reference pdb-files</a>.
If actual structures are very different from this reference, alignment may be poor, and - more importantly
- may appear inconsistent for members of the same cluster. This may mask the homogeneity of
a cluster. These errors are more likely to occur if the reference structure is of low complexity,
<i>e.g.</i>, fully extended.</li>
<li>The alignment does not use the subset of degrees of freedom that the user potentially selected
for clustering (via <a href="keywords.html#CFILE">FMCSC_CFILE</a>). Both this and some aspects of
the aforementioned error can be remedied by manually editing the last line of this visualization script
where the align function is called (for further details see <a href="outputfiles.html#basename_VIS.vmd">above</a>).</li>
<li>By default, all representations are turned off initially. They can be turned on one-by-one
through VMD's "Graphical Representations" window.</li>
<li>For each cluster, two representations are created: The first is a collection of all the viewable
frames that are part of a cluster in a macromolecular cartoon representation. The second is 
a licorice representation of the structure deemed to be closest to the geometric center of the 
cluster.</li>
<li>The color indices for representations cycle through. Currently, it is assumed that VMD has 33 color
indices ranging from 0 to 32. Note that the numbering of representations and/or the color indices
do not match the actual numbering of clusters used in the <a href="outputfiles.html#STRUCT_CLUSTERING.clu">ASCII output file</a>,
since not all clusters may be represented (and those are then simply skipped).</li>
<li>For large datasets, the many and long lists of frames provided for each representation may
cause performance and/or stability problems with VMD even at the loading stage. In such a case,
it is recommended to remove blocks of content from this file on a temporary basis. Performance
problems will also occur if the simultaneous display of all the member of a cluster involves
too many graphical objects.</li>
</ol>

Note that a lot of the global options (type of representations to use, alignment details, <i>etc.</i>) 
can be modified rather easily by editing this file. This is certainly more tedious for anything
involving modifications as to what frames to display for each cluster.<br/>
<br/>In MPI averaging simulations, structural clustering is available, but the visualization file will not be produced.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the data accumulation for clustering does not take note of whether particles
are physically present or not. This ensures coherent calculation of conformational distances. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<a class="NoHi" id="STRUCT_CLUSTERING.graphml"><b>STRUCT_CLUSTERING.graphml</b></a><br/>
<br/>
Complete graph of the results of structural clustering (&rarr; <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a> and
related keywords)<br/>
<br/>
Format: The graphml format is an xml-based format (therefore human-readable) and the user
is referred to <a href="http://graphml.graphdrawing.org/primer/graphml-primer.html" target="_TOP">external
resources</a> for further details. Many software packages meant to deal with
graphs will be able to read/process/convert this output file.<br/>
<br/>
The output contains up to 10 node (cluster)-based attributes: as integers, the index of 
the snapshot of its centroid (<i>i0</i>) and the index of the strongly connected component it is a
part of (<i>i2</i>); as real numbers, the cluster's radius, diameter, and "quality" (<i>d1</i>, <i>d2</i>,
and <i>d3</i>, respectively), its statistical weights derived from the raw sampling weight and 
iterative solutions to forward and backward time <a href="keywords.html#CREWEIGHT">steady states
via the iterative algorithm</a> (<i>d0</i>, <i>d6</i>, and <i>d7</i>, respectively), and
lastly <a href="keywords.html#EIGVAL_MD">linear algebra solutions</a> to the same forward
and backward time problems (<i>d8</i> and <i>d9</i>, respectively).
The quality of a cluster (<i>d3</i>) is a measure of tightness. The availability of forward
and backward time solutions depends on keyword <a href="keywords.html#TMAT_MD">FMCSC_TMAT_MD</a>.
The linear algebra solutions
may be available from output file <a href="outputfiles.html#EIGEN_xxx.dat">EIGEN_xxx.dat</a> as well
and are generally restricted to a single strongly connected component. This means that values of strictly
zero indicate that the steady state was not determined for the component this cluster belongs to.
Importantly, the sampling weights are always assumed to sup to 1.0 for <i>all</i> clusters. Because 
components cannot be reweighted against each other, every component retains its
raw sampling weight and the sum of cluster weights for the clusters in each component
preserves this sampling weight. For edges, the output contains 3 attributes, the raw sampling count (<i>i1</i>), the 
mean square geometric distance associated with the transition (<i>d4</i>), and 
a real number weight that is currently not used meaningfully (<i>d5</i>).
Details of how some of these attributes are interpreted may depend on the algorithm chosen
(&rarr; <a href="keywords.html#CMODE">FMCSC_CMODE</a>). In particular, for pure
file-based clustering (<a href="keywords.html#CMODE">FMCSC_CMODE</a> is 7), none
of the attributes are available that have geometric meanings.<br/>
Briefly, the representation of the data as a graph or network make sense primarily if
they were generated by a process that (at least partially) preserves a meaningful
continuous evolution of the system on the underlying (free) energy landscape. If this
is not the case, transition rates cannot report on the barriers separating individual
basins on that landscape. For instance, it would be quite meaningless to represent
a scrambled trajectory as a graph. Similarly, certain Monte Carlo moves 
(&rarr; <a href="keywords.html#movesetcontrols">corresponding documentation</a>)
and in particular the replica-exchange technique (&rarr; <a href="keywords.html#REMC">FMCSC_REMC</a>)
can give rise to spurious transitions that do not respect or represent the underlying barriers.
In such cases, techniques that attempt to map out the underlying free energy landscape
using network transition rates have to be evaluated with caution.<br/>
The output in this file is responsive to several other keywords. In particular, the link
structure will reflect the changes requested through keywords <a href="keywords.html#TRAJBREAKSFILE">FMCSC_TRAJBREAKSFILE</a>,
<a href="keywords.html#TRACEFILE">FMCSC_TRACEFILE</a>, <a href="keywords.html#TRAJLINKSFILE">FMCSC_TRAJLINKSFILE</a>,
and <a href="keywords.html#CADDLINKMODE">FMCSC_CADDLINKMODE</a>.
Furthermore, except for the raw sampling count, the reported statistical weights of each cluster
are available only if the corresponding keywords are used (<a href="keywords.html#CREWEIGHT">FMCSC_CREWEIGHT</a>
and <a href="keywords.html#EIGVAL_MD">FMCSC_EIGVAL_MD</a>).<br/>
<br/>In MPI averaging mode, structural clustering will automatically concatenate the individual
trajectories (ordered by increasing replica number). By default, the resultant graph will contain spurious transitions 
(at least) at the trajectory breaks that also affect the output of other functionalities.
These can be corrected by the aforementioned modifications to the link structure.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the data accumulation for clustering does not take note of whether particles
are physically present or not. This ensures coherent calculation of conformational distances. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>


<br/>
<br/>
<a class="NoHi" id="basename_yyy.fwts"><b>{basename}_yyy.fwts</b></a><br/>
<br/>
Resultant weights per analyzed snapshot from the Markov model steady state solutions assuming either
<a href="keywords.html#TMAT_MD">forward (yyy is "FWD") or backward (yyy is "BWD") time</a>
(&rarr; <a href="keywords.html#CREWEIGHT">FMCSC_CREWEIGHT</a> and <a href="keywords.html#EIGVAL_MD">FMCSC_EIGVAL_MD</a>):<br/>
<br/>
Format: (i14,1x,g19.11E3)<br/>
<br/>
This output file lists indices of simulation snapshots with their associated weights. This file is in the same general format
required for snapshot-based reweighting via <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a>. Output is only
produced for snapshots that were part of a <a href="keywords.html#CCOLLECT">structural clustering analysis</a>
yielding a conformational transition network. Snapshots indices refer to this relative numbering and not
to the original simulation or trajectory data irrespective of whether 
<a href="keywords.html#PDB_ANALYZE">trajectory analysis mode</a> is turned on or whether 
an <a href="keywords.html#MPIAVG">MPI-parallel calculation</a> combining data from all replicas
is performed. The weights printed to these files can also be computed manually from the combined data in output files
<a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a>
and <a href="outputfiles.html#STRUCT_CLUSTERING.clu">STRUCT_CLUSTERING.clu</a>.
An individual snapshot weight is calculated as the ratio of the steady state population of the cluster
it belongs to (assuming the conformational space network can be treated as a Markov state model)
divided by the raw sampling weight (cluster size divided by number of snapshots). Both of those
numbers are reported in <a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a>, and
<a href="outputfiles.html#STRUCT_CLUSTERING.clu">STRUCT_CLUSTERING.clu</a> provides the map from snapshot to
cluster.<br/>
<br/>
For any <a href="keywords.html#MPI_settings">MPI parallel run</a>, these files are produced and printed only by the master
MPI process but using the standard prefix. The step numbers in the file will refer to the processed data 
(depending on <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a> and <a href="keywords.html#NEQUIL">FMCSC_NEQUIL</a>)
as if concatenated by replica and <i>not</i> to the source data (see above).<br/>
<br/>
<br/>
<!-- ************************************************************************************************* -->
<!-- ************************************ tmat_XXX.dat *********************************************** -->
<a class="NoHi" id="TMAT_xxxxxx_yyy.dat"><b>TMAT_xxxxxx_yyy.dat (or TMAT_yyy.dat)</b></a><br/>
<br/>
Simple transition matrix output file for one (or all) strongly connected components of a graph.<br/>
<br/>
Format: "<i>Number of edges in the <a href="keywords.html#TMAT_MD">transition (sub)network</a></i>" rows, <i>6</i> columns. 
Each row is formatted as (I7,1x,I7,1x,I7,1x,I7,1x,G19.11,2x,I9) or (I7,1x,I7,1x,G19.11,2x,I9)<br/> 
<br/>
This file is produced only if <a href="keywords.html#TMATREPORT">FMCSC_TMATREPORT</a> is set to 1 and 
depends on structural clustering being performed (&rarr; <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>).
Here we write the non-null entries t<sub>ij</sub> of the <a href="keywords.html#TMAT_MD">selected transition matrix(ces)</a>, 
which correspond to the edges of the transition network. For the forward time matrix yyy is 'FWD', and
for the backward time matrix yyy is 'BWD'. For the component-specific file, the xxxxxx gives the number (leading zeros) of the 
strongly connected component the matrix corresponds to. This numbering can be found and mapped in output file 
<a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a>, which is always produced. (One-way) transitions
connecting other components are not listed. The global file will simply list all transition regardless of component.
The file is primarily meant to simplify subsequent analysis for individual components using independent tools.
<br/>
The first column stores the indexes of the clusters <i>i</i> of t<sub>ij</sub> relative to the 
reference strongly connected component. Similarly, the second column refers to the clusters <i>j</i>.
For the component-specific file type, the third and fourth columns index the same clusters in absolute terms, <i>i.e.</i> 
numbered as in the whole data set (<a href="outputfiles.html#STRUCT_CLUSTERING.clu">STRUCT_CLUSTERING.clu</a>).
The second-to-last column reports the floating point value of t<sub>ij</sub>, <i>viz.</i> it 
contains the transition probability. The last column stores the transition counts, <i>i.e.</i> 
the number of directed transitions from node <i>i</i> to node <i>j</i>.
All user-requested alterations to the connectivity, <i>e.g.</i> <a href="keywords.html#TRAJBREAKSFILE">FMCSC_TRAJBREAKSFILE</a>
or <a href="keywords.html#CADDLINKMODE">FMCSC_CADDLINKMODE</a> are reflected in these files. The only
exception to this is the iterative constrained maximum likelihood algorithm enabled when
<a href="keywords.html#CADDLINKMODE">FMCSC_CADDLINKMODE</a> is 6 (or -6). Since this algorithm has an inherent
non-integer logic, the last column here just reports the values as if <a href="keywords.html#CADDLINKMODE">FMCSC_CADDLINKMODE</a>
were 0 (other changes are reflected, however). <br/>
Note that this output is generally recoverable from <a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a>
as well.<br/>
<br/>In MPI averaging mode, structural clustering will automatically concatenate the individual
trajectories (ordered by increasing replica number). By default, the resultant graph will contain spurious transitions 
(at least) at the trajectory breaks that also affect the output of other functionalities.
These can be corrected by the aforementioned modifications to the link structure.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the data accumulation for clustering does not take note of whether particles
are physically present or not. This ensures coherent calculation of conformational distances. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<!-- ************************************************************************************************* -->
<!-- **************************************** EIGEN_xxx.dat ****************************************** -->
<a class="NoHi" id="EIGEN_xxx.dat"><b>EIGEN_xxx.dat</b></a><br/>
<br/>
Eigenvalues and (possibly) eigenvectors computed from derived transition matrix(ces).<br/>
<br/>
Format: "<i><a href="keywords.html#NEIGV">FMCSC_NEIGV</a> or <a href="keywords.html#NEIGV">FMCSC_NEIGV</a> + 1</i>"
rows, "<i>2 or 2 + Number of clusters in the reference component</i>" columns.
Each value is formatted as "G19.11" with "2x" separator. <br/>
<br/>
This output file hosts the results of the <a href="keywords.html#EIGVAL_MD">spectral decomposition</a>
of the <a href="keywords.html#TMAT_MD">transition matrix(ces)</a>. The string "XXX" in the file name maps to
"FWD" in case <a href="keywords.html#TMAT_MD">FMCSC_TMAT_MD</a> is set to 1, to "BWD" if <a href="keywords.html#TMAT_MD">FMCSC_TMAT_MD</a>
is 2. If <a href="keywords.html#TMAT_MD">FMCSC_TMAT_MD</a> is set to 3, both files are written.
The first column contains the real part of the <a href="keywords.html#NEIGV">FMCSC_NEIGV</a> eigenvalues
identified with <a href="keywords.html#EIGVAL_MD">FMCSC_EIGVAL_MD</a> while the second column holds their
imaginary parts. If the user asks for the computation of the eigenvectors as well (<a href="keywords.html#DOEIGVECT">FMCSC_DOEIGVECT</a>),
there will be as many additional columns as there are clusters in the reference component in each row.
The additional entries correspond to the values of the eigenvector associated
with the eigenvalue that is in the first two columns of a given row. In general, it is assumed 
that complex conjugate pairs of eigenvalues are printed in direct sequence, the one with
positive imaginary part first. Since the eigenvector is also complex in this case, 
it is spread across both of those rows (real parts first, then imaginary parts), and only the eigenvector
corresponding to the eigenvalue with positive imaginary part is printed.  
If the last computed eigenvalue is the first one in a pair of complex conjugates, <a href="keywords.html#NEIGV">FMCSC_NEIGV</a>
+ 1 rows are written to prevent loss of information regarding the eigenvector. <br/>
The eigenvectors' entries (from right to left) refer to the clusters of the reference component as ordered by population.
The map to the original numbering as found in output files <a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a> 
and <a href="outputfiles.html#STRUCT_CLUSTERING.clu">STRUCT_CLUSTERING.clu</a> can be recovered easily from
<a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a>, which lists both
raw sampling weights and component indices, or from 
<a href="outputfiles.html#TMAT_xxxxxx_yyy.dat">the simplified transition matrix output</a>. 
The reference component is generally chosen by keyword <a href="keywords.html#INISYNSNAP">FMCSC_INISYNSNAP</a>.<br/>
<br/>In MPI averaging mode, structural clustering will automatically concatenate the individual
trajectories (ordered by increasing replica number). By default, the resultant transition matrix(ces) can contain spurious transitions 
(at least) at the trajectory breaks. These can be corrected by modifications to the link structure, <i>e.g.</i>,
<a href="keywords.html#TRAJBREAKSFILE">custom breaks</a>.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the data accumulation for clustering does not take note of whether particles
are physically present or not. This ensures coherent calculation of conformational distances. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<!-- ************************************************************************************************* -->
<!-- *************************************** pfold_XXX_plus.dat ************************************** -->
<a class="NoHi" id="PFOLD_PLUS_xxx.dat"><b>PFOLD_PLUS_xxx.dat</b></a><br/>
<br/>
Forward (+) committor probabilities computed from derived transition matrix(ces).<br/>
<br/>
Format: "<i>Number of clusters that are part of the strongly connected component selected by <a href="keywords.html#CLUFOLDFILE">FMCSC_CLUFOLDFILE</a>
or <a href="keywords.html#INISYNSNAP">FMCSC_INISYNSNAP</a> in <a href="keywords.html#DOPFOLD">(+) committor</a> analyses</i>"
rows, <i>2</i> columns.
Each row is formatted as (I9,2x,F12.6).<br/>
<br/>
In this output file, the values of (+) committors are printed per cluster. The indices of the clusters refer
to the whole data set, <i>i.e.</i>, they follow the same numbering one can find in output files
<a href="outputfiles.html#STRUCT_CLUSTERING.clu">STRUCT_CLUSTERING.clu</a> and <a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a>.
The string "xxx" reflects the <a href="keywords.html#TMAT_MD">assumed direction of time</a>
and it can be either "FWD" or "BWD".<br/>
The set of linear equations solved here is described <a href="keywords.html#DOPFOLD">elsewhere</a>. The solutions
correspond to the probabilities that a random walker initialized in a given cluster reaches the <a href="keywords.html#CLUFOLDFILE">target set (B)</a>
before reaching the <a href="keywords.html#CLUUNFOLDFILE">alternative set (A)</a>. By definition, clusters comprising
sets A and B have values of 0.0 and 1.0, respectively. They are included in this output file for clarity.
Any values outside of the interval [0:1] would indicate a failure to solve the system (either for numerical or 
conceptual reasons). Note that it easy to confuse the forward/backward time direction with the (+)/(-) committor notation.
If <a href="keywords.html#CADDLINKMODE">detailed balance holds</a>, the sum of (+) and (-) committors is always 1.0,
and the results do not depend on <a href="keywords.html#TMAT_MD">time direction</a> <br/>
<br/>In MPI averaging mode, structural clustering will automatically concatenate the individual
trajectories (ordered by increasing replica number). By default, the resultant transition matrix(ces) can contain spurious transitions 
(at least) at the trajectory breaks. These can be corrected by modifications to the link structure, <i>e.g.</i>,
<a href="keywords.html#TRAJBREAKSFILE">custom breaks</a>.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the data accumulation for clustering does not take note of whether particles
are physically present or not. This ensures coherent calculation of conformational distances. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<a class="NoHi" id="PFOLD_MINUS_xxx.dat"><b>PFOLD_MINUS_xxx.dat</b></a><br/>
<br/>
Backward (-) committor probabilities computed from derived transition matrix(ces).<br/>
<br/>
Format: "<i>Number of clusters that are part of the strongly connected component selected by <a href="keywords.html#CLUFOLDFILE">FMCSC_CLUFOLDFILE</a>
or <a href="keywords.html#INISYNSNAP">FMCSC_INISYNSNAP</a> in <a href="keywords.html#DOPFOLD_MINUS">(-) committor</a> analyses</i>"
rows, <i>2</i> columns.
Each row is formatted as (I9,2x,F12.6).<br/>
<br/>
This output file is identical to the one for the (+) committor (&rarr; <a href="outputfiles.html#PFOLD_PLUS_xxx.dat">PFOLD_PLUS_xxx.dat</a>)
only that it contains the values for the (-) committors per cluster. The set of linear equations solved here is described <a href="keywords.html#DOPFOLD_MINUS">elsewhere</a>.
The (-) committor corresponds to the probability
per cluster to have previously been in the alternative (A) rather than the target (B) set.
If <a href="keywords.html#CADDLINKMODE">detailed balance holds</a>, the sum of (+) and (-) committors is always 1.0,
and the results do not depend on <a href="keywords.html#TMAT_MD">time direction</a>.<br/>
<br/>
<br/>
<!-- ************************************************************************************************* -->
<!-- ********************************* MSM_SYN_TRAJ_xxxxx_yyy.frames ********************************** -->
<a class="NoHi" id="MSM_SYN_TRAJ_xxxxx_yyy.frames"><b>MSM_SYN_TRAJ_xxxxx_yyy.frames</b></a><br/>
<br/>
Coarse-grained (cluster/state) trajectories from random walks on a transition network assumed to be Markovian<br/>
<br/>
Format: A descriptive header line followed by a variable number of rows in a single column. Each of these rows is formatted as (I9). <br/>
<br/>
In the header, these files give the length of the random walk along with starting and ending states (clusters). 
Subsequently, they hold sequences of integers that identify the clusters that compose a <a href="keywords.html#SYNTRAJ_MD">synthetic trajectory</a>
with a selected output frequency chosen by keyword  <a href="keywords.html#SYNTRAJOUT">FMCSC_SYNTRAJOUT</a>. 
These integers correspond to the clusters in the same overall numbering that is used in output files
<a href="outputfiles.html#STRUCT_CLUSTERING.clu">STRUCT_CLUSTERING.clu</a> and 
<a href="outputfiles.html#STRUCT_CLUSTERING.graphml">STRUCT_CLUSTERING.graphml</a> (as well as being printed to the log-output).
Random walks either count <a href="keywords.html#NSYNSNAPS">FMCSC_NSYNSNAPS</a> cluster visits or less (in case <a href="keywords.html#SYNTRAJ_MD">FMCSC_SYNTRAJ_MD</a> is 1).
For a walk with a length of 1000 steps and an <a href="keywords.html#SYNTRAJOUT">output frequency</a> of 50, 20 lines would be written
in addition to the header line.These would correspond to steps 50, 100, and so on. 
The creation of these files can be completely suppressed if <a href="keywords.html#SYNTRAJ_MD">FMCSC_SYNTRAJ_MD</a> is 1 even
though random walks are performed.<br/>
The number of synthetic trajectories that can be written is set by keyword <a href="keywords.html#NSYNTRAJS">NSYNTRAJS</a>
and is currently limited to 10<sup>4</sup>. Therefore, the string "xxxxx" in the file name ranges from 00000 to 09999. If 
<a href="keywords.html#SYNTRAJ_MD">FMCSC_SYNTRAJ_MD</a> is 1, the number indexes the attempts and not the successful trajectories.
The string "yyy" refers to the <a href="keywords.html#TMAT_MD">assumed direction of time</a> taken for the transition matrix that
underlies the generation of a given <a href="keywords.html#SYNTRAJ_MD">random walk</a>,
<i>viz.</i> it can either be "FWD" (forward-time transition matrix) or "BWD" (backward-time transition matrix). If <a href="keywords.html#TMAT_MD">FMCSC_TMAT_MD</a>
is set to 3, both types of propagators are used and up to <i>2&#183;<a href="keywords.html#NSYNTRAJS">FMCSC_NSYNTRAJS</a></i>
may be generated (see <a href="keywords.html#SYNTRAJ_MD">SYNTRAJ_MD</a>).<br/>
<br/>In MPI averaging mode, structural clustering will automatically concatenate the individual
trajectories (ordered by increasing replica number). By default, the resultant transition matrix(ces) can contain spurious transitions 
(at least) at the trajectory breaks. These can be corrected by modifications to the link structure, <i>e.g.</i>,
<a href="keywords.html#TRAJBREAKSFILE">custom breaks</a>.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the data accumulation for clustering does not take note of whether particles
are physically present or not. This ensures coherent calculation of conformational distances. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<a class="NoHi" id="PRINCIPAL_COMPONENTS.dat"><b>PRINCIPAL_COMPONENTS.dat</b></a><br/>
<br/>
<i>n_variables</i> columns, <i>n_collected</i>+1 rows:<br/>
<br/>
Format: <i>n_variables</i>(1x,g10.4) with one header line<br/>
<br/>
This file contains the data collected during structural clustering (see <a href="keywords.html#CCOLLECT">FMCSC_CCOLLECT</a>)
after being linearly transformed into either principal component (PCA) space (<a href="keywords.html#PCAMODE">FMCSC_PCAMODE</a> is 3)
or into time structure-based independent component (tICA) space (<a href="keywords.html#PCAMODE">FMCSC_PCAMODE</a> is 5).
The dimensionality of the subspace is <i>n_variables</i>,
and the number of collected/analyzed instances (snapshots) is <i>n_collected</i>. Naturally, this file
can become quite large for large input data sets. The data subspace is the same as that
selected via <a href="keywords.html#CFILE">FMCSC_CFILE</a> and <a href="keywords.html#CDISTANCE">FMCSC_CDISTANCE</a>
for structural clustering. The associated units are the same as in structural clustering (generally &#8491; and &deg;).
Linear transformations of this type are performed in a post-processing setting.
They share their data collection routines entirely with structural clustering, and execution is therefore
coupled. It is possible, however, to suppress the analysis entirely or to specifically suppress printing of this
output file (and of computing the necessary data, &rarr; <a href="keywords.html#PCAMODE">FMCSC_PCAMODE</a>).<br/>
The header lines gives the eigenvalue of the relevant matrix for each component. This is the quantity that is
expected to be maximal for the component with the largest value. This quantity is the variance in PCA and the
autocorrelation at fixed lag time in tICA. In the output file, these values should
be strictly ordered such that the magnitude decreases from left to right. Note that autocorrelation values can
be negative which means that the last component may actually be interesting.
The corresponding transformed data set is then found for each
corresponding column in the remaining <i>n_collected</i> rows (with the natural simulation sequence
order being preserved from top to bottom). A histogram-based analysis of 
the first few columns of this file provides a way to compute probability (free energy) surfaces in component space
as they are often found in the literature. The transformed data are produced with the help of
the eigenvectors themselves that are written to a separate output file
(&rarr; <a href="outputfiles.html#PRINCIPAL_COMPONENTS.evs">PRINCIPAL_COMPONENTS.evs</a>).
Note that PCA and tICA are performed on centered data (for each variable, sample mean
is subtracted), and that the data written to this file are also centered. It is important to keep in mind that eigenvectors
have arbitrary sign, which means that the transformed data can also have arbitrary sign.<br/>
<br/>In MPI averaging mode, computations of linear transforms operate on the joint set of concatenated trajectories.
Consequently, the resultant time series in PC space is longer by a factor corresponding to the number of processes
(&rarr; <a href="keywords.html#REPLICAS">FMCSC_REPLICAS</a>). Trajectories are arranged by replica number from low 
to high numbers.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the necessary data accumulation does not take note of whether particles
are physically present or not. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<a class="NoHi" id="PRINCIPAL_COMPONENTS.evs"><b>PRINCIPAL_COMPONENTS.evs</b></a><br/>
<br/>
<i>n_variables</i> columns, <i>n_variables</i>+1(2) rows:<br/>
<br/>
Format: <i>n_variables</i>(1x,g10.4) with one header line<br/>
<br/>
This file contains the eigenvectors resulting from multivariate analyses of this type that are enabled by
keyword <a href="keywords.html#PCAMODE">FMCSC_PCAMODE</a>. The source data are those collected 
during the course of structural clustering and selected via <a href="keywords.html#CFILE">FMCSC_CFILE</a>
and <a href="keywords.html#CDISTANCE">FMCSC_CDISTANCE</a>. The dimensionality of the subspace is <i>n_variables</i>.
Like structural clustering, these analyses are performed in a post-processing setting.
The associated output file <a href="outputfiles.html#PRINCIPAL_COMPONENTS.dat">PRINCIPAL_COMPONENTS.dat</a>
provides the transformed data based on the vectors supplied here. The header line gives the singular values or
eigenvalues of a matrix, which should decrease cateogrically from left to right. For tICA, this is the
same quantity as the header line written to <a href="outputfiles.html#PRINCIPAL_COMPONENTS.dat">PRINCIPAL_COMPONENTS.dat</a>.
For PCA, the singular values of the data matrix (from singular value decomposition) are written unless the <a href="keywords.html#MPIAVG">MPI-parallel</a>
clustering workflow is used. This is because in the latter case CAMPARI resorts to the actual eigenvalue
problem with an explicit calculation of the covariance matrix.
The corresponding eigenvector (coefficients for linear transform) is then found in each
corresponding column in the remaining <i>n_variables</i> rows. Note that eigenvectors have arbitrary sign.<br/>
<br/>In MPI averaging mode, computations of linear transforms operate on the joint set of concatenated trajectories.<br/>
<br/>
In <a href="keywords.html#PDBANALYZE">trajectory analysis runs</a> using <a href="keywords.html#FRAMESFILE">frame weights</a>,
this output file does not reflect the weights in any way.<br/>
<br/>In (S)GC ensemble calculations, the necessary data accumulation does not take note of whether particles
are physically present or not. It is not
recommended to use these two functionalities simultaneously unless there are very specific reasons for doing so.<br/>
<br/>
<br/>
<a class="NoHi" id="WANGLANDAU_G.dat"><b>WANGLANDAU_G.dat</b></a><br/>
<br/>
Estimate of the <a href="keywords.html#WL_MODE">target distribution</a> produced by application of the
<a href="keywords.html#MC_ACCEPT">Wang-Landau</a> (WL) method (always logarithmic, <i>i.e.</i>, ln g(E), ln p(E), ln p(Z),
or selected two-dimensional PMFs):<br/>
<br/>
Format:<br/>
One-dimensional case: (g24.14,1x,g25.15)<br/>
Two-dimensional case: (g24.14,1x,N<sub>d2</sub>(g25.15)) where N<sub>d2</sub> is the number of bins for the second dimension;
header line is provided in addition<br/>
<br/>
This output file lists the <a href="keywords.html#WL_MODE">target distribution</a> (logarithmic), <i>e.g.</i>,
the logarithm of the (energy) density of states. The bin centers for the first dimension are
always provided in the first column, and the bin-wise estimates are provided in subsequent columns. In the two-dimensional case, there
is an additional header line (starting with '#') giving the N<sub>d2</sub> bin centers for the second dimension.<br/>
The estimate of the target distribution is the key result of the WL method. The accuracy of the 
estimate is fundamentally limited by the final value of the <a href="keywords.html#WL_F0">f parameter</a>, which
serves as the increment for the estimate (see log-output for this information). Successive reduction of this parameter according to 
<a href="keywords.html#WL_HVMODE">certain rules</a> is what both prescribes and diagnoses convergence of the
WL method. Because the simulated ensemble is not the canonical one (instead it ideally is a flat walk ensemble),
any other quantities extracted from the simulations require <a href="keywords.html#FRAMESFILE">snapshot-based reweighting</a>,
although it remains unclear under which exact circumstances this is appropriate. The most critical issues to keep in mind
are:<ol type='a'>
<li>Discretization errors in particular in regions where the target distribution varies quickly</li>
<li>Degeneracy errors, where the dimensions orthogonal to the reaction coordinate are not explored properly, yet 
the simulation displays apparent convergence</li>
<li>Range restriction errors for quantities with unknown bounds (particularly relevant for reweighting)</li>
<li>Reproducibility (the nature of the method renders typical error estimates inapplicable)</li>
</ol>
<br/>
For <a href="keywords.html#REMC">replica exchange calculations</a>, the estimates of the target distribution are produced and
printed for individual replicas.
If the <a href="keywords.html#MPIAVG">parallel Wang-Landau implementation</a> is used, the result represents the 
joint estimate obtained by combining data from all replicas.<br/>
<br/>
Concurrent use of (S)GC ensembles and the WL method is not recommended. It is not clear how the indirect coupling between
chemostat and WL procedure would influence the results.<br/>
<br/>
<br/>
<a class="NoHi" id="WANGLANDAU_GH.dat"><b>WANGLANDAU_GH.dat</b></a><br/>
<br/>
Total (overall) visitation histogram for the <a href="keywords.html#MC_ACCEPT">Wang-Landau</a> (WL) method:<br/>
<br/>
Format:<br/>
One-dimensional case: (g24.14,1x,i12)<br/>
Two-dimensional case: (g24.14,1x,N<sub>d2</sub>(i12)) where N<sub>d2</sub> is the number of bins for the second dimension;
header line is provided in addition<br/>
<br/>
This output file lists integer counts for the bins visited by a WL simulation. The bin centers for the first dimension are
always provided in the first column, and the counts are provided in subsequent columns. In the two-dimensional case, there
is an additional header line (starting with '#') giving the N<sub>d2</sub> bin centers for the second dimension. Bins
without any counts should be completely disregarded for analysis. Boundaries of relevant bins for the two-dimensional
case will generally not be rectangular. A well-converged WL simulation should have spent most of its time with
a solid estimate of the <a href="keywords.html#WL_MODE">target distribution</a> in place, and the counts can 
therefore be expected to be flat. However, since this is not true initially, the level of flatness cannot be used
as a diagnostic (it is slightly more appropriate to use <a href="outputfiles.html#WL_TMP">temporary visitation histograms</a>
for this). This is particularly true if the target distribution contains very steep boundaries.<br/>
<br/>
For <a href="keywords.html#REMC">replica exchange calculations</a>, the histograms are produced and printed for individual replicas.
If the <a href="keywords.html#MPIAVG">parallel Wang-Landau implementation</a> is used, the histogram represents the 
accumulated total from all replicas.<br/>
<br/>
Concurrent use of (S)GC ensembles and the WL method is not recommended. It is not clear how the indirect coupling between
chemostat and WL procedure would influence the results.<br/>
<br/>
<br/>
<a class="NoHi" id="WL_TMP"><b>Temporary Histogram Output for Wang-Landau Runs</b></a><br/>
<br/>
If keyword <a href="keywords.html#WL_DEBUG">WL_DEBUG</a> is set to 1 (true), and if the  
<a href="keywords.html#WL_MODE">target distribution</a> for the <a href="keywords.html#MC_ACCEPT">Wang-Landau</a> method
is one-dimensional, CAMPARI will write temporary output files similar to
<a href="outputfiles.html#WANGLANDAU_G.dat">WANGLANDAU_G.dat</a> (called _TMPL_WL_G_XX where XX is an integer
number (leading zeros) used to eventually recycle these temporary files) and 
<a href="outputfiles.html#WANGLANDAU_GH.dat">WANGLANDAU_GH.dat</a> (called _TMPL_WL_GH_XX). The latter
report the <i>temporary</i> histograms, <i>i.e.</i>, the counts since the
<a href="keywords.html#WL_FLATCHECK">last update of the f parameter</a>. All these files are written
primarily for debugging purposes and are not documented in detail.<br/>
<br/>
<br/>
<a class="NoHi" id="ELS_WFRAMES.dat"><b>ELS_WFRAMES.dat</b></a><br/>
<br/>
Resultant weights per simulation step for the <a href="keywords.html#SCULPT">energy landscape sculpting</a> (ELS) method:<br/>
<br/>
Format: (i14,1x,g19.11E3)<br/>
<br/>
This output file lists indices of simulation snapshots with their associated weights. This file is in the same format
required for snapshot-based reweighting via <a href="keywords.html#FRAMESFILE">FMCSC_FRAMESFILE</a>. Output is only
produced for snapshots that match the frequency chosen via <a href="keywords.html#ELS_PRINT_WEIGHTS">FMCSC_ELS_PRINT_WEIGHTS</a>.
In most cases, this should be matched to the frequency of <a href="outputfiles.html#Trajectory_output">trajectory output</a> (&rarr;
<a href="keywords.html#XYZOUT">FMCSC_XYZOUT</a>), since otherwise options for reanalysis are limited. The same weights printed to
this file can also be computed manually from the data in <a href="outputfiles.html#ENERGY.dat">ENERGY.dat</a> (see comments there).
This is because the generating distribution is known, <i>i.e.</i>, for the i<sup>th</sup> snapshot,
w<sub>i</sub> = p<sub>i,desired</sub>/p<sub>i,generating</sub>. Importantly, the first column lists the simulation step number
referring to the original simulation, <i>i.e.</i>, it is usually required to post-process the data in this column for 
a subsequent <a href="keywords.html#PDB_ANALYZE">trajectory analysis run</a>.<br/>
<br/>
<br/>
For any <a href="keywords.html#MPI_settings">parallel run</a>, this file is produced and printed for all individual replicas
using the standard naming conventions.<br/>
<br/>
<br/>

</div>
                    
                </div>
                </div>
                </div>
                
            <!-- Side Navigation Bar -->
            <!--<div class="art-sidebar1">-->                                    
                <!-- Contributors Block -->
                <!--<div class="art-Block">
                    <div class="art-Block-tl"></div>
                    <div class="art-Block-tr"></div>
                    <div class="art-Block-bl"></div>
                    <div class="art-Block-br"></div>
                    <div class="art-Block-tc"></div>
                    <div class="art-Block-bc"></div>
                    <div class="art-Block-cl"></div>
                    <div class="art-Block-cr"></div>
                    <div class="art-Block-cc"></div>
                    <div class="art-Block-body">
                        <div class="art-BlockHeader">
                            <div class="l"></div>
                            <div class="r"></div>
                            <div class="art-header-tag-icon">
                                <div class="t">Contributors - Links</div>
                            </div>
                        </div><div class="art-BlockContent">
                            <div class="art-BlockContent-body">
                                <div>
                                <b>Pappu Lab</b><br />
                                Washington University in St. Louis, MO<br />
                                Website: <a href="lima.wustl.edu">lima.wustl.edu</a><br />
                                Phone: (999) 999-9999 <br/>
                                <br/>
                                <b>Andreas Vitalis</b><br />
                                Unistat de Zurich<br />
                                Website: <a href="papajohns.com">vitalis.net</a><br />
                                Phone: (111) 999-9999 <br/>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>-->
                
                <!-- Search Block -->
                <!--
                <div class="art-Block">
                    <div class="art-Block-tl"></div>
                    <div class="art-Block-tr"></div>
                    <div class="art-Block-bl"></div>
                    <div class="art-Block-br"></div>
                    <div class="art-Block-tc"></div>
                    <div class="art-Block-bc"></div>
                    <div class="art-Block-cl"></div>
                    <div class="art-Block-cr"></div>
                    <div class="art-Block-cc"></div>
                    <div class="art-Block-body">
                        <div class="art-BlockHeader">
                            <div class="l"></div>
                            <div class="r"></div>
                            <div class="art-header-tag-icon">
                                <div class="t">Find Text</div>
                            </div>
                        </div><div class="art-BlockContent">
                            <div class="art-BlockContent-body">
                                <div><form method="get" id="newsletterform" action="javascript:void(0)">
                                <input type="text" value="" name="email" id="s" style="width: 95%;" />
                                <span class="art-button-wrapper">
                                    <span class="l"> </span>
                                    <span class="r"> </span>
                                    <input class="art-button" type="submit" name="search" value="Search"/>
                                </span>
                                </form></div>
                            </div>
                        </div>
                    </div>
                </div> 
                -->
                
                <!-- News/Updates Block -->
                <!--
                <div class="art-Block">
                    <div class="art-Block-tl"></div>
                    <div class="art-Block-tr"></div>
                    <div class="art-Block-bl"></div>
                    <div class="art-Block-br"></div>
                    <div class="art-Block-tc"></div>
                    <div class="art-Block-bc"></div>
                    <div class="art-Block-cl"></div>
                    <div class="art-Block-cr"></div>
                    <div class="art-Block-cc"></div>
                    <div class="art-Block-body">
                        <div class="art-BlockHeader">
                            <div class="l"></div>
                            <div class="r"></div>
                            <div class="art-header-tag-icon">
                                <div class="t">Recent News</div>
                            </div>
                        </div><div class="art-BlockContent">
                            <div class="art-BlockContent-body">
                                <div>
                                    <p><b>Jun 14, 2008</b><br/>
                                    Aliquam sit amet felis. Mauris semper, 
                                    velit semper laoreet dictum, quam 
                                    diam dictum urna, nec placerat elit 
                                    nisl in quam. Etiam augue pede, 
                                    molestie eget, rhoncus at, convallis 
                                    ut, eros. Aliquam pharetra.<br/>
                                    <a href="javascript:void(0)">Read more...</a></p>
                                                          
                                    <p><b>Aug 24, 2008</b><br/>
                                    Aliquam sit amet felis. Mauris semper, 
                                    velit semper laoreet dictum, quam 
                                    diam dictum urna, nec placerat elit 
                                    nisl in quam. Etiam augue pede, 
                                    molestie eget, rhoncus at, convallis 
                                    ut, eros. Aliquam pharetra.<br/>
                                    <a href="javascript:void(0)">Read more...</a></p>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div> --><!-- End Sidebar -->

            </div>
        </div>
    </div> <!-- End main page contents -->

            
    <!-- Footer -->
    <div class="art-Footer">
    <div class="art-Footer-background">
    <div class="art-Footer-inner">
        <a href="#" class="art-rss-tag-icon" title="RSS"></a>
        <div class="art-Footer-text">
        <p><a href="#">Contact Us</a> | <a href="#">Terms of Use</a>
        | <a href="#">License Information</a><br />
        Copyright &copy; 2010 . All Rights Reserved.</p>
        </div>
    </div>
    </div>
    </div>
    <p class="art-page-footer">XHTML and CSS valid. Powered by <a href="http://www.flashmint.com/">FlashMint</a> - flash templates provider.</p>
    <div style="text-align: center; font-size: 0.75em;">Design downloaded from <a href="http://www.freewebtemplates.com/">free website templates</a>.</div>
                
</div>
</body>
</html>
