<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<!--
############################################################################
# LICENSE INFO:                                                            #
############################################################################
#    This file is part of CAMPARI.                                         #
#                                                                          #
#    Version 3.0                                                           #
#                                                                          #
#    Copyright (C) 2017, The CAMPARI development team (current and former  #
#                        contributors)                                     #
#                        Andreas Vitalis, Adam Steffen, Rohit Pappu, Hoang #
#                        Tran, Albert Mao, Xiaoling Wang, Jose Pulido,     #
#                        Nicholas Lyle, Nicolas Bloechliger, Marco Bacci,  #
#                        Davide Garolini, Jiri Vymetal                     #
#                                                                          #
#    Website: http://sourceforge.net/projects/campari/                     #
#                                                                          #
#    CAMPARI is free software: you can redistribute it and/or modify       #
#    it under the terms of the GNU General Public License as published by  #
#    the Free Software Foundation, either version 3 of the License, or     #
#    (at your option) any later version.                                   #
#                                                                          #
#    CAMPARI is distributed in the hope that it will be useful,            #
#    but WITHOUT ANY WARRANTY; without even the implied warranty of        #
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
#    GNU General Public License for more details.                          #
#                                                                          #
#    You should have received a copy of the GNU General Public License     #
#    along with CAMPARI.  If not, see <http://www.gnu.org/licenses/>.      #
############################################################################
# AUTHORSHIP INFO:                                                         #
############################################################################
#                                                                          #
# MAIN AUTHOR:   Adam Steffen                                              #
# CONTRIBUTIONS: Andreas Vitalis, Nicholas Lyle                            #
#                                                                          #
############################################################################
-->
<html xml:lang="en" dir="ltr" xmlns="http://www.w3.org/1999/xhtml" lang="en-US">
<head>
<!--
    Designed by CAMPARI Development Group
    Base template (without user's data) checked by http://validator.w3.org : "This page is valid XHTML 1.0 Transitional"
    -->
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"></meta>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge"></meta>
  <title>CAMPARI Tutorial 1</title>
  <link rel="stylesheet" href="style_red.css" type="text/css" media="screen"></link>
</head>
<body>
<div id="art-page-background-simple-gradient"></div>
<div id="art-main"><!-- Header -->
<div class="art-Sheet2"><!-- Banner Art -->
<div class="art-Header">
<div class="art-Header-png"></div>
</div>
<!-- Top Navigation Bar -->
<div class="art-nav">
<div class="l"></div>
<div class="r"></div>
<ul class="art-menu">
  <li><a href="index.html"><span class="l"></span><span class="r"></span><span
 class="t">Home</span></a></li>
  <li><a href="documentation.html" class=" active"><span class="l"></span><span
 class="r"></span><span class="t">Documentation</span></a></li>
  <li><a href="references.html"><span class="l"></span><span class="r"></span><span
 class="t">References</span></a></li>
  <li><a href="contact.html"><span class="l"></span><span class="r"></span><span
 class="t">Support</span></a></li>
  <li><a href="download.html"><span class="l"></span><span class="r"></span><span
 class="t">Download</span></a></li>
</ul>
</div>
<!-- Top Sub Navigation Bar -->
<div class="nav-menu">
<ul class="navlist">
  <li><a href="documentation.html">Overview</a></li>
  <li><a href="install.html">Install</a></li>
  <li><a href="tutorials.html" class=" active">Tutorials</a></li>
  <li><a href="inputfiles.html">Inputfiles</a></li>
  <li><a href="keywords.html">Keywords</a></li>
  <li><a href="outputfiles.html">Outputfiles</a></li>
  <li><a href="parameters.html">Parameters</a></li>
  <li><a href="development.html">Development</a></li>
  <li><br/>
  </li>
</ul>
</div>
</div>
<!-- Main Page Contents -->
<div class="art-Sheet">
<div class="art-Sheet-bl"></div>
<div class="art-Sheet-br"></div>
<div class="art-Sheet-bc"></div>
<div class="art-Sheet-cl"></div>
<div class="art-Sheet-cr"></div>
<div class="art-Sheet-cc"></div>
<div class="art-Sheet-body">
<div class="art-contentLayout"><!-- Page Contents -->
<div class="art-content">
<div class="art-Post">
<div class="art-Post-body">
<div class="art-PostContent">
<br/>
<h2>Tutorial 1: First CAMPARI Simulation - Torsional Space Monte Carlo
Sampling of a Q<sub>N</sub>-Peptide ("Polyglutamine") in Implicit
Solvent</h2>
<h3>Preface:</h3>
A single, highly-flexible peptide chain like polyglutamine (Q<sub>N</sub>)
is an example of a system that can be simulated efficiently by Monte Carlo (MC)
algorithms given that N is small enough. Researchers in the Pappu
lab have studied this system extensively, and it is one of the most-often
simulated system within CAMPARI. It is therefore the perfect example system
for an introductory tutorial to CAMPARI.<br/>
A homopolymer such as Q<sub>N</sub> is an easy system to setup and run using CAMPARI.
From a polymeric point of view, polyglutamine adopts collapsed (globular) and disordered conformations
in aqueous milieus at ambient temperature and pressure. Despite being an exclusively polar system,
experimentally polyglutamine is extremely prone to aggregate and - in this capacity - associated with
several neurodegenerative diseases such as Huntington's. Polymeric properties and 
association of monomeric and dimeric Q<sub>N</sub> were analyzed, for example, in this
<a href="references.html#ref24_4">reference</a>. 
At the end of this tutorial, you should be able to set up simulations that
attempt to reproduce those data at least for the monomeric case.<br/>
<br/>
Before we can start, however, make sure you have a working version of the serial
or the <a href="keywords.html#NRTHREADS">threads-parallel</a> 
CAMPARI executable and parameter files (normally in the subdirectory params/ )
available. If you still need these, please refer to the <a href="install.html">Installation
section of the documentation</a>.<br/>
<br/>
If you have CAMPARI installed and compiled, there are three
main steps that need to be done to run a successful CAMPARI simulation:<br/>
<ol>
  <li><span style="font-style: italic;">Create a key-file</span>.
The key-file contains all the instructions and parameters needed to run
the simulation you want. An entire section of the documentation
is dedicated to clearly laying out the purpose of every keyword (<a href="keywords.html">keywords</a>).
This understandably very verbose, and it is precisely the role of tutorials to introduce
groups of keywords in a direct application context.
Some (but very few) keywords are essential
and cannot be left out of any key-file. Most keywords have default values set
or are not applicable to a specific calculation, so in a lot of cases they can be omitted.
Often times, however, it will help to write clearer key-files that contain 
more than the necessary keywords, especially for those that affect the 
majority of simulations. That way you can ensure that you are familiar with
the most relevant options, which may save a lot of time and avoid
running "bad" (as in, containing inadvertent errors) simulations. 
The template key-file used in this tutorial is generally a
good starting point for Monte Carlo simulations within CAMPARI. Not surprisingly, we will 
(further below) start this tutorial by looking at this key-file and explaining its
components.</li>
  <li><span style="font-style: italic;">Check and/or create auxiliary input files</span>.
The only auxiliary input files that are always needed are 
the <a href="inputfiles.html#FMCSC_SEQFILE">sequence file</a>
and the parameter file (&rarr;<a href="keywords.html#PARAMETERS">PARAMETERS</a>). These
define the system to be simulated and allow CAMPARI to collect the necessary 
parameters to define the interactions amongst its components.
Many additional keywords are references to file system locations where
further auxiliary input files are placed. These often have very simple formats, and are
only needed for specialized calculations (such as <a href="keywords.html#REMC">replica-exchange runs</a>).
Obviously, if any necessary input file does not exist, is not readable, or is in the incorrect format,
CAMPARI will not run correctly and most likely exit with an indicative message. It is always
recommended to scan the initial portion of the log-output produced by CAMPARI (for non-MPI runs,
this, by default, is written to the terminal) to check for warnings. Many features are automatically
disabled if required inputs are missing, and this may not always be desirable.
In addition, segmentation faults sometimes occur with bad input files because CAMPARI uses Fortran formatted reads
(<i>i.e.</i>, variable type mismatches are poorly tolerated, empty lines are not tolerated everywhere,
and very long strings can be problematic (<i>e.g.</i>, file names: there is a macro in source file
"<a href="../source/macros.i">macros.i</a>" defining maximum string lengths). These language limitations are clearly annoying, and their
only usefulness lies in pointing out inadvertent errors in input files.</li>
  <li><span style="font-style: italic;">Run CAMPARI</span>.
If all input files are present and the key-file complete, the
command is as simple as:<br/>
<br/>
<tt>
&lt;full path to CAMPARI&gt;/bin/x86_64/campari -k foo.key<br/>
</tt>
<br/>
Note that the "x86_64" is an architecture or compiler string chosen during installation, so it may differ 
for your specific case. For the threads-parallel executable:<br/>
<br/>
<tt>
&lt;full path to CAMPARI&gt;/bin/x86_64/campari_threads -k foo.key<br/>
</tt>
<br/>
All output files will be created in the working directory. Remember that
CAMPARI (by design) will overwrite almost all of these output files if the 
simulation is rerun in the same directory. Therefore, make sure to back up / move data
from older runs that you wish to preserve.</li>
</ol>
Note that this tutorial is primarily meant to familiarize new users
with the use of CAMPARI in general. The example chosen is more or less arbitrary.
Therefore, it will not always be clear as to why certain choices for run
parameters (keywords) are suggested. Please refer to the other tutorials
for cases that are constructed to accomplish very specific tasks.<br/>
<br/>
<h3>Step-by-Step:</h3>
<h5>Step 1 - Create (by modification) an appropriate key-file for Monte Carlo simulations of Q<sub>N</sub></h5>
Download or copy the <a href="../examples/tutorial1/TEMPLATE.key">key-file found 
in the subdirectory examples/tutorial1/</a> to your current working directory and rename the file
as "tutorial1.key". This can be used as a starting point for future simulations.
CAMPARI offers many options for customizing the simulation, but often
very little needs to be modified from key-file to key-file if the simulations
are analogous (for example, simulations on the same system, but with different analyses or
sampling methods).<br/>
<br/>
First, open the key-file and inspect its contents. Notice the universal
prefix used by almost all keywords, <i>viz.</i>, "FMCSC_".
This tag, which is largely historical, makes it somewhat easier to automatically find keywords both in the source code
and in key-files. Its single most important role is that it allows the key-file parser to 
differentiate unused keywords from other lines lacking a leading comment character ('#').
We will only examine some of the more important keywords in this section. For all others,
you can and should use the <a href="keywords.html">comprehensive documentation</a>. 
The descriptions there are usually relatively detailed and contain many links to
related keywords or input/output files allowing you to navigate efficiently through
topically connected subjects found in different parts of the documentation:<br/>
Two keywords of fundamental interest pertain to the only two essential input files
required by a standard CAMPARI run:<br/>
<ul>
<li><a href="keywords.html#PARAMETERS">PARAMETERS</a> -
This is one of two keywords (the other being <a href="keywords.html#SEQFILE">FMCSC_SEQFILE</a>) that
are absolutely essential to run CAMPARI in any fashion (both are no longer needed when
using the <a href="keywords.html#netcdf_anal">data mining executable</a>). This keyword
gives the path to the parameter file that will be used in the
simulation to define the interaction model between the particles present, and to
provide the most fundamental atomic properties needed to even build biomolecules.
Several canonical, biomolecular force fields are supported, such as <a href="references.html#ref1_7">OPLS-AA/L</a>,
<a href="references.html#ref1_8">CHARMM22</a>, <a href="references.html#ref1_1">AMBER99</a>, or
<a href="references.html#ref1_6">GROMOS53</a>. This keyword is one of two exceptions to the
use of "FMCSC_" as a standard prefix, and for our
simulation should be written as:<br/>
<br/>
<tt>
PARAMETERS &lt;full path to folder&gt;/campari/params/abs3.2_opls.prm<br/>
</tt>
<br/>
You can read up on the <a href="keywords.html#abs32_opls_prm">corresponding documentation</a>
to find out what the specified file corresponds to. It should become apparent
later why we are using it in this example.</li>
<li><a href="keywords.html#SEQFILE">FMCSC_SEQFILE</a> -
This is the path to the <a href="inputfiles.html#FMCSC_SEQFILE">sequence file</a>
for the simulation. Being the second essential input file, it
defines the composition of the system as a sequence of "residues". Residues
are one of the fundamental organizational units in CAMPARI 
As mentioned, this file is always needed regardless of whether you are
reading in an annotated coordinate file (via <a href="keywords.html#PDBFILE">FMCSC_PDBFILE</a>
as a starting structure) or when generating a random starting structure from
the sequence file itself. CAMPARI will never automatically infer the system's sequence
from an input file other than the <a href="inputfiles.html#FMCSC_SEQFILE">sequence file</a> itself.
There is, however, a <a href="../tools/convert_SEQRES_toseq.sh">tool available</a> to construct an appropriate sequence file
from a PDB file as found in the data bank itself.
We will discuss constructing the sequence file from scratch in detail in step 2 of this tutorial. Set the value
for the keyword to q20.in.</li>
</ul>
Below you can find short descriptions of keywords in 5 separate groups. We here list only those
of particular interest to the simulation example chosen for this tutorial.<br/>
<h4>Basic setup</h4>
There are some very fundamental and consequently very important keywords that control the type, extent,
and initial conditions for the calculation to be undertaken.
<ul>
<li><a href="keywords.html#PDBANALYZE">FMCSC_PDBANALYZE</a> -
This is the major switch between running a "fresh" CAMPARI simulation that
uses an actual sampling engine <i>vs.</i> running CAMPARI in trajectory analysis mode.
The latter refers to the possibility of feeding in trajectory data
that can then be analyzed by CAMPARI, for example by its various <a href="outputfiles.html">analysis routines</a>.
Keep this keyword at 0 (default) to run a fresh simulation. This keyword
is an example of a "simple logical" as it is often referred to in the 
<a href="keywords.html">documentation of keywords</a>. This means that
specifying a value of 1 instructs CAMPARI to turn the corresponding functionality
on ("true") whereas all other specified values are interpreted as "false", and disable
the corresponding functionality.</li>
<li><a href="keywords.html#DYNAMICS">FMCSC_DYNAMICS</a> -
Use the default setting of 1 corresponding to a pure Monte Carlo (MC) simulation. This keyword
allows the use of gradient-based methods such as molecular dynamics.</li>
<li><a href="keywords.html#CARTINT">FMCSC_CARTINT</a> - This controls the basic choice of degrees
of freedom that the system is supposed to evolve in. It can be either rigid-body/torsional (internal) coordinates
or Cartesian coordinates. In MC, only the former is available (value of 1).</li>
<li><a href="keywords.html#ENSEMBLE">FMCSC_ENSEMBLE</a> -
The thermodynamic ensemble assumed to underlie the simulation is specified by this keyword.
The default setting of 1 corresponds to the canonical (NVT) ensemble, which is the most convenient
intuitive ensemble for a system in a implicit bath of solvent.</li>
<li><a href="keywords.html#RANDOMIZE">FMCSC_RANDOMIZE</a> - Setting
this keyword to option 1, which is also the
default, will instruct CAMPARI to attempt to find a set of degrees of 
freedom that randomizes the starting structure in such a way that excluded
volume is approximately obeyed. Randomization by this option is generally useful 
since it will supplement any structural input with randomized parts as needed.
When running multiple simulations for the same system, 
randomized starting structures are in important condition for keeping the final results 
interpretable in statistical terms (they help understand the role of initial state bias).
Randomization is controllable at different levels: options 0-3 for <a href="keywords.html#RANDOMIZE">FMCSC_RANDOMIZE</a>
and keywords <a href="keywords.html#RANDOMTHRESH">FMCSC_RANDOMTHRESH</a> and <a href="keywords.html#RANDOMATTS">FMCSC_RANDOMATTS</a>.</li>
<li><a href="keywords.html#NRSTEPS">FMCSC_NRSTEPS</a> - This is the
total number of elementary simulation (here MC) steps taken by the
simulation. Here we will set this keyword to
21x10<sup>6</sup>. Based on prior experience, this simulation lengths ensures
that we have sufficient data for the readouts we consider
(if you are interested in simply following the technical steps
involved in this tutorial, lower this and the setting for 
<a href="keywords.html#EQUIL">FMCSC_EQUIL</a> below by two orders of magnitude to 
create a modified "tutorial" that finishes in a few minutes).</li>
<li><a href="keywords.html#EQUIL">FMCSC_EQUIL</a> - This is the
number of total elementary simulation (here MC) steps discarded as
equilibration steps. NRSTEPS minus EQUIL will be the total number
of steps over which analysis will be run, with the frequency of
analysis controlled by output settings discussed below. Note that 
no trajectory data will be produced during equilibration, either
(&rarr; <a href="keywords.html#XYZOUT">FMCSC_XYZOUT</a>).</li>
<li><a href="keywords.html#TEMP">FMCSC_TEMP</a> - For constant temperature simulations,
this defines the system (bath) temperature, and we should set it
to the physiological temperature (310K) for this example.</li>
</ul>
<h4>Box settings</h4>
These settings define the simulation container, and they almost always matter.
<ul>
<li><a href="keywords.html#SHAPE">FMCSC_SHAPE</a> - Setting 2 corresponds
to using a spherical droplet. Note that with a single molecule in the system (see below),
we essentially create conditions of "infinite dilution" assuming that
the radius of the droplet is large enough, that the molecule is sufficiently far away from the boundary,
 and that it cannot move away on account of the Monte Carlo move set.</li>
<li><a href="keywords.html#SIZE">FMCSC_SIZE</a> - For our simulation
of a 20-mer of polyglutamine (Q<sub>20</sub>), we will
use a radius of 400&#8491;, which gives a droplet diameter one order of magnitude larger than the contour
length of the fully extended molecule.</li>
<li><a href="keywords.html#BOUNDARY">FMCSC_BOUNDARY</a> - By using setting
4, we create a half-harmonic potential defining the droplet's boundary that acts on all atoms
within the system. Because of the size (see above), for this particular system this choice is not of importance (but this is
clearly the exception rather than the rule).</li>
<li><a href="keywords.html#SOFTWALL">FMCSC_SOFTWALL</a> - For a half-harmonic boundary potential, this
keyword sets the force constant, While this value cannot be exactly zero, it can be made arbitrarily small
to minimize the impact of the boundary in cases where this would be undesirable. As discussed below,
we will create a move set that keeps the polymer in the same place, and thus it is advisable to
set a tiny value (1.0e-9) here.</li>
</ul>
Note that a mismatch of the size of a <a href="keywords.html#BOUNDARY">periodic system</a> relative
to structural input, <i>e.g.</i>, trajectory data being <a href="keywords.html#PDBANALYZE">analyzed</a>,
can lead to errors that are difficult to understand at first.<br/>
<h4>Energy parameters</h4>
Look over the Hamiltonian section of the key-file.
The specified values essentially correspond to the use of the
<a href="references.html#ref19_1">ABSINTH implicit solvation model</a> as published.
You should not change them within the scope of this tutorial.<br/>
In general, Hamiltonian keywords consist of outside scaling factors
that can assume any positive real number including zero (the
latter disabling this term in the Hamiltonian), <i>e.g.</i>,
keyword <a href="keywords.html#SC_ATTLJ">FMCSC_SC_ATTLJ</a>
controls whether dispersive interactions assumed to be scaling with r<sup>-6</sup> act between nonbonded
particles in the system.<br/>
The second class are dependent parameters that only matter for 
one or more specific terms in the Hamiltonian. These vary widely in
their applicability, scope, and impact. For instance, <a href="keywords.html#SIGRULE">FMCSC_SIGRULE</a>
can be used to set the combination rule for how to construct
pairwise Lennard-Jones parameters &sigma;<sub>ij</sub> from terms
&sigma;<sub>ii</sub> and &sigma;<sub>jj</sub>. Some of these keywords
may be difficult to understand for someone not familiar with
molecular force fields. In any case, we still strongly recommend reading
the <a href="keywords.html#energy_terms">corresponding parts of the documentation</a>.<br/>
<h4>Monte Carlo move set controls</h4>
Monte Carlo simulations in CAMPARI are performed in rigid-body / torsional space.
For a single polypeptide chain, the native CAMPARI degrees of freedom
are the backbone (&phi;, &psi;, and &omega;) and sidechain (&chi;<sub>1-3</sub> for glutamine)
dihedral angles. A list of native degrees of freedom is found in the
<a href="inputfiles.html#FMCSC_SEQFILE">description of sequence input</a>.
Exceptions include but are not limited to the sampling of ring pucker states in 
proline and sugars where bond angles of the ring will change slightly. 
Rigid-body motions of the chain can usually be turned off for the monomer,
unless systems are meant to be studied under explicit confinement conditions.<br/>
Some of the occurring keywords are used to illustrate how to control the MC move set via
the key-file. A comprehensive description is found <a href="keywords.html#movesetcontrols">elsewhere</a>. <br/>
<ul>
<li><a href="keywords.html#ALIGN">FMCSC_ALIGN</a> - This is an important keyword that
applies to almost all pivot-like moves, <i>i.e.</i>, moves where a change in a single dihedral
angle requires a decision as to which end to swivel around. The default option, 4, is not ideal
here because stochastic swiveling means that there is no fixed base of motion. This would mean that the
molecule moves around the droplet, which implies that it will contact the boundary at some point.
Thus change it to option 3 and check that the <a href="outputfiles.html#basename_START.pdb">initial structure</a>
is far away enough from the boundary (otherwise simply restart).</li>
<li><a href="keywords.html#RIGIDFREQ">FMCSC_RIGIDFREQ</a> - This is the frequency setting for any type
of move sampling exclusively the rigid body coordinates of one or more molecules. In our example,
sampling of the rigid-body motion of the chain is wasteful so you should turn off (set it to 0.0).
Bear in mind, however, that sometimes unfortunate
settings may place part of a molecule such that it overlaps with the boundary of the droplet.
Such a conflict should be avoided as it may bias the result, for example
by choosing a larger droplet.
You can eliminate this problem altogether by switching to <a href="keywords.html#BOUNDARY">periodic boundaries</a>
or by setting <a href="keywords.html#SOFTWALL">FMCSC_SOFTWALL</a> to an extremely small value.<br/>
In general, in the MC sampler within CAMPARI, the probability of individual types
of moves is determined in a hierarchical decision tree. Turning <a href="keywords.html#RIGIDFREQ">FMCSC_RIGIDFREQ</a>
off would therefore also disable a set of related move types, such
as cluster rigid-body moves (&rarr; <a href="keywords.html#CLURBFREQ">FMCSC_CLURBFREQ</a>).
This principle applies in general, and every move set frequency keyword should indicate
in its documentation how to calculate the expected number of moves of that given type.
This should also clarify the hierarchy. The disadvantage of the binary decision tree
is that frequency settings are only relative to the respective branch (see below).</li>
<li><a href="keywords.html#CHIFREQ">FMCSC_CHIFREQ</a> - Frequency
of sidechain moves. To give an example of the decision tree in action, consider the following:
If <a href="keywords.html#PARTICLEFLUCFREQ">FMCSC_PARTICLEFLUCFREQ</a> is zero (which it must be here
since we have a constant particle number ensemble) and assuming that 
<a href="keywords.html#RIGIDFREQ">FMCSC_RIGIDFREQ</a> was set to zero above, we can infer
that a setting of 0.2 with a <a href="keywords.html#NRSTEPS">total simulation length</a> of 21x10<sup>6</sup> steps
will produce ca. 4.2x10<sup>6</sup> side chain moves. This serves as an example for how
to compute net expected attempt totals for moves of a given type (note
that we disregard the extremely specialized moves controlled by 
<a href="keywords.html#PHFREQ">FMCSC_PHFREQ</a> here, <i>i.e.</i>, we rely on the default being zero).</li>
<li><a href="keywords.html#CRFREQ">FMCSC_CRFREQ</a> - In general, concerted 
rotation moves describe a class of moves for which stretches of a polymer
are sampled while the ends are either restrained or constrained to their
original positions. By choosing a nonzero frequency here, we would rely on many default choices
for dependent keywords that would go beyond the scope of this tutorial to explain.
Therefore, keep this entire class of moves disabled (setting of 0.0).</li>
<li><a href="keywords.html#OMEGAFREQ">FMCSC_OMEGAFREQ</a> - The amide
(&omega;-)bond of the peptide chain is sampled
separately from the &phi;/&psi;-angles based on the frequency setting of this keyword.
This is an example for an improved move set that can take into account the peculiarities
of specific degrees of freedom. Here, the large barrier associated with <i>cis</i>/<i>trans</i>
isomerization and the narrow basins mean that move parameters should be different from
those for, <i>e.g.</i>, &phi;/&psi;-angles. These move parameters are controlled by their own keywords:
<a href="keywords.html#OMEGARDFREQ">FMCSC_OMEGARDFREQ</a> and <a href="keywords.html#OMEGASTEPSZ">FMCSC_OMEGASTEPSZ</a>.
These parameters are named analogously for many move types and explained for the next item.</li>
<li><a href="keywords.html#PIVOTRDFREQ">FMCSC_PIVOTRDFREQ</a> -
<a href="keywords.html#PIVOTMODE">Pivot-type moves of the &phi;/&psi;-angles</a>
are the "default" move in CAMPARI,
and their expected number is obtained from the remaining frequency settings (that is,
they are the last remaining option in the decision tree).
The role of this keyword is to control the frequency for a subset of those,
specifically the fraction of moves randomly resetting the corresponding
degrees of freedom on the interval from -180&deg; to 180&deg; (maximal step size).
This is an example of a keyword that occurs in analogous form for other move types
 as well (see for example <a href="keywords.html#RIGIDRDFREQ">FMCSC_RIGIDRDFREQ</a>
or <a href="keywords.html#CHIRDFREQ">FMCSC_CHIRDFREQ</a>).</li>
<li><a href="keywords.html#OTHERFREQ">FMCSC_OTHERFREQ</a> -
The list of native CAMPARI degrees of freedom obviously does not include degrees of freedom in unsupported residues.
However, it does not include certain symmetric dihedral angles in supported residues either,
<i>e.g.</i>, the primary amide C-N bond as found also in glutamine. To sample either of these
types of degrees freedom it is necessary to use this keyword, which controls what
are all single dihedral angle pivot moves (again with subordinate parameters).
These moves can also work on native degrees of freedom (excepting &phi;/&psi;-angles 
in polypeptides and bonds in flexible rings. Here, the settings in the key-file mean that we will
use them on native degrees of freedom only.</li>
<li><a href="keywords.html#PIVOTSTEPSZ">FMCSC_PIVOTSTEPSZ</a> - The
remaining &phi;/&psi;-pivot moves are stepwise perturbations from the current values
for the particular torsions. This
keyword controls the magnitudes of these small perturbations. Again, there are
several analogous keywords for other move types. As alluded to, for example, the <a href="keywords.html#OMEGASTEPSZ">
step size for &omega;-pivot moves</a> should generally be smaller.</li>
</ul>
Note for all of the above that CAMPARI will parse the selections made
for the move set, determine its applicability to the system at hand, and
disable those components that cannot be used. In rare cases, it may also exit at the very beginning
and ask for an explicit adjustment of the move set to avoid confusion.
Also note that there are a few dihedral angles degrees of freedom not native to CAMPARI 
(<i>i.e.</i>, they are frozen by default). These include the N-C bond in the sidechain primary amide groups
and methyl groups at the terminal caps. They could be sampled with the help of keywords
<a href="keywords.html#OTHERFREQ">FMCSC_OTHERFREQ</a> and associated keywords, but this
is not explored here.
<h4>Output frequency and files for initial analysis</h4>
CAMPARI does a lot of analysis during the simulation depending on how
you set keywords that control the frequency with which quantities of interest
are evaluated, data are accumulated, and/or instantaneous output is written
to corresponding <a href="outputfiles.html">output files</a>. 
Probably the most ubiquitous and important output are <a href="outputfiles.html#Trajectory_output">trajectory
data</a> since they - as mentioned above - allow later reanalysis of quantities
missed in the original simulation (&rarr; <a href="keywords.html#PDBANALYZE">FMCSC_PDBANALYZE</a>).
This is the traditional work flow in molecular simulations, <i>viz.</i>, to separate completely trajectory generation
and analysis.<br/>
Only 5 analysis keywords are discussed here in some detail:<br/>
<ul>
<li><a href="keywords.html#BASENAME">FMCSC_BASENAME</a> - This
keyword will apply a particular prefix to a few of
the output files. The most important output in this class is <a href="outputfiles.html#Trajectory_output">trajectory output</a>.</li>
<li><a href="keywords.html#XYZOUT">FMCSC_XYZOUT</a> - This
keyword controls the frequency at which the Cartesian coordinates
of the system are output to a trajectory file. Like every other analysis
keyword ending in "OUT" or "CALC", setting this for example to 5000 means that 
the corresponding analysis/output is performed every 5000 steps. Here, a
snapshot of the system's Cartesian coordinates will be output every 5000 steps.</li>
<li><a href="keywords.html#XYZPDB">FMCSC_XYZPDB</a> - This
keyword controls the type of trajectory output generated. By setting it to 3, we
request the trajectory to be written in dcd-format, a common binary trajectory file format
also used by programs such as CHARMM or NAMD.</li>
<li><a href="keywords.html#ENOUT">FMCSC_ENOUT</a> - This keyword controls
the output interval for instantaneous energy data split by term (see
<a href="outputfiles.html#ENERGY.dat">ENERGY.dat</a>). Set this to 5000, which 
will be sufficient to diagnose the mean boundary energy in particular.</li>
<li><a href="keywords.html#POLCALC">FMCSC_POLCALC</a> - This keyword
controls the calculation and data accumulation interval (frequency) for
polymeric descriptors of low computational cost (those that scale linearly
with system size). For instance, this will accumulate data to produce in the end
a histogram of the observed radii of gyration throughout the production phase
of the simulation. Change this to be 500.</li>
</ul>
You should now have a complete key-file with appropriate settings. 
There are many other default keywords that were not discussed, so remember
to refer to the <a href="keywords.html">full documentation of all keywords</a> to understand what
they can be used for and how they work.<br/>
<br/>
<h5>Step 2 - Create a sequence file</h5>
The sequence file is a linear series of 3-letter codes (with possible suffixes) that define
the molecular components of the simulation. CAMPARI
natively supports standard amino acids, some post-translationally modified and nonstandard amino acids, capping groups,
standard nucleotides, and many small molecules (including ions). Native support means that CAMPARI is, at the source code level,
aware of the molecular topologies of these residues and how they can be connected (for polymers). Simulating
residues that are not supported is also possible by means of a structural input file and, possibly, patches
(see <a href="inputfiles.html#FMCSC_SEQFILE">FMCSC_SEQFILE</a> and <a href="keywords.html#PDB_TEMPLATE">FMCSC_PDB_TEMPLATE</a>).
There are specific capping groups like the acetyl unit (ACE) for polypeptide N-termini and the N-methylamide group (NME)
for polypeptide C-termini. All terminal amino acids can also be made into (charged) N- or C-terminal residues simply
by adding "_N" or "_C" after the 3-letter code, respectively. When constructing a
chain, CAMPARI will connect residues into a single chain until it meets
a valid C-terminal residue (either a capping group or a normal residue explicitly made C-terminal),
at which point it begins to form a new molecule (that could be a small molecule or another polymer chain).
A polyglutamine sequence with appropriate caps would therefore look like this:<br/>
<br/>
ACE<br/>
GLN<br/>
GLN<br/>
....etc...<br/>
GLN<br/>
GLN<br/>
NME<br/>
END<br/>
<br/>
Notice the "END" statement at the end of the sequence file. This line terminates the processing of 
sequence input exactly at that point, which means that any polymers started before must have been
terminated. To learn more about sequence file construction and the available residues and small molecules, 
consult the <a href="inputfiles.html#FMCSC_SEQFILE">corresponding documentation</a>.
For this tutorial, create a sequence file with the sequence for capped Q<sub>20</sub> in the same directory
that your key-file resides in and name this file q20.in. As a final step, adjust other required paths 
in the key-file to match your local working environment.<br/>
<h5>Step 3 - Run CAMPARI</h5>
Simply run the CAMPARI executable while providing the key-file
from that folder as the only argument:<br/>
<br/>
<tt>
&lt;full path to CAMPARI&gt;/bin/x86_64/campari -k tutorial1.key &gt;&amp; log &amp;<br/>
</tt>
<br/>
If you want to use the threads-parallel executable, be sure to set the number of threads
(keyword <a href="keywords.html#NRTHREADS">FMCSC_NRTHREADS</a>) to the number of cores you can spare 
for the job. This also works with just a single core (<i>i.e.</i>, there is no need to compile and use
the fully serial version unless you intend to use features not (yet) supported by the threads parallelization):<br/>
<br/>
<tt>
&lt;full path to CAMPARI&gt;/bin/x86_64/campari_threads -k tutorial1.key &gt;&amp; log &amp;<br/>
</tt>
<br/>
This will run the job in the background, and all log-output is redirected to the file "log".
If something is incorrect, the program will in all likelihood terminate right away and
tell you in hopefully comprehensible fashion what the problem is. Common problems
include missing or incorrect input files, bad paths, and, as mentioned above, format mismatches
in the key-file (<i>.e.g.</i>, providing a character variable to a keyword expecting an integer).<br/>
At the full number of steps, and depending on your processor speed, the
simulation should run for several hours. If you just want to
quickly see whether the simulation runs, and what it outputs, you can lower the
<a href="keywords.html#NRSTEPS">number of steps</a> by two orders of magnitude as suggested above. CAMPARI
outputs a summary of the attempted calculation when starting a simulation, and provides a few
global statistics at completion. Both of those are obtained as part of a standard output. This can
be directed to a log-file, which is a good place to look
for explanations if CAMPARI crashes or terminates unexpectedly. It is always useful to look at
the initial summary right away, <i>i.e.</i>, it is better to find setup errors <i>before</i> the bulk
of the computing time has been invested.<br/>
<h5>Step 4 - Have a look at the output provided by CAMPARI</h5>
First check the last lines of said log file to see if CAMPARI completed
correctly. It should have an end time and date of execution
indicating that everything went well. Because we requested a number of reports,
the initial output will contain what is essentially debugging information.
It will be advisable to disable most standard reports for most calculations (although it is often
helpful to keep reports for user-requested features, <i>e.g.</i>, <a href="keywords.html#FRZFILE">custom constraints</a>
and their <a href="keywords.html#FRZREPORT">corresponding report flag</a>).<br/>
<br/>
Now list the contents of the folder where CAMPARI was executed. Some files are
always created if the simulation ran correctly, such as <a href="outputfiles.html#basename_START.pdb">
initial</a> and <a href="outputfiles.html#basename_END.pdb">final</a> pdb-files
(called POLYQ_START.pdb and 
POLYQ_END.pdb unless you changed the setting for <a href="keywords.html#BASENAME">FMCSC_BASENAME</a>). You should also see the 
<a href="outputfiles.html#Trajectory_output">trajectory file</a> that will be
called POLYQ_traj.dcd (unless you changed the chosen <a href="keywords.html#XYZPDB">file format</a>). The trajectory with the
full Cartesian (xyz-)coordinates will usually be obtained in binary form, for example
in the highly compressed xtc-format (if permissible) or the more space-consuming 
dcd- or NetCDF formats. CAMPARI interfaces easily with
the visualization software <a href="references.html#ref23_9">VMD</a> that can open and manipulate most trajectory formats.
These binary formats are often used by other simulation software as well (CHARMM
for instance uses the dcd-format that we also use here setting <a href="keywords.html#XYZPDB">FMCSC_XYZPDB</a> to 3).
For convenience, there is an <a href="outputfiles.html#basename_VIS.vmd">automatic visualization
script</a> to be used with VMD that is going to be called POLYQ_VIS.vmd. 
It loads the CAMPARI-generated data and defines a few representations to view the structure. Note that
trajectory output, unlike almost all other output in CAMPARI, is appended if you rerun the simulation
several times in the same folder with the same settings.<br/>
<br/>
CAMPARI can also take all the binary trajectory formats it writes as input for post-processing
in <a href="keywords.html#PDBANALYZE">trajectory analysis mode</a>. This is often useful
to refine or add certain analyses or change the settings, and can even be used to recompute energies
for the structures present using a different Hamiltonian.<br/>
<br/>
In the working directory, you should also see a file called <a href="outputfiles.html#POLYAVG.dat">POLYAVG.dat</a>. 
This is the simplest output file provided on account of us choosing a value of 500 for <a href="keywords.html#POLCALC">FMCSC_POLCALC</a>. 
By opening it, you should be able to read off the average radius of gyration (&lt;R<sub>g</sub>&gt;) for the
full simulation (minus equilibration) as the first data entry. The distribution of the radius of gyration
is printed to <a href="outputfiles.html#RGHIST.dat">RGHIST.dat</a>, and further files being produced
are <a href="outputfiles.html#RETEHIST.dat">RETEHIST.dat</a> (histogram of end-to-end distance), 
<a href="outputfiles.html#RDHIST.dat">RDHIST.dat</a> (two-dimensional histogram of a reduced radius of gyration and asphericity),
<a href="outputfiles.html#DENSPROF.dat">DENSPROF.dat</a> (radial density profile),
<a href="outputfiles.html#PERSISTENCE.dat">PERSISTENCE.dat</a> (angular correlation function along backbone), and
<a href="outputfiles.html#TURNS_RES.dat">TURNS_RES.dat</a> (average bending angles along backbone). If you ran
the simulation to its full length, these data could reveal to you that capped Q<sub>20</sub> is collapsed under these conditions
(roughly globular) with some preference for &alpha;-helices. 
<br/>
<br/>
There are many other analyses that can be performed within CAMPARI.
<a href="outputfiles.html">The section of the documentation on output files</a> provides - at the top - a list
of groups of analyses that are available within CAMPARI including links to the respective entries. By following these
chains of links, you will usually be able to collect all the information (including needed keywords) required
to perform, understand, and control these analyses. Whether these files are produced or not, will at the most fundamental
level depend on their applicability to the system and on the choices for the various calculation interval
flags (such as <a href="keywords.html#POLCALC">FMCSC_POLCALC</a> discussed above).<br/>
<br/>
This is the end of this tutorial. You should now have a good idea of what setting up a Monte Carlo simulation
in CAMPARI entails (or at least where to find the necessary information to be able to do so). Other tutorials
will expand on this knowledge and/or introduce you to new functionality within CAMPARI that was not touched
upon at all this far.<br/>
<br/>
</div>
</div>
</div>
</div>
<!-- Side Navigation Bar --><!--<div class="art-sidebar1">--><!-- Contributors Block --><!--<div class="art-Block">
                    <div class="art-Block-tl"></div>
                    <div class="art-Block-tr"></div>
                    <div class="art-Block-bl"></div>
                    <div class="art-Block-br"></div>
                    <div class="art-Block-tc"></div>
                    <div class="art-Block-bc"></div>
                    <div class="art-Block-cl"></div>
                    <div class="art-Block-cr"></div>
                    <div class="art-Block-cc"></div>
                    <div class="art-Block-body">
                        <div class="art-BlockHeader">
                            <div class="l"></div>
                            <div class="r"></div>
                            <div class="art-header-tag-icon">
                                <div class="t">Contributors - Links</div>
                            </div>
                        </div><div class="art-BlockContent">
                            <div class="art-BlockContent-body">
                                <div>
                                <b>Pappu Lab</b><br />
                                Washington University in St. Louis, MO<br />
                                Website: <a href="lima.wustl.edu">lima.wustl.edu</a><br />
                                Phone: (999) 999-9999 <br/>
                                <br/>
                                <b>Andreas Vitalis</b><br />
                                Unistat de Zurich<br />
                                Website: <a href="papajohns.com">vitalis.net</a><br />
                                Phone: (111) 999-9999 <br/>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>--><!-- Search Block --><!--
                <div class="art-Block">
                    <div class="art-Block-tl"></div>
                    <div class="art-Block-tr"></div>
                    <div class="art-Block-bl"></div>
                    <div class="art-Block-br"></div>
                    <div class="art-Block-tc"></div>
                    <div class="art-Block-bc"></div>
                    <div class="art-Block-cl"></div>
                    <div class="art-Block-cr"></div>
                    <div class="art-Block-cc"></div>
                    <div class="art-Block-body">
                        <div class="art-BlockHeader">
                            <div class="l"></div>
                            <div class="r"></div>
                            <div class="art-header-tag-icon">
                                <div class="t">Find Text</div>
                            </div>
                        </div><div class="art-BlockContent">
                            <div class="art-BlockContent-body">
                                <div><form method="get" id="newsletterform" action="javascript:void(0)">
                                <input type="text" value="" name="email" id="s" style="width: 95%;" />
                                <span class="art-button-wrapper">
                                    <span class="l"> </span>
                                    <span class="r"> </span>
                                    <input class="art-button" type="submit" name="search" value="Search"/>
                                </span>
                                </form></div>
                            </div>
                        </div>
                    </div>
                </div> 
                --><!-- News/Updates Block --><!--
                <div class="art-Block">
                    <div class="art-Block-tl"></div>
                    <div class="art-Block-tr"></div>
                    <div class="art-Block-bl"></div>
                    <div class="art-Block-br"></div>
                    <div class="art-Block-tc"></div>
                    <div class="art-Block-bc"></div>
                    <div class="art-Block-cl"></div>
                    <div class="art-Block-cr"></div>
                    <div class="art-Block-cc"></div>
                    <div class="art-Block-body">
                        <div class="art-BlockHeader">
                            <div class="l"></div>
                            <div class="r"></div>
                            <div class="art-header-tag-icon">
                                <div class="t">Recent News</div>
                            </div>
                        </div><div class="art-BlockContent">
                            <div class="art-BlockContent-body">
                                <div>
                                    <p><b>Jun 14, 2008</b><br/>
                                    Aliquam sit amet felis. Mauris semper, 
                                    velit semper laoreet dictum, quam 
                                    diam dictum urna, nec placerat elit 
                                    nisl in quam. Etiam augue pede, 
                                    molestie eget, rhoncus at, convallis 
                                    ut, eros. Aliquam pharetra.<br/>
                                    <a href="javascript:void(0)">Read more...</a></p>
                                                          
                                    <p><b>Aug 24, 2008</b><br/>
                                    Aliquam sit amet felis. Mauris semper, 
                                    velit semper laoreet dictum, quam 
                                    diam dictum urna, nec placerat elit 
                                    nisl in quam. Etiam augue pede, 
                                    molestie eget, rhoncus at, convallis 
                                    ut, eros. Aliquam pharetra.<br/>
                                    <a href="javascript:void(0)">Read more...</a></p>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div> --><!-- End Sidebar -->
</div>
</div>
</div>
<!-- End main page contents --><!-- Footer -->
<div class="art-Footer">
<div class="art-Footer-background">
<div class="art-Footer-inner"> <a href="#" class="art-rss-tag-icon"
 title="RSS"></a>
<div class="art-Footer-text">
<p><a href="#">Contact Us</a> | <a href="#">Terms of Use</a> | <a
 href="#">License Information</a><br/>
Copyright © 2010 . All Rights Reserved.</p>
</div>
</div>
</div>
</div>
<p class="art-page-footer">XHTML and CSS valid. Powered by <a
 href="http://www.flashmint.com/">FlashMint</a> - flash templates
provider.</p>
<div style="text-align: center; font-size: 0.75em;">Design downloaded
from <a href="http://www.freewebtemplates.com/">free website templates</a>.</div>
</div>
</body>
</html>
